<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-GB" xml:lang="en-GB">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="The Anopheles gambiae 1000 Genomes Consortium" />
  <meta name="dcterms.date" content="2020-10-19" />
  <meta name="keywords" content="malaria, anopheles, genomics" />
  <title>Genome variation and population structure in three African malaria vector species within the Anopheles gambiae complex</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/master/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta name="dc.title" content="Genome variation and population structure in three African malaria vector species within the *Anopheles gambiae* complex" />
  <meta name="citation_title" content="Genome variation and population structure in three African malaria vector species within the *Anopheles gambiae* complex" />
  <meta property="og:title" content="Genome variation and population structure in three African malaria vector species within the *Anopheles gambiae* complex" />
  <meta property="twitter:title" content="Genome variation and population structure in three African malaria vector species within the *Anopheles gambiae* complex" />
  <meta name="dc.date" content="2020-10-19" />
  <meta name="citation_publication_date" content="2020-10-19" />
  <meta name="dc.language" content="en-GB" />
  <meta name="citation_language" content="en-GB" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="The Anopheles gambiae 1000 Genomes Consortium" />
  <link rel="canonical" href="https://malariagen.github.io/ag1000g-phase3-data-paper/" />
  <meta property="og:url" content="https://malariagen.github.io/ag1000g-phase3-data-paper/" />
  <meta property="twitter:url" content="https://malariagen.github.io/ag1000g-phase3-data-paper/" />
  <meta name="citation_fulltext_html_url" content="https://malariagen.github.io/ag1000g-phase3-data-paper/" />
  <meta name="citation_pdf_url" content="https://malariagen.github.io/ag1000g-phase3-data-paper/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://malariagen.github.io/ag1000g-phase3-data-paper/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://malariagen.github.io/ag1000g-phase3-data-paper/v/8dde732399499bad86d4639833065b210e7b3b24/" />
  <meta name="manubot_html_url_versioned" content="https://malariagen.github.io/ag1000g-phase3-data-paper/v/8dde732399499bad86d4639833065b210e7b3b24/" />
  <meta name="manubot_pdf_url_versioned" content="https://malariagen.github.io/ag1000g-phase3-data-paper/v/8dde732399499bad86d4639833065b210e7b3b24/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Genome variation and population structure in three African malaria vector species within the <em>Anopheles gambiae</em> complex</h1>
</header>
<p><small><em>
This manuscript
(<a href="https://malariagen.github.io/ag1000g-phase3-data-paper/v/8dde732399499bad86d4639833065b210e7b3b24/">permalink</a>)
was automatically generated
from <a href="https://github.com/malariagen/ag1000g-phase3-data-paper/tree/8dde732399499bad86d4639833065b210e7b3b24">malariagen/ag1000g-phase3-data-paper@8dde732</a>
on October 19, 2020.
</em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><strong>The Anopheles gambiae 1000 Genomes Consortium</strong><br><br>
<small>
</small></li>
</ul>
<h2 class="page_break_before" id="abstract">Abstract</h2>
<h2 id="population-sampling">Population Sampling</h2>
<p>The third and final phase of the Ag1000g project data resource contains wild-caught <em>Anopheles</em> mosquito genomes collected from a total of 124 sites across 19 countries, six of which are novel, in Sub-Saharan Africa.</p>
<p>Collections from Malawi increase the density of coverage in West Africa, Central African Republic and Democratic Republic of Congo begin to fill the gap previously present in Central Africa while Malawi, Mozambique and Tanzania provide much more power to analyse East African malaria vectors.</p>
<p>New quality control and filtering pipelines were developed for this phase, so all 4,693 samples submitted across the three project phases were evaluated with these new methods.</p>
<p>High quality whole genome sequences from 2784 unique individuals passed all sample filters, 2532 females, 223 males and 29 samples with unknown sex (see Methods).</p>
<p>1823 of these wild-caught individuals are novel to phase 3.</p>
<p>181 samples from phase 2 did not pass the new quality thresholds are not included in this data release (@@ either - see Supplementary table OR explain 172 were lost via FILTER_second_rep_hi_skew &lt;- then need to explain what this is?).</p>
<p>Alongside sampling from natural populations, 699 individuals (@@?? meta data??)(parents and offspring) from 15 lab crosses have been sequenced.</p>
<p>Five of these crosses are novel to phase 3, adding power to measure the quality of wild sample variation and methods validation more generally.</p>
<p>Parents of crosses were drawn from the @@ and @@… colonies.</p>
<div id="fig:sample_collection_map" class="fignos">
<figure>
<img src="images/sample_collection_map.svg" style="width:100.0%" alt="" /><figcaption><span>Figure 1:</span> Sample Collection Map</figcaption>
</figure>
</div>
<h2 class="page_break_before" id="whole-genome-sequencing-and-alignment">Whole Genome Sequencing and Alignment</h2>
<p>4,693 individual mosquitoes were sequenced on either Illumina HiSeq2000 (n=3,130) or Illumina HiSeqX (n=1,563) to a target coverage of 30X.</p>
<p>Between machine types the median number of bases sequenced per sample was 9.76Gb and 10.33Gb respectively, representing a difference in yield (two-tailed mann-whitney U p &lt; 0.0001).</p>
<p>These values correspond to a yield per reference base (vs AgamP4) of 35.76X and 37.82X.</p>
<p>91.9% of HiSeqX runs and 80.5% of HiSeq2000 runs met the target yield of 30X.</p>
<p>Reads were aligned to the AgamP4 reference and Single Nucleotide Polymophisms (SNPs) called using GATK UnifiedGenotyper.</p>
<p>All samples successfully completed the pipeline and entered the sample quality control (QC) process.</p>
<h3 id="sample-qc">Sample QC</h3>
<p>The sample QC process was composed of three stages, sequence quality assurance, replicate handling, and anomaly detection.</p>
<p>668 samples were removed where sequencing was of insufficient quality to accurately call genotypes across the whole genome.</p>
<p>Exclusions were due to poor coverage (n=410), potential contamination (n=229), and an ambiguous sex call (n=29).</p>
<p>Where technical replicates were available, we excluded 4 pairs (8 samples) with low genotype concordance.</p>
<p>Where pairs met the concordance threshold we excluded the lower quality sample (n=407).</p>
<p>Samples were also screened pairwise within submission sets for unexpected pairs, though none were detected.</p>
<p>The AG1000G-X submission set, made up of laboratory experimental crosses, was exempted from this step due to familial similarity and high levels of inbreeding.</p>
<p>The third stage used principal component analysis (PCA) to identify and exclude individual samples that were outliers based on available metadata.</p>
<p>A review process identified samples that could not be explained parsimoniously, and were therefore likely to be sample mix ups or instances of mislabelling.</p>
<p>28 samples were excluded as they respectively dominated the first principal components, indicating high divergence from all other samples and therefore likely members of other Anopheline species.</p>
<p>A further 82 samples were excluded as potential sample mix ups.</p>
<p>Following all sample QC steps, 3,483 samples (74.2%) were retained from the original cohort for analysis.</p>
<h3 id="coverage">Coverage</h3>
<p>Summary of site coverage post QC exclusions.</p>
<h3 id="species-assignment-and-sex-calling">Species assignment and sex calling</h3>
<p>The Anopheles gambiae complex is a crypic group of sibling species, with no single locus offering unambiguous resolution of species.</p>
<p>To identify species we looked beyond the conventional set of PCR based markers and applied a wider set of ancestry informative markers (AIMs).</p>
<p>Species were not assigned to samples from laboratory colony crosses due to inbreeding and high levels of genetic drift.</p>
<p>To distinguish <em>A. arabiensis</em> from <em>A. gambiae s.l</em> a set of novel markers was derived from data from the 16 genomes project (ref).</p>
<p>Using cut offs based on agreement with the established PCR marker, 368 individuals were classed as <em>A. arabiensis</em> and 2415 as <em>A. gambiae s.l</em>.</p>
<p>A single individual collected in Tororo, Uganda is classed as intermediate- given the majority (XX%) of AIM SNPs in the genome are heterozygous between the gambiae-like and arabiensis-like alleles, this individual is likely to be an F1 hybrid.</p>
<p>To resolve the <em>A. gambiae s.l</em> individuals as <em>A. gambiae</em> and <em>A. coluzzii</em> we applied 729 AIMs identified by Neafsey et al (ref), and used by us in previous analyses (ref paper2, paper1).</p>
<p>Of the 2415 <em>A. gambiae s.l</em> individuals, 1571 were called as <em>A. gambiae s.s</em>, 675 as <em>A. coluzzii</em> and 169 as intermediate (ref collection map).</p>
<p>Many intermediate samples were sampled from the Western coast of West Africa (particularly The Gambia and Guinea Bissau), and given distinct populations of <em>A. gambiae s.l.</em> and <em>A. coluzzii</em> are also found in this region, highlights the complexity of species relationships here.</p>
<h3 id="snp-filtering-and-quality">SNP filtering and quality</h3>
<p>Site filtering is necessary to ensure that reported variation is of high quality.</p>
<p>Features of specific regions of the Anopheles genome contribute to calling errors in short-read technologies; such features include regions of high divergence from the reference, high homology between regions, copy number variation, or the presence of transposable elements.</p>
<p>Where previously we have used manually curated cutoffs based on observed mendelian error rates to filter sites (ref phase1, phase2), here we built a statistical model where cohort level summary statistics were used to identify sites likely to contain genotyping errors.</p>
<p>Using the 15 available Anopheles pedigrees previously described, we used the presence of mendelian error at sites as a proxy for genotype discordance.</p>
<p>10 of the 15 crosses were used to train the model while 5 were held out for validation.</p>
<p>Each of the 5 pedigrees represent independent evaluation sets.</p>
<p>Before applying the site filters, the false discovery rate (FDR) of the 5 crosses over all autosomal sites ranged between 0.74% and 1.10% (table XXX).</p>
<p>The application of the site filters defines the accessible fraction of the genome at 72.58%, and the range of false discovery rates is 0.04% to 0.10%.</p>
<p>The median fold change of FDR was -3.71.</p>
<p>On the hemizygous X chromosome we used the more direct measure of heterozygote calls in males to ascertain mendelian error.</p>
<p>In the dataset are 220 <em>A. gambiae s.l</em> male samples, each of which represent an independent proxy for genotype discordance.</p>
<p>Pre-application of the site filters, subject to a Genotype Quality (GQ) threshold of 30, the median heterozygosity rate was 0.244%, and post filtering this drops to 0.023% (table XX).</p>
<p>The median fold change in error rate was -3.33, with 69.97% of the X chromosome passing site filters.</p>
<p>The new model based method represents a marked improvement over the site filters generated as part of phase 2.</p>
<p>All 5 evaluation pedigrees showed a small reduction in FDR, but coupled with the lower rate of accessibility in phase 2 (62.05%) resulted in an significant improvement in Youden score (Table XXX) across autosomes.</p>
<p>The X chromosome showed a similar pattern, the heterozygosity rate is similar to the new site filters (0.028%), but the higher accessibility in the updated filter set (69.97% vs 62.46%) yields improved sensitivity.</p>
<p>As genomic features vary between species, different sets of site filters were generated to allow high quality analyses both within and between species.</p>
<p>The <code>gamb_colu</code> site filters were generated as above, and are appropriate for analyses that include <em>gambiae</em> and <em>coluzzii</em> samples only.</p>
<p>The <code>arab</code> site filters were generated following application of the model to the summary statistics from arabiensis samples in the cohort (n=XXX), this set of site filters are appropriate when working with <em>A. arabiensis</em> samples only.</p>
<p>Finally, the <code>gamb_colu_arab</code> site filters allow analyses across all three species and are the intersection of the <code>gamb_colu</code> and the <code>arab</code> site filters.</p>
<p>Place holders for tables.</p>
<ul>
<li><p>Table A: Mendel errors per cross per autosome.
row indices: chromosome and raw/filtered
column indexes: crosses + frac accessible.
ie 8 rows, and 6 columns.</p></li>
<li><p>Table B: comparison of cross and X
row indices: raw/filtered
column indices: MER, frac accessible, Youden, each for 2 and 3.
column indexes: crosses + frac accessible.
ie 2 rows, and 6 columns.</p></li>
</ul>
<div style="font-size: 8pt">
<pre class="table"><code>---
caption: &#39;Result of heterozygote calls on male X chromosome&#39;
alignment: LLLLLLLL
include: content/tables/mer_X.csv
csv-kwargs:
  dialect: unix
width: [0.2, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1]
---</code></pre>
</div>
<h3 id="genome-accessibility">Genome accessibility</h3>
<p>We define accessibility as the fraction of sites in a region passing the appropriate set of site filters.</p>
<p>Overall, 70% of the genome, and ??% of the exome are considered accessible in the <code>gamb_colu</code> set.</p>
<p>This is an improvement from phase 2, where XXX of the genome, and YYY of the exome was considered accessible.</p>
<p>As expected, accessiblity was generally lower around the centromeres, and in regions of heterochromatin (table ref).</p>
<p>One notable region of low accessibility spans 40-41Mbp of chromosome 3R, this corresponds to ??.</p>
<p>Accessibility of the <code>arab</code> site filters closely follows that of <code>gamb_colu</code>, with the exception of the X chromosome where we see substantially lower values.</p>
<p>This appears to be driven by high divergence between AgamP4 and our <em>A. arabiensis</em> samples, particularly around the Xag inversion at Q-Q Mbp, ref figure.</p>
<p>On the autosomes the divergence from the reference is comparable between <em>A.arabiensis</em> and <em>A. gambiae</em>/ <em>A. coluzzii</em> samples, suggesting a strong basis for comparison accross species.</p>
<p>The median divergence (Dxy) of 100kbp windows is XXX (5%/95% TTT/SSS) for gambiae/coluzzii and YYY (TTT/YYY) for arabiensis.</p>
<p>On the X chromosome these values are XXX ( / ) for gambiae/coluzzii and YYY ( / ) for arabiensis.</p>
<h3 id="snp-discovery">SNP discovery</h3>
<p>Overall, we report XX,XXX,SSS single nucleotide polymophisms (SNPs) segregating in this cohort that pass filters, of which XX,XXX (%) are multiallelic.</p>
<p>12,223 SNPs are segregating in both species groups, while XXX are private to gambiae_/coluzzii_ and YYY to arabiensis [fig ref].</p>
<p>This phase of the study reports an additional XXX SNPs from phase 2.</p>
<p>In XXX gambiae and coluzzii individuals we report 12,222,222 SNPs (Q% multiallelic), corresponding to a SNP every 1.6 accessible bases.</p>
<p>In XXX arabiensis individuals we identify 10,000,000 SNPs (Q% multiallelic), a SNP every 2.5 accessible bases.</p>
<h2 id="population-structure">Population Structure</h2>
<h2 id="genetic-diversity-within-populations">Genetic Diversity within Populations</h2>
<h2 id="insecticide-resistance">Insecticide Resistance</h2>
<h2 id="gene-drive">Gene Drive</h2>
<h1 class="page_break_before" id="methods">Methods</h1>
<h2 id="population-sampling-1">Population Sampling</h2>
<p>Mosquitoes, from natural populations, were collected at 124 sites (unique latitude/longitudes) in 19 sub-Saharan African countries (Figure 1; Supplemental Table @@??).</p>
<p>95 of these sites are novel to Ag1000G phase 3, including 19 sites in six newly sampled countries, the remainder were previously sampled in phases 1 and 2 of the project (<span class="citation" data-cites="doi.org/10.1038/nature24995">[<span class="citeproc-not-found" data-reference-id="doi.org/10.1038/nature24995"><strong>???</strong></span>]</span>; @<span class="citation" data-cites="phase">[<span class="citeproc-not-found" data-reference-id="phase"><strong>???</strong></span>]</span> 2 doi when published).</p>
<p>New samples present in phase 3 comprised the following:</p>
<p><strong>Burkina Faso</strong></p>
<p>Two new submissions of collections from Burkina Faso are included here. The first added collections made in three villages separated by at most 30km: Bana (11.233, -4.472), Souroukoudinga (11.235, -4.535) and Pala (11.150, -4.235).</p>
<p>These collections were made in July and October 2014, and January, February and April 2015. The area is agricultural, with rice-growing areas near Bana and Souroukoudinga, and a large mango grove near Pala.</p>
<p>Female mosquitoes were collected by human landing catch, pyrethrum spray collection or aspiration; males were collected by swarm netting. Both <em>An. gambiae</em> and <em>An. coluzzii</em> (<span class="citation" data-cites="70em6Bgx"><a href="#ref-70em6Bgx" role="doc-biblioref">1</a></span>) were collected.</p>
<p>Specimens were stored in 80% ethanol and DNA was extracted using the DNeasy Tissue Kit (Qiagen).</p>
<p>The second new submission from Burkina Faso added collections of indoor resting adults made by spray catch from Monomtenga in central Burkina Faso (12.06, -1.17).</p>
<p>These specimens were sorted morphologically to <em>An. gambiae</em> s.l.</p>
<p>Ovaries of half-gravid females were dissected and placed in numbered individual micro-tubes containing modified Carnoy’s solution (1:3 glacial acetic acid: 100% ethanol).</p>
<p>Carcasses were placed in correspondingly numbered micro-tubes over desiccant. Genomic DNA was isolated from individual mosquitoes using one of the following: DNeasy Extraction Kit (Qiagen, Valencia, CA), Puregene kit (Gentra Systems, Inc., Minneapolis, MN), DNAzol kit (Molecular Research Center, Inc., Cincinnati, OH.) or Easy-DNA kit (Invitrogen, Carlsbad, CA).</p>
<p><em>An. gambiae</em> s.s. and its molecular forms were identified using one of two rDNA-based PCR/RFLP assays (<span class="citation" data-cites="1G04AzsFo"><a href="#ref-1G04AzsFo" role="doc-biblioref">2</a></span>; <span class="citation" data-cites="Z00Fb5xf"><a href="#ref-Z00Fb5xf" role="doc-biblioref">3</a></span>).</p>
<p>Ovaries from specimens of the desired species were subject to polytene chromosome analysis.</p>
<p><strong>Democratic Rebublic of the Congo</strong></p>
<p>Samples were collected from Gbadolite (4.283, 21.017), a town located in the far north of the Democratic Republic of Congo (DRC) near the border with the Central African Republic, surrounded by forest.</p>
<p>In common with much of DRC, malaria transmission rates are high, and the samples are <em>An. gambiae</em> s.s., which is the dominant vector.</p>
<p>Samples were collected as larvae from temporary pools within and around the town by dipping in early August 2015. All larvae were reared to adults and females preserved over silica for DNA extraction using Qiagen DNAEasy kits.</p>
<p><strong>Central African Republic</strong></p>
<p>Collections were carried out in Bangui (4.367, 18.583), during December 1993, by indoor resting aspiration or pyrethrum spray catch.</p>
<p><strong>Cameroon</strong></p>
<p>Two new submissions of samples from Cameroon are included in this phase.</p>
<p>In the first submission anopheline mosquitoes were taken from 64 locations covering a 1,500 km north-to-south transect that crossed all eco-geographical areas of Cameroon (<span class="citation" data-cites="daG5Pj6c"><a href="#ref-daG5Pj6c" role="doc-biblioref">4</a></span>).</p>
<p>Mosquito collection involved spraying aerosols of pyrethroid insecticides inside human dwellings, dead mosquitoes were retrieved from white sheets that were laid on the floor.</p>
<p>Anopheline mosquitoes were identified using morphological identification keys (&gt;&gt;Gillies and De Meillon 1968&lt;&lt;@<span class="citation" data-cites="how">[<span class="citeproc-not-found" data-reference-id="how"><strong>???</strong></span>]</span> reference books with no ISBN?; <span class="citation" data-cites="14XMzb1n7"><a href="#ref-14XMzb1n7" role="doc-biblioref">5</a></span>).</p>
<p>Ovaries from half-gravid <em>An. gambiae</em> s.l. females were dissected and stored in Carnoy’s fixative solution (absolute ethanol:glacial acetic acid 3:1) for cytogenetic analyses.</p>
<p>Carcasses were stored individually in tubes containing a desiccant and kept at -20°C until they were molecularly processed. All half-gravid specimens collected in each village were identified to species and molecular forms using PCR-RFLP (<span class="citation" data-cites="1G04AzsFo"><a href="#ref-1G04AzsFo" role="doc-biblioref">2</a></span>).</p>
<p>The second Cameroonian submission came from pyrethrum spray collections, larval sampling and human landing catches conducted in twenty locations during October 2013.</p>
<p>These villages are scattered throughout the country and reflect a gradient of human-dominated environments, for example, forest (Manda: 5.726, 10.868 and Campo: 2.367, 9.817); forest/savanna transition (Tibati: 6.469, 12.629); savanna (Lagdo: 9.049, 13.656); suburban area (Nkolondom: 3.972, 11.516) and
urban areas (Douala: 4.055, 9.721 and Yaoundé: 3.880, 11.506).</p>
<p>Contributed specimens were <em>An. gambiae</em> and <em>An. coluzzii</em> (<span class="citation" data-cites="1G04AzsFo"><a href="#ref-1G04AzsFo" role="doc-biblioref">2</a></span>).</p>
<p>Population genomics studies indicated the presence of relatively differentiated subgroups within both species as well as clusters thriving in polluted breeding sites in large cities (<span class="citation" data-cites="doi:10.1093/molbev/msx0877">[<span class="citeproc-not-found" data-reference-id="doi:10.1093/molbev/msx0877"><strong>???</strong></span>]</span>).</p>
<p>Specimens were stored on silica gel, and DNA was extracted using a Zymo research kit for adults and a Qiagen kit for larvae.</p>
<p><strong>Mayotte</strong></p>
<p>Phase 3 adds collections from three novel sites on the island of Mayotte.</p>
<p>Samples were collected as larvae during March-April 2011 in temporary pools by dipping in Tsounzou (-12.797, 45.185), Tsinkoura (-12.936, 45.138) and aerogare (-12.803, 45.283).</p>
<p>Larvae were stored in 80% ethanol prior to DNA extraction.</p>
<p>All specimens contributed were <em>An. gambiae</em> (<span class="citation" data-cites="Z00Fb5xf"><a href="#ref-Z00Fb5xf" role="doc-biblioref">3</a></span>) with the standard 2L<sup>+a</sup>/2L<sup>+a</sup> or inverted 2L<sup>a</sup>/2L<sup>a</sup> karyotype as determined by the molecular PCR diagnostics (<span class="citation" data-cites="ISDKf1P8"><a href="#ref-ISDKf1P8" role="doc-biblioref">6</a></span>).</p>
<p>The samples were identified as males or females by the sequencing read coverage of the X chromosome using LookSeq (<span class="citation" data-cites="19Z5CFAYk"><a href="#ref-19Z5CFAYk" role="doc-biblioref">7</a></span>).</p>
<p><strong>Gabon</strong> @<span class="citation" data-cites="was">[<span class="citeproc-not-found" data-reference-id="was"><strong>???</strong></span>]</span> this submission “GA-B” included in the release?? I can’t see it in the meta data, delete?</p>
<p><em>Anopheles</em> sampling was carried out in three new locations in Gabon @<span class="citation" data-cites="year">[<span class="citeproc-not-found" data-reference-id="year"><strong>???</strong></span>]</span>?.</p>
<p>Adult <em>An. gambiae</em> females were collected in Benguia (-1.633, 13.492) by human landing catches (National Research Ethics Committee of Gabon n°. 0031/2014/SG/CNE) in September 2015.</p>
<p>Benguia is a forest-savanna village with less than 300 inhabitants. Malaria is endemic across the year.</p>
<p>The specimens were stored immediately at -80°C.</p>
<p><em>An. coluzzii</em> larvae were sampled in Libreville (0.390, 9.454) and Cocobeach (0.992, 9.576) by dipping in natural breeding sites in January 2016.</p>
<p>Libreville is the capital of the country and it is urban and polluted site, Cocobeach is a coastal village north of Libreville.</p>
<p>Across both malaria is endemic across the year.</p>
<p>The specimens were stored in alcohol at -20°C.</p>
<p>Total genomic DNA was extracted for all the mosquitoes using the CTAB protocol (<span class="citation" data-cites="XUSF8pjQ"><a href="#ref-XUSF8pjQ" role="doc-biblioref">8</a></span>).</p>
<p><strong>The Gambia</strong></p>
<p>Two new submissions of samples from The Gambia are present in phase 3.</p>
<p>The first were collected along the Gambia River from the western coastal region of The Gambia, (Low River Area; Caputo et al. 2008), in August 2006.</p>
<p><em>An. gambiae</em> and <em>An. coluzzii</em> specimens were identified to species following the PCR-RFLP protocol (<span class="citation" data-cites="1G04AzsFo"><a href="#ref-1G04AzsFo" role="doc-biblioref">2</a></span>) using DNA extracted from the mosquito leg.</p>
<p>Only <em>An. coluzzii</em> specimens were collected from villages of Tankular (13.417, -16.033) and Kalataba (13.550, -15.617).</p>
<p><em>An. gambiae</em> and <em>An. coluzzii</em> specimens were found in sympatry and collected from villages of Yallal Tankonjala (13.550, -15.700), Sare Samba Sowe (13.583, -15.900) and Hamdalai (13.567, -16.0167).</p>
<p>PCR-RFLP protocol also revealed the presence of mosquitoes with hybrid <em>An. gambiae</em>/<em>An. coluzzii</em> genotype in Yallal Tankonjala and Sare Samba Sowe.</p>
<p>Collections of indoor daytime-resting half gravid mosquitoes were carried out mainly in human dwellings and, in few cases, in animal shelters.</p>
<p>Collections were carried out by pyrethroid and/or paper-cup mouth aspirators from 12 AM to sunset, and kept in vials with desiccant.</p>
<p>Ovaries were dissected, maintained into Carnoy fixative (three parts pure ethanol:one part glacial acetic acid) and stored at -20<sup>o</sup>C before polytene chromosome preparations (<span class="citation" data-cites="1EjgFGIf9"><a href="#ref-1EjgFGIf9" role="doc-biblioref">9</a></span>).</p>
<p>Chromosome scoring was carried out under a phase‐contrast optical microscope.</p>
<p>Paracentric inversion karyotypes were scored according to the nomenclature and conventions of Coluzzi et al. (<span class="citation" data-cites="doi:10.1016/0035-9203">[<span class="citeproc-not-found" data-reference-id="doi:10.1016/0035-9203"><strong>???</strong></span>]</span>(79)90036-1) and Touré et al. (<span class="citation" data-cites="Z6MC0qXg"><a href="#ref-Z6MC0qXg" role="doc-biblioref">10</a></span>).</p>
<p>The second new submission consists of adult mosquitoes collected at Wali Kunda in the rural, central river region of The Gambia (13.567, -14.917).</p>
<p>The area is 180 km from the sea, on the south bank of the River Gambia, in flat Sudan savannah with a small fishing village (and a research field station) as well as rice fields and swamplands.</p>
<p>The dominant <em>Anopheles</em> vector species is <em>An. coluzzii</em> (<span class="citation" data-cites="Px2G7gld"><a href="#ref-Px2G7gld" role="doc-biblioref">11</a></span>).</p>
<p>Mosquitoes were captured using human landing collections both inside and outside huts for 19 days in October and November 2012. Mosquitoes were stored in RNAlater or dried over silica gel and stored at -20<sup>o</sup>C.</p>
<p><strong>Guinea and Mali</strong></p>
<p>A novel sample submission to phase 3 included collections from both Guinea and Mali.</p>
<p>Mosquitoes were collected from four different study sites at the border with Mali and in Guinea Conakry. Takan (11.47, -8.33) and Toumani Oulena (10.83, -7.81) are both small villages in the Yanfolila district of southern Mali and represent the Sudanian savannah ecological zone.</p>
<p>Takan is arid savannah, while Toumani Oulena is humid savannah.</p>
<p>In Guinea Conakry, we sampled in Koraboh, (9.28, -10.03) a small village in the Kissidougou district in the Faranah region representing a semi-forest site with intermediate ecology, a mix of savannah and forest, and in Koundara, (8.48, -9.53), a small village in the Macenta district in the Nzerekore region representing deep forest ecology.</p>
<p>All reported collections occurred in October and November in 2012.</p>
<p>At each site, mosquitoes were collected using three different methods: human-landing capture, indoor manual aspirator or pyrethroid spray catch, and larval capture - where the first and second instar larvae were raised to adult in a field insectary under standard insectary conditions prior to DNA isolation from the adults, and the third and fourth instar larvae were preserved directly for DNA isolation, without rearing in the insectary.</p>
<p>The two distinct methods of larval collection were used to control for possible genetic bias inherent in lab rearing of captured larvae.</p>
<p>Across sites, all types of larval sites were sampled, including both temporary and permanent sites.</p>
<p>Human-landing captures were performed both inside dwellings and outside (&gt;10 m from dwelling) at night between 18:00 and 06:30.</p>
<p>The indoor aspirator or spray catches were done in the morning between 06:00 and 12:00.</p>
<p>Adult specimens or third and fourth instar larvae were preserved immediately in 80% ethanol until later DNA extraction.</p>
<p>First and second instar larvae were raised to adults in nearby field insectaries and upon emergence were preserved in 80% ethanol.</p>
<p>DNA was extracted from mosquitoes using DNAzol by the provided protocol (Invitrogen, CA, USA).</p>
<p>Further details on sampling and basic Southern Mali/Guinea vector biology are presented in Coulibaly et al. (doi:10.1186/s12936-016-1242-5).</p>
<p><strong>Guinea Bissau</strong></p>
<p>Two new Guinea Bissau collection sites are added here, both performed in October 2010 by indoor CDC light traps.</p>
<p>Samples were also collected from Ga-Mbana (12.052, -14.902) and Leibala (12.272, -14.222).</p>
<p>Ga-Mbana is a rural village located along a main road in central Guinea Bissau, while Leibala is a neighbourhood of the eastern town of Gabu.</p>
<p>The samples of Ga-Mbana comprised <em>An. coluzzii</em> whereas the samples of Leibala comprised <em>An. gambiae</em>, all being identified by IGS and SINEX markers as described in Vicente et al. (<span class="citation" data-cites="YX2M3jrq"><a href="#ref-YX2M3jrq" role="doc-biblioref">12</a></span>).</p>
<p>The <em>kdr</em> pyrethroid target site resistance mutation L1014F occurs at high frequency in Leibala but at very low frequency in Ga-Mbana (<span class="citation" data-cites="YX2M3jrq"><a href="#ref-YX2M3jrq" role="doc-biblioref">12</a></span>).</p>
<p>Malaria is meso-hyperendemic (<span class="citation" data-cites="16as3jwaa"><a href="#ref-16as3jwaa" role="doc-biblioref">13</a></span>) and sporozoite rates are below 1% in the region.</p>
<p>Specimens were stored on silica gel and DNA extraction was performed by a phenol-chloroform protocol described in Donnelly et al. (<span class="citation" data-cites="DnnjILYG"><a href="#ref-DnnjILYG" role="doc-biblioref">14</a></span>).</p>
<p><strong>Kenya</strong></p>
<p>New Kenyan specimens were obtained from villages located in Kilifi County near the Kenyan coast between 2000 and 2014.</p>
<p>All Anopheles mosquito sampling was conducted indoors using CDC light traps.</p>
<p><em>An. gambiae</em>, <em>An. funestus</em>, <em>An. arabiensis</em> and <em>An. merus</em> were present at sampling locations (<span class="citation" data-cites="14yZJJBPD"><a href="#ref-14yZJJBPD" role="doc-biblioref">15</a></span>).</p>
<p>Sporozoite rates for the area during previous studies were 1.47% (<span class="citation" data-cites="1C7qEzk6S"><a href="#ref-1C7qEzk6S" role="doc-biblioref">16</a></span>).</p>
<p><strong>Mali</strong></p>
<p>Two further submissions containing collections from 11 sites in Mali are new for phase 3.</p>
<p>In the first, collections were made in four villages in the Koulikoro region; Tieneguebougou (12.810, -8.080) approximately 20 km north of Bamako,
and Kababougou (12.890, -8.150), Ouassorola (12.900, -8.160), Sogolombougou ( 12.880, -8.140), approximately 30 km north of Bamako.</p>
<p>The collections were made in August 2014 by human landing catch and pyrethrum spray catch.</p>
<p>Both <em>An. gambiae</em> and <em>An. coluzzii</em> (<span class="citation" data-cites="1G04AzsFo"><a href="#ref-1G04AzsFo" role="doc-biblioref">2</a></span>) were collected.</p>
<p>Specimens were stored in 80% ethanol.</p>
<p>In the second submission, collections of indoor resting adults were made by spray catch from seven villages in the southern part of Mali in August-September 2004: Banambani (12.800, -8.050), Bancoumana (12.200, -8.200), Douna (13.210, -5.900), Fanzana (13.200, -6.130), Kela (11.880, -8.450), Moribobougou (12.690, -7.870) and N’Gabakoro (12.680, -7.840).</p>
<p>Specimens were sorted morphologically to <em>An. gambiae</em> s.l.</p>
<p>Ovaries of half-gravid females were dissected and placed in numbered individual micro-tubes containing modified Carnoy’s solution (1:3 glacial acetic acid: 100% ethanol).</p>
<p>Carcasses were placed in correspondingly numbered micro-tubes over desiccant.</p>
<p>Genomic DNA was isolated from individual mosquitoes using one of the following: DNeasy Extraction Kit (Qiagen, Valencia, CA), Puregene kit (Gentra Systems, Inc., Minneapolis, MN), DNAzol kit (Molecular Research Center, Inc., Cincinnati, OH.) or Easy-DNA kit (Invitrogen, Carlsbad, CA).</p>
<p><em>An. gambiae</em> s.s. and its molecular forms were identified using one of two rDNA-based PCR/RFLP assays (<span class="citation" data-cites="1G04AzsFo"><a href="#ref-1G04AzsFo" role="doc-biblioref">2</a></span>; <span class="citation" data-cites="Z00Fb5xf"><a href="#ref-Z00Fb5xf" role="doc-biblioref">3</a></span>).</p>
<p>Ovaries from specimens of the desired species were subject to polytene chromosome analysis.</p>
<p><strong>Malawi</strong></p>
<p>Specimens were obtained from villages within the catchment of the Majete Malaria Project, Chikhwawa District, Malawi (-15.933, 34.755) (<span class="citation" data-cites="xsFRxYp6"><a href="#ref-xsFRxYp6" role="doc-biblioref">17</a></span>).</p>
<p>Mosquitoes were collected indoors and outdoors by Suna light trap in May 2015.</p>
<p>Chickhwawa District is an area with perennial and intense malaria transmission (<span class="citation" data-cites="vckO98IK"><a href="#ref-vckO98IK" role="doc-biblioref">18</a></span>).</p>
<p>All specimens were <em>An. arabiensis</em> (<span class="citation" data-cites="1G04AzsFo"><a href="#ref-1G04AzsFo" role="doc-biblioref">2</a></span>).</p>
<p>Additional details of vector population bionomics may be found in (<span class="citation" data-cites="xsFRxYp6"><a href="#ref-xsFRxYp6" role="doc-biblioref">17</a></span>; <span class="citation" data-cites="s77kHoJc"><a href="#ref-s77kHoJc" role="doc-biblioref">19</a></span>).</p>
<p>Specimens were stored over silica and DNA was extracted using the Qiagen plate protocol.</p>
<p><strong>Mozambique</strong></p>
<p>Mosquito samples were collected in Furvela (-23.716, 35.299), Mozambique, by CDC light traps between December 2003 and April 2004.</p>
<p>Specimens were stored on silica gel and DNA was extracted according to Collins et al. (<span class="citation" data-cites="pSmuJkYQ"><a href="#ref-pSmuJkYQ" role="doc-biblioref">20</a></span>).</p>
<p>Contributed specimens consisted of <em>An. gambiae</em> individuals identified according to Fanello et al. (<span class="citation" data-cites="1G04AzsFo"><a href="#ref-1G04AzsFo" role="doc-biblioref">2</a></span>).</p>
<p>Furvela is a rural village located in Inhambane Province, where malaria is transmitted mainly by <em>An. gambiae</em> and <em>An. funestus</em> (<span class="citation" data-cites="f8uRMyY6"><a href="#ref-f8uRMyY6" role="doc-biblioref">21</a></span>).</p>
<p><em>An. arabiensis</em> and <em>An. merus</em> are also found at low frequency.</p>
<p>Sporozoite rates around 4% have been reported in <em>An. gambiae</em> from Furvela (<span class="citation" data-cites="f8uRMyY6"><a href="#ref-f8uRMyY6" role="doc-biblioref">21</a></span>).</p>
<p><strong>Tanzania</strong></p>
<p>Tanzanian samples were collected from four distinct locations.</p>
<p>Moshi samples came from lower Mabogini (-3.400, 37.350), rice fields near lower Moshi on the southern slope of Mount Kilimanjaro, a region shown to have increasing resistance to pyrethroids (<span class="citation" data-cites="FVN9nrFP"><a href="#ref-FVN9nrFP" role="doc-biblioref">22</a></span>).</p>
<p>Mosquitoes were collected as larvae, during the rice growing season in August-September 2012, raised to adults and females bioassayed in WHO tubes for one hour with 0.05% lambda cyhalothrin (<span class="citation" data-cites="isbn:9789241511575">[<span class="citeproc-not-found" data-reference-id="isbn:9789241511575"><strong>???</strong></span>]</span>).</p>
<p>Alive and dead mosquitoes were preserved over silica.</p>
<p>In Tanzanian samples screened in Kabula et al. (<span class="citation" data-cites="ca2mjT0Y"><a href="#ref-ca2mjT0Y" role="doc-biblioref">23</a></span>), Moshi was the most pyrethroid resistant population, they were found to be completely DDT susceptible, only in one out of 642 mosquitoes assayed by Matowo et al. (<span class="citation" data-cites="FVN9nrFP"><a href="#ref-FVN9nrFP" role="doc-biblioref">22</a></span>) was found to carry a <em>kdr</em> resistance mutation (Vgsc-995F).</p>
<p>Tarime collections took place in the village of Komaswa (-1.417, 34.183) about 410 km north west of Moshi, during August 2012.</p>
<p>Mosquito larvae were collected, raised to adults and females bioassayed with a range of insecticides in WHO tubes for one hour (<span class="citation" data-cites="isbn:9789241511575">[<span class="citeproc-not-found" data-reference-id="isbn:9789241511575"><strong>???</strong></span>]</span>),
finding almost complete multi-insecticide susceptibility: permethrin (100% mortality), lambda cyhalothrin (97%), fenitrothion (100%), DDT (100%) and bendiocarb (100%) (Nyka, T. unpublished data – Insecticide Resistance Monitoring Report 2012. NIMR Tanzania).</p>
<p>Mulheza samples were collected from Zeneti village (-5.217, 38.650), northeast Tanzania.</p>
<p>Malaria is intense and perennial with transmission peaking after the rainy season in May and June (<span class="citation" data-cites="ca2mjT0Y"><a href="#ref-ca2mjT0Y" role="doc-biblioref">23</a></span>).
Mosquitoes were sampled between November 2012 and May 2013.</p>
<p>Indoor resting collections were used to obtain live females for deltamethrin susceptibility testing and pyrethrum spray catches were used for mosquitoes that were collected for blood meal analysis.</p>
<p>Collections were conducted between 06:00 and 09:00 from randomly selected houses.</p>
<p>Live mosquitoes collected for susceptibility testing were provided with 10% glucose solution and transported to the field insectary.</p>
<p>Mosquitoes were sorted and morphologically identified to species, carcasses were stored individually over desiccant for laboratory processing.
Muleba (1.750, 31.667), the final collection region, is in the North-western part of Tanzania.</p>
<p>The district is known to be a malaria epidemic prone area with unstable transmission of varying seasonality.</p>
<p>The highest peak of malaria transmission is usually reached between May-July and November-January, which results from proceeding rain seasons.</p>
<p>There have been malaria vector control efforts since 2007 when indoor residual spraying using Lambdacyhalothrin was introduced.</p>
<p>Insecticide resistance in this district is coupled with high frequency of <em>kdr</em> pyrethroid target site mutations in the <em>An. gambiae</em> s.s population (<span class="citation" data-cites="ca2mjT0Y"><a href="#ref-ca2mjT0Y" role="doc-biblioref">23</a></span>; <span class="citation" data-cites="dB1ebxOj"><a href="#ref-dB1ebxOj" role="doc-biblioref">24</a></span>).</p>
<p>Sampling was conducted over six months, which include both dry and rainy season and covers 6 villages selected to represent all major ecological systems in the district.</p>
<p><strong>Uganda</strong></p>
<p>In Uganda, a single new site is sampled in phase 3.</p>
<p>In Kihihi subcounty, Kanungu District (-0.751, 29.701), resting mosquitoes were collected during October and November 2012.</p>
<p>Kihihi is located in an upland area with seasonal malaria transmission (<span class="citation" data-cites="m3y4gwzb"><a href="#ref-m3y4gwzb" role="doc-biblioref">25</a></span>).</p>
<p>All specimens were <em>An. gambiae</em> (<span class="citation" data-cites="1G04AzsFo"><a href="#ref-1G04AzsFo" role="doc-biblioref">2</a></span>).</p>
<p>Additional details of vector population bionomics may be found in (<span class="citation" data-cites="doi:10.1186/1475-2875-13-1114">[<span class="citeproc-not-found" data-reference-id="doi:10.1186/1475-2875-13-1114"><strong>???</strong></span>]</span>).</p>
<p>Specimens were stored in 80% ethanol and DNA was extracted using the Qiagen plate protocol.</p>
<p>Details of natural population samples previously described in phases 1 and 2 can be found in the Supplementary Information of The Anopheles gambiae 1000 Genomes Consortium (<span class="citation" data-cites="doi.org/10.1038/nature24995">[<span class="citeproc-not-found" data-reference-id="doi.org/10.1038/nature24995"><strong>???</strong></span>]</span>; @<span class="citation" data-cites="phase">[<span class="citeproc-not-found" data-reference-id="phase"><strong>???</strong></span>]</span> 2 doi when published - respectively).</p>
<p><strong>Colony Crosses</strong></p>
<p>15@@? crosses were contributed to Ag1000G phase 1: @@…</p>
<p>The labels, e.g. “29-2”, are identifiers used for each of the crosses within the project and have no special meaning.</p>
<p>The crosses were generated using parents from seven different colonies: G3(MRA-112); Kisumu(MRA-762); Pimperena (canonical representative of <em>An. gambiae</em> species; MRA-861); Ghana (recent colony of <em>An. coluzzii</em> from Okyereko, southern Ghana (<span class="citation" data-cites="z9HqxYF3"><a href="#ref-z9HqxYF3" role="doc-biblioref">26</a></span>)); Mali-NIH (canonical representative of <em>An. coluzzii</em> species; Niono, MRA-860); (P)Akron (Benin, MRA-913); and Tiassalé (southern Cote d’Ivoire (<span class="citation" data-cites="z9HqxYF3"><a href="#ref-z9HqxYF3" role="doc-biblioref">26</a></span>)).</p>
<p>Full details of cross production, sequencing and quality control is described in the Supplementary Information of The Anopheles gambiae 1000 Genomes Consortium (<span class="citation" data-cites="doi.org/10.1038/nature24995">[<span class="citeproc-not-found" data-reference-id="doi.org/10.1038/nature24995"><strong>???</strong></span>]</span>).</p>
<h2 id="whole-genome-sequencing-and-alignment-1">Whole Genome Sequencing and Alignment</h2>
<p>4,693 individual mosquitoes were sequenced using the Illumina HiSeq2000 (n=3,130) and the Illumina HiSeqX (n=1,563) to a target coverage of 30X.</p>
<p>Reads were aligned to the AgamP4 reference genome using <code>bwa</code> version <code>0.7.15</code>.</p>
<p>Indel realignment was performed using GATK <code>v3.7-0</code> RealignerTargetCreator and IndelRealigner.</p>
<p>Single nucleotide polymophisms were called against AgamP4 using GATK UnifiedGenotyper <code>v3.7-0</code>.</p>
<p>Sample genotypes were called independently, in genotyping mode, given all possible alleles at each site, allowing parallelisation over samples.</p>
<p>Coverage considered at individual sites was capped at 250X by random downsampling.</p>
<p>Full details of pipelines including all parameter settings are available in supplementary information.</p>
<p>Following successfull completion of the pipeline samples entered the sample quality control (QC) process.</p>
<h1 id="sample-qc-1">Sample QC</h1>
<p>The sample QC process was composed of three distinct stages: sequence quality assurance, replicate handling, and anomaly detection.</p>
<p>To meet the requirements of sequence quality assurance median coverage had to be at least 10X, and minumum 50% of the genome covered by at least 1X.</p>
<p>We also implemented the test for contamination in NGS alignments described in Jun et al (https://doi.org/10.1016/j.ajhg.2012.09.004).</p>
<p>Briefly the method estimates the likelihood of the observed alternate and reference allele counts under different contamination fractions given population allele frequencies.</p>
<p>Population allele frequencies were estimated from the Ag1000G phase 2 data.</p>
<p>The model computes a maximum likelihood value for a parameter representing percentage contamination (alpha).</p>
<p>Where this parameter was 4.5% or greater the sample was excluded.</p>
<p>We also made sex calls based on the modal coverage ratio X:3R.</p>
<p>3R was selected as representative of autosomal coverage as it is free from inversions and large regions of heterochromatin.</p>
<p>The modal coverage was used owing to concerns around the high skewdness of coverage distributions.</p>
<p>To mitigate this further, when computing the modal coverage we only considered sites where coverage was at least 2X.</p>
<p>The sample was classed as male where the coverage ratio was between 0.4-0.6, and female between 0.8-1.2.</p>
<p>Where the ratio was outside these limits, the sex call was not made and the sample dropped.</p>
<p>One of the submission sets from The Gambia, was composed entirely of Whole Genome Amplified (WGA) samples.</p>
<p>These received a sex call where possible but the decision was made not to exclude based on ratio, due to the inherent value of this submission.</p>
<p>The fact this set is entirely WGA was considered throughout the analysis of these samples.</p>
<p>The sample QC process also included assessment of technical replicates.</p>
<p>We computed pairwise genetic distance between all sample pairs within a submission set.</p>
<p>The metric used was Hamming between alleles, so a genotype of 0/1 records a value of 1 against the genotype 1/2, allowing straightforward handling of multiallelic SNPs.</p>
<p>Only sites where both samples had a genotype call were included, mean distance was calculated over a denominator of the number of assessed sites multiplied by 2 for diploidy.</p>
<p>Computations were initially carried out on a downsampled set of 10 x 100,000 contiguous sites genome wide to be computationally feasible.</p>
<p>This use of chunks was convenient to leverage the underlying storage of data in <code>zarr</code> format.</p>
<p>Where a pair of samples fell beneath a conservative threshold of 0.012, the true genetic distance (i.e. without downsampling) was computed across all sites.</p>
<p>For each pair of technical replicates, we excluded both members of the pair where genetic distance was above 0.006.</p>
<p>Where replicate pairs met the concordance threshold we excluded the lower quality sample.</p>
<p>Quality was determined based on the skewdness of mean to median, i.e. <span class="math inline">\(1 - |mean_{cov}/median_{cov}|\)</span>.</p>
<p>The sample with the lower value was preferred as it suggests a more normal coverage distribution.</p>
<p>To identify unknown replicate pairs as a result of sample mix ups or mislabelling, we screened within submission sets for unexpected pairs, using the genetic distance cut off of 0.006 as above.</p>
<p>We did not attempt to identify unknown replicate pairs in the AG1000G-X submission set, made up of laboratory experimental crosses, due to familial similarity and high levels of inbreeding.</p>
<p>The third stage used principal component analysis (PCA) to identify and exclude individual samples that were outliers based on available metadata.</p>
<p>A review process identified samples that could not be explained parsimoniously, and were therefore likely to be sample mix ups or instances of mislabelling.</p>
<p>Using the PCA implementation in <code>scikit-allel v2.1.0</code> we downsampled to 100,000 segregating non-singleton sites from chromosomes 3R and 3L, to avoid regions complicated by known introgression loci or paracentric inversions.</p>
<p>Multiallelic sites were included as dummy rows by melting the data structure.</p>
<p>A careful review process identified: a) samples that dominated single principal components, either individually or in very small numbers.</p>
<p>This suggests an individual belonging to another Anopheline species, or some inherent problem with the sample.</p>
<p>Or b) samples that clustered with other samples inconsistently with metadata.</p>
<p>This was a subjective assessment, but bore in mind the given collection location, time, and PCR species assignment (where available) of the sample.</p>
<p>Multiple geographical sites submitted by the same partner were also considered, where sample mix ups formed the most parsimonious explanations of incongruent clustering.</p>
<h3 id="species-assignment-and-sex-calling-1">Species assignment and sex calling</h3>
<p>Ancestry informative markers were used to assign species in our cohort.</p>
<p>To derive markers informative between <em>A. arabiensis</em> and <em>A. gambiae s.l.</em>, we used publicly available data from the 16 genomes project (ref).</p>
<p>Whole genome SNP calls called against the AgamP3 reference for 12 <em>A. arabiensis</em> and 38 <em>A. gambiae s.l</em> individuals were available.</p>
<p>Alleles were mapped onto the same alternate allele space, the frequencies of which were computed across both groups.</p>
<p>Sites that were multiallelic in either group were excluded, as well as sites where any genotypes were missing.</p>
<p>565,329 SNPs were identified as potentially informative where no shared alleles were present between groups.</p>
<p>These were spread throughout the genome, but were concentrated on the X chromosome (63.2%), particularly around the Xag inversion.</p>
<p>The full AIM set of positions and alleles are available as part of the phase 3 data release.</p>
<p>Called genotypes for each individual in the dataset were cross referenced against a random subset of 50,000 ancestry informative marker alleles, genotype alleles were accordingly classified as gambiae-like or arabiensis-like.</p>
<p>AIM fractions were cross referenced against PCR results available from a subset of inviduals.</p>
<p>Given the relatively small number of <em>A. arabiensis</em> samples in the 16 genomes project- it was clear that a significant proportion of putative AIMs were not likely to be truly informative.</p>
<p>Therefore, classification requirements are less rigourous than other sets of validated markers; individuals were classed as <em>A. arabiensis</em> where a fraction &gt;0.6 of alleles were arabiensis-like (n=368), and as <em>A. gambiae s.l</em> where this value was &lt; 0.03 (n=2415).</p>
<p>Species were not assigned to samples from the AG1000G-X submission due to inbreeding and high levels of genetic drift.</p>
<p>A single individual collected in Tororo, Uganda is classed as intermediate- given the majority (XX%) of sites in the genome are heterozygous between the gambiae and arabiensis allele, this individual is likely to be an F1 hybrid.</p>
<p>To resove the <em>A. gambiae s.l</em> individuals into <em>A. gambiae</em> and <em>A. coluzzii</em> we applied the 729 AIMs previously identified by Neafsey et al (ref).</p>
<p>?? Need to find code that creates gambiae from neafsey set</p>
<p>Cutoffs were made at &lt;0.12 (gambiae) and &gt;0.9 (coluzzii), with individuals between classed as intermediate.</p>
<p>Of the 2415 <em>s.l</em> individuals, 1571 were called as gambiae, 675 as coluzzii and 169 as intermediate (ref collection map).</p>
<h3 id="snp-filtering-and-quality-1">SNP filtering and quality</h3>
<p>Site filtering ensures that reported variation is of high quality.</p>
<p>As genomic features vary between species, different sets of site filters were generated to allow high quality analyses both within and between species.</p>
<p>The <code>gamb_colu</code> site filters were generated using a decision tree model, and are appropriate for analyses that include <em>gambiae</em> and <em>coluzzii</em> samples only.</p>
<p>Inputs to the decision tree model are summary statistics from the set of SNP calls and genotype alignments.</p>
<p>The <code>arab</code> site filters were generated following application of the resulting model to the summary statistics from arabiensis samples in the cohort, this set of site filters are appropriate when working with <em>A. arabiensis</em> samples only.</p>
<p>Finally, the <code>gamb_colu_arab</code> site filters allow analyses across all three species and are the intersection of the <code>gamb_colu</code> and the <code>arab</code> site filters.</p>
<p>Using the 15 available Anopheles pedigrees previously described, we used the presence of mendelian error at sites as a proxy for genotype discordance.</p>
<p>Where previously we have used manually curated cutoffs based on observed mendelian error rates to filter sites (ref phase1, phase2), here we built a statistical model where cohort level summary statistics were used to predict the presence or absence of mendelian error, becoming a binary classification problem.</p>
<p>Summary statistics used as input to the model are presented in table (ref).</p>
<p>Pedigrees included <em>A. gambiae</em> and <em>A. coluzzii</em> mosquitoes only, and summary statistics to build the initial site filters model came from females of these species (n=2415).</p>
<p>Males were excluded, so that the model could also be applied without modification to the X chromosome.</p>
<p>5 of the 15 crosses were held out for validation, so performance could be evaluated objectively.</p>
<p>Sites were defined as PASS where all genotypes across all 10 remaining crosses were called, and no mendelian inconsistencies were observed.</p>
<p>Sites were defined as FAIL where a mendelian inconsistency was observed in any pedigree.</p>
<p>All other sites were not considered eligible for inclusion in model training.</p>
<p>A balanced training set was generated from the remaining 10 crosses containing XXX autosomal(?) sites.</p>
<p>We applied a decision tree, as it provides decisions with clear explanations, and is similar in concept to the set of hard thresholds commonly used in SNP calling in non-model organisms.</p>
<p>A set of trees with different parameter settings were learned, exploring the depth of trees and the number of samples allowed at a terminal node.</p>
<p>Parameter settings were evaluated on an unbalanced evaluation set, consisting of XXX sites randomly from sampled from the whole genome.</p>
<p>Leaves of the trained models contain different proportions of PASS sites, by increasing the cutoff for these proportions required to label a leaf as PASS, we were able to compute the area under the receiver operating curve (AUROC) for each parameter set.</p>
<p>The best performing parameter set based on AUROC was selected as the final model, the classification cutoff used was optimised based on the Youden statistic.</p>
<p>The resulting model was a decision tree of depth 8, with a maximum of XX (CHECK NUMBERS) terminal nodes, where leaves were assigned to PASS where &gt; 0.533 of training data in that leaf were PASS.</p>
<p>All sites in the genome were then assigned to PASS or FAIL given the model inputs.</p>
<p>The 5 remaining cross pedigrees were used to perform a final evaluation of the approach.</p>
<p>Above definitions of PASS sites were retained, but independently within pedigrees, providing 5 distinct evaluation sets.</p>
<p>To evaluate performance on the hemizygous X chromosome we use the more direct measure of heterozygote calls in males.</p>
<p>In the dataset are 220 <em>A. gambiae s.l.</em> male samples, each of which represent an independent proxy for genotype discordance.</p>
<p>Male error rates were estimated from genotype calls with a minimum Genotype Quality (GQ) value of 30.</p>
<p>A non heterozygote call is labelled PASS, and a heterozygote call FAIL.</p>
<p>Error rates were computed for all crosses, over all chromosomes before and after application of site filters.</p>
<p>We report the false positive rate (FPR), (i.e proportion of the genome considered accessible), and the Youden statistic (ratio of sensitivity to specificity).</p>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references hanging-indent" role="doc-bibliography">
<div id="ref-70em6Bgx">
<p>1.<strong>:{unav)</strong><br />
Elien E Wilkins, Paul I Howell, Mark Q Benedict<br />
<em>Malaria Journal</em> (2006) <a href="https://doi.org/dbtfkr">https://doi.org/dbtfkr</a><br />
DOI: <a href="https://doi.org/10.1186/1475-2875-5-125">10.1186/1475-2875-5-125</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/17177993">17177993</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1769388">PMC1769388</a></p>
</div>
<div id="ref-1G04AzsFo">
<p>2. <strong>Simultaneous identification of species and molecular forms of the Anopheles gambiae complex by PCR-RFLP</strong><br />
C. Fanello, F. Santolamazza, A. Della Torre<br />
<em>Medical and Veterinary Entomology</em> (2002-12) <a href="https://doi.org/ds5pmz">https://doi.org/ds5pmz</a><br />
DOI: <a href="https://doi.org/10.1046/j.1365-2915.2002.00393.x">10.1046/j.1365-2915.2002.00393.x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/12510902">12510902</a></p>
</div>
<div id="ref-Z00Fb5xf">
<p>3. <strong>SHORT REPORT: A NEW POLYMERASE CHAIN REACTION-RESTRICTION FRAGMENT LENGTH POLYMORPHISM METHOD TO IDENTIFY ANOPHELES ARABIENSIS FROM AN. GAMBIAE AND ITS TWO MOLECULAR FORMS FROM DEGRADED DNA TEMPLATES OR MUSEUM SAMPLES</strong><br />
ALESSANDRA DELLA TORRE, ADALGISA CACCONE, FEDERICA SANTOLAMAZZA<br />
<em>The American Journal of Tropical Medicine and Hygiene</em> (2004-06-01) <a href="https://doi.org/ghcqgn">https://doi.org/ghcqgn</a><br />
DOI: <a href="https://doi.org/10.4269/ajtmh.2004.70.604">10.4269/ajtmh.2004.70.604</a></p>
</div>
<div id="ref-daG5Pj6c">
<p>4. <strong>Ecological niche partitioning between Anopheles gambiae molecular forms in Cameroon: the ecological side of speciation</strong><br />
Frédéric Simard, Diego Ayala, Guy Kamdem, Marco Pombi, Joachim Etouna, Kenji Ose, Jean-Marie Fotsing, Didier Fontenille, Nora J Besansky, Carlo Costantini<br />
<em>BMC Ecology</em> (2009) <a href="https://doi.org/bd8bz5">https://doi.org/bd8bz5</a><br />
DOI: <a href="https://doi.org/10.1186/1472-6785-9-17">10.1186/1472-6785-9-17</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19460146">19460146</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2698860">PMC2698860</a></p>
</div>
<div id="ref-14XMzb1n7">
<p>5. <strong>The Anophelinae of Africa south of the Sahara. Suppl: Afrotropical region</strong><br />
M. T. Gillies, Botha de Meillon<br />
<em>Publications of the South African Institute for Medical Research</em> (1987)<br />
ISBN: <a href="https://worldcat.org/isbn/9780620103213">9780620103213</a></p>
</div>
<div id="ref-ISDKf1P8">
<p>6. <strong>MOLECULAR KARYOTYPING OF THE 2LA INVERSION IN ANOPHELES GAMBIAE</strong><br />
NORA J. BESANSKY, OLGA GRUSHKO, ALESSANDRA DELLA TORRE, IGOR SHARAKHOV, CECILE BRENGUES, JONATHAN K. KAYONDO, FEDERICA SANTOLAMAZZA, FREDERIC SIMARD, MARCO POMBI, MAMADOU COULIBALY, … KARINE MOULINE<br />
<em>The American Journal of Tropical Medicine and Hygiene</em> (2007-02-01) <a href="https://doi.org/ghcqgq">https://doi.org/ghcqgq</a><br />
DOI: <a href="https://doi.org/10.4269/ajtmh.2007.76.334">10.4269/ajtmh.2007.76.334</a></p>
</div>
<div id="ref-19Z5CFAYk">
<p>7. <strong>LookSeq: A browser-based viewer for deep sequencing data</strong><br />
H. M. Manske, D. P. Kwiatkowski<br />
<em>Genome Research</em> (2009-08-13) <a href="https://doi.org/b5hmbh">https://doi.org/b5hmbh</a><br />
DOI: <a href="https://doi.org/10.1101/gr.093443.109">10.1101/gr.093443.109</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19679872">19679872</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2775587">PMC2775587</a></p>
</div>
<div id="ref-XUSF8pjQ">
<p>8. <strong>POPULATION STRUCTURE OF ANOPHELES ARABIENSIS ON LA RÉUNION ISLAND, INDIAN OCEAN</strong><br />
FRÉDÉRIC SIMARD, RICHARD HUNT, DIDIER FONTENILLE, ISABELLE MORLAIS, ROMAIN GIROD<br />
<em>The American Journal of Tropical Medicine and Hygiene</em> (2005-12-01) <a href="https://doi.org/ghcqgp">https://doi.org/ghcqgp</a><br />
DOI: <a href="https://doi.org/10.4269/ajtmh.2005.73.1077">10.4269/ajtmh.2005.73.1077</a></p>
</div>
<div id="ref-1EjgFGIf9">
<p>9. <strong>Anopheles gambiae complex along The Gambia river, with particular reference to the molecular forms of An. gambiae s.s</strong><br />
Beniamino Caputo, Davis Nwakanma, Musa Jawara, Majidah Adiamoh, Ibrahima Dia, Lassana Konate, Vincenzo Petrarca, David J Conway, Alessandra della Torre<br />
<em>Malaria Journal</em> (2008) <a href="https://doi.org/b97c6c">https://doi.org/b97c6c</a><br />
DOI: <a href="https://doi.org/10.1186/1475-2875-7-182">10.1186/1475-2875-7-182</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/18803885">18803885</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2569043">PMC2569043</a></p>
</div>
<div id="ref-Z6MC0qXg">
<p>10. <strong>The distribution and inversion polymorphism of chromosomally recognized taxa of the Anopheles gambiae complex in Mali, West Africa.</strong><br />
YT Touré, V Petrarca, SF Traoré, A Coulibaly, HM Maiga, O Sankaré, M Sow, MA Di Deco, M Coluzzi<br />
<em>Parassitologia</em> (1998-12) <a href="https://www.ncbi.nlm.nih.gov/pubmed/10645562">https://www.ncbi.nlm.nih.gov/pubmed/10645562</a><br />
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/10645562">10645562</a></p>
</div>
<div id="ref-Px2G7gld">
<p>11. <strong>Does insecticide resistance contribute to heterogeneities in malaria transmission in The Gambia?</strong><br />
Kevin Ochieng’ Opondo, David Weetman, Musa Jawara, Mathurin Diatta, Amfaal Fofana, Florence Crombe, Julia Mwesigwa, Umberto D’Alessandro, Martin James Donnelly<br />
<em>Malaria Journal</em> (2016-03-15) <a href="https://doi.org/ghcqgh">https://doi.org/ghcqgh</a><br />
DOI: <a href="https://doi.org/10.1186/s12936-016-1203-z">10.1186/s12936-016-1203-z</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26980461">26980461</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4793517">PMC4793517</a></p>
</div>
<div id="ref-YX2M3jrq">
<p>12. <strong>Massive introgression drives species radiation at the range limit of Anopheles gambiae</strong><br />
José L. Vicente, Christopher S. Clarkson, Beniamino Caputo, Bruno Gomes, Marco Pombi, Carla A. Sousa, Tiago Antao, João Dinis, Giordano Bottà, Emiliano Mancini, … João Pinto<br />
<em>Scientific Reports</em> (2017-04-18) <a href="https://doi.org/f93m36">https://doi.org/f93m36</a><br />
DOI: <a href="https://doi.org/10.1038/srep46451">10.1038/srep46451</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28417969">28417969</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5394460">PMC5394460</a></p>
</div>
<div id="ref-16as3jwaa">
<p>13. <strong>TRANSMISSION OF MIXED PLASMODIUM SPECIES AND PLASMODIUM FALCIPARUM GENOTYPES</strong><br />
VIRGÍLIO E. DO ROSÁRIO, KATINKA PÅLSSON, THOMAS G. T. JAENSON, GEORGES SNOUNOU, JOÃO PINTO, ANA PAULA AREZ<br />
<em>The American Journal of Tropical Medicine and Hygiene</em> (2003-02-01) <a href="https://doi.org/ghcqgm">https://doi.org/ghcqgm</a><br />
DOI: <a href="https://doi.org/10.4269/ajtmh.2003.68.2.0680161">10.4269/ajtmh.2003.68.2.0680161</a></p>
</div>
<div id="ref-DnnjILYG">
<p>14. <strong>Population structure in the malaria vector, Anopheles arabiensis Patton, in East Africa</strong><br />
MJ Donnelly, N Cuamba, JD Charlwood, FH Collins, H Townson<br />
<em>Heredity</em> (1999-10-01) <a href="https://doi.org/bg4xmm">https://doi.org/bg4xmm</a><br />
DOI: <a href="https://doi.org/10.1038/sj.hdy.6885930">10.1038/sj.hdy.6885930</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/10583542">10583542</a></p>
</div>
<div id="ref-14yZJJBPD">
<p>15. <strong>Identification of Single Specimens of the Anopheles Gambiae Complex by the Polymerase Chain Reaction</strong><br />
Julie A. Scott, William G. Brogdon, Frank H. Collins<br />
<em>The American Journal of Tropical Medicine and Hygiene</em> (1993-10-01) <a href="https://doi.org/ghcqgk">https://doi.org/ghcqgk</a><br />
DOI: <a href="https://doi.org/10.4269/ajtmh.1993.49.520">10.4269/ajtmh.1993.49.520</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/8214283">8214283</a></p>
</div>
<div id="ref-1C7qEzk6S">
<p>16. <strong>Wind direction and proximity to larval sites determines malaria risk in Kilifi District in Kenya</strong><br />
Janet T. Midega, Dave L. Smith, Ally Olotu, Joseph M. Mwangangi, Joseph G. Nzovu, Juliana Wambua, George Nyangweso, Charles M. Mbogo, George K. Christophides, Kevin Marsh, Philip Bejon<br />
<em>Nature Communications</em> (2012-02-14) <a href="https://doi.org/ghcqgd">https://doi.org/ghcqgd</a><br />
DOI: <a href="https://doi.org/10.1038/ncomms1672">10.1038/ncomms1672</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22334077">22334077</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3292715">PMC3292715</a></p>
</div>
<div id="ref-xsFRxYp6">
<p>17. <strong>Assessment of the effect of larval source management and house improvement on malaria transmission when added to standard malaria control strategies in southern Malawi: study protocol for a cluster-randomised controlled trial</strong><br />
Robert S. McCann, Henk van den Berg, Peter J. Diggle, Michèle van Vugt, Dianne J. Terlouw, Kamija S. Phiri, Aurelio Di Pasquale, Nicolas Maire, Steven Gowelo, Monicah M. Mburu, … Willem Takken<br />
<em>BMC Infectious Diseases</em> (2017-09-22) <a href="https://doi.org/ggr5g7">https://doi.org/ggr5g7</a><br />
DOI: <a href="https://doi.org/10.1186/s12879-017-2749-2">10.1186/s12879-017-2749-2</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28938876">28938876</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5610449">PMC5610449</a></p>
</div>
<div id="ref-vckO98IK">
<p>18. <strong>Mapping Malaria Transmission Intensity in Malawi, 2000–2010</strong><br />
Adam Bennett, Lawrence Kazembe, Don P. Mathanga, Damaris Kinyoki, Doreen Ali, Robert W. Snow, Abdisalan M. Noor<br />
<em>The American Journal of Tropical Medicine and Hygiene</em> (2013-11-06) <a href="https://doi.org/f5h9zz">https://doi.org/f5h9zz</a><br />
DOI: <a href="https://doi.org/10.4269/ajtmh.13-0028">10.4269/ajtmh.13-0028</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24062477">24062477</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3820324">PMC3820324</a></p>
</div>
<div id="ref-s77kHoJc">
<p>19. <strong>Entomological indices of malaria transmission in Chikhwawa district, Southern Malawi</strong><br />
Themba Mzilahowa, Ian M Hastings, Malcolm E Molyneux, Philip J McCall<br />
<em>Malaria Journal</em> (2012-11-21) <a href="https://doi.org/gbcb3h">https://doi.org/gbcb3h</a><br />
DOI: <a href="https://doi.org/10.1186/1475-2875-11-380">10.1186/1475-2875-11-380</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23171123">23171123</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3536595">PMC3536595</a></p>
</div>
<div id="ref-pSmuJkYQ">
<p>20. <strong>A Ribosomal RNA Gene Probe Differentiates Member Species of the Anopheles gambiae Complex</strong><br />
Frank H. Collins, Nora J. Besansky, M. Alina Mendez, Melissa O. Rasmussen, Victoria Finnerty, Philip C. Mehaffey<br />
<em>The American Journal of Tropical Medicine and Hygiene</em> (1987-07-01) <a href="https://doi.org/ghcqgj">https://doi.org/ghcqgj</a><br />
DOI: <a href="https://doi.org/10.4269/ajtmh.1987.37.37">10.4269/ajtmh.1987.37.37</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/2886070">2886070</a></p>
</div>
<div id="ref-f8uRMyY6">
<p>21. <strong>Analysis of the sporozoite ELISA for estimating infection rates in Mozambican anophelines</strong><br />
J. D. CHARLWOOD, E. V. E. TOMÁS, N. CUAMBA, J. PINTO<br />
<em>Medical and Veterinary Entomology</em> (2015-03) <a href="https://doi.org/f62ckd">https://doi.org/f62ckd</a><br />
DOI: <a href="https://doi.org/10.1111/mve.12084">10.1111/mve.12084</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25088021">25088021</a></p>
</div>
<div id="ref-FVN9nrFP">
<p>22. <strong>Genetic basis of pyrethroid resistance in a population of Anopheles arabiensis, the primary malaria vector in Lower Moshi, north-eastern Tanzania</strong><br />
Johnson Matowo, Christopher M Jones, Bilali Kabula, Hilary Ranson, Keith Steen, Franklin Mosha, Mark Rowland, David Weetman<br />
<em>Parasites &amp; Vectors</em> (2014) <a href="https://doi.org/ghcqgg">https://doi.org/ghcqgg</a><br />
DOI: <a href="https://doi.org/10.1186/1756-3305-7-274">10.1186/1756-3305-7-274</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24946780">24946780</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4082164">PMC4082164</a></p>
</div>
<div id="ref-ca2mjT0Y">
<p>23. <strong>Susceptibility status of malaria vectors to insecticides commonly used for malaria control in Tanzania</strong><br />
Bilali Kabula, Patrick Tungu, Johnson Matowo, Jovin Kitau, Clement Mweya, Basiliana Emidi, Denis Masue, Calvin Sindato, Robert Malima, Jubilate Minja, … William Kisinza<br />
<em>Tropical Medicine &amp; International Health</em> (2012-06) <a href="https://doi.org/f3zsbt">https://doi.org/f3zsbt</a><br />
DOI: <a href="https://doi.org/10.1111/j.1365-3156.2012.02986.x">10.1111/j.1365-3156.2012.02986.x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22519840">22519840</a></p>
</div>
<div id="ref-dB1ebxOj">
<p>24. <strong>High level of resistance in the mosquito Anopheles gambiae to pyrethroid insecticides and reduced susceptibility to bendiocarb in north-western Tanzania</strong><br />
Natacha Protopopoff, Johnson Matowo, Robert Malima, Reginald Kavishe, Robert Kaaya, Alexandra Wright, Philippa A West, Immo Kleinschmidt, William Kisinza, Franklin W Mosha, Mark Rowland<br />
<em>Malaria Journal</em> (2013-05-02) <a href="https://doi.org/ghcqgf">https://doi.org/ghcqgf</a><br />
DOI: <a href="https://doi.org/10.1186/1475-2875-12-149">10.1186/1475-2875-12-149</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23638757">23638757</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3655935">PMC3655935</a></p>
</div>
<div id="ref-m3y4gwzb">
<p>25. <strong>Estimating the annual entomological inoculation rate for Plasmodium falciparum transmitted by Anopheles gambiae s.l. using three sampling methods in three sites in Uganda</strong><br />
Maxwell Kilama, David L Smith, Robert Hutchinson, Ruth Kigozi, Adoke Yeka, Geoff Lavoy, Moses R Kamya, Sarah G Staedke, Martin J Donnelly, Chris Drakeley, … Steve W Lindsay<br />
<em>Malaria Journal</em> (2014-03-21) <a href="https://doi.org/gdkzsz">https://doi.org/gdkzsz</a><br />
DOI: <a href="https://doi.org/10.1186/1475-2875-13-111">10.1186/1475-2875-13-111</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24656206">24656206</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4001112">PMC4001112</a></p>
</div>
<div id="ref-z9HqxYF3">
<p>26. <strong>CYP6 P450 Enzymes and ACE-1 Duplication Produce Extreme and Multiple Insecticide Resistance in the Malaria Mosquito Anopheles gambiae</strong><br />
Constant V. Edi, Luc Djogbénou, Adam M. Jenkins, Kimberly Regna, Marc A. T. Muskavitch, Rodolphe Poupardin, Christopher M. Jones, John Essandoh, Guillaume K. Kétoh, Mark J. I. Paine, … David Weetman<br />
<em>PLoS Genetics</em> (2014-03-20) <a href="https://doi.org/f56k77">https://doi.org/f56k77</a><br />
DOI: <a href="https://doi.org/10.1371/journal.pgen.1004236">10.1371/journal.pgen.1004236</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24651294">24651294</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3961184">PMC3961184</a></p>
</div>
</div>
<!-- default theme -->

<style>
    /* import google fonts */
    @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
    @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

    /* -------------------------------------------------- */
    /* global */
    /* -------------------------------------------------- */

    /* all elements */
    * {
        /* force sans-serif font unless specified otherwise */
        font-family: "Open Sans", "Helvetica", sans-serif;

        /* prevent text inflation on some mobile browsers */
        -webkit-text-size-adjust: none !important;
        -moz-text-size-adjust: none !important;
        -o-text-size-adjust: none !important;
        text-size-adjust: none !important;
    }

    @media only screen {
        /* "page" element */
        body {
            position: relative;
            box-sizing: border-box;
            font-size: 12pt;
            line-height: 1.5;
            max-width: 8.5in;
            margin: 20px auto;
            padding: 40px;
            border-radius: 5px;
            border: solid 1px #bdbdbd;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            background: #ffffff;
        }
    }

    /* when on screen < 8.5in wide */
    @media only screen and (max-width: 8.5in) {
        /* "page" element */
        body {
            padding: 20px;
            margin: 0;
            border-radius: 0;
            border: none;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
            background: none;
        }
    }

    /* -------------------------------------------------- */
    /* headings */
    /* -------------------------------------------------- */

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
        margin: 20px 0;
        padding: 0;
        font-weight: bold;
    }

    /* biggest heading */
    h1 {
        margin: 40px 0;
        text-align: center;
    }

    /* second biggest heading */
    h2 {
        margin-top: 30px;
        padding-bottom: 5px;
        border-bottom: solid 1px #bdbdbd;
    }

    /* -------------------------------------------------- */
    /* manuscript header */
    /* -------------------------------------------------- */

    /* manuscript title */
    header > h1 {
        margin: 0;
    }

    /* manuscript title caption text (ie "automatically generated on") */
    header + p {
        text-align: center;
        margin-top: 10px;
    }

    /* -------------------------------------------------- */
    /* text elements */
    /* -------------------------------------------------- */

    /* links */
    a {
        color: #2196f3;
        overflow-wrap: break-word;
    }

    /* normal links (not empty, not button link, not syntax highlighting link) */
    a:not(:empty):not(.button):not(.sourceLine) {
        padding-left: 1px;
        padding-right: 1px;
    }

    /* superscripts and subscripts */
    sub,
    sup {
        /* prevent from affecting line height */
        line-height: 0;
    }

    /* unordered and ordered lists*/
    ul,
    ol {
        padding-left: 20px;
    }

    /* class for styling text semibold */
    .semibold {
        font-weight: 600;
    }

    /* class for styling elements horizontally left aligned */
    .left {
        display: block;
        text-align: left;
        margin-left: auto;
        margin-right: 0;
        justify-content: left;
    }

    /* class for styling elements horizontally centered */
    .center {
        display: block;
        text-align: center;
        margin-left: auto;
        margin-right: auto;
        justify-content: center;
    }

    /* class for styling elements horizontally right aligned */
    .right {
        display: block;
        text-align: right;
        margin-left: 0;
        margin-right: auto;
        justify-content: right;
    }

    /* -------------------------------------------------- */
    /* section elements */
    /* -------------------------------------------------- */

    /* horizontal divider line */
    hr {
        border: none;
        height: 1px;
        background: #bdbdbd;
    }

    /* paragraphs, horizontal dividers, figures, tables, code */
    p,
    hr,
    figure,
    table,
    pre {
        /* treat all as "paragraphs", with consistent vertical margins */
        margin-top: 20px;
        margin-bottom: 20px;
    }

    /* -------------------------------------------------- */
    /* figures */
    /* -------------------------------------------------- */

    /* figure */
    figure {
        max-width: 100%;
        margin-left: auto;
        margin-right: auto;
    }

    /* figure caption */
    figcaption {
        padding: 0;
        padding-top: 10px;
    }

    /* figure image element */
    figure img {
        max-width: 100%;
        display: block;
        margin-left: auto;
        margin-right: auto;
    }

    /* figure auto-number */
    img + figcaption > span:first-of-type {
        font-weight: bold;
        margin-right: 5px;
    }

    /* -------------------------------------------------- */
    /* tables */
    /* -------------------------------------------------- */

    /* table */
    table {
        border-collapse: collapse;
        border-spacing: 0;
        width: 100%;
        margin-left: auto;
        margin-right: auto;
    }

    /* table cells */
    th,
    td {
        border: solid 1px #bdbdbd;
        padding: 10px;
        /* squash table if too wide for page by forcing line breaks */
        overflow-wrap: break-word;
        word-break: break-word;
    }

    /* header row and even rows */
    th,
    tr:nth-child(2n) {
        background-color: #fafafa;
    }

    /* odd rows */
    tr:nth-child(2n + 1) {
        background-color: #ffffff;
    }

    /* table caption */
    caption {
        text-align: left;
        padding: 0;
        padding-bottom: 10px;
    }

    /* table auto-number */
    table > caption > span:first-of-type,
    div.table_wrapper > table > caption > span:first-of-type {
        font-weight: bold;
        margin-right: 5px;
    }

    /* -------------------------------------------------- */
    /* code */
    /* -------------------------------------------------- */

    /* multi-line code block */
    pre {
        padding: 10px;
        background-color: #eeeeee;
        color: #000000;
        border-radius: 5px;
        break-inside: avoid;
        text-align: left;
    }

    /* inline code, ie code within normal text */
    :not(pre) > code {
        padding: 0 4px;
        background-color: #eeeeee;
        color: #000000;
        border-radius: 5px;
    }

    /* code text */
    /* apply all children, to reach syntax highlighting sub-elements */
    code,
    code * {
        /* force monospace font */
        font-family: "Source Code Pro", "Courier New", monospace;
    }

    /* -------------------------------------------------- */
    /* quotes */
    /* -------------------------------------------------- */

    /* quoted text */
    blockquote {
        margin: 0;
        padding: 0;
        border-left: 4px solid #bdbdbd;
        padding-left: 16px;
        break-inside: avoid;
    }

    /* -------------------------------------------------- */
    /* banners */
    /* -------------------------------------------------- */

    /* info banners */
    .banner {
        box-sizing: border-box;
        display: block;
        position: relative;
        width: 100%;
        margin-top: 20px;
        margin-bottom: 20px;
        padding: 20px;
        text-align: center;
    }

    /* paragraph in banner */
    .banner > p {
        margin: 0;
    }

    /* -------------------------------------------------- */
    /* highlight colors */
    /* -------------------------------------------------- */

    .white {
        background: #ffffff;
    }
    .lightgrey {
        background: #eeeeee;
    }
    .grey {
        background: #757575;
    }
    .darkgrey {
        background: #424242;
    }
    .black {
        background: #000000;
    }
    .lightred {
        background: #ffcdd2;
    }
    .lightyellow {
        background: #ffecb3;
    }
    .lightgreen {
        background: #dcedc8;
    }
    .lightblue {
        background: #e3f2fd;
    }
    .lightpurple {
        background: #f3e5f5;
    }
    .red {
        background: #f44336;
    }
    .orange {
        background: #ff9800;
    }
    .yellow {
        background: #ffeb3b;
    }
    .green {
        background: #4caf50;
    }
    .blue {
        background: #2196f3;
    }
    .purple {
        background: #9c27b0;
    }
    .white,
    .lightgrey,
    .lightred,
    .lightyellow,
    .lightgreen,
    .lightblue,
    .lightpurple,
    .orange,
    .yellow,
    .white a,
    .lightgrey a,
    .lightred a,
    .lightyellow a,
    .lightgreen a,
    .lightblue a,
    .lightpurple a,
    .orange a,
    .yellow a {
        color: #000000;
    }
    .grey,
    .darkgrey,
    .black,
    .red,
    .green,
    .blue,
    .purple,
    .grey a,
    .darkgrey a,
    .black a,
    .red a,
    .green a,
    .blue a,
    .purple a {
        color: #ffffff;
    }

    /* -------------------------------------------------- */
    /* buttons */
    /* -------------------------------------------------- */

    /* class for styling links like buttons */
    .button {
        display: inline-flex;
        justify-content: center;
        align-items: center;
        margin: 5px;
        padding: 10px 20px;
        font-size: 0.75em;
        font-weight: 600;
        text-transform: uppercase;
        text-decoration: none;
        letter-spacing: 1px;
        background: none;
        color: #2196f3;
        border: solid 1px #bdbdbd;
        border-radius: 5px;
    }

    /* buttons when hovered */
    .button:hover:not([disabled]),
    .icon_button:hover:not([disabled]) {
        cursor: pointer;
        background: #f5f5f5;
    }

    /* buttons when disabled */
    .button[disabled],
    .icon_button[disabled] {
        opacity: 0.35;
        pointer-events: none;
    }

    /* class for styling buttons containg only single icon */
    .icon_button {
        display: inline-flex;
        justify-content: center;
        align-items: center;
        text-decoration: none;
        margin: 0;
        padding: 0;
        background: none;
        border-radius: 5px;
        border: none;
        width: 20px;
        height: 20px;
        min-width: 20px;
        min-height: 20px;
    }

    /* icon button inner svg image */
    .icon_button > svg {
        height: 16px;
    }

    /* -------------------------------------------------- */
    /* icons */
    /* -------------------------------------------------- */

    /* class for styling icons inline with text */
    .inline_icon {
        height: 1em;
        position: relative;
        top: 0.125em;
    }

    /* -------------------------------------------------- */
    /* print control */
    /* -------------------------------------------------- */

    @media print {
        @page {
            /* suggested printing margin */
            margin: 0.5in;
        }

        /* document and "page" elements */
        html, body {
            margin: 0;
            padding: 0;
            width: 100%;
            height: 100%;
        }

        /* "page" element */
        body {
            font-size: 11pt !important;
            line-height: 1.35;
        }

        /* all headings */
        h1,
        h2,
        h3,
        h4,
        h5,
        h6 {
            margin: 15px 0;
        }

        /* heading 1 */
        h1 {
            font-size: 1.75em;
        }

        /* heading 2 */
        h2 {
            font-size: 1.25em;
            margin-top: 0;
        }

        /* heading 3 */
        h3 {
            font-size: 1.10em;
        }

        /* figures and tables */
        figure, table {
            font-size: 0.85em;
        }

        /* table cells */
        th,
        td {
            padding: 5px;
        }

        /* shrink font awesome icons */
        i.fas,
        i.fab,
        i.far,
        i.fal {
            transform: scale(0.85);
        }

        /* decrease banner margins */
        .banner {
            margin-top: 15px;
            margin-bottom: 15px;
            padding: 15px;
        }

        /* class for centering an element vertically on its own page */
        .page_center {
            margin: auto;
            width: 100%;
            height: 100%;
            display: flex;
            align-items: center;
            vertical-align: middle;
            break-before: page;
            break-after: page;
        }

        /* always insert a page break before the element */
        .page_break_before {
            break-before: page;
        }

        /* always insert a page break after the element */
        .page_break_after {
            break-after: page;
        }

        /* avoid page break before the element */
        .page_break_before_avoid {
            break-before: avoid;
        }

        /* avoid page break after the element */
        .page_break_after_avoid {
            break-after: avoid;
        }

        /* avoid page break inside the element */
        .page_break_inside_avoid {
            break-inside: avoid;
        }
    }

    /* -------------------------------------------------- */
    /* override pandoc css quirks */
    /* -------------------------------------------------- */

    .sourceCode {
        /* prevent unsightly overflow in wide code blocks */
        overflow: auto !important;
    }

    div.sourceCode {
        /* prevent background fill on top-most code block  container */
        background: none !important;
    }

    .sourceCode * {
        /* force consistent line spacing */
        line-height: 1.5 !important;
    }

    div.sourceCode {
        /* style code block margins same as <pre> element */
        margin-top: 20px;
        margin-bottom: 20px;
    }

    /* -------------------------------------------------- */
    /* tablenos */
    /* -------------------------------------------------- */

    /* tablenos wrapper */
    .tablenos {
        /* show scrollbar on tables if necessary to prevent overflow */
        width: 100%;
        margin: 20px 0;
    }

    .tablenos > table {
        /* move margins from table to table_wrapper to allow margin collapsing */
        margin: 0;
    }

    @media only screen {
        /* tablenos wrapper */
        .tablenos {
            /* show scrollbar on tables if necessary to prevent overflow */
            overflow-x: auto !important;
        }

        .tablenos th,
        .tablenos td {
            overflow-wrap: unset !important;
            word-break: unset !important;
        }

        /* table in wrapper */
        .tablenos table,
        .tablenos table * {
            /* don't break table words */
            overflow-wrap: normal !important;
        }
    }

    /* -------------------------------------------------- */
    /* mathjax */
    /* -------------------------------------------------- */

    /* mathjax containers */
    .math.display > span:not(.MathJax_Preview) {
        /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
        display: flex !important;
        overflow-x: auto !important;
        overflow-y: hidden !important;
        justify-content: center;
        align-items: center;
        margin: 0 !important;
    }

    /* right click menu */
    .MathJax_Menu {
        border-radius: 5px !important;
        border: solid 1px #bdbdbd !important;
        box-shadow: none !important;
    }

    /* equation auto-number */
    span[id^="eq:"] > span.math.display + span {
        font-weight: 600;
    }

    /* equation */
    span[id^="eq:"] > span.math.display > span {
        /* nudge to make room for equation auto-number and anchor */
        margin-right: 60px !important;
    }

    /* -------------------------------------------------- */
    /* anchors plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* anchor button */
        .anchor {
            opacity: 0;
            margin-left: 5px;
        }

        /* anchor buttons within <h2>'s */
        h2 .anchor {
            margin-left: 10px;
        }

        /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
        *:hover > .anchor,
        .anchor:hover,
        .anchor:focus {
            opacity: 1;
        }

        /* anchor button when hovered */
        .anchor:hover {
            cursor: pointer;
        }
    }

    /* always show anchor button on devices with no mouse/hover ability */
    @media (hover: none) {
        .anchor {
            opacity: 1;
        }
    }

    /* always hide anchor button on print */
    @media only print {
        .anchor {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* accordion plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* accordion arrow button */
        .accordion_arrow {
            margin-right: 10px;
        }

        /* arrow icon when <h2> data-collapsed attribute true */
        h2[data-collapsed="true"] > .accordion_arrow > svg {
            transform: rotate(-90deg);
        }

        /* all elements (except <h2>'s) when data-collapsed attribute true */
        *:not(h2)[data-collapsed="true"] {
            display: none;
        }

        /* accordion arrow button when hovered and <h2>'s when hovered */
        .accordion_arrow:hover,
        h2[data-collapsed="true"]:hover,
        h2[data-collapsed="false"]:hover {
            cursor: pointer;
        }
    }

    /* always hide accordion arrow button on print */
    @media only print {
        .accordion_arrow {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* tooltips plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* tooltip container */
        #tooltip {
            position: absolute;
            width: 50%;
            min-width: 240px;
            max-width: 75%;
            z-index: 1;
        }

        /* tooltip content */
        #tooltip_content {
            margin-bottom: 5px;
            padding: 20px;
            border-radius: 5px;
            border: solid 1px #bdbdbd;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            background: #ffffff;
            overflow-wrap: break-word;
        }

        /* tooltip copy of paragraphs and figures */
        #tooltip_content > p,
        #tooltip_content > figure {
            margin: 0;
            max-height: 320px;
            overflow-y: auto;
        }

        /* tooltip copy of <img> */
        #tooltip_content > figure > img {
            max-height: 260px;
        }

        /* navigation bar */
        #tooltip_nav_bar {
            margin-top: 10px;
            text-align: center;
        }

        /* navigation bar previous/next buton */
        #tooltip_nav_bar > .icon_button {
            position: relative;
            top: 3px;
        }

        /* navigation bar previous button */
        #tooltip_nav_bar > .icon_button:first-of-type {
            margin-right: 5px;
        }

        /* navigation bar next button */
        #tooltip_nav_bar > .icon_button:last-of-type {
            margin-left: 5px;
        }
    }

    /* always hide tooltip on print */
    @media only print {
        #tooltip {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* jump to first plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* jump button */
        .jump_arrow {
            position: relative;
            top: 0.125em;
            margin-right: 5px;
        }
    }

    /* always hide jump button on print */
    @media only print {
        .jump_arrow {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* link highlight plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* anything with data-highlighted attribute true */
        [data-highlighted="true"] {
            background: #ffeb3b;
        }

        /* anything with data-selected attribute true */
        [data-selected="true"] {
            background: #ff8a65 !important;
        }

        /* animation definition for glow */
        @keyframes highlight_glow {
            0% {
                background: none;
            }
            10% {
                background: #bbdefb;
            }
            100% {
                background: none;
            }
        }

        /* anything with data-glow attribute true */
        [data-glow="true"] {
            animation: highlight_glow 2s;
        }
    }

    /* -------------------------------------------------- */
    /* table of contents plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* toc panel */
        #toc_panel {
            box-sizing: border-box;
            position: fixed;
            top: 0;
            left: 0;
            background: #ffffff;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            z-index: 2;
        }

        /* toc panel when closed */
        #toc_panel[data-open="false"] {
            min-width: 60px;
            width: 60px;
            height: 60px;
            border-right: solid 1px #bdbdbd;
            border-bottom: solid 1px #bdbdbd;
        }

        /* toc panel when open */
        #toc_panel[data-open="true"] {
            min-width: 260px;
            max-width: 480px;
            /* keep panel edge consistent distance away from "page" edge */
            width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
            bottom: 0;
            border-right: solid 1px #bdbdbd;
        }

        /* toc panel header */
        #toc_header {
            box-sizing: border-box;
            display: flex;
            flex-direction: row;
            align-items: center;
            height: 60px;
            margin: 0;
            padding: 20px;
        }

        /* toc panel header when hovered */
        #toc_header:hover {
            cursor: pointer;
        }

        /* toc panel header when panel open */
        #toc_panel[data-open="true"] > #toc_header {
            border-bottom: solid 1px #bdbdbd;
        }

        /* toc open/close header button */
        #toc_button {
            margin-right: 20px;
        }

        /* hide toc list and header text when closed */
        #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
        #toc_panel[data-open="false"] > #toc_list {
            display: none;
        }

        /* toc list of entries */
        #toc_list {
            box-sizing: border-box;
            width: 100%;
            padding: 20px;
            position: absolute;
            top: calc(60px + 1px);
            bottom: 0;
            overflow: auto;
        }

        /* toc entry, link to section in document */
        .toc_link {
            display: block;
            padding: 5px;
            position: relative;
            font-weight: 600;
            text-decoration: none;
        }

        /* toc entry when hovered or when "viewed" */
        .toc_link:hover,
        .toc_link[data-viewing="true"] {
            background: #f5f5f5;
        }

        /* toc entry, level 1 indentation */
        .toc_link[data-level="1"] {
            margin-left: 0;
        }

        /* toc entry, level 2 indentation */
        .toc_link[data-level="2"] {
            margin-left: 20px;
        }

        /* toc entry, level 3 indentation */
        .toc_link[data-level="3"] {
            margin-left: 40px;
        }

        /* toc entry, level 4 indentation */
        .toc_link[data-level="4"] {
            margin-left: 60px;
        }

        /* toc entry bullets */
        #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
            position: absolute;
            left: -15px;
            top: -1px;
            font-size: 1.5em;
        }

        /* toc entry, level 2 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
            content: "\2022";
        }

        /* toc entry, level 3 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
            content: "\25AB";
        }

        /* toc entry, level 4 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
            content: "-";
        }
    }

    /* when on screen < 8.5in wide */
    @media only screen and (max-width: 8.5in) {
        /* push <body> ("page") element down to make room for toc icon */
        .toc_body_nudge {
            padding-top: 60px;
        }

        /* toc icon when panel closed and not hovered */
        #toc_panel[data-open="false"]:not(:hover) {
            background: rgba(255, 255, 255, 0.75);
        }
    }

    /* always hide toc panel on print */
    @media only print {
        #toc_panel {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* lightbox plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* regular <img> in document when hovered */
        .lightbox_document_img:hover {
            cursor: pointer;
        }

        .body_no_scroll {
            overflow: hidden !important;
        }

        /* screen overlay */
        #lightbox_overlay {
            display: flex;
            flex-direction: column;
            position: fixed;
            left: 0;
            top: 0;
            right: 0;
            bottom: 0;
            background: rgba(0, 0, 0, 0.75);
            z-index: 3;
        }

        /* middle area containing lightbox image */
        #lightbox_image_container {
            flex-grow: 1;
            display: flex;
            justify-content: center;
            align-items: center;
            overflow: hidden;
            position: relative;
            padding: 20px;
        }

        /* bottom area containing caption */
        #lightbox_bottom_container {
            display: flex;
            justify-content: center;
            align-items: center;
            height: 100px;
            min-height: 100px;
            max-height: 100px;
            background: rgba(0, 0, 0, 0.5);
        }

        /* image number info text box */
        #lightbox_number_info {
            position: absolute;
            color: #ffffff;
            font-weight: 600;
            left: 2px;
            top: 0;
            z-index: 4;
        }

        /* zoom info text box */
        #lightbox_zoom_info {
            position: absolute;
            color: #ffffff;
            font-weight: 600;
            right: 2px;
            top: 0;
            z-index: 4;
        }

        /* copy of image caption */
        #lightbox_caption {
            box-sizing: border-box;
            display: inline-block;
            width: 100%;
            max-height: 100%;
            padding: 10px 0;
            text-align: center;
            overflow-y: auto;
            color: #ffffff;
        }

        /* navigation previous/next button */
        .lightbox_button {
            width: 100px;
            height: 100%;
            min-width: 100px;
            min-height: 100%;
            color: #ffffff;
        }

        /* navigation previous/next button when hovered */
        .lightbox_button:hover {
            background: none !important;
        }

        /* navigation button icon */
        .lightbox_button > svg {
            height: 25px;
        }

        /* figure auto-number */
        #lightbox_caption > span:first-of-type {
            font-weight: bold;
            margin-right: 5px;
        }

        /* lightbox image when hovered */
        #lightbox_img:hover {
            cursor: grab;
        }

        /* lightbox image when grabbed */
        #lightbox_img:active {
            cursor: grabbing;
        }
    }

    /* when on screen < 480px wide */
    @media only screen and (max-width: 480px) {
        /* make navigation buttons skinnier on small screens to make more room for caption text */
        .lightbox_button {
            width: 50px;
            min-width: 50px;
        }
    }

    /* always hide lightbox on print */
    @media only print {
        #lightbox_overlay {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* hypothesis (annotations) plugin */
    /* -------------------------------------------------- */

    /* side panel */
    .annotator-frame {
        width: 280px !important;
        z-index: 0 !important;
    }

    /* match highlight color to rest of theme */
    .annotator-highlights-always-on .annotator-hl {
        background-color: #ffeb3b !important;
    }

    /* match focused color to rest of theme */
    .annotator-hl.annotator-hl-focused {
        background-color: #ff8a65 !important;
    }

    /* match bucket bar color to rest of theme */
    .annotator-bucket-bar {
        background: #f5f5f5 !important;
    }

    /* always hide toolbar and tooltip on print */
    @media only print {
        .annotator-frame {
            display: none !important;
        }

        hypothesis-adder {
            display: none !important;
        }
    }
</style>
<!-- anchors plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin adds an anchor next to each of a certain type
        // of element that provides a human-readable url to that specific
        // item/position in the document (eg "manuscript.html#abstract"). It
        // also makes it such that scrolling out of view of a target removes
        // its identifier from the url.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'anchors';

        // default plugin options
        const options = {
            // which types of elements to add anchors next to, in
            // "document.querySelector" format
            typesQuery: 'h1, h2, h3, [id^="fig:"], [id^="tbl:"], [id^="eq:"]',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // add anchor to each element of specified types
            const elements = document.querySelectorAll(options.typesQuery);
            for (const element of elements)
                addAnchor(element);

            // attach scroll listener to window
            window.addEventListener('scroll', onScroll);
        }

        // when window is scrolled
        function onScroll() {
            // if url has hash and user has scrolled out of view of hash
            // target, remove hash from url
            const tolerance = 100;
            const target = getHashTarget();
            if (target) {
                if (
                    target.getBoundingClientRect().top >
                        window.innerHeight + tolerance ||
                    target.getBoundingClientRect().bottom < 0 - tolerance
                )
                    history.pushState(null, null, ' ');
            }
        }

        // add anchor to element
        function addAnchor(element) {
            let addTo; // element to add anchor button to

            // if figure or table, modify withId and addTo to get expected
            // elements
            if (element.id.indexOf('fig:') === 0) {
                addTo = element.querySelector('figcaption');
            } else if (element.id.indexOf('tbl:') === 0) {
                addTo = element.querySelector('caption');
            } else if (element.id.indexOf('eq:') === 0) {
                addTo = element.querySelector('.eqnos-number');
            }

            addTo = addTo || element;
            const id = element.id || null;

            // do not add anchor if element doesn't have assigned id.
            // id is generated by pandoc and is assumed to be unique and
            // human-readable
            if (!id)
                return;

            // create anchor button
            const anchor = document.createElement('a');
            anchor.innerHTML = document.querySelector('.icon_link').innerHTML;
            anchor.title = 'Link to this part of the document';
            anchor.classList.add('icon_button', 'anchor');
            anchor.dataset.ignore = 'true';
            anchor.href = '#' + id;
            addTo.appendChild(anchor);
        }

        // get element that is target of link or url hash
        function getHashTarget() {
            const hash = window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            // if figure or table, modify target to get expected element
            if (id.indexOf('fig:') === 0)
                target = target.querySelector('figure');
            if (id.indexOf('tbl:') === 0)
                target = target.querySelector('table');

            return target;
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- link icon -->

<template class="icon_link">
    <!-- modified from: https://fontawesome.com/icons/link -->
    <svg width="16" height="16" viewBox="0 0 512 512">
        <path
            fill="currentColor"
            d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
        ></path>
    </svg>
</template>
<!-- accordion plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin allows sections of content under <h2> headings
        // to be collapsible.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'accordion';

        // default plugin options
        const options = {
            // whether to always start expanded ('false'), always start
            // collapsed ('true'), or start collapsed when screen small ('auto')
            startCollapsed: 'auto',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // run through each <h2> heading
            const headings = document.querySelectorAll('h2');
            for (const heading of headings) {
                addArrow(heading);

                // start expanded/collapsed based on option
                if (
                    options.startCollapsed === 'true' ||
                    (options.startCollapsed === 'auto' && isSmallScreen())
                )
                    collapseHeading(heading);
                else
                    expandHeading(heading);
            }

            // attach hash change listener to window
            window.addEventListener('hashchange', onHashChange);
        }

        // when hash (eg manuscript.html#introduction) changes
        function onHashChange() {
            const target = getHashTarget();
            if (target)
                goToElement(target);
        }

        // add arrow to heading
        function addArrow(heading) {
            // add arrow button
            const arrow = document.createElement('button');
            arrow.innerHTML = document.querySelector(
                '.icon_angle_down'
            ).innerHTML;
            arrow.classList.add('icon_button', 'accordion_arrow');
            heading.insertBefore(arrow, heading.firstChild);

            // attach click listener to heading and button
            heading.addEventListener('click', onHeadingClick);
            arrow.addEventListener('click', onArrowClick);
        }

        // determine if on mobile-like device with small screen
        function isSmallScreen() {
            return Math.min(window.innerWidth, window.innerHeight) < 480;
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // get element that is target of hash
        function getHashTarget(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            // if figure or table, modify target to get expected element
            if (id.indexOf('fig:') === 0)
                target = target.querySelector('figure');
            if (id.indexOf('tbl:') === 0)
                target = target.querySelector('table');

            return target;
        }

        // when <h2> heading is clicked
        function onHeadingClick(event) {
            // only collapse if <h2> itself is target of click (eg, user did
            // not click on anchor within <h2>)
            if (event.target === this)
                toggleCollapse(this);
        }

        // when arrow button is clicked
        function onArrowClick() {
            toggleCollapse(this.parentNode);
        }

        // collapse section if expanded, expand if collapsed
        function toggleCollapse(heading) {
            if (heading.dataset.collapsed === 'false')
                collapseHeading(heading);
            else
                expandHeading(heading);
        }

        // elements to exclude from collapse, such as table of contents panel,
        // hypothesis panel, etc
        const exclude = '#toc_panel, div.annotator-frame, #lightbox_overlay';

        // collapse section
        function collapseHeading(heading) {
            heading.setAttribute('data-collapsed', 'true');
            const children = getChildren(heading);
            for (const child of children)
                child.setAttribute('data-collapsed', 'true');
        }

        // expand section
        function expandHeading(heading) {
            heading.setAttribute('data-collapsed', 'false');
            const children = getChildren(heading);
            for (const child of children)
                child.setAttribute('data-collapsed', 'false');
        }

        // get list of elements between this <h2> and next <h2> or <h1>
        // ("children" of the <h2> section)
        function getChildren(heading) {
            return nextUntil(heading, 'h2, h1', exclude);
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get list of elements after a start element up to element matching
        // query
        function nextUntil(element, query, exclude) {
            const elements = [];
            while (element = element.nextElementSibling, element) {
                if (element.matches(query))
                    break;
                if (!element.matches(exclude))
                    elements.push(element);
            }
            return elements;
        }

        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
    <!-- modified from: https://fontawesome.com/icons/angle-down -->
    <svg width="16" height="16" viewBox="0 0 448 512">
        <path
            fill="currentColor"
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
        ></path>
    </svg>
</template>
<!-- tooltips plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when the user hovers or
        // focuses a link to a citation or figure, a tooltip appears with a
        // preview of the reference content, along with arrows to navigate
        // between instances of the same reference in the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'tooltips';

        // default plugin options
        const options = {
            // whether user must click off to close tooltip instead of just
            // un-hovering
            clickClose: 'false',
            // delay (in ms) between opening and closing tooltip
            delay: '100',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            const links = getLinks();
            for (const link of links) {
                // attach hover and focus listeners to link
                link.addEventListener('mouseover', onLinkHover);
                link.addEventListener('mouseleave', onLinkUnhover);
                link.addEventListener('focus', onLinkFocus);
                link.addEventListener('touchend', onLinkTouch);
            }

            // attach mouse, key, and resize listeners to window
            window.addEventListener('mousedown', onClick);
            window.addEventListener('touchstart', onClick);
            window.addEventListener('keyup', onKeyUp);
            window.addEventListener('resize', onResize);
        }

        // when link is hovered
        function onLinkHover() {
            // function to open tooltip
            const delayOpenTooltip = function() {
                openTooltip(this);
            }.bind(this);

            // run open function after delay
            this.openTooltipTimer = window.setTimeout(
                delayOpenTooltip,
                options.delay
            );
        }

        // when mouse leaves link
        function onLinkUnhover() {
            // cancel opening tooltip
            window.clearTimeout(this.openTooltipTimer);

            // don't close on unhover if option specifies
            if (options.clickClose === 'true')
                return;

            // function to close tooltip
            const delayCloseTooltip = function() {
                // if tooltip open and if mouse isn't over tooltip, close
                const tooltip = document.getElementById('tooltip');
                if (tooltip && !tooltip.matches(':hover'))
                    closeTooltip();
            };

            // run close function after delay
            this.closeTooltipTimer = window.setTimeout(
                delayCloseTooltip,
                options.delay
            );
        }

        // when link is focused (tabbed to)
        function onLinkFocus(event) {
            openTooltip(this);
        }

        // when link is touched on touch screen
        function onLinkTouch(event) {
            // attempt to force hover state on first tap always, and trigger
            // regular link click (and navigation) on second tap
            if (event.target === document.activeElement)
                event.target.click();
            else {
                document.activeElement.blur();
                event.target.focus();
            }
            if (event.cancelable)
                event.preventDefault();
            event.stopPropagation();
            return false;
        }

        // when mouse is clicked anywhere in window
        function onClick(event) {
            closeTooltip();
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            switch (event.key) {
                // trigger click of prev button
                case 'ArrowLeft':
                    const prevButton = document.getElementById(
                        'tooltip_prev_button'
                    );
                    if (prevButton)
                        prevButton.click();
                    break;
                // trigger click of next button
                case 'ArrowRight':
                    const nextButton = document.getElementById(
                        'tooltip_next_button'
                    );
                    if (nextButton)
                        nextButton.click();
                    break;
                // close on esc
                case 'Escape':
                    closeTooltip();
                    break;
            }
        }

        // when window is resized or zoomed
        function onResize() {
            closeTooltip();
        }

        // get all links of types we wish to handle
        function getLinks() {
            const queries = [];
            // exclude buttons, anchor links, toc links, etc
            const exclude =
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            queries.push('a[href^="#ref-"]' + exclude); // citation links
            queries.push('a[href^="#fig:"]' + exclude); // figure links
            const query = queries.join(', ');
            return document.querySelectorAll(query);
        }

        // get links with same target, get index of link in set, get total
        // same links
        function getSameLinks(link) {
            const sameLinks = [];
            const links = getLinks();
            for (const otherLink of links) {
                if (
                    otherLink.getAttribute('href') === link.getAttribute('href')
                )
                    sameLinks.push(otherLink);
            }

            return {
                elements: sameLinks,
                index: sameLinks.indexOf(link),
                total: sameLinks.length
            };
        }

        // open tooltip
        function openTooltip(link) {
            // delete tooltip if it exists, start fresh
            closeTooltip();

            // make tooltip element
            const tooltip = makeTooltip(link);

            // if source couldn't be found and tooltip not made, exit
            if (!tooltip)
                return;

            // make navbar elements
            const navBar = makeNavBar(link);
            if (navBar)
                tooltip.firstElementChild.appendChild(navBar);

            // attach tooltip to page
            document.body.appendChild(tooltip);

            // position tooltip
            const position = function() {
                positionTooltip(link);
            };
            position();

            // if tooltip contains images, position again after they've loaded
            const imgs = tooltip.querySelectorAll('img');
            for (const img of imgs)
                img.addEventListener('load', position);
        }

        // close (delete) tooltip
        function closeTooltip() {
            const tooltip = document.getElementById('tooltip');
            if (tooltip)
                tooltip.remove();
        }

        // make tooltip
        function makeTooltip(link) {
            // get target element that link points to
            const source = getSource(link);

            // if source can't be found, exit
            if (!source)
                return;

            // create new tooltip
            const tooltip = document.createElement('div');
            tooltip.id = 'tooltip';
            const tooltipContent = document.createElement('div');
            tooltipContent.id = 'tooltip_content';
            tooltip.appendChild(tooltipContent);

            // make copy of source node and put in tooltip
            const sourceCopy = makeCopy(source);
            tooltipContent.appendChild(sourceCopy);

            // attach mouse event listeners
            tooltip.addEventListener('click', onTooltipClick);
            tooltip.addEventListener('mousedown', onTooltipClick);
            tooltip.addEventListener('touchstart', onTooltipClick);
            tooltip.addEventListener('mouseleave', onTooltipUnhover);

            // (for interaction with lightbox plugin)
            // transfer click on tooltip copied img to original img
            const sourceImg = source.querySelector('img');
            const sourceCopyImg = sourceCopy.querySelector('img');
            if (sourceImg && sourceCopyImg) {
                const clickImg = function() {
                    sourceImg.click();
                    closeTooltip();
                };
                sourceCopyImg.addEventListener('click', clickImg);
            }

            return tooltip;
        }

        // make carbon copy of html dom element
        function makeCopy(source) {
            const sourceCopy = source.cloneNode(true);

            // delete elements marked with ignore (eg anchor and jump buttons)
            const deleteFromCopy = sourceCopy.querySelectorAll(
                '[data-ignore="true"]'
            );
            for (const element of deleteFromCopy)
                element.remove();

            // delete certain element attributes
            const attributes = [
                'id',
                'data-collapsed',
                'data-selected',
                'data-highlighted',
                'data-glow'
            ];
            for (const attribute of attributes) {
                sourceCopy.removeAttribute(attribute);
                const elements = sourceCopy.querySelectorAll(
                    '[' + attribute + ']'
                );
                for (const element of elements)
                    element.removeAttribute(attribute);
            }

            return sourceCopy;
        }

        // when tooltip is clicked
        function onTooltipClick(event) {
            // when user clicks on tooltip, stop click from transferring
            // outside of tooltip (eg, click off to close tooltip, or eg click
            // off to unhighlight same refs)
            event.stopPropagation();
        }

        // when tooltip is unhovered
        function onTooltipUnhover(event) {
            if (options.clickClose === 'true')
                return;

            // make sure new mouse/touch/focus no longer over tooltip or any
            // element within it
            const tooltip = document.getElementById('tooltip');
            if (!tooltip)
                return;
            if (this.contains(event.relatedTarget))
                return;

            closeTooltip();
        }

        // make nav bar to go betwen prev/next instances of same reference
        function makeNavBar(link) {
            // find other links to the same source
            const sameLinks = getSameLinks(link);

            // don't show nav bar when singular reference
            if (sameLinks.total <= 1)
                return;

            // find prev/next links with same target
            const prevLink = getPrevLink(link, sameLinks);
            const nextLink = getNextLink(link, sameLinks);

            // create nav bar
            const navBar = document.createElement('div');
            navBar.id = 'tooltip_nav_bar';
            const text = sameLinks.index + 1 + ' of ' + sameLinks.total;

            // create nav bar prev/next buttons
            const prevButton = document.createElement('button');
            const nextButton = document.createElement('button');
            prevButton.id = 'tooltip_prev_button';
            nextButton.id = 'tooltip_next_button';
            prevButton.title =
                'Jump to the previous occurence of this item in the document [←]';
            nextButton.title =
                'Jump to the next occurence of this item in the document [→]';
            prevButton.classList.add('icon_button');
            nextButton.classList.add('icon_button');
            prevButton.innerHTML = document.querySelector(
                '.icon_caret_left'
            ).innerHTML;
            nextButton.innerHTML = document.querySelector(
                '.icon_caret_right'
            ).innerHTML;
            navBar.appendChild(prevButton);
            navBar.appendChild(document.createTextNode(text));
            navBar.appendChild(nextButton);

            // attach click listeners to buttons
            prevButton.addEventListener('click', function() {
                onPrevNextClick(link, prevLink);
            });
            nextButton.addEventListener('click', function() {
                onPrevNextClick(link, nextLink);
            });

            return navBar;
        }

        // get previous link with same target
        function getPrevLink(link, sameLinks) {
            if (!sameLinks)
                sameLinks = getSameLinks(link);
            // wrap index to other side if < 1
            let index;
            if (sameLinks.index - 1 >= 0)
                index = sameLinks.index - 1;
            else
                index = sameLinks.total - 1;
            return sameLinks.elements[index];
        }

        // get next link with same target
        function getNextLink(link, sameLinks) {
            if (!sameLinks)
                sameLinks = getSameLinks(link);
            // wrap index to other side if > total
            let index;
            if (sameLinks.index + 1 <= sameLinks.total - 1)
                index = sameLinks.index + 1;
            else
                index = 0;
            return sameLinks.elements[index];
        }

        // get element that is target of link or url hash
        function getSource(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            // if ref or figure, modify target to get expected element
            if (id.indexOf('ref-') === 0)
                target = target.querySelector('p');
            else if (id.indexOf('fig:') === 0)
                target = target.querySelector('figure');

            return target;
        }

        // when prev/next arrow button is clicked
        function onPrevNextClick(link, prevNextLink) {
            if (link && prevNextLink)
                goToElement(prevNextLink, window.innerHeight * 0.5);
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // determine position to place tooltip based on link position in
        // viewport and tooltip size
        function positionTooltip(link, left, top) {
            const tooltipElement = document.getElementById('tooltip');
            if (!tooltipElement)
                return;

            // get convenient vars for position/dimensions of
            // link/tooltip/page/view
            link = getRectInPage(link);
            const tooltip = getRectInPage(tooltipElement);
            const view = getRectInPage();

            // horizontal positioning
            if (left)
                // use explicit value
                left = left;
            else if (link.left + tooltip.width < view.right)
                // fit tooltip to right of link
                left = link.left;
            else if (link.right - tooltip.width > view.left)
                // fit tooltip to left of link
                left = link.right - tooltip.width;
            // center tooltip in view
            else
                left = (view.right - view.left) / 2 - tooltip.width / 2;

            // vertical positioning
            if (top)
                // use explicit value
                top = top;
            else if (link.top - tooltip.height > view.top)
                // fit tooltip above link
                top = link.top - tooltip.height;
            else if (link.bottom + tooltip.height < view.bottom)
                // fit tooltip below link
                top = link.bottom;
            else {
                // center tooltip in view
                top = view.top + view.height / 2 - tooltip.height / 2;
                // nudge off of link to left/right if possible
                if (link.right + tooltip.width < view.right)
                    left = link.right;
                else if (link.left - tooltip.width > view.left)
                    left = link.left - tooltip.width;
            }

            tooltipElement.style.left = left + 'px';
            tooltipElement.style.top = top + 'px';
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get position of element relative to page
        function getRectInPage(element) {
            const rect = getRectInView(element);
            const body = getRectInView(document.body);

            const newRect = {};
            newRect.left = rect.left - body.left;
            newRect.top = rect.top - body.top;
            newRect.right = rect.right - body.left;
            newRect.bottom = rect.bottom - body.top;
            newRect.width = rect.width;
            newRect.height = rect.height;

            return newRect;
        }

        // (for interaction with accordion plugin)
        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // (for interaction with accordion plugin)
        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // (for interaction with accordion plugin)
        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
    <!-- modified from: https://fontawesome.com/icons/caret-left -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
        ></path>
    </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
    <!-- modified from: https://fontawesome.com/icons/caret-right -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        ></path>
    </svg>
</template>
<!-- jump to first plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin adds a button next to each reference entry,
        // figure, and table that jumps the page to the first occurrence of a
        // link to that item in the manuscript.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'jumpToFirst';

        // default plugin options
        const options = {
            // whether to add buttons next to reference entries
            references: 'true',
            // whether to add buttons next to figures
            figures: 'true',
            // whether to add buttons next to tables
            tables: 'true',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            if (options.references !== 'false')
                makeReferenceButtons();
            if (options.figures !== 'false')
                makeFigureButtons();
            if (options.tables !== 'false')
                makeTableButtons();
        }

        // when jump button clicked
        function onButtonClick() {
            const first = getFirstOccurrence(this.dataset.id);
            if (!first)
                return;

            // update url hash so navigating "back" in history will return
            // user to jump button
            window.location.hash = this.dataset.id;
            // scroll to link
            window.setTimeout(function() {
                goToElement(first, window.innerHeight * 0.5);
            }, 0);
        }

        // get first occurence of link to item in document
        function getFirstOccurrence(id) {
            let query = 'a';
            query += '[href="#' + id + '"]';
            // exclude buttons, anchor links, toc links, etc
            query +=
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            return document.querySelector(query);
        }

        // add button next to each reference entry
        function makeReferenceButtons() {
            const references = document.querySelectorAll('div[id^="ref-"]');
            for (const reference of references) {
                // get reference id and element to add button to
                const id = reference.id;
                const container = reference.firstElementChild;
                const first = getFirstOccurrence(id);

                // if can't find link to reference, ignore
                if (!first)
                    continue;

                // make jump button
                let button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this reference in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.innerHTML = button.outerHTML + container.innerHTML;
                button = container.firstElementChild;
                button.addEventListener('click', onButtonClick);
            }
        }

        // add button next to each figure
        function makeFigureButtons() {
            const figures = document.querySelectorAll('[id^="fig:"]');
            for (const figure of figures) {
                // get figure id and element to add button to
                const id = figure.id;
                const container = figure.querySelector('figcaption') || figure;
                const first = getFirstOccurrence(id);

                // if can't find link to figure, ignore
                if (!first)
                    continue;

                // make jump button
                const button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this figure in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.insertBefore(button, container.firstElementChild);
                button.addEventListener('click', onButtonClick);
            }
        }

        // add button next to each figure
        function makeTableButtons() {
            const tables = document.querySelectorAll('[id^="tbl:"]');
            for (const table of tables) {
                // get ref id and element to add button to
                const id = table.id;
                const container = table.querySelector('caption') || table;
                const first = getFirstOccurrence(id);

                // if can't find link to table, ignore
                if (!first)
                    continue;

                // make jump button
                const button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this table in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.insertBefore(button, container.firstElementChild);
                button.addEventListener('click', onButtonClick);
            }
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // (for interaction with accordion plugin)
        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
    <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
    <svg width="16" height="16" viewBox="0 0 320 512">
        <path
            fill="currentColor"
            d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
        ></path>
    </svg>
</template>
<!-- link highlight plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when a user hovers or
        // focuses a link, other links that have the same target will be
        // highlighted. It also makes it such that when clicking a link, the
        // target of the link (eg reference, figure, table) is briefly
        // highlighted.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'linkHighlight';

        // default plugin options
        const options = {
            // whether to also highlight links that go to external urls
            externalLinks: 'false',
            // whether user must click off to unhighlight instead of just
            // un-hovering
            clickUnhighlight: 'false',
            // whether to also highlight links that are unique
            highlightUnique: 'true',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            const links = getLinks();
            for (const link of links) {
                // attach mouse and focus listeners to link
                link.addEventListener('mouseenter', onLinkFocus);
                link.addEventListener('focus', onLinkFocus);
                link.addEventListener('mouseleave', onLinkUnhover);
            }

            // attach click and hash change listeners to window
            window.addEventListener('click', onClick);
            window.addEventListener('touchstart', onClick);
            window.addEventListener('hashchange', onHashChange);

            // run hash change on window load in case user has navigated
            // directly to hash
            onHashChange();
        }

        // when link is focused (tabbed to) or hovered
        function onLinkFocus() {
            highlight(this);
        }

        // when link is unhovered
        function onLinkUnhover() {
            if (options.clickUnhighlight !== 'true')
                unhighlightAll();
        }

        // when the mouse is clicked anywhere in window
        function onClick(event) {
            unhighlightAll();
        }

        // when hash (eg manuscript.html#introduction) changes
        function onHashChange() {
            const target = getHashTarget();
            if (target)
                glowElement(target);
        }

        // get element that is target of link or url hash
        function getHashTarget(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            return target;
        }

        // start glow sequence on an element
        function glowElement(element) {
            const startGlow = function() {
                onGlowEnd();
                element.dataset.glow = 'true';
                element.addEventListener('animationend', onGlowEnd);
            };
            const onGlowEnd = function() {
                element.removeAttribute('data-glow');
                element.removeEventListener('animationend', onGlowEnd);
            };
            startGlow();
        }

        // highlight link and all others with same target
        function highlight(link) {
            // force unhighlight all to start fresh
            unhighlightAll();

            // get links with same target
            if (!link)
                return;
            const sameLinks = getSameLinks(link);

            // if link unique and option is off, exit and don't highlight
            if (sameLinks.length <= 1 && options.highlightUnique !== 'true')
                return;

            // highlight all same links, and "select" (special highlight) this
            // one
            for (const sameLink of sameLinks) {
                if (sameLink === link)
                    sameLink.setAttribute('data-selected', 'true');
                else
                    sameLink.setAttribute('data-highlighted', 'true');
            }
        }

        // unhighlight all links
        function unhighlightAll() {
            const links = getLinks();
            for (const link of links) {
                link.setAttribute('data-selected', 'false');
                link.setAttribute('data-highlighted', 'false');
            }
        }

        // get links with same target
        function getSameLinks(link) {
            const results = [];
            const links = getLinks();
            for (const otherLink of links) {
                if (
                    otherLink.getAttribute('href') === link.getAttribute('href')
                )
                    results.push(otherLink);
            }
            return results;
        }

        // get all links of types we wish to handle
        function getLinks() {
            let query = 'a';
            if (options.externalLinks !== 'true')
                query += '[href^="#"]';
            // exclude buttons, anchor links, toc links, etc
            query +=
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            return document.querySelectorAll(query);
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>
<!-- table of contents plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin provides a "table of contents" (toc) panel on
        // the side of the document that allows the user to conveniently
        // navigate between sections of the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'tableOfContents';

        // default plugin options
        const options = {
            // which types of elements to add links for, in
            // "document.querySelector" format
            typesQuery: 'h1, h2, h3',
            // whether default behavior is to be closed ('false'), open
            // ('true'), or only open when screen wide enough to fit panel
            // ('auto'). note: still always starts closed when page loads.
            open: 'auto',
            // if list item is more than this many characters, text will be
            // truncated
            charLimit: '50',
            // whether or not to show bullets next to each toc item
            bullets: 'false',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // make toc panel and populate with entries (links to document
            // sections)
            const panel = makePanel();
            if (!panel)
                return;
            makeEntries(panel);
            document.body.insertBefore(panel, document.body.firstChild);

            closePanel();

            // attach click, scroll, and hash change listeners to window
            window.addEventListener('click', onClick);
            window.addEventListener('touchstart', onClick);
            window.addEventListener('scroll', onScroll);
            window.addEventListener('hashchange', onScroll);
            window.addEventListener('keyup', onKeyUp);
            onScroll();

            // add class to push document body down out of way of toc button
            document.body.classList.add('toc_body_nudge');
        }

        // determine if screen wide enough to fit toc panel
        function isSmallScreen() {
            // in default theme:
            // 816px = 8.5in = width of "page" (<body>) element
            // 260px = min width of toc panel (*2 for both sides of <body>)
            return window.innerWidth < 816 + 260 * 2;
        }

        // open/close panel based on option and screen size
        function openOrClosePanel() {
            if (
                options.open === 'true' ||
                (options.open === 'auto' && !isSmallScreen())
            )
                openPanel();
            else
                closePanel();
        }

        // when mouse is clicked anywhere in window
        function onClick() {
            const panel = document.getElementById('toc_panel');
            if (!panel)
                return;

            if (panel.dataset.open === 'true')
                openOrClosePanel();
        }

        // when window is scrolled or hash changed
        function onScroll() {
            highlightViewed();
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            // close on esc
            if (event.key === 'Escape')
                closePanel();
        }

        // find entry of currently viewed document section in toc and highlight
        function highlightViewed() {
            const firstId = getFirstInView(options.typesQuery);

            // get toc entries (links), unhighlight all, then highlight viewed
            const list = document.getElementById('toc_list');
            if (!firstId || !list)
                return;
            const links = list.querySelectorAll('a');
            for (const link of links)
                link.dataset.viewing = 'false';
            const link = list.querySelector('a[href="#' + firstId + '"]');
            if (!link)
                return;
            link.dataset.viewing = 'true';
        }

        // get first or previous toc listed element in top half of view
        function getFirstInView(query) {
            // get all elements matching query and with id
            const elements = document.querySelectorAll(query);
            const elementsWithIds = [];
            for (const element of elements) {
                if (element.id)
                    elementsWithIds.push(element);
            }


            // get first or previous element in top half of view
            for (let i = 0; i < elementsWithIds.length; i++) {
                const element = elementsWithIds[i];
                const prevElement = elementsWithIds[Math.max(0, i - 1)];
                if (element.getBoundingClientRect().top >= 0) {
                    if (
                        element.getBoundingClientRect().top <
                        window.innerHeight / 2
                    )
                        return element.id;
                    else
                        return prevElement.id;
                }
            }
        }

        // make panel
        function makePanel() {
            // create panel
            const panel = document.createElement('div');
            panel.id = 'toc_panel';
            if (options.bullets === 'true')
                panel.dataset.bullets = 'true';

            // create header
            const header = document.createElement('div');
            header.id = 'toc_header';

            // create toc button
            const button = document.createElement('button');
            button.id = 'toc_button';
            button.innerHTML = document.querySelector(
                '.icon_th_list'
            ).innerHTML;
            button.classList.add('icon_button');

            // create header text
            const text = document.createElement('h3');
            text.innerHTML = 'Table of Contents';

            // create container for toc list
            const list = document.createElement('div');
            list.id = 'toc_list';

            // attach click listeners
            panel.addEventListener('click', onPanelClick);
            header.addEventListener('click', onHeaderClick);
            button.addEventListener('click', onButtonClick);

            // attach elements
            header.appendChild(button);
            header.appendChild(text);
            panel.appendChild(header);
            panel.appendChild(list);

            return panel;
        }

        // create toc entries (links) to each element of the specified types
        function makeEntries(panel) {
            const elements = document.querySelectorAll(options.typesQuery);
            for (const element of elements) {
                // do not add link if element doesn't have assigned id
                if (!element.id)
                    continue;

                // create link/list item
                const link = document.createElement('a');
                link.classList.add('toc_link');
                switch (element.tagName.toLowerCase()) {
                    case 'h1':
                        link.dataset.level = '1';
                        break;
                    case 'h2':
                        link.dataset.level = '2';
                        break;
                    case 'h3':
                        link.dataset.level = '3';
                        break;
                    case 'h4':
                        link.dataset.level = '4';
                        break;
                }
                link.title = element.innerText;
                let text = element.innerText;
                if (text.length > options.charLimit)
                    text = text.slice(0, options.charLimit) + '...';
                link.innerHTML = text;
                link.href = '#' + element.id;
                link.addEventListener('click', onLinkClick);

                // attach link
                panel.querySelector('#toc_list').appendChild(link);
            }
        }

        // when panel is clicked
        function onPanelClick(event) {
            // stop click from propagating to window/document and closing panel
            event.stopPropagation();
        }

        // when header itself is clicked
        function onHeaderClick(event) {
            togglePanel();
        }

        // when button is clicked
        function onButtonClick(event) {
            togglePanel();
            // stop header underneath button from also being clicked
            event.stopPropagation();
        }

        // when link is clicked
        function onLinkClick() {
            openOrClosePanel();
        }

        // open panel if closed, close if opened
        function togglePanel() {
            const panel = document.getElementById('toc_panel');
            if (!panel)
                return;

            if (panel.dataset.open === 'true')
                closePanel();
            else
                openPanel();
        }

        // open panel
        function openPanel() {
            const panel = document.getElementById('toc_panel');
            if (panel)
                panel.dataset.open = 'true';
        }

        // close panel
        function closePanel() {
            const panel = document.getElementById('toc_panel');
            if (panel)
                panel.dataset.open = 'false';
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- th list icon -->

<template class="icon_th_list">
    <!-- modified from: https://fontawesome.com/icons/th-list -->
    <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
        <path
            fill="currentColor"
            d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            tabindex="-1"
        ></path>
    </svg>
</template>
<!-- lightbox plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when a user clicks on an
        // image, the image fills the screen and the user can pan/drag/zoom
        // the image and navigate between other images in the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'lightbox';

        // default plugin options
        const options = {
            // list of possible zoom/scale factors
            zoomSteps:
                '0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1,' +
                '1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8',
            // whether to fit image to view ('fit'), display at 100% and shrink
            // if necessary ('shrink'), or always display at 100% ('100')
            defaultZoom: 'fit',
            // whether to zoom in/out toward center of view ('true') or mouse
            // ('false')
            centerZoom: 'false',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // run through each <img> element
            const imgs = document.querySelectorAll('figure > img');
            let count = 1;
            for (const img of imgs) {
                img.classList.add('lightbox_document_img');
                img.dataset.number = count;
                img.dataset.total = imgs.length;
                img.addEventListener('click', openLightbox);
                count++;
            }

            // attach mouse and key listeners to window
            window.addEventListener('mousemove', onWindowMouseMove);
            window.addEventListener('keyup', onKeyUp);
        }

        // when mouse is moved anywhere in window
        function onWindowMouseMove(event) {
            window.mouseX = event.clientX;
            window.mouseY = event.clientY;
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            switch (event.key) {
                // trigger click of prev button
                case 'ArrowLeft':
                    const prevButton = document.getElementById(
                        'lightbox_prev_button'
                    );
                    if (prevButton)
                        prevButton.click();
                    break;
                // trigger click of next button
                case 'ArrowRight':
                    const nextButton = document.getElementById(
                        'lightbox_next_button'
                    );
                    if (nextButton)
                        nextButton.click();
                    break;
                // close on esc
                case 'Escape':
                    closeLightbox();
                    break;
            }
        }

        // open lightbox
        function openLightbox() {
            const lightbox = makeLightbox(this);
            if (!lightbox)
                return;

            blurBody(lightbox);
            document.body.appendChild(lightbox);
        }

        // make lightbox
        function makeLightbox(img) {
            // delete lightbox if it exists, start fresh
            closeLightbox();

            // create screen overlay containing lightbox
            const overlay = document.createElement('div');
            overlay.id = 'lightbox_overlay';

            // create image info boxes
            const numberInfo = document.createElement('div');
            const zoomInfo = document.createElement('div');
            numberInfo.id = 'lightbox_number_info';
            zoomInfo.id = 'lightbox_zoom_info';

            // create container for image
            const imageContainer = document.createElement('div');
            imageContainer.id = 'lightbox_image_container';
            const lightboxImg = makeLightboxImg(
                img,
                imageContainer,
                numberInfo,
                zoomInfo
            );
            imageContainer.appendChild(lightboxImg);

            // create bottom container for caption and navigation buttons
            const bottomContainer = document.createElement('div');
            bottomContainer.id = 'lightbox_bottom_container';
            const caption = makeCaption(img);
            const prevButton = makePrevButton(img);
            const nextButton = makeNextButton(img);
            bottomContainer.appendChild(prevButton);
            bottomContainer.appendChild(caption);
            bottomContainer.appendChild(nextButton);

            // attach top middle and bottom to overlay
            overlay.appendChild(numberInfo);
            overlay.appendChild(zoomInfo);
            overlay.appendChild(imageContainer);
            overlay.appendChild(bottomContainer);

            return overlay;
        }

        // make <img> object that is intuitively draggable and zoomable
        function makeLightboxImg(
            sourceImg,
            container,
            numberInfoBox,
            zoomInfoBox
        ) {
            // create copy of source <img>
            const img = sourceImg.cloneNode(true);
            img.classList.remove('lightbox_document_img');
            img.removeAttribute('id');
            img.removeAttribute('width');
            img.removeAttribute('height');
            img.style.position = 'unset';
            img.style.margin = '0';
            img.style.padding = '0';
            img.style.width = '';
            img.style.height = '';
            img.style.minWidth = '';
            img.style.minHeight = '';
            img.style.maxWidth = '';
            img.style.maxHeight = '';
            img.id = 'lightbox_img';

            // build sorted list of unique zoomSteps, always including a 100%
            let zoomSteps = [];
            const optionsZooms = options.zoomSteps.split(/[^0-9.]/);
            for (const optionZoom of optionsZooms) {
                const newZoom = parseFloat(optionZoom);
                if (newZoom && !zoomSteps.includes(newZoom))
                    zoomSteps.push(newZoom);
            }
            if (!zoomSteps.includes(1))
                zoomSteps.push(1);
            zoomSteps = zoomSteps.sort(function sortNumber(a, b) {
                return a - b;
            });

            // <img> object property variables
            let zoom = 1;
            let translateX = 0;
            let translateY = 0;
            let clickMouseX = undefined;
            let clickMouseY = undefined;
            let clickTranslateX = undefined;
            let clickTranslateY = undefined;

            updateNumberInfo();

            // update image numbers displayed in info box
            function updateNumberInfo() {
                numberInfoBox.innerHTML =
                    sourceImg.dataset.number + ' of ' + sourceImg.dataset.total;
            }

            // update zoom displayed in info box
            function updateZoomInfo() {
                let zoomInfo = zoom * 100;
                if (!Number.isInteger(zoomInfo))
                    zoomInfo = zoomInfo.toFixed(2);
                zoomInfoBox.innerHTML = zoomInfo + '%';
            }

            // move to closest zoom step above current zoom
            const zoomIn = function() {
                for (const zoomStep of zoomSteps) {
                    if (zoomStep > zoom) {
                        zoom = zoomStep;
                        break;
                    }
                }
                updateTransform();
            };

            // move to closest zoom step above current zoom
            const zoomOut = function() {
                zoomSteps.reverse();
                for (const zoomStep of zoomSteps) {
                    if (zoomStep < zoom) {
                        zoom = zoomStep;
                        break;
                    }
                }
                zoomSteps.reverse();

                updateTransform();
            };

            // update display of <img> based on scale/translate properties
            const updateTransform = function() {
                // set transform
                img.style.transform =
                    'translate(' +
                    (translateX || 0) +
                    'px,' +
                    (translateY || 0) +
                    'px) scale(' +
                    (zoom || 1) +
                    ')';

                // get new width/height after scale
                const rect = img.getBoundingClientRect();
                // limit translate
                translateX = Math.max(translateX, -rect.width / 2);
                translateX = Math.min(translateX, rect.width / 2);
                translateY = Math.max(translateY, -rect.height / 2);
                translateY = Math.min(translateY, rect.height / 2);

                // set transform
                img.style.transform =
                    'translate(' +
                    (translateX || 0) +
                    'px,' +
                    (translateY || 0) +
                    'px) scale(' +
                    (zoom || 1) +
                    ')';

                updateZoomInfo();
            };

            // fit <img> to container
            const fit = function() {
                // no x/y offset, 100% zoom by default
                translateX = 0;
                translateY = 0;
                zoom = 1;

                // widths of <img> and container
                const imgWidth = img.naturalWidth;
                const imgHeight = img.naturalHeight;
                const containerWidth = parseFloat(
                    window.getComputedStyle(container).width
                );
                const containerHeight = parseFloat(
                    window.getComputedStyle(container).height
                );

                // how much zooming is needed to fit <img> to container
                const xRatio = imgWidth / containerWidth;
                const yRatio = imgHeight / containerHeight;
                const maxRatio = Math.max(xRatio, yRatio);
                const newZoom = 1 / maxRatio;

                // fit <img> to container according to option
                if (options.defaultZoom === 'shrink') {
                    if (maxRatio > 1)
                        zoom = newZoom;
                } else if (options.defaultZoom === 'fit')
                    zoom = newZoom;

                updateTransform();
            };

            // when mouse wheel is rolled anywhere in container
            const onContainerWheel = function(event) {
                if (!event)
                    return;

                // let ctrl + mouse wheel to zoom behave as normal
                if (event.ctrlKey)
                    return;

                // prevent normal scroll behavior
                event.preventDefault();
                event.stopPropagation();

                // point around which to scale img
                const viewRect = container.getBoundingClientRect();
                const viewX = (viewRect.left + viewRect.right) / 2;
                const viewY = (viewRect.top + viewRect.bottom) / 2;
                const originX = options.centerZoom === 'true' ? viewX : mouseX;
                const originY = options.centerZoom === 'true' ? viewY : mouseY;

                // get point on image under origin
                const oldRect = img.getBoundingClientRect();
                const oldPercentX = (originX - oldRect.left) / oldRect.width;
                const oldPercentY = (originY - oldRect.top) / oldRect.height;

                // increment/decrement zoom
                if (event.deltaY < 0)
                    zoomIn();
                if (event.deltaY > 0)
                    zoomOut();

                // get offset between previous image point and origin
                const newRect = img.getBoundingClientRect();
                const offsetX =
                    originX - (newRect.left + newRect.width * oldPercentX);
                const offsetY =
                    originY - (newRect.top + newRect.height * oldPercentY);

                // translate image to keep image point under origin
                translateX += offsetX;
                translateY += offsetY;

                // perform translate
                updateTransform();
            };

            // when container is clicked
            function onContainerClick(event) {
                // if container itself is target of click, and not other
                // element above it
                if (event.target === this)
                    closeLightbox();
            }

            // when mouse button is pressed on image
            const onImageMouseDown = function(event) {
                // store original mouse position relative to image
                clickMouseX = window.mouseX;
                clickMouseY = window.mouseY;
                clickTranslateX = translateX;
                clickTranslateY = translateY;
                event.stopPropagation();
                event.preventDefault();
            };

            // when mouse button is released anywhere in window
            const onWindowMouseUp = function(event) {
                // reset original mouse position
                clickMouseX = undefined;
                clickMouseY = undefined;
                clickTranslateX = undefined;
                clickTranslateY = undefined;

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('mouseup', onWindowMouseUp);
            };

            // when mouse is moved anywhere in window
            const onWindowMouseMove = function(event) {
                if (
                    clickMouseX === undefined ||
                    clickMouseY === undefined ||
                    clickTranslateX === undefined ||
                    clickTranslateY === undefined
                )
                    return;

                // offset image based on original and current mouse position
                translateX = clickTranslateX + window.mouseX - clickMouseX;
                translateY = clickTranslateY + window.mouseY - clickMouseY;
                updateTransform();
                event.preventDefault();

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('mousemove', onWindowMouseMove);
            };

            // when window is resized
            const onWindowResize = function(event) {
                fit();

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('resize', onWindowResize);
            };

            // attach the necessary event listeners
            img.addEventListener('dblclick', fit);
            img.addEventListener('mousedown', onImageMouseDown);
            container.addEventListener('wheel', onContainerWheel);
            container.addEventListener('mousedown', onContainerClick);
            container.addEventListener('touchstart', onContainerClick);
            window.addEventListener('mouseup', onWindowMouseUp);
            window.addEventListener('mousemove', onWindowMouseMove);
            window.addEventListener('resize', onWindowResize);

            // run fit() after lightbox atttached to document and <img> Loaded
            // so needed container and img dimensions available
            img.addEventListener('load', fit);

            return img;
        }

        // make caption
        function makeCaption(img) {
            const caption = document.createElement('div');
            caption.id = 'lightbox_caption';
            const captionSource = img.nextElementSibling;
            if (captionSource.tagName.toLowerCase() === 'figcaption') {
                const captionCopy = makeCopy(captionSource);
                caption.innerHTML = captionCopy.innerHTML;
            }

            caption.addEventListener('touchstart', function(event) {
                event.stopPropagation();
            });

            return caption;
        }

        // make carbon copy of html dom element
        function makeCopy(source) {
            const sourceCopy = source.cloneNode(true);

            // delete elements marked with ignore (eg anchor and jump buttons)
            const deleteFromCopy = sourceCopy.querySelectorAll(
                '[data-ignore="true"]'
            );
            for (const element of deleteFromCopy)
                element.remove();

            // delete certain element attributes
            const attributes = [
                'id',
                'data-collapsed',
                'data-selected',
                'data-highlighted',
                'data-glow'
            ];
            for (const attribute of attributes) {
                sourceCopy.removeAttribute(attribute);
                const elements = sourceCopy.querySelectorAll(
                    '[' + attribute + ']'
                );
                for (const element of elements)
                    element.removeAttribute(attribute);
            }

            return sourceCopy;
        }

        // make button to jump to previous image in document
        function makePrevButton(img) {
            const prevButton = document.createElement('button');
            prevButton.id = 'lightbox_prev_button';
            prevButton.title = 'Jump to the previous image in the document [←]';
            prevButton.classList.add('icon_button', 'lightbox_button');
            prevButton.innerHTML = document.querySelector(
                '.icon_caret_left'
            ).innerHTML;

            // attach click listeners to button
            prevButton.addEventListener('click', function() {
                getPrevImg(img).click();
            });

            return prevButton;
        }

        // make button to jump to next image in document
        function makeNextButton(img) {
            const nextButton = document.createElement('button');
            nextButton.id = 'lightbox_next_button';
            nextButton.title = 'Jump to the next image in the document [→]';
            nextButton.classList.add('icon_button', 'lightbox_button');
            nextButton.innerHTML = document.querySelector(
                '.icon_caret_right'
            ).innerHTML;

            // attach click listeners to button
            nextButton.addEventListener('click', function() {
                getNextImg(img).click();
            });

            return nextButton;
        }

        // get previous image in document
        function getPrevImg(img) {
            const imgs = document.querySelectorAll('.lightbox_document_img');

            // find index of provided img
            let index;
            for (index = 0; index < imgs.length; index++) {
                if (imgs[index] === img)
                    break;
            }


            // wrap index to other side if < 1
            if (index - 1 >= 0)
                index--;
            else
                index = imgs.length - 1;
            return imgs[index];
        }

        // get next image in document
        function getNextImg(img) {
            const imgs = document.querySelectorAll('.lightbox_document_img');

            // find index of provided img
            let index;
            for (index = 0; index < imgs.length; index++) {
                if (imgs[index] === img)
                    break;
            }


            // wrap index to other side if > total
            if (index + 1 <= imgs.length - 1)
                index++;
            else
                index = 0;
            return imgs[index];
        }

        // close lightbox
        function closeLightbox() {
            focusBody();

            const lightbox = document.getElementById('lightbox_overlay');
            if (lightbox)
                lightbox.remove();
        }

        // make all elements behind lightbox non-focusable
        function blurBody(overlay) {
            const all = document.querySelectorAll('*');
            for (const element of all)
                element.tabIndex = -1;
            document.body.classList.add('body_no_scroll');
        }

        // make all elements focusable again
        function focusBody() {
            const all = document.querySelectorAll('*');
            for (const element of all)
                element.removeAttribute('tabIndex');
            document.body.classList.remove('body_no_scroll');
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
    <!-- modified from: https://fontawesome.com/icons/caret-left -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
        ></path>
    </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
    <!-- modified from: https://fontawesome.com/icons/caret-right -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        ></path>
    </svg>
</template>
<!-- attributes plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin allows arbitrary HTML attributes to be attached
        // to (almost) any element. Place an HTML comment inside or next to the
        // desired element in the format <!-- $attribute="value" -->

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'attributes';

        // default plugin options
        const options = {
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // get list of comments in document
            const comments = findComments();

            for(const comment of comments)
                if (comment.parentElement)
                    addAttributes(
                        comment.parentElement,
                        comment.nodeValue.trim()
                    );
        }

        // add html attributes to specified element based on string of 
        // html attributes and values
        function addAttributes(element, text) {
            // regex's for finding attribute/value pairs in the format of
            // attribute="value" or attribute='value
            const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
            const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

            // loop through attribute/value pairs
            let match;
            while(match = text.match(regex2) || text.match(regex1)) {
                // get attribute and value from regex capture groups
                let attribute = match[1];
                let value = match[2];

                // remove from string
                text = text.substring(match.index + match[0].length);

                if (!attribute || !value)
                    break;

                // set attribute of parent element
                try {
                    element.setAttribute(attribute, value);
                } catch(error) {
                    console.log(error);
                }

                // special case for colspan
                if (attribute === 'colspan')
                    removeTableCells(element, value);
            }
        }

        // get list of comment elements in document
        function findComments() {
            const comments = [];

            // iterate over comment nodes in document
            function acceptNode(node) {
                return NodeFilter.FILTER_ACCEPT;
            }
            const iterator = document.createNodeIterator(
                document.body,
                NodeFilter.SHOW_COMMENT,
                acceptNode
            );
            let node;
            while(node = iterator.nextNode())
                comments.push(node);

            return comments;
        }

        // remove certain number of cells after specified cell
        function removeTableCells(cell, number) {
            number = parseInt(number);
            if (!number)
                return;

            // remove elements
            for(; number > 1; number--) {
                if (cell.nextElementSibling)
                    cell.nextElementSibling.remove();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>
<!-- math plugin configuration -->

<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        "CommonHTML": { linebreaks: { automatic: true } },
        "HTML-CSS": { linebreaks: { automatic: true } },
        "SVG": { linebreaks: { automatic: true } },
        "fast-preview": { disabled: true }
    });
</script>

<!-- math plugin -->

<script src='https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML'>
    // /////////////////////////
    // DESCRIPTION
    // /////////////////////////

    // This third-party plugin 'MathJax' allows the proper rendering of
    // math/equations written in LaTeX.

    // https://www.mathjax.org/
</script>
<!-- annotations plugin configuration -->

<script>
    window.hypothesisConfig = function() {
        return {
            branding: {
                accentColor: '#2196f3',
                appBackgroundColor: '#f8f8f8',
                ctaBackgroundColor: '#f8f8f8',
                ctaTextColor: '#000000',
                selectionFontFamily: 'Open Sans, Helvetica, sans serif',
                annotationFontFamily: 'Open Sans, Helvetica, sans serif'
            }
        };
    };
</script>

<!-- annotations plugin -->

<script src='https://hypothes.is/embed.js'>
    // /////////////////////////
    // DESCRIPTION
    // /////////////////////////

    // This third-party plugin 'Hypothesis' allows public annotation of the
    // manuscript.

    // https://web.hypothes.is/
</script>
<!-- analytics plugin -->

<!-- copy and paste code from Google Analytics or similar service here -->
</body>
</html>
