## Population Sampling


## Whole Genome Sequencing and Alignment

Mosquitoes were submitted to the Ag1000G consortium in 23 samplesets (ref to table).

A total of XXXX individual mosquitoes were sequenced on either Illumina hi-seq (n=XXX) or Illumina 10X (n=XXX) to a target coverage of 30X. 

(relevant sequencing statistics)
- how many reads?

Reads were aligned to the AgamP4 reference genome using `bwa` version `XXX`.

(other steps)

Single nucleotide polymophisms were called using GATK UnifiedGenotyper `v3.2`.

Sample genotypes were called independently, in genotyping mode given all possible alleles at each site, allowing complete parallelisation.

Coverage considered at individual sites was capped at 250.

All samples successfully completed this pipeline and entered the sample quality control process.

### Sample QC

The sample quality control (QC) process was composed of three stages. 

Firstly the exclusion of samples where sequencing was of insufficient quality to accurately call genotypes across the whole genome.

Secondly, exclusion of samples where higher quality technical replicates were available, and pairs that unexpectedly showed high levels of similarity.

The AG1000G-X sampleset, made up of laboratory experimental crosses, was exempted from the requirements of this stage due to familial similarity and high levels of inbreeding.

Thirdly, exclusion of samples with cryptic relationships to other samples, that could not be explained parsimoniously, and were therefore likely to be sample mix ups.

Of the 4000 samples, XXX passed the first stage of QC.

Exclusions were due to fraction of the genome covered < 0.1 (n=99), sex call being undetermined, median coverage etc.

For the second stage, all technical replicate pairs were within samplesets, and we computed pairwise Euclidean genetic distance between all possible pairs within a sampleset.

In total XXX samples in technical replicate pairs were excluded in favour of better quality samples, based on the fraction of the genome covered at YYY.

XXX samples (YY pairs) were excluded as although they were marked as technical replicates they exceeded the maximum allowable distance of XXX.

XXX samples (YY pairs) were excluded owing to a distance of XXX, below the allowed distance of XXX.

The third stage used principal component analysis to identify and exclude individual samples that were clear outliers from their sampleset.

3 samples were excluded as they respectively dominated the first principal components, indicating high divergence from all other samples and therefore likely members of other Anopheline species.

A further XX samples were excluded as potential sample mix ups.

Following all sample QC steps, 3,483 samples (78.X%) remained from the original cohort of 4100.

### Coverage

Summary of site coverage post QC exclusions.

### SNP filtering and quality

### Genome accessibility

### SNP discovery



## Species Assignment

## Population Structure

## Genetic Diversity within Populations

## Insecticide Resistance

## Gene Drive
