{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import dask.array as da"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/lib/python3.7/site-packages/dask/dataframe/utils.py:14: FutureWarning: pandas.util.testing is deprecated. Use the functions in the public API at pandas.testing instead.\n",
      "  import pandas.util.testing as tm\n"
     ]
    }
   ],
   "source": [
    "import ag2, ag3\n",
    "\n",
    "ag2_release = ag2.release_data()\n",
    "ag3_release = ag3.release_data()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def load_accessibility(chrom, mask_id='gamb_colu'):\n",
    "    \n",
    "    if mask_id == \"phase2\":\n",
    "\n",
    "        phase2_is_accessible = ag2_release.load_is_accessible(chrom)\n",
    "        filter_n = ag2_release.load_filter_n(chrom)\n",
    "        phase2_is_accessible_nonN = da.compress(\n",
    "            ~filter_n, phase2_is_accessible, axis=0)\n",
    "\n",
    "        # Compute chunk sizes avoids this error when c\n",
    "        # concatenating arrays: ValueError: Arrays chunk sizes are unknown: (nan,)\n",
    "        phase2_is_accessible_nonN.compute_chunk_sizes() \n",
    "        return phase2_is_accessible_nonN\n",
    "    \n",
    "    else:\n",
    "        return ag3_release.load_mask(chrom, mask_id=mask_id)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## For given genomic regions, summarize accessibility\n",
    "\n",
    "### What about \"N\" sites?\n",
    "\n",
    "N sites are currently NOT included. \n",
    "\n",
    "If a feature contains many N sites- these are not considered in the % calculation.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "features =  \"\"\"\n",
    "2La     inv     2L      20528089        42165182\n",
    "2Rb     inv     2R      19444433        26313071\n",
    "2Rj     inv     2R      3262186 15750717\n",
    "2Rc     inv     2R      26780000        31450000\n",
    "2Ru     inv     2R      31480000        35500000\n",
    "2Rd     inv     2R      33575891        41360919\n",
    "2Rk     inv     2R      25146360        30717395\n",
    "CHX     chro    X       20009764        24393108\n",
    "CH2R    chro    2R      58984778        61545105\n",
    "CH2L    chro    2L      1       2431617\n",
    "PEU2L   chro    2L      2487770 5042389\n",
    "IH2L    chro    2L      5078962 5788875\n",
    "IH3R    chro    3R      38988757        41860198\n",
    "CH3R    chro    3R      52161877        53200684\n",
    "CH3L    chro    3L      1       1815119\n",
    "PEU3L   chro    3L      1896830 4235209\n",
    "IH3L    chro    3L      4264713 5031692\n",
    "\"\"\"\n",
    "# via Thiago Antao"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "features_df = pd.DataFrame(\n",
    "    [x.split() for x in features.strip().split(\"\\n\")], columns=[\"id\", \"type\", \"seq_id\", \"start\", \"stop\"])\n",
    "\n",
    "features_df.start = features_df.start.astype(np.int32)\n",
    "features_df.stop = features_df.stop.astype(np.int32)\n",
    "    \n",
    "    #dtype={\"start\": np.int32, \"stop\": np.int32})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "features_df[\"type\"] = features_df.type.map({\"inv\": \"inversion\", \"chro\": \"chromatin\"})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "chromosomes = [\"2R\", \"2L\", \"3R\", \"3L\", \"X\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "avail_masks = [\"gamb_colu\", \"arab\", \"gamb_colu_arab\", \"phase2\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "features_by_chrom = features_df.groupby(\"seq_id\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "import allel"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "out = pd.DataFrame(columns=features_df.columns)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "out_list = []\n",
    "\n",
    "for chrom, df in features_by_chrom:\n",
    "    \n",
    "    pos = allel.SortedIndex(ag3_release.load_variants(chrom).compute())\n",
    "    \n",
    "    for mask_name in avail_masks:\n",
    "        \n",
    "        mask = load_accessibility(chrom, mask_name)\n",
    "        assert mask.shape == pos.shape\n",
    "        \n",
    "        for _, row in df.iterrows():\n",
    "            \n",
    "            ix = pos.locate_range(row.start, row.stop)\n",
    "            \n",
    "            # computing the mean is ok- as it allows us to silently handle Ns, which are ignored.\n",
    "            access = mask[ix].mean().compute()\n",
    "            out_list.append([mask_name, chrom, row.id, row.type, row.start, row.stop, access])\n",
    "            \n",
    "        out_list.append([mask_name, chrom, chrom, \"contig\", 1, pos[-1], mask.mean().compute()])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "result_df = pd.DataFrame(out_list, columns=[\"mask_id\", \"chrom\", \"feature_name\", \"type\", \"start\", \"stop\", \"frac_accessible\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "result_df[\"type\"] = pd.Categorical(result_df.type, [\"contig\", \"inversion\", \"chromatin\"], ordered=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "result_df.sort_values([\"mask_id\", \"type\", \"chrom\", \"start\"], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>mask_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>feature_name</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>frac_accessible</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>arab</td>\n",
       "      <td>2L</td>\n",
       "      <td>2L</td>\n",
       "      <td>contig</td>\n",
       "      <td>1</td>\n",
       "      <td>49364325</td>\n",
       "      <td>0.725773</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>arab</td>\n",
       "      <td>2R</td>\n",
       "      <td>2R</td>\n",
       "      <td>contig</td>\n",
       "      <td>1</td>\n",
       "      <td>61545105</td>\n",
       "      <td>0.735970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>arab</td>\n",
       "      <td>3L</td>\n",
       "      <td>3L</td>\n",
       "      <td>contig</td>\n",
       "      <td>1</td>\n",
       "      <td>41963435</td>\n",
       "      <td>0.696811</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>arab</td>\n",
       "      <td>3R</td>\n",
       "      <td>3R</td>\n",
       "      <td>contig</td>\n",
       "      <td>1</td>\n",
       "      <td>53200684</td>\n",
       "      <td>0.697803</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>arab</td>\n",
       "      <td>X</td>\n",
       "      <td>X</td>\n",
       "      <td>contig</td>\n",
       "      <td>1</td>\n",
       "      <td>24393108</td>\n",
       "      <td>0.533184</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   mask_id chrom feature_name    type  start      stop  frac_accessible\n",
       "9     arab    2L           2L  contig      1  49364325         0.725773\n",
       "35    arab    2R           2R  contig      1  61545105         0.735970\n",
       "59    arab    3L           3L  contig      1  41963435         0.696811\n",
       "73    arab    3R           3R  contig      1  53200684         0.697803\n",
       "83    arab     X            X  contig      1  24393108         0.533184"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "for label, df in result_df.groupby(\"mask_id\"):\n",
    "    df.drop(\"mask_id\", axis=1).to_csv(f\"../content/tables/accessibility/accessibility_features_{label}.csv\", float_format=\"%.3f\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "output_df = result_df.pivot(\n",
    "    index=[\"chrom\", \"feature_name\", \"type\", \"start\", \"stop\"], \n",
    "    columns=\"mask_id\", \n",
    "    values=\"frac_accessible\").reset_index()\n",
    "output_df.index.name = \"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "output_df.to_csv(\n",
    "    \"../content/tables/accessibility/table4_accessibility_features_by_mask.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th>mask_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th>feature_name</th>\n",
       "      <th>type</th>\n",
       "      <th>start</th>\n",
       "      <th>stop</th>\n",
       "      <th>arab</th>\n",
       "      <th>gamb_colu</th>\n",
       "      <th>gamb_colu_arab</th>\n",
       "      <th>phase2</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>2L</td>\n",
       "      <td>2L</td>\n",
       "      <td>contig</td>\n",
       "      <td>1</td>\n",
       "      <td>49364325</td>\n",
       "      <td>0.725773</td>\n",
       "      <td>0.741980</td>\n",
       "      <td>0.670365</td>\n",
       "      <td>0.580992</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>2L</td>\n",
       "      <td>2La</td>\n",
       "      <td>inversion</td>\n",
       "      <td>20528089</td>\n",
       "      <td>42165182</td>\n",
       "      <td>0.753209</td>\n",
       "      <td>0.765582</td>\n",
       "      <td>0.701509</td>\n",
       "      <td>0.511216</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2L</td>\n",
       "      <td>CH2L</td>\n",
       "      <td>chromatin</td>\n",
       "      <td>1</td>\n",
       "      <td>2431617</td>\n",
       "      <td>0.298523</td>\n",
       "      <td>0.377953</td>\n",
       "      <td>0.225203</td>\n",
       "      <td>0.200037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>2L</td>\n",
       "      <td>IH2L</td>\n",
       "      <td>chromatin</td>\n",
       "      <td>5078962</td>\n",
       "      <td>5788875</td>\n",
       "      <td>0.338672</td>\n",
       "      <td>0.341595</td>\n",
       "      <td>0.264138</td>\n",
       "      <td>0.246504</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>2L</td>\n",
       "      <td>PEU2L</td>\n",
       "      <td>chromatin</td>\n",
       "      <td>2487770</td>\n",
       "      <td>5042389</td>\n",
       "      <td>0.642531</td>\n",
       "      <td>0.643251</td>\n",
       "      <td>0.575857</td>\n",
       "      <td>0.575392</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2R</td>\n",
       "      <td>2R</td>\n",
       "      <td>contig</td>\n",
       "      <td>1</td>\n",
       "      <td>61545105</td>\n",
       "      <td>0.735970</td>\n",
       "      <td>0.739031</td>\n",
       "      <td>0.674539</td>\n",
       "      <td>0.657463</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>2R</td>\n",
       "      <td>2Rb</td>\n",
       "      <td>inversion</td>\n",
       "      <td>19444433</td>\n",
       "      <td>26313071</td>\n",
       "      <td>0.809948</td>\n",
       "      <td>0.830475</td>\n",
       "      <td>0.759510</td>\n",
       "      <td>0.734145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>2R</td>\n",
       "      <td>2Rc</td>\n",
       "      <td>inversion</td>\n",
       "      <td>26780000</td>\n",
       "      <td>31450000</td>\n",
       "      <td>0.801247</td>\n",
       "      <td>0.815872</td>\n",
       "      <td>0.745410</td>\n",
       "      <td>0.741728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>2R</td>\n",
       "      <td>2Rd</td>\n",
       "      <td>inversion</td>\n",
       "      <td>33575891</td>\n",
       "      <td>41360919</td>\n",
       "      <td>0.716491</td>\n",
       "      <td>0.693729</td>\n",
       "      <td>0.635669</td>\n",
       "      <td>0.615729</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>2R</td>\n",
       "      <td>2Rj</td>\n",
       "      <td>inversion</td>\n",
       "      <td>3262186</td>\n",
       "      <td>15750717</td>\n",
       "      <td>0.786997</td>\n",
       "      <td>0.793172</td>\n",
       "      <td>0.730491</td>\n",
       "      <td>0.696147</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>2R</td>\n",
       "      <td>2Rk</td>\n",
       "      <td>inversion</td>\n",
       "      <td>25146360</td>\n",
       "      <td>30717395</td>\n",
       "      <td>0.813327</td>\n",
       "      <td>0.831740</td>\n",
       "      <td>0.759192</td>\n",
       "      <td>0.752275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>2R</td>\n",
       "      <td>2Ru</td>\n",
       "      <td>inversion</td>\n",
       "      <td>31480000</td>\n",
       "      <td>35500000</td>\n",
       "      <td>0.676314</td>\n",
       "      <td>0.651264</td>\n",
       "      <td>0.588420</td>\n",
       "      <td>0.561496</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>2R</td>\n",
       "      <td>CH2R</td>\n",
       "      <td>chromatin</td>\n",
       "      <td>58984778</td>\n",
       "      <td>61545105</td>\n",
       "      <td>0.447814</td>\n",
       "      <td>0.504804</td>\n",
       "      <td>0.367949</td>\n",
       "      <td>0.295004</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>3L</td>\n",
       "      <td>3L</td>\n",
       "      <td>contig</td>\n",
       "      <td>1</td>\n",
       "      <td>41963435</td>\n",
       "      <td>0.696811</td>\n",
       "      <td>0.704341</td>\n",
       "      <td>0.634700</td>\n",
       "      <td>0.612056</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>3L</td>\n",
       "      <td>CH3L</td>\n",
       "      <td>chromatin</td>\n",
       "      <td>1</td>\n",
       "      <td>1815119</td>\n",
       "      <td>0.273445</td>\n",
       "      <td>0.396110</td>\n",
       "      <td>0.221327</td>\n",
       "      <td>0.262528</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>3L</td>\n",
       "      <td>IH3L</td>\n",
       "      <td>chromatin</td>\n",
       "      <td>4264713</td>\n",
       "      <td>5031692</td>\n",
       "      <td>0.264274</td>\n",
       "      <td>0.350127</td>\n",
       "      <td>0.211377</td>\n",
       "      <td>0.195048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>3L</td>\n",
       "      <td>PEU3L</td>\n",
       "      <td>chromatin</td>\n",
       "      <td>1896830</td>\n",
       "      <td>4235209</td>\n",
       "      <td>0.721518</td>\n",
       "      <td>0.758577</td>\n",
       "      <td>0.678704</td>\n",
       "      <td>0.702815</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>3R</td>\n",
       "      <td>3R</td>\n",
       "      <td>contig</td>\n",
       "      <td>1</td>\n",
       "      <td>53200684</td>\n",
       "      <td>0.697803</td>\n",
       "      <td>0.712270</td>\n",
       "      <td>0.639346</td>\n",
       "      <td>0.621070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>3R</td>\n",
       "      <td>CH3R</td>\n",
       "      <td>chromatin</td>\n",
       "      <td>52161877</td>\n",
       "      <td>53200684</td>\n",
       "      <td>0.254093</td>\n",
       "      <td>0.319606</td>\n",
       "      <td>0.190784</td>\n",
       "      <td>0.172572</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>3R</td>\n",
       "      <td>IH3R</td>\n",
       "      <td>chromatin</td>\n",
       "      <td>38988757</td>\n",
       "      <td>41860198</td>\n",
       "      <td>0.417199</td>\n",
       "      <td>0.464178</td>\n",
       "      <td>0.335847</td>\n",
       "      <td>0.315776</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>X</td>\n",
       "      <td>CHX</td>\n",
       "      <td>chromatin</td>\n",
       "      <td>20009764</td>\n",
       "      <td>24393108</td>\n",
       "      <td>0.192680</td>\n",
       "      <td>0.304518</td>\n",
       "      <td>0.130716</td>\n",
       "      <td>0.161594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>X</td>\n",
       "      <td>X</td>\n",
       "      <td>contig</td>\n",
       "      <td>1</td>\n",
       "      <td>24393108</td>\n",
       "      <td>0.533184</td>\n",
       "      <td>0.699703</td>\n",
       "      <td>0.479696</td>\n",
       "      <td>0.624621</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "mask_id chrom feature_name       type     start      stop      arab  \\\n",
       "                                                                      \n",
       "0          2L           2L     contig         1  49364325  0.725773   \n",
       "1          2L          2La  inversion  20528089  42165182  0.753209   \n",
       "2          2L         CH2L  chromatin         1   2431617  0.298523   \n",
       "3          2L         IH2L  chromatin   5078962   5788875  0.338672   \n",
       "4          2L        PEU2L  chromatin   2487770   5042389  0.642531   \n",
       "5          2R           2R     contig         1  61545105  0.735970   \n",
       "6          2R          2Rb  inversion  19444433  26313071  0.809948   \n",
       "7          2R          2Rc  inversion  26780000  31450000  0.801247   \n",
       "8          2R          2Rd  inversion  33575891  41360919  0.716491   \n",
       "9          2R          2Rj  inversion   3262186  15750717  0.786997   \n",
       "10         2R          2Rk  inversion  25146360  30717395  0.813327   \n",
       "11         2R          2Ru  inversion  31480000  35500000  0.676314   \n",
       "12         2R         CH2R  chromatin  58984778  61545105  0.447814   \n",
       "13         3L           3L     contig         1  41963435  0.696811   \n",
       "14         3L         CH3L  chromatin         1   1815119  0.273445   \n",
       "15         3L         IH3L  chromatin   4264713   5031692  0.264274   \n",
       "16         3L        PEU3L  chromatin   1896830   4235209  0.721518   \n",
       "17         3R           3R     contig         1  53200684  0.697803   \n",
       "18         3R         CH3R  chromatin  52161877  53200684  0.254093   \n",
       "19         3R         IH3R  chromatin  38988757  41860198  0.417199   \n",
       "20          X          CHX  chromatin  20009764  24393108  0.192680   \n",
       "21          X            X     contig         1  24393108  0.533184   \n",
       "\n",
       "mask_id  gamb_colu  gamb_colu_arab    phase2  \n",
       "                                              \n",
       "0         0.741980        0.670365  0.580992  \n",
       "1         0.765582        0.701509  0.511216  \n",
       "2         0.377953        0.225203  0.200037  \n",
       "3         0.341595        0.264138  0.246504  \n",
       "4         0.643251        0.575857  0.575392  \n",
       "5         0.739031        0.674539  0.657463  \n",
       "6         0.830475        0.759510  0.734145  \n",
       "7         0.815872        0.745410  0.741728  \n",
       "8         0.693729        0.635669  0.615729  \n",
       "9         0.793172        0.730491  0.696147  \n",
       "10        0.831740        0.759192  0.752275  \n",
       "11        0.651264        0.588420  0.561496  \n",
       "12        0.504804        0.367949  0.295004  \n",
       "13        0.704341        0.634700  0.612056  \n",
       "14        0.396110        0.221327  0.262528  \n",
       "15        0.350127        0.211377  0.195048  \n",
       "16        0.758577        0.678704  0.702815  \n",
       "17        0.712270        0.639346  0.621070  \n",
       "18        0.319606        0.190784  0.172572  \n",
       "19        0.464178        0.335847  0.315776  \n",
       "20        0.304518        0.130716  0.161594  \n",
       "21        0.699703        0.479696  0.624621  "
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "output_df"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Over the exome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "exon_data = {}\n",
    "wg_data = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "gff_geneset = \"../../Anopheles-gambiae-PEST_BASEFEATURES_AgamP4.12.gff3.gz\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "for chrom in chromosomes:\n",
    "    \n",
    "    exons = gene_features = allel.gff3_to_dataframe(\n",
    "        gff_geneset, \n",
    "        attributes=[\"ID\"]).query(\"(seqid == @chrom) & (type == 'exon')\")\n",
    "    \n",
    "    pos = allel.SortedIndex(ag3_release.load_variants(chrom).compute())    \n",
    "    \n",
    "    for mask_name in avail_masks:\n",
    "        \n",
    "        mask = load_accessibility(chrom, mask_name).compute()\n",
    "        \n",
    "        locv, _ = pos.locate_intersection_ranges(exons.start, exons.end)\n",
    "        \n",
    "        v = np.compress(locv, mask, axis=0)\n",
    "        \n",
    "        exon_data[mask_name, chrom] = pd.Series([v.sum(), locv.sum()], index=[\"n_accessible\", \"n_total_bases\"], dtype=np.int32)\n",
    "        \n",
    "        wg_data[mask_name, chrom] = pd.Series([mask.sum(), mask.shape[0]], index=[\"n_accessible\", \"n_total_bases\"], dtype=np.int32)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## whole exome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>n_accessible</th>\n",
       "      <th>n_total_bases</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mask_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>gamb_colu</th>\n",
       "      <th>2R</th>\n",
       "      <td>8326769</td>\n",
       "      <td>9434770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>arab</th>\n",
       "      <th>2R</th>\n",
       "      <td>8251192</td>\n",
       "      <td>9434770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu_arab</th>\n",
       "      <th>2R</th>\n",
       "      <td>7796125</td>\n",
       "      <td>9434770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phase2</th>\n",
       "      <th>2R</th>\n",
       "      <td>8327539</td>\n",
       "      <td>9434770</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu</th>\n",
       "      <th>2L</th>\n",
       "      <td>6718168</td>\n",
       "      <td>7423546</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>arab</th>\n",
       "      <th>2L</th>\n",
       "      <td>6578902</td>\n",
       "      <td>7423546</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu_arab</th>\n",
       "      <th>2L</th>\n",
       "      <td>6284272</td>\n",
       "      <td>7423546</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phase2</th>\n",
       "      <th>2L</th>\n",
       "      <td>6373764</td>\n",
       "      <td>7423546</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu</th>\n",
       "      <th>3R</th>\n",
       "      <td>6221607</td>\n",
       "      <td>7043977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>arab</th>\n",
       "      <th>3R</th>\n",
       "      <td>6096759</td>\n",
       "      <td>7043977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu_arab</th>\n",
       "      <th>3R</th>\n",
       "      <td>5810909</td>\n",
       "      <td>7043977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phase2</th>\n",
       "      <th>3R</th>\n",
       "      <td>6090867</td>\n",
       "      <td>7043977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu</th>\n",
       "      <th>3L</th>\n",
       "      <td>4899622</td>\n",
       "      <td>5729235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>arab</th>\n",
       "      <th>3L</th>\n",
       "      <td>4864846</td>\n",
       "      <td>5729235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu_arab</th>\n",
       "      <th>3L</th>\n",
       "      <td>4576239</td>\n",
       "      <td>5729235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phase2</th>\n",
       "      <th>3L</th>\n",
       "      <td>4814627</td>\n",
       "      <td>5729235</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu</th>\n",
       "      <th>X</th>\n",
       "      <td>2725468</td>\n",
       "      <td>3269838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>arab</th>\n",
       "      <th>X</th>\n",
       "      <td>2565284</td>\n",
       "      <td>3269838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu_arab</th>\n",
       "      <th>X</th>\n",
       "      <td>2305933</td>\n",
       "      <td>3269838</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phase2</th>\n",
       "      <th>X</th>\n",
       "      <td>2875899</td>\n",
       "      <td>3269838</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      n_accessible  n_total_bases\n",
       "mask_id        chrom                             \n",
       "gamb_colu      2R          8326769        9434770\n",
       "arab           2R          8251192        9434770\n",
       "gamb_colu_arab 2R          7796125        9434770\n",
       "phase2         2R          8327539        9434770\n",
       "gamb_colu      2L          6718168        7423546\n",
       "arab           2L          6578902        7423546\n",
       "gamb_colu_arab 2L          6284272        7423546\n",
       "phase2         2L          6373764        7423546\n",
       "gamb_colu      3R          6221607        7043977\n",
       "arab           3R          6096759        7043977\n",
       "gamb_colu_arab 3R          5810909        7043977\n",
       "phase2         3R          6090867        7043977\n",
       "gamb_colu      3L          4899622        5729235\n",
       "arab           3L          4864846        5729235\n",
       "gamb_colu_arab 3L          4576239        5729235\n",
       "phase2         3L          4814627        5729235\n",
       "gamb_colu      X           2725468        3269838\n",
       "arab           X           2565284        3269838\n",
       "gamb_colu_arab X           2305933        3269838\n",
       "phase2         X           2875899        3269838"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "exon_counts = pd.DataFrame.from_dict(exon_data, orient=\"index\")\n",
    "exon_counts.index.names = [\"mask_id\", \"chrom\"]\n",
    "exon_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>n_accessible</th>\n",
       "      <th>n_total_bases</th>\n",
       "      <th>frac_accessible</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mask_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>arab</th>\n",
       "      <td>28356983</td>\n",
       "      <td>32901366</td>\n",
       "      <td>0.861879</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu</th>\n",
       "      <td>28891634</td>\n",
       "      <td>32901366</td>\n",
       "      <td>0.878129</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu_arab</th>\n",
       "      <td>26773478</td>\n",
       "      <td>32901366</td>\n",
       "      <td>0.813750</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phase2</th>\n",
       "      <td>28482696</td>\n",
       "      <td>32901366</td>\n",
       "      <td>0.865699</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                n_accessible  n_total_bases  frac_accessible\n",
       "mask_id                                                     \n",
       "arab                28356983       32901366         0.861879\n",
       "gamb_colu           28891634       32901366         0.878129\n",
       "gamb_colu_arab      26773478       32901366         0.813750\n",
       "phase2              28482696       32901366         0.865699"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "whole_exome_sum = exon_counts.groupby(level=0).agg(sum)\n",
    "whole_exome_sum[\"frac_accessible\"] = whole_exome_sum[\"n_accessible\"] / whole_exome_sum[\"n_total_bases\"]\n",
    "whole_exome_sum"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## whole genome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>n_accessible</th>\n",
       "      <th>n_total_bases</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mask_id</th>\n",
       "      <th>chrom</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>gamb_colu</th>\n",
       "      <th>2R</th>\n",
       "      <td>44439759</td>\n",
       "      <td>60132453</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>arab</th>\n",
       "      <th>2R</th>\n",
       "      <td>44255681</td>\n",
       "      <td>60132453</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu_arab</th>\n",
       "      <th>2R</th>\n",
       "      <td>40561667</td>\n",
       "      <td>60132453</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phase2</th>\n",
       "      <th>2R</th>\n",
       "      <td>39534863</td>\n",
       "      <td>60132453</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu</th>\n",
       "      <th>2L</th>\n",
       "      <td>36005131</td>\n",
       "      <td>48525747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>arab</th>\n",
       "      <th>2L</th>\n",
       "      <td>35218697</td>\n",
       "      <td>48525747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu_arab</th>\n",
       "      <th>2L</th>\n",
       "      <td>32529983</td>\n",
       "      <td>48525747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phase2</th>\n",
       "      <th>2L</th>\n",
       "      <td>28193050</td>\n",
       "      <td>48525747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu</th>\n",
       "      <th>3R</th>\n",
       "      <td>37199402</td>\n",
       "      <td>52226568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>arab</th>\n",
       "      <th>3R</th>\n",
       "      <td>36443830</td>\n",
       "      <td>52226568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu_arab</th>\n",
       "      <th>3R</th>\n",
       "      <td>33390862</td>\n",
       "      <td>52226568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phase2</th>\n",
       "      <th>3R</th>\n",
       "      <td>32436343</td>\n",
       "      <td>52226568</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu</th>\n",
       "      <th>3L</th>\n",
       "      <td>28707856</td>\n",
       "      <td>40758473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>arab</th>\n",
       "      <th>3L</th>\n",
       "      <td>28400950</td>\n",
       "      <td>40758473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu_arab</th>\n",
       "      <th>3L</th>\n",
       "      <td>25869385</td>\n",
       "      <td>40758473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phase2</th>\n",
       "      <th>3L</th>\n",
       "      <td>24946488</td>\n",
       "      <td>40758473</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu</th>\n",
       "      <th>X</th>\n",
       "      <td>16362809</td>\n",
       "      <td>23385349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>arab</th>\n",
       "      <th>X</th>\n",
       "      <td>12468702</td>\n",
       "      <td>23385349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu_arab</th>\n",
       "      <th>X</th>\n",
       "      <td>11217869</td>\n",
       "      <td>23385349</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phase2</th>\n",
       "      <th>X</th>\n",
       "      <td>14606989</td>\n",
       "      <td>23385349</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                      n_accessible  n_total_bases\n",
       "mask_id        chrom                             \n",
       "gamb_colu      2R         44439759       60132453\n",
       "arab           2R         44255681       60132453\n",
       "gamb_colu_arab 2R         40561667       60132453\n",
       "phase2         2R         39534863       60132453\n",
       "gamb_colu      2L         36005131       48525747\n",
       "arab           2L         35218697       48525747\n",
       "gamb_colu_arab 2L         32529983       48525747\n",
       "phase2         2L         28193050       48525747\n",
       "gamb_colu      3R         37199402       52226568\n",
       "arab           3R         36443830       52226568\n",
       "gamb_colu_arab 3R         33390862       52226568\n",
       "phase2         3R         32436343       52226568\n",
       "gamb_colu      3L         28707856       40758473\n",
       "arab           3L         28400950       40758473\n",
       "gamb_colu_arab 3L         25869385       40758473\n",
       "phase2         3L         24946488       40758473\n",
       "gamb_colu      X          16362809       23385349\n",
       "arab           X          12468702       23385349\n",
       "gamb_colu_arab X          11217869       23385349\n",
       "phase2         X          14606989       23385349"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wg_counts = pd.DataFrame.from_dict(wg_data, orient=\"index\")\n",
    "wg_counts.index.names = [\"mask_id\", \"chrom\"]\n",
    "wg_counts"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>n_accessible</th>\n",
       "      <th>n_total_bases</th>\n",
       "      <th>frac_accessible</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mask_id</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>arab</th>\n",
       "      <td>156787860</td>\n",
       "      <td>225028590</td>\n",
       "      <td>0.696746</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu</th>\n",
       "      <td>162714957</td>\n",
       "      <td>225028590</td>\n",
       "      <td>0.723086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>gamb_colu_arab</th>\n",
       "      <td>143569766</td>\n",
       "      <td>225028590</td>\n",
       "      <td>0.638007</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>phase2</th>\n",
       "      <td>139717733</td>\n",
       "      <td>225028590</td>\n",
       "      <td>0.620889</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                n_accessible  n_total_bases  frac_accessible\n",
       "mask_id                                                     \n",
       "arab               156787860      225028590         0.696746\n",
       "gamb_colu          162714957      225028590         0.723086\n",
       "gamb_colu_arab     143569766      225028590         0.638007\n",
       "phase2             139717733      225028590         0.620889"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "whole_genome_sum = wg_counts.groupby(level=0).agg(sum)\n",
    "whole_genome_sum[\"frac_accessible\"] = whole_genome_sum[\"n_accessible\"] / whole_genome_sum[\"n_total_bases\"]\n",
    "whole_genome_sum"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
