<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-GB" xml:lang="en-GB">
<head>
  <meta charset="utf-8" />
  <meta name="generator" content="pandoc" />
  <meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes" />
  <meta name="author" content="The Anopheles gambiae 1000 Genomes Consortium" />
  <meta name="dcterms.date" content="2021-02-24" />
  <meta name="keywords" content="malaria, anopheles, genomics" />
  <title>Genome variation and population structure in three African malaria vector species within the Anopheles gambiae complex</title>
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    span.underline{text-decoration: underline;}
    div.column{display: inline-block; vertical-align: top; width: 50%;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
  </style>
  <!--[if lt IE 9]>
    <script src="//cdnjs.cloudflare.com/ajax/libs/html5shiv/3.7.3/html5shiv-printshiv.min.js"></script>
  <![endif]-->
  <!--
  Manubot generated metadata rendered from header-includes-template.html.
  Suggest improvements at https://github.com/manubot/manubot/blob/master/manubot/process/header-includes-template.html
  -->
  <meta name="dc.format" content="text/html" />
  <meta name="dc.title" content="Genome variation and population structure in three African malaria vector species within the *Anopheles gambiae* complex" />
  <meta name="citation_title" content="Genome variation and population structure in three African malaria vector species within the *Anopheles gambiae* complex" />
  <meta property="og:title" content="Genome variation and population structure in three African malaria vector species within the *Anopheles gambiae* complex" />
  <meta property="twitter:title" content="Genome variation and population structure in three African malaria vector species within the *Anopheles gambiae* complex" />
  <meta name="dc.date" content="2021-02-24" />
  <meta name="citation_publication_date" content="2021-02-24" />
  <meta name="dc.language" content="en-GB" />
  <meta name="citation_language" content="en-GB" />
  <meta name="dc.relation.ispartof" content="Manubot" />
  <meta name="dc.publisher" content="Manubot" />
  <meta name="citation_journal_title" content="Manubot" />
  <meta name="citation_technical_report_institution" content="Manubot" />
  <meta name="citation_author" content="The Anopheles gambiae 1000 Genomes Consortium" />
  <link rel="canonical" href="https://malariagen.github.io/ag1000g-phase3-data-paper/" />
  <meta property="og:url" content="https://malariagen.github.io/ag1000g-phase3-data-paper/" />
  <meta property="twitter:url" content="https://malariagen.github.io/ag1000g-phase3-data-paper/" />
  <meta name="citation_fulltext_html_url" content="https://malariagen.github.io/ag1000g-phase3-data-paper/" />
  <meta name="citation_pdf_url" content="https://malariagen.github.io/ag1000g-phase3-data-paper/manuscript.pdf" />
  <link rel="alternate" type="application/pdf" href="https://malariagen.github.io/ag1000g-phase3-data-paper/manuscript.pdf" />
  <link rel="alternate" type="text/html" href="https://malariagen.github.io/ag1000g-phase3-data-paper/v/ce136c8e9ab6fcc6b072830cbf3a00de69918933/" />
  <meta name="manubot_html_url_versioned" content="https://malariagen.github.io/ag1000g-phase3-data-paper/v/ce136c8e9ab6fcc6b072830cbf3a00de69918933/" />
  <meta name="manubot_pdf_url_versioned" content="https://malariagen.github.io/ag1000g-phase3-data-paper/v/ce136c8e9ab6fcc6b072830cbf3a00de69918933/manuscript.pdf" />
  <meta property="og:type" content="article" />
  <meta property="twitter:card" content="summary_large_image" />
  <link rel="icon" type="image/png" sizes="192x192" href="https://manubot.org/favicon-192x192.png" />
  <link rel="mask-icon" href="https://manubot.org/safari-pinned-tab.svg" color="#ad1457" />
  <meta name="theme-color" content="#ad1457" />
  <!-- end Manubot generated metadata -->
</head>
<body>
<header id="title-block-header">
<h1 class="title">Genome variation and population structure in three African malaria vector species within the <em>Anopheles gambiae</em> complex</h1>
</header>
<p><small><em>
This manuscript
(<a href="https://malariagen.github.io/ag1000g-phase3-data-paper/v/ce136c8e9ab6fcc6b072830cbf3a00de69918933/">permalink</a>)
was automatically generated
from <a href="https://github.com/malariagen/ag1000g-phase3-data-paper/tree/ce136c8e9ab6fcc6b072830cbf3a00de69918933">malariagen/ag1000g-phase3-data-paper@ce136c8</a>
on February 24, 2021.
</em></small></p>
<h2 id="authors">Authors</h2>
<ul>
<li><strong>The Anopheles gambiae 1000 Genomes Consortium</strong><br><br>
<small>
</small></li>
</ul>
<h2 class="page_break_before" id="abstract">Abstract</h2>

<h2 id="population-sampling">Population Sampling</h2>
<p>The third and final phase of the Ag1000g project data resource contains wild-caught <em>Anopheles</em> mosquito genomes from Sub-Saharan Africa, collected from a total of 124 sites across 19 countries, 6 of which are novel.</p>
<p>Collections from Mali increase the density of coverage in West Africa, Central African Republic and Democratic Republic of Congo begin to fill the gap previously present in Central Africa while Malawi, Mozambique and Tanzania provide much more power to analyse East African malaria vectors, including <em>A. arabiensis</em> an important vector species not previously sequenced in the project.</p>
<p>Alongside sampling from natural populations, we include colony individuals from a number of laboratory crosses, comprising 11 crosses that were released as part of phase 2, and 4 additional pedigrees.</p>
<div id="fig:sample_collection_map" class="fignos">
<figure>
<img src="images/sample_collection_map.svg" style="width:100.0%" alt="" /><figcaption><span>Figure 1:</span> Sample Collection Map</figcaption>
</figure>
</div>
<h2 class="page_break_before" id="whole-genome-sequencing-and-alignment">Whole Genome Sequencing and Alignment</h2>
<p>4,693 individual mosquito genomes were sequenced on either Illumina HiSeq2000 (n=3,130) or Illumina HiSeqX (n=1,563) to a target coverage of 30X.</p>
<p>Between machine types the median number of bases sequenced per sample was 9.76Gb and 10.33Gb respectively, representing a difference in yield (two-tailed mann-whitney U p &lt; 0.0001).</p>
<p>These values correspond to a yield per reference base (vs AgamP4) of 35.76X and 37.82X.</p>
<p>91.9% of HiSeqX runs and 80.5% of HiSeq2000 runs met the target yield of 30X.</p>
<p>Reads were aligned to the AgamP4 reference and Single Nucleotide Polymophisms (SNPs) called using GATK UnifiedGenotyper.</p>
<p>All samples successfully completed the pipeline and entered the sample quality control (QC) process.</p>
<h3 id="sample-qc">Sample QC</h3>
<p>For wild-caught samples (n=XXXX), the QC process was composed of three stages, sequence quality assurance, replicate handling, and anomaly detection.</p>
<p>A total of 668 samples were removed where sequencing was of insufficient quality to accurately call genotypes across the whole genome.</p>
<p>Exclusions were due to poor coverage (n=410), potential contamination (n=229), and an ambiguous sex call (n=29).</p>
<p>Where technical replicates were available, we excluded 4 pairs (8 samples) with low genotype concordance.</p>
<p>Where pairs met the concordance threshold we excluded the lower quality sample (n=407).</p>
<p>Samples were also screened pairwise within submission sets for unexpected pairs, though none were detected.</p>
<p>The third stage used principal component analysis (PCA) to identify and exclude individual samples that were outliers based on available metadata.</p>
<p>A review process identified samples that could not be explained parsimoniously, and were therefore likely to be sample mix ups or instances of mislabelling.</p>
<p>28 samples were excluded as they respectively dominated the first principal components, indicating high divergence from all other samples and therefore likely members of other Anopheline species.</p>
<p>A further 82 samples were excluded as potential sample mix ups.</p>
<p>Following all sample QC steps, 3,483 wild-caught samples (74.2%) were retained from the original cohort for analysis.</p>
<p>This represents an additional 1,823 mosquitoes relative to the phase 2 release.</p>
<p>Due to a change in assessment of sample quality where technical replicates are available, the preferred replicate was changed for 172 mosquitoes between phase 2 and phase 3.</p>
<p>9 samples included in phase 2 are not present in this release (sup XXX).</p>
<p>The AG1000G-X submission set, made up of laboratory experimental crosses, was subject to a slightly different QC process.</p>
<p>Firstly an analysis based on rates of Mendelian error identified true fathers of crosses (where multiple males were introduced to cages), and validated provided pedigrees.</p>
<p>Of the 7XX samples provided we were able to validate 15 crosses to a high level of confidence, comprising 299 samples.</p>
<p>4 of these crosses are novel relative to phase 2.</p>
<p>These samples went through a modified sequence quality assurance process, where 1 sample was removed for insufficient coverage (methods).</p>
<p>The final data release therefore comprises 3,XXX samples, XXX from laboratory crosses, and YYY wild collected samples.</p>
<h3 id="coverage">Coverage</h3>
<p>%% TO DO
%% (PLOTS DONE, but numbers needed).</p>
<p>Summary of site coverage post QC exclusions.</p>
<ul>
<li>ie what frac of the genome is at 1X median</li>
<li>what frac at 10X.</li>
<li>What frac of exome</li>
<li>what frac of X</li>
</ul>
<p>At this point we do not mention arabiensis.</p>
<h3 id="species-assignment-and-sex-calling">Species assignment and sex calling</h3>
<p>The <em>Anopheles gambiae</em> complex is a crypic group of sibling species,
with no single locus offering unambiguous resolution of species.
To identify species we looked beyond the conventional set of PCR based markers
and applied a wider set of ancestry informative markers (AIMs).
Species were not assigned to samples from laboratory colony crosses
due to inbreeding and high levels of genetic drift.
To distinguish <em>A. arabiensis</em> from <em>A. gambiae s.l</em> a set of novel markers
was derived from data from the 16 genomes project <span class="citation" data-cites="9XFa7ZJB">[<a href="#ref-9XFa7ZJB" role="doc-biblioref">1</a>]</span>.
Using cut offs based on agreement with the established PCR marker,
368 individuals were classed as <em>A. arabiensis</em> and 2415 as <em>A. gambiae s.l</em>.
A single individual collected in Tororo, Uganda is classed as intermediate-
given the majority (93.9%) of AIM SNPs in the genome are heterozygous between
the gambiae-like and arabiensis-like alleles, this individual is likely to be an F1 hybrid.
To resolve the <em>A. gambiae s.l</em> individuals as <em>A. gambiae</em> and <em>A. coluzzii</em>
we applied 729 AIMs previously identified by Neafsey et al <span class="citation" data-cites="2xdynpZK">[<a href="#ref-2xdynpZK" role="doc-biblioref">2</a>]</span>.
and used in previous analyses of Ag1000G data <span class="citation" data-cites="nvrjDUOw Rho6nYsT">[<a href="#ref-nvrjDUOw" role="doc-biblioref">3</a>,<a href="#ref-Rho6nYsT" role="doc-biblioref">4</a>]</span>.
Of the 2,415 <em>A. gambiae s.l</em> individuals,
1571 were called as <em>A. gambiae s.s</em>, 675 as <em>A. coluzzii</em> and 169 as intermediate.
Many intermediate samples are from the Western coast of West Africa (particularly The Gambia and Guinea Bissau),
and given distinct populations of <em>A. gambiae s.l.</em> and <em>A. coluzzii</em> are also found in this region,
this result highlights the complexity of species relationships here.
Additionally a number of samples were classed as intermediate in coastal populations of East Africa,
in Kilifi Kenya, and Muleba Tanzania.</p>
<p>It is established that species barriers between members of the <em>An. gambiae</em> complex are porous,
and numerous instances of introgression associated with selection have been observed in West Africa,
particularly of the <em>kdr</em> allele <span class="citation" data-cites="zjUaokUj PFIjKsQ5">[<a href="#ref-zjUaokUj" role="doc-biblioref">5</a>,<a href="#ref-PFIjKsQ5" role="doc-biblioref">6</a>]</span>.
In West African coluzzii populations (2011 onwards),
frequency of gambiae-like alleles around this locus reach 90%,
<em>e.g.</em> Burkina Faso, Cote d’Ivoire, Cameroon, Ghana, Guinea and Mali.
In Southern Africa, <em>An. coluzzii</em> from Luanda, Angola (2009)
also show high frequency of gambiae alleles at this locus (83.9%).
However no introgression is observed in <em>coluzzii</em> collections prior to this;
Cameroon (2005, n=7), Mali (2004, n=36) and the Central African Republic (1993/4, n=18)
all have gambiae alleles present at rates below 1%.
Due to this known introgression, chromosome 2L was not considered when assigning species within <em>An. gambiae s.l.</em>.</p>
<h3 id="snp-filtering-and-quality">SNP filtering and quality</h3>
<p>Site filtering is necessary to ensure that reported variation is of high quality.</p>
<p>Features of specific regions of the Anopheles genome contribute to calling errors in short-read technologies; such features include regions of high divergence from the reference, high homology between regions, copy number variation, or the presence of transposable elements.</p>
<p>Where previously we have used manually curated cutoffs based on observed mendelian error rates to filter sites (ref phase1, phase2), here we built a statistical model where cohort level summary statistics were used to identify sites likely to contain genotyping errors.</p>
<p>Using the 15 available Anopheles pedigrees previously described, we used the presence of mendelian error at sites as a proxy for genotype discordance.</p>
<p>10 of the 15 crosses were used to train the model while 5 were held out for validation.</p>
<p>Each of the 5 pedigrees represent independent evaluation sets.</p>
<p>Before applying the site filters, the false discovery rate (FDR) of the 5 crosses over all autosomal sites ranged between 0.74% and 1.10% (table XXX).</p>
<p>The application of the site filters defines the accessible fraction of the autosomes at 72.58%, and the range of false discovery rates is 0.04% to 0.10%.</p>
<p>The median fold change of FDR was -3.71.</p>
<p>On the hemizygous X chromosome we used the more direct measure of heterozygote calls in males to ascertain mendelian error.</p>
<p>In the dataset are 220 <em>A. gambiae s.l</em> male samples, each of which represent an independent proxy for genotype discordance.</p>
<p>Pre-application of the site filters, subject to a Genotype Quality (GQ) threshold of 30, the median heterozygosity rate was 0.244%, and post filtering this drops to 0.023% (table XX).</p>
<p>The median fold change in error rate was -3.33, with 69.97% of the X chromosome passing site filters.</p>
<p>The new model based method represents a marked improvement over the site filters generated as part of phase 2; all 5 evaluation pedigrees showed a modest reduction in FDR, but the higher rate of accessibility in this release (72.58% vs 62.05%) resulted in an significant improvement in Youden score (Table XXX) across autosomes.</p>
<p>The X chromosome showed a similar pattern, the median heterozygosity rate in phase 2 is similar to the new site filters (0.028%), but the higher accessibility in the updated filter set (69.97% vs 62.46%) yields improved sensitivity.</p>
<p>As genomic features vary between species, different sets of site filters were generated to allow high quality analyses both within and between species.</p>
<p>The <code>gamb_colu</code> site filters were generated as above, and are appropriate for analyses that include <em>gambiae</em> and <em>coluzzii</em> samples only.</p>
<p>%% TODO Add accessibility of other site filters.
The <code>arab</code> site filters were generated following application of the model to the summary statistics from arabiensis samples in the cohort (n=XXX), this set of site filters are appropriate when working with <em>A. arabiensis</em> samples only.</p>
<p>Finally, the <code>gamb_colu_arab</code> site filters allow analyses across all three species and are the intersection of the <code>gamb_colu</code> and the <code>arab</code> site filters.</p>
<p>Place holders for tables.</p>
<ul>
<li><p>Table A: Mendel errors per cross per autosome.
row indices: chromosome and raw/filtered
column indexes: crosses + frac accessible.
ie 8 rows, and 6 columns.</p></li>
<li><p>Table B: comparison of cross and X
row indices: raw/filtered
column indices: MER, frac accessible, Youden, each for 2 and 3.
column indexes: crosses + frac accessible.
ie 2 rows, and 6 columns.</p></li>
</ul>
<div style="font-size: 8pt">
<pre class="table"><code>---
caption: &#39;Result of heterozygote calls on male X chromosome&#39;
alignment: LLLLLLLL
include: content/tables/mer_X.csv
csv-kwargs:
  dialect: unix
width: [0.2, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1, 0.1]
---</code></pre>
</div>
<h3 id="genome-accessibility">Genome accessibility</h3>
<p>We define accessibility as the fraction of sites in a region passing the appropriate set of site filters.</p>
<p>Overall, 70% of the genome, and ??% of the exome are considered accessible in the <code>gamb_colu</code> set.</p>
<p>This is an improvement from phase 2, where XXX of the genome, and YYY of the exome was considered accessible.</p>
<p>As expected, accessiblity was generally lower around the centromeres, and in regions of heterochromatin (table ref).</p>
<p>One notable region of low accessibility spans 40-41Mbp of chromosome 3R, this corresponds to ??.</p>
<p>Accessibility of the <code>arab</code> site filters closely follows that of <code>gamb_colu</code>, with the exception of the X chromosome where we see substantially lower values.</p>
<p>This appears to be driven by high divergence between AgamP4 and our <em>A. arabiensis</em> samples, particularly around the Xag inversion at Q-Q Mbp, ref figure.</p>
<p>On the autosomes the divergence from the reference is comparable between <em>A.arabiensis</em> and <em>A. gambiae</em>/ <em>A. coluzzii</em> samples, suggesting a strong basis for comparison accross species.</p>
<p>The median divergence (Dxy) of 100kbp windows is XXX (5%/95% TTT/SSS) for gambiae/coluzzii and YYY (TTT/YYY) for arabiensis.</p>
<p>On the X chromosome these values are XXX ( / ) for gambiae/coluzzii and YYY ( / ) for arabiensis.</p>
<h3 id="snp-discovery">SNP discovery</h3>
<p>Overall, we report XX,XXX,SSS single nucleotide polymophisms (SNPs) segregating in this cohort that pass filters, of which XX,XXX (%) are multiallelic.</p>
<p>12,223 SNPs are segregating in both species groups, while XXX are private to gambiae_/coluzzii_ and YYY to arabiensis [fig ref].</p>
<p>This phase of the study reports an additional XXX SNPs from phase 2.</p>
<p>In XXX gambiae and coluzzii individuals we report 12,222,222 SNPs (Q% multiallelic), corresponding to a SNP every 1.6 accessible bases.</p>
<p>In XXX arabiensis individuals we identify 10,000,000 SNPs (Q% multiallelic), a SNP every 2.5 accessible bases.</p>
<h2 id="population-structure">Population Structure</h2>
<h3 id="genome-wide-patterns">Genome wide patterns</h3>
<p>%% TODO NOtes currently.</p>
<p>Re-introduce key idea of structure being different across the genome.</p>
<p>How does arabiensis fit into this? Are there regions of the genome where arabiensis ancestry is secondary?</p>
<h3 id="pca-umap">PCA / UMAP</h3>
<p>To highlight population structure we performed principal component analysis across all wild-caught samples in the dataset.</p>
<p>To avoid confounding of structure in genomic regions including paracentric inversions, extremely low diversity and regions under strong selection, we limited our analysis to euchromatic regions of chromosome 3L.</p>
<p>The most apparent signal in the dataset is PC1 clearly being driven by Arabiensis, with clear separation of arabiensis samples from gambiae/coluzzii.</p>
<p>The apparent hybrid sits between gambiae and arabiensis samples.</p>
<p>To view population structure within gambiae/coluzzii and arabiensis more independently, we performed subsequent PCA analysis Arabiensis and gambiae/coluzzii individuals separately.</p>
<p>Population structure between gambiae and coluzzii is significantly more complex.</p>
<p>Separately between species. What are the major findings?</p>
<ul>
<li><p>Arabiensis drives PC1.</p></li>
<li><p>East Africa: Seems to be clear population structure between <em>gambiae</em> in KE and TZ.</p></li>
<li><p>According to AIM analysis, a significant proportion of samples in these groups are classed as IM between gambiae. Certainly not coluzzii, but some kind of complex ancestry.</p></li>
<li><p>Relevance to TENEGLRA</p></li>
<li><p>West Africa- in far west Africa we see intermediate population. Not gambiae coluzzii, unlikely to be hybrids, but a related subspecies.</p></li>
</ul>
<p>Interestingly seems to be stable in the presence of both gamb and colu. Although they sit close to col in the PCA they are distinct from coluzzii, given they are founf at the same site.</p>
<h3 id="genetic-diversity-within-sampling-sites">Genetic Diversity within sampling sites</h3>
<p>Better to avoid use of population.</p>
<p>Using species groupings above, i.e. PCA clusters of samples not clearly gambcolu, but sympatric with them are classed as intermediate.</p>
<p>First look at diversity at a regional level within species. ie gambiae is more diverse in west than east africa. Central?</p>
<p>Coluzzii is similar within its range.</p>
<p>Arabiensis only found in EA, but do we see differences in diversity?</p>
<p>Justification of using wattersons theta.</p>
<p>THEN, we can start to speak about differences between species, within regions.</p>
<p>West African gambiae have higher diversity than coluzzii.</p>
<p>Then how do west african intermediate compare to these?</p>
<p>In east africa, we compare gambiae to arabiensis.</p>
<h2 id="insecticide-resistance">Insecticide Resistance</h2>
<ul>
<li>kdr frequencies in different sampling groups</li>
<li>we don’t have CNVs… so? We can use markers?</li>
</ul>
<h2 id="gene-drive">Gene Drive</h2>
<ul>
<li>repeat of phase 2 analysis.</li>
</ul>
<h1 class="page_break_before" id="methods">Methods</h1>
<h2 id="population-sampling-1">Population Sampling</h2>
<h3 id="summary">Summary</h3>
<p>The Ag1000G project is coordinated by a consortium of partners from a range of different research institutions and countries.
This includes consortium members who are carrying out independent research studies in malaria endemic regions, and who have contributed mosquito specimens or mosquito DNA samples collected in the course of their own research.
The methods presented here describe the studies that have contributed samples to phase 3 of the Ag1000G project, including wild-caught samples from 19 African countries.
This section also provides information about the collection locations and methods, the people involved in the studies, and references to any published articles providing further information about the studies.
Throughout this document we use species nomenclature following Coetzee <em>et al.</em> <span class="citation" data-cites="R1m6hsR0">[<a href="#ref-R1m6hsR0" role="doc-biblioref">7</a>]</span>.
Unless otherwise stated, the DNA extraction method used for the collections described below was Qiagen DNeasy Blood and Tissue Kit (Qiagen Science, MD, USA).</p>
<h3 class="page_break_before" id="angola">Angola</h3>
<h4 id="sample-sets">Sample sets</h4>
<p><code>AG1000G-AO</code>.</p>
<h4 id="study-information">Study information</h4>
<p>Adult mosquitoes were obtained by rearing larvae collected from breeding sites along the main roads connecting the municipalities of Kilamba-Kiaxi and Viana, Luanda province <strong><a href="https://www.openstreetmap.org/?mlat=-8.821&amp;mlon=13.291&amp;zoom=12&amp;layers=M">(-8.821,13.291)</a></strong>, in April/May 2009.
These are peri-urban areas where malaria reaches hyperendemic levels.
All specimens collected in the study area were typed as <em>A. coluzzii</em> <span class="citation" data-cites="1G04AzsFo">[<a href="#ref-1G04AzsFo" role="doc-biblioref">8</a>]</span> although <em>An. melas</em> and <em>A. arabiensis</em> have also been recorded in the province <span class="citation" data-cites="BAhAVfFc 2Iw2A7Bd">[<a href="#ref-BAhAVfFc" role="doc-biblioref">9</a>,<a href="#ref-2Iw2A7Bd" role="doc-biblioref">10</a>]</span>.
Specimens were stored on silica gel and DNA extraction was performed by a phenol-chloroform protocol described in <span class="citation" data-cites="DnnjILYG">[<a href="#ref-DnnjILYG" role="doc-biblioref">11</a>]</span>.</p>
<h4 id="contributors">Contributors</h4>
<ul>
<li>Arlete D. Troco
<ul>
<li>Programa Nacional de Controle da Malária, Direcção Nacional de Saúde Pública, Ministério da Saúde, Luanda, Angola.</li>
</ul></li>
<li>João Pinto (jpinto@ihmt.unl.pt)
<ul>
<li>Global Health and Tropical Medicine, GHTM, Instituto de Higiene e Medicina Tropical, IHMT, Universidade Nova de Lisboa, UNL, Rua da Junqueira 100, 1349-008 Lisbon, Portugal.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="burkina-faso-1">Burkina Faso (1)</h3>
<h4 id="sample-sets-1">Sample sets</h4>
<p><code>AG1000G-BF-A</code>, <code>AG1000G-BF-B</code></p>
<h4 id="study-information-1">Study information</h4>
<p>The Target Malaria project contributed samples from collections made in three villages separated by at most 30km: Bana <strong><a href="https://www.openstreetmap.org/?mlat=11.233&amp;mlon=-4.472&amp;zoom=12&amp;layers=M">(11.233, -4.472)</a></strong>, Souroukoudinga <strong><a href="https://www.openstreetmap.org/?mlat=11.235&amp;mlon=-4.535&amp;zoom=12&amp;layers=M">(11.235, -4.535)</a></strong> and Pala <strong><a href="https://www.openstreetmap.org/?mlat=11.150&amp;mlon=-4.235&amp;zoom=12&amp;layers=Mi">(11.150, -4.235)</a></strong>.
These collections were made in July-August 2012, July and October 2014, and January, February and April 2015.
The area is agricultural, with rice-growing areas near Bana and Souroukoudinga, and a large mango grove near Pala.
Female mosquitoes were collected by human landing catch, pyrethrum spray collection or aspiration.
Males were collected by swarm netting.
Both <em>An. gambiae</em> and <em>An. coluzzii</em> <span class="citation" data-cites="QhOf6jCa">[<a href="#ref-QhOf6jCa" role="doc-biblioref">12</a>]</span> were collected.
Specimens were stored in 80% ethanol and DNA was extracted using the DNeasy Tissue Kit (Qiagen) or using a simple CTAB method.</p>
<h4 id="external-resources">External resources</h4>
<ul>
<li><strong><a href="%5Bhttps://targetmalaria.org/">https://targetmalaria.org/</a></strong></li>
</ul>
<h4 id="contributors-1">Contributors</h4>
<ul>
<li>Abdoulaye Diabaté
<ul>
<li>Institut de Recherche en Sciences de la Santé (IRSS), Bobo Dioulasso, B.P.545, Burkina Faso.</li>
</ul></li>
<li>Patric Stephane Epopa
<ul>
<li>Institut de Recherche en Sciences de la Santé (IRSS), Bobo Dioulasso, B.P.545, Burkina Faso.</li>
</ul></li>
<li>Franck Yao
<ul>
<li>Institut de Recherche en Sciences de la Santé (IRSS), Bobo Dioulasso, B.P.545, Burkina Faso.</li>
</ul></li>
<li>Samantha O’Loughlin (s.oloughlin@imperial.ac.uk)
<ul>
<li>Department of Life Sciences, Imperial College, Silwood Park, Ascot, Berkshire SL5 7PY, UK.</li>
</ul></li>
<li>Austin Burt
<ul>
<li>Department of Life Sciences, Imperial College, Silwood Park, Ascot, Berkshire SL5 7PY, UK.</li>
</ul></li>
</ul>
<h4 id="acknowledgements">Acknowledgements</h4>
<p>We would like to thank the technicians of the Institute de Recherche en Sciences de la Santé/Target Malaria Burkina Faso, including Guel Hyacinthe, Diabate Brama, Ilboudo Seni, Kabre Rasmane, Diabate Noufou and Yeye Pascal, for their contributions to sample collections.</p>
<h3 class="page_break_before" id="burkina-faso-2">Burkina Faso (2)</h3>
<h4 id="sample-sets-2">Sample sets</h4>
<p><code>AG1000G-BF-C</code></p>
<h4 id="study-information-2">Study information</h4>
<p>Samples were contributed from collections of indoor resting adults made by spray catch from Monomtenga in central Burkina Faso <strong><a href="https://www.openstreetmap.org/?mlat=12.06&amp;mlon=-1.17&amp;zoom=12&amp;layers=M">(12.06, -1.17)</a></strong>.
These specimens were sorted morphologically to <strong>An. gambiae s.l.</strong>.
Ovaries of half-gravid females were dissected and placed in numbered individual micro-tubes containing modified Carnoy’s solution (1:3 glacial acetic acid: 100% ethanol).
Carcasses were placed in correspondingly numbered micro-tubes over desiccant.
Genomic DNA was isolated from individual mosquitoes using one of the following: DNeasy Extraction Kit (Qiagen, Valencia, CA), Puregene kit (Gentra Systems, Inc., Minneapolis, MN), DNAzol kit (Molecular Research Center, Inc., Cincinnati, OH.) or Easy-DNA kit (Invitrogen, Carlsbad, CA).
<strong>An. gambiae s.s.</strong> and its molecular forms were identified using one of two rDNA-based PCR/RFLP assays, <span class="citation" data-cites="1G04AzsFo mTkFlBOi">[<a href="#ref-1G04AzsFo" role="doc-biblioref">8</a>,<a href="#ref-mTkFlBOi" role="doc-biblioref">13</a>]</span>.
Ovaries from specimens of the desired species were subject to polytene chromosome analysis.</p>
<h4 id="contributors-2">Contributors</h4>
<ul>
<li>Carlo Costantini (carlo.costantini@ird.fr)
<ul>
<li>UMR MIVEGEC, Univ. Montpellier, CNRS, IRD, Montpellier, France.</li>
</ul></li>
<li>N’Fale Sagnon
<ul>
<li>Centre National de Recherche et Formation sur le Paludisme (CNRFP), 01BP 2208 Ouagadougou 01, Burkina Faso.</li>
</ul></li>
<li>Nora J. Besansky (nbesansk@nd.edu)
<ul>
<li>Eck Institute for Global Health &amp; Department of Biological Sciences, University of Notre Dame, IN 46556, USA.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="cameroon-1">Cameroon (1)</h3>
<h4 id="sample-sets-3">Sample sets</h4>
<p><code>AG1000G-CM-A</code></p>
<h4 id="study-information-3">Study information</h4>
<p>Pyrethrum spray collections were conducted in three villages in Cameroon during September and October 2009.
These villages comprise a transect from forest (village of Mayos: <strong><a href="https://www.openstreetmap.org/?mlat=4.341&amp;mlon=13.558&amp;zoom=12&amp;layers=M">(4.341, 13.558)</a>)</strong> to forest/savanna transition (village of Daiguene: <strong><a href="https://www.openstreetmap.org/?mlat=4.777&amp;mlon=13.844&amp;zoom=12&amp;layers=M">(4.777, 13.844)</a>)</strong> to savanna (villages of Gado-Badzere and Zembe-Borongo: <strong><a href="https://www.openstreetmap.org/?mlat=5.747&amp;mlon=14.442&amp;zoom=12&amp;layers=M">(5.747, 14.442)</a></strong>) (<span class="citation" data-cites="1rFnXZkd"><a href="#ref-1rFnXZkd" role="doc-biblioref">14</a></span>).
All contributed specimens were <strong>An. gambiae s.s.</strong> <span class="citation" data-cites="1G04AzsFo">[<a href="#ref-1G04AzsFo" role="doc-biblioref">8</a>]</span>.
A proportion of specimens were karyotyped via scoring of polytene chromosomes <span class="citation" data-cites="122DDCVbs">[<a href="#ref-122DDCVbs" role="doc-biblioref">15</a>]</span>.
Specimens were stored on silica gel, and DNA was extracted using a simple CTAB protocol and run over Qiagen columns.</p>
<h4 id="related-publications">Related publications</h4>
<ul>
<li>NF Lobo et al. 2010. Breakpoint structure of the Anopheles gambiae 2Rb chromosomal inversion. Malaria Journal. 9:293. (<span class="citation" data-cites="1rFnXZkd"><a href="#ref-1rFnXZkd" role="doc-biblioref">14</a></span>).</li>
</ul>
<h4 id="contributors-3">Contributors</h4>
<ul>
<li>Carlo Costantini (carlo.costantini@ird.fr)
<ul>
<li>UMR MIVEGEC, Univ. Montpellier, CNRS, IRD, Montpellier, France.</li>
</ul></li>
<li>Kyanne R. Rohatgi
<ul>
<li>Eck Institute for Global Health &amp; Department of Biological Sciences, University of Notre Dame, IN 46556, USA.</li>
</ul></li>
<li>Nora J. Besansky (nbesansk@nd.edu)
<ul>
<li>Eck Institute for Global Health &amp; Department of Biological Sciences, University of Notre Dame, IN 46556, USA.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="cameroon-2">Cameroon (2)</h3>
<h4 id="sample-sets-4">Sample sets</h4>
<p><code>AG1000G-CM-B</code></p>
<h4 id="study-information-4">Study information</h4>
<p>These samples were collected as part of a study which took place in Cameroon in Central Africa.
The country is commonly referred to as “miniature Africa”, owing to the diversity of its climate, topography, landscape,
and bio-ecological settings: arid savannas in the north gradually turn into rain forest in the south,
along with highland areas, contribute to increase diversity of ecological settings.
Anopheline mosquitoes were collected in 2005 from 64 locations covering a 1,500 km north-to-south transect
that crossed all eco-geographical areas of Cameroon <span class="citation" data-cites="daG5Pj6c">[<a href="#ref-daG5Pj6c" role="doc-biblioref">16</a>]</span>
Mosquito collection involved spraying aerosols of pyrethroid insecticides inside human dwellings,
dead mosquitoes were retrieved from white sheets that were laid on the floor.
Anopheline mosquitoes were identified using morphological identification keys <span class="citation" data-cites="14XMzb1n7 Zqohe5sj">[<a href="#ref-14XMzb1n7" role="doc-biblioref">17</a>,<a href="#ref-Zqohe5sj" role="doc-biblioref">18</a>]</span>.
Ovaries from half-gravid An. gambiae s.l. females were dissected and stored in Carnoy’s fixative solution
(absolute ethanol:glacial acetic acid 3:1) for cytogenetic analyses.
Carcasses were stored individually in tubes containing a desiccant and kept at -20<sup>o</sup> C
until they were processed for molecular analysis.
All half-gravid specimens collected in each village were identified to species and molecular forms using PCR-RFLP <span class="citation" data-cites="1G04AzsFo">[<a href="#ref-1G04AzsFo" role="doc-biblioref">8</a>]</span>.</p>
<h4 id="related-publications-1">Related publications</h4>
<ul>
<li>Simard, Frédéric, et al. “Ecological Niche Partitioning between Anopheles Gambiae Molecular Forms in Cameroon: The Ecological Side of Speciation.” BMC Ecology, vol. 9, no. 1, 2009, p. 17, 10.1186/1472-6785-9-17. <span class="citation" data-cites="daG5Pj6c">[<a href="#ref-daG5Pj6c" role="doc-biblioref">16</a>]</span>.</li>
</ul>
<h4 id="contributors-4">Contributors</h4>
<ul>
<li>Frédéric Simard (frederic.simard@ird.fr)
<ul>
<li>UMR MIVEGEC, Univ. Montpellier, CNRS, IRD, Montpellier, France.</li>
</ul></li>
<li>Diego Ayala (diego.ayala@ird.fr)
<ul>
<li>UMR MIVEGEC, Univ. Montpellier, CNRS, IRD, Montpellier, France.</li>
</ul></li>
<li>Nora J. Besansky (nbesansk@nd.edu)
<ul>
<li>Eck Institute for Global Health &amp; Department of Biological Sciences, University of Notre Dame, IN 46556, USA.</li>
</ul></li>
<li>Carlo Costantini (carlo.costantini@ird.fr)
<ul>
<li>UMR MIVEGEC, Univ. Montpellier, CNRS, IRD, Montpellier, France.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="cameroon-3">Cameroon (3)</h3>
<h4 id="sample-sets-5">Sample sets</h4>
<p><code>AG1000G-CM-C</code></p>
<h4 id="study-information-5">Study information</h4>
<p>Samples were contributed from pyrethrum spray collections, larval sampling and human landing catches conducted in twenty locations during October 2013.</p>
<p>These villages are scattered throughout the country and reflect a gradient of human-dominated environments, for example, forest (Manda: <strong><a href="https://www.openstreetmap.org/?mlat=5.726&amp;mlon=10.868&amp;zoom=12&amp;layers=M">(5.726, 10.868)</a></strong> and Campo: <strong><a href="https://www.openstreetmap.org/?mlat=2.367&amp;mlon=9.817&amp;zoom=12&amp;layers=M">(2.367, 9.817)</a>; forest/savanna transition (Tibati: </strong><a href="https://www.openstreetmap.org/?mlat=6.469&amp;mlon=12.629&amp;zoom=12&amp;layers=M">(6.469, 12.629)</a><strong>); savanna (Lagdo: </strong><a href="https://www.openstreetmap.org/?mlat=9.049&amp;mlon=13.656&amp;zoom=12&amp;layers=M">(9.049, 13.656)</a>); suburban area (Nkolondom: <strong><a href="https://www.openstreetmap.org/?mlat=3.972&amp;mlon=11.516&amp;zoom=12&amp;layers=M">(3.972, 11.516)</a></strong>) and urban areas (Douala: <strong><a href="https://www.openstreetmap.org/?mlat=4.055&amp;mlon=9.721&amp;zoom=12&amp;layers=M">(4.055, 9.721)</a></strong> and Yaoundé: <strong><a href="https://www.openstreetmap.org/?mlat=3.880&amp;mlon=11.506&amp;zoom=12&amp;layers=M">(3.880, 11.506)</a></strong>.</p>
<p>Contributed specimens were <em>An. gambiae</em> or <em>An.coluzzii</em> <span class="citation" data-cites="1G04AzsFo">[<a href="#ref-1G04AzsFo" role="doc-biblioref">8</a>]</span>.</p>
<p>Population genomics studies indicated the presence of relatively dierentiated subgroups within both species as well as clusters thriving in polluted breeding sites in large cities <span class="citation" data-cites="kGCikEHW">[<a href="#ref-kGCikEHW" role="doc-biblioref">19</a>]</span>.</p>
<p>Specimens were stored on silica gel.</p>
<p>DNA was extracted using a Zymo research kit for adults, and a Qiagen kit for larvae.</p>
<h4 id="related-publications-2">Related publications</h4>
<ul>
<li>C Kamdem et al. 2017. Pollutants and Insecticides Drive Local Adaptation in African Malaria Mosquitoes. Molecular Biology and Evolution. 34: 1261-1275. <span class="citation" data-cites="kGCikEHW">[<a href="#ref-kGCikEHW" role="doc-biblioref">19</a>]</span>.</li>
</ul>
<h4 id="contributors-5">Contributors</h4>
<ul>
<li>Colince Kamdem
<ul>
<li>Laboratoire de Recherche sur le Paludisme, Organisation de Coordination pour la lutte contre les Endémies en Afrique Centrale (OCEAC), B.P. 288, Yaoundé, Cameroon.</li>
</ul></li>
<li>Caroline Fouet
<ul>
<li>Department of Entomology, University of California, Riverside, CA, USA.</li>
</ul></li>
<li>Bradley J. White (bradwhite@verily.com)
<ul>
<li>Verily Life Sciences, South San Francisco, CA 94080, USA.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="central-african-republic">Central African Republic</h3>
<h4 id="sample-sets-6">Sample sets</h4>
<p><code>AG1000G-CF</code></p>
<h4 id="study-information-6">Study information</h4>
<p>Collections were carried out in Bangui <strong><a href="https://www.openstreetmap.org/?mlat=4.367&amp;mlon=18.583&amp;zoom=12&amp;layers=M">(4.367, 18.583)</a></strong>, during December 1993, by indoor
resting aspiration or pyrethrum spray catch.</p>
<h4 id="contributors-6">Contributors</h4>
<ul>
<li>Alessandra della Torre (alessandra.dellatorre@uniroma1.it)
<ul>
<li>Istituto Pasteur Italia Fondazione Cenci Bolognetti, Dipartimento di Sanita Pubblica e Malattie Infettive, Università di Roma SAPIENZA, Rome, Italy.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="bioko-island-equatorial-guinea">Bioko Island, Equatorial Guinea</h3>
<h4 id="sample-sets-7">Sample sets</h4>
<p><code>AG1000G-GQ</code></p>
<h4 id="study-information-7">Study information</h4>
<p>Collections were performed during the rainy season in September 2002 by overnight CDC light traps in Sacriba of Bioko island <strong><a href="https://www.openstreetmap.org/?mlat=3.7&amp;mlon=8.7&amp;zoom=12&amp;layers=M">(3.7, 8.7)</a></strong>.</p>
<p>Specimens were stored dry on silica gel before DNA extraction.</p>
<p>Specimens contributed from this site were <em>An. gambiae</em> females, genotype determined by two assays <span class="citation" data-cites="14yZJJBPD">[<a href="#ref-14yZJJBPD" role="doc-biblioref">20</a>]</span>.</p>
<p>All specimens had the 2L<sup>+a</sup>/2L<sup>+a</sup> karyotype as determined by the molecular PCR diagnostics <span class="citation" data-cites="WKPmPPDE">[<a href="#ref-WKPmPPDE" role="doc-biblioref">21</a>]</span>.</p>
<p>These mosquitoes represent a population that inhabited Bioko Island before a comprehensive malaria control intervention initiated in February 2004 <span class="citation" data-cites="EfCseGxd">[<a href="#ref-EfCseGxd" role="doc-biblioref">22</a>]</span>.</p>
<p>After the intervention <em>An. gambiae</em> was declining, and more recently almost only <em>An. coluzzii</em> can be found <span class="citation" data-cites="DaNgNJ6t">[<a href="#ref-DaNgNJ6t" role="doc-biblioref">23</a>]</span>.</p>
<h4 id="contributors-7">Contributors</h4>
<ul>
<li>Jorge Cano
<ul>
<li>London School of Hygiene and Tropical Medicine, Keppel Street, Bloomsbury, London WC1E 7HT, UK.</li>
</ul></li>
<li>Maryam Kamali
<ul>
<li>Department of Medical Entomology and Parasitology, Faculty of Medical Sciences, Tarbiat Modares University, Tehran, Iran.</li>
<li>Department of Entomology, Virginia Tech, Blacksburg, Virginia 24061, USA.</li>
</ul></li>
<li>Igor Sharakhov (igor@vt.edu)
<ul>
<li>Department of Entomology, Virginia Tech, Blacksburg, Virginia 24061, USA.</li>
<li>Department of Genetics and Cell Biology, Tomsk State University, Tomsk 634050, Russia.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="côte-divoire">Côte d’Ivoire</h3>
<h4 id="sample-sets-8">Sample sets</h4>
<p><code>AG1000G-CI</code></p>
<h4 id="study-information-8">Study information</h4>
<p>Samples were collected in Tiassale <strong><a href="https://www.openstreetmap.org/?mlat=5.898&amp;mlon=-4.823&amp;zoom=12&amp;layers=M">(5.898, -4.823)</a></strong>, located in the evergreen forest zone of southern Côte d’Ivoire.
The primary agricultural activity is rice cultivation in irrigated fields.
High malaria transmission occurs during the rainy seasons, between May and November.
Samples were collected as larvae from irrigated rice fields by dipping between May and September 2012.
All larvae were reared to adults and females preserved over silica for DNA extraction.
Specimens from this site were all <em>An. coluzzii</em> , determined by PCR assay <span class="citation" data-cites="mTkFlBOi">[<a href="#ref-mTkFlBOi" role="doc-biblioref">13</a>]</span>.</p>
<h4 id="related-publications-3">Related publications</h4>
<ul>
<li>Grau-Bové, Xavier, et al. “Resistance to Pirimiphos-Methyl in West African Anopheles Is Spreading via Duplication and Introgression of the Ace1 Locus.” PLOS Genetics, vol. 17, no. 1, 21 Jan. 2021, p. e1009253, 10.1371/journal.pgen.1009253. Accessed 1 Feb. 2021. <span class="citation" data-cites="meNim6ix">[<a href="#ref-meNim6ix" role="doc-biblioref">24</a>]</span>.</li>
</ul>
<h4 id="contributors-8">Contributors</h4>
<ul>
<li>David Weetman (david.weetman@lstmed.ac.uk)
<ul>
<li>Department of Vector Biology, Liverpool School of Tropical Medicine, Liverpool, United Kingdom.</li>
</ul></li>
<li>Edi Constant
<ul>
<li>Centre Suisse de Recherches Scientifiques. Yopougon, Abidjan - 01 BP 1303 Abidjan, Côte d’Ivoire.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="ghana">Ghana</h3>
<h4 id="sample-sets-9">Sample sets</h4>
<p><code>AG1000G-GH</code></p>
<h4 id="study-information-9">Study information</h4>
<p>Samples were collected in Twifo Praso <strong><a href="https://www.openstreetmap.org/?mlat=5.609&amp;mlon=-1.549&amp;zoom=12&amp;layers=M">(5.609,-1.549)</a></strong>, a peri-urban community located in semi-deciduous forest in the Central Region of Ghana.
It is an extensive agricultural area characterised by small-scale (vegetable growing) and large-scale commercial farms such as oil palm and cocoa plantations.
Mosquito samples were collected as larvae from puddles near farms between September and October, 2012.
Madina <strong><a href="https://www.openstreetmap.org/?mlat=5.668&amp;mlon=-0.219&amp;zoom=12&amp;layers=M">(5.668,-0.219)</a></strong> is suburb of Accra within a coastal savanna zone of Ghana.
It is an urban community characterised by myriad vegetable-growing areas.
The vegetation consists of mainly grassland interspersed with dense short thickets often less than 5m high with a few trees.
Specimens were sampled from puddles near roadsides and farms between October and December 2012.
Takoradi <strong><a href="https://www.openstreetmap.org/?mlat=4.912&amp;mlon=-1.774&amp;zoom=12&amp;layers=M">(4.912,-1.774)</a></strong> is the capital city of the Western Region of Ghana.
It is an urban community located in the coastal savanna zone.
Mosquito samples were collected from puddles near road construction and farms between August and September 2012.
Koforidua <strong><a href="https://www.openstreetmap.org/?mlat=6.094&amp;mlon=-0.261&amp;zoom=12&amp;layers=M">(6.094,-0.261)</a></strong> is the capital city of the Eastern Region of Southern Ghana and is located in semi-deciduous forest.
It is an urban community characterized by numerous small-scale vegetable farms.
Samples were collected from puddles near road construction sites and farms between August and September 2012.
Larvae from all collection sites were reared to adults and females preserved over silica for DNA extraction.
Both <em>An. gambiae</em> and <em>An. coluzzii</em> were collected from these sites, determined by PCR assay <span class="citation" data-cites="mTkFlBOi">[<a href="#ref-mTkFlBOi" role="doc-biblioref">13</a>]</span>.</p>
<h4 id="related-publications-4">Related publications</h4>
<ul>
<li>Essandoh, John, et al. “Acetylcholinesterase (Ace-1) Target Site Mutation 119S Is Strongly Diagnostic of Carbamate and Organophosphate Resistance in Anopheles Gambiae S.s. And Anopheles Coluzzii across Southern Ghana.” Malaria Journal, vol. 12, no. 1, 2013, p. 404, 10.1186/1475-2875-12-404. <span class="citation" data-cites="szUswio1">[<a href="#ref-szUswio1" role="doc-biblioref">25</a>]</span>.</li>
</ul>
<h4 id="contributors-9">Contributors</h4>
<ul>
<li>David iWeetman (david.weetman@lstmed.ac.uk)
<ul>
<li>Department of Vector Biology, Liverpool School of Tropical Medicine, Liverpool, United Kingdom.</li>
</ul></li>
<li>John Essandoh
<ul>
<li>Department of Wildlife and Entomology, University of Cape Coast, Cape Coast, Ghana.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="democratic-republic-of-the-congo">Democratic Republic of the Congo</h3>
<h4 id="sample-sets-10">Sample sets</h4>
<p><code>AG1000G-CD</code></p>
<h4 id="study-information-10">Study information</h4>
<p>Samples were collected from Gbadolite <strong><a href="https://www.openstreetmap.org/?mlat=4.283&amp;mlon=21.017&amp;zoom=12&amp;layers=M">(4.283,21.017)</a></strong>,
a town located in the far north of the Democratic Republic of Congo (DRC)
near the border with the Central African Republic, surrounded by forest.
In common with much of DRC, malaria transmission rates are high, and the samples are <em>An. gambiae s.s.</em> , which is the dominant vector.
Samples were collected as larvae from temporary pools within and around the town by dipping in early August 2015.
All larvae were reared to adults and females preserved over silica for DNA extraction using Qiagen DNAEasy kits.</p>
<h4 id="contributors-10">Contributors</h4>
<ul>
<li>David Weetman (david.weetman@lstmed.ac.uk)
<ul>
<li>Department of Vector Biology, Liverpool School of Tropical Medicine, Liverpool, United Kingdom.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="gabon">Gabon</h3>
<h4 id="sample-sets-11">Sample sets</h4>
<p><code>AG1000G-GA-A</code></p>
<h4 id="study-information-11">Study information</h4>
<p>Mosquitoes were collected by landing catches in the capital city Libreville <strong><a href="https://www.openstreetmap.org/?mlat=0.384&amp;mlon=9.455&amp;zoom=12&amp;layers=M">(0.384,9.455)</a></strong> in December 2000 <span class="citation" data-cites="TPRu2NBx">[<a href="#ref-TPRu2NBx" role="doc-biblioref">26</a>]</span>, an urban and polluted site.
Malaria is endemic throughout the year.
Specimens were stored in alcohol at -20<sup>o</sup>C.
Co-occurrence of both <em>kdr</em> resistance alleles and absence of wild-type susceptible alleles have been reported in this population <span class="citation" data-cites="TPRu2NBx">[<a href="#ref-TPRu2NBx" role="doc-biblioref">26</a>]</span>.
<em>An coluzzii</em> and <em>An. melas</em> are also present in the region but at frequencies &lt;1% <span class="citation" data-cites="s65aawfm">[<a href="#ref-s65aawfm" role="doc-biblioref">27</a>]</span>.
Specimens were stored on silica gel and DNA extraction was performed by a phenol-chloroform protocol
described in <span class="citation" data-cites="8S1Lluej">[<a href="#ref-8S1Lluej" role="doc-biblioref">28</a>]</span></p>
<h4 id="contributors-11">Contributors</h4>
<ul>
<li>Nohal Elissa
<ul>
<li>Institut Pasteur de Madagascar, Avaradoha, BP 1274, 101 Antananarivo, Madagascar.</li>
</ul></li>
<li>João Pinto (jpinto@ihmt.unl.pt)
<ul>
<li>Global Health and Tropical Medicine, GHTM, Instituto de Higiene e Medicina Tropical, IHMT, Universidade Nova de Lisboa, UNL, Rua da Junqueira 100, 1349-008 Lisbon, Portugal.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="guinea-bissau">Guinea Bissau</h3>
<h4 id="sample-sets-12">Sample sets</h4>
<p><code>AG1000G-GW</code></p>
<h4 id="study-information-12">Study information</h4>
<p>Guinea Bissau samples were collected from three sites in October 2010 by indoor CDC light traps.
Safim <strong><a href="https://www.openstreetmap.org/?mlat=11.957&amp;mlon=-15.649&amp;zoom=12&amp;layers=M">(11.957,-15.649)</a></strong>
and Antula <strong><a href="https://www.openstreetmap.org/?mlat=11.891&amp;mlon=-15.582&amp;zoom=12&amp;layers=M">(11.891,-15.582)</a></strong>,
from a south-western coastal region, characterised mainly by mixed flooded forests and croplands.
Leibala is a neighbourhood of the eastern town of Gabu where shrubland and open deciduous forest predominate
<strong><a href="https://www.openstreetmap.org/?mlat=12.272&amp;mlon=-14.222&amp;zoom=12&amp;layers=M">(12.272,-14.222)</a></strong>.
According to PCR-RFLP of the IGS <span class="citation" data-cites="1G04AzsFo">[<a href="#ref-1G04AzsFo" role="doc-biblioref">8</a>]</span> and SINEX <span class="citation" data-cites="mTkFlBOi">[<a href="#ref-mTkFlBOi" role="doc-biblioref">13</a>]</span> all samples were identified as <em>An. gambiae</em>.
The <em>kdr</em> pyrethroid target site resistance mutation L995F occurs at high frequency in Leibala
but at very low frequency in the western coastal region <span class="citation" data-cites="YX2M3jrq">[<a href="#ref-YX2M3jrq" role="doc-biblioref">29</a>]</span>
Malaria is meso-hyperendemic <span class="citation" data-cites="16as3jwaa">[<a href="#ref-16as3jwaa" role="doc-biblioref">30</a>]</span> and sporozoite rates are below 1% in the region.
Specimens were stored on silica gel and DNA extraction was performed by a phenol-chloroform protocol described in <span class="citation" data-cites="8S1Lluej">[<a href="#ref-8S1Lluej" role="doc-biblioref">28</a>]</span>.</p>
<h4 id="related-publications-5">Related publications</h4>
<ul>
<li>Vicente, José L., et al. “Massive Introgression Drives Species Radiation at the Range Limit of Anopheles Gambiae.” Scientific Reports, vol. 7, no. 1, 18 Apr. 2017, 10.1038/srep46451. <span class="citation" data-cites="YX2M3jrq">[<a href="#ref-YX2M3jrq" role="doc-biblioref">29</a>]</span></li>
</ul>
<h4 id="contributors-12">Contributors</h4>
<ul>
<li>Amabélia Rodrigues
<ul>
<li>Instituto Nacional de Saaúde Paública, Ministaério da Saaúde Paública, Bissau, Guinaé-Bissau</li>
</ul></li>
<li>João Dinis
<ul>
<li>Instituto Nacional de Saaúde Paública, Ministaério da Saaúde Paública, Bissau, Guinaé-Bissau</li>
</ul></li>
<li>Marco Pombi
<ul>
<li>Istituto Pasteur Italia-Fondazione Cenci Bolognetti, Dipartimento di Sanita Pubblica e Malattie Infettive, Università di Roma SAPIENZA, Rome, Italy.</li>
</ul></li>
<li>Beniamino Caputo (beniamino.caputo@uniroma1.it)
<ul>
<li>Istituto Pasteur Italia-Fondazione Cenci Bolognetti, Dipartimento di Sanita Pubblica e Malattie Infettive, Università di Roma SAPIENZA, Rome, Italy.</li>
</ul></li>
<li>João Pinto (jpinto@ihmt.unl.pt)
<ul>
<li>Global Health and Tropical Medicine, GHTM, Instituto de Higiene e Medicina Tropical, IHMT, Universidade Nova de Lisboa, UNL, Rua da Junqueira 100, 1349-008 Lisbon, Portugal.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="guinea-and-mali-1">Guinea and Mali (1)</h3>
<h4 id="sample-sets-13">Sample sets</h4>
<p><code>AG1000G-GN-A</code>, <code>AG1000G-GN-B</code></p>
<h4 id="study-information-13">Study information</h4>
<p>Collections were made from four different study sites around the border between Guinea and Mali.
From Mali; Takan <strong><a href="https://www.openstreetmap.org/?mlat=11.47&amp;mlon=-8.33&amp;zoom=12&amp;layers=M">(11.47,-8.33)</a></strong>
and Toumani Oulena <strong><a href="https://www.openstreetmap.org/?mlat=10.83&amp;mlon=-7.81&amp;zoom=12&amp;layers=M">(10.83,-7.81)</a></strong>
are both small villages in the Yanfolila district of southern Mali and represent the Sudanian savannah ecological zone.
Takan is arid savannah, while Toumani Oulena is humid savannah.
In Guinea Conakry, mosquitoes were sampled from Koraboh <strong><a href="https://www.openstreetmap.org/?mlat=9.28&amp;mlon=-10.03&amp;zoom=12&amp;layers=M">(9.28,-10.03)</a></strong>,
a small village in the Kissidougou district in the Faranah region representing a semi-forest site with intermediate ecology,
a mix of savannah and forest, and in Koundara <strong><a href="https://www.openstreetmap.org/?mlat=8.48&amp;mlon=-9.53&amp;zoom=12&amp;layers=M">(8.48,-9.53)</a></strong>,
a small village in the Macenta district in the Nzerekore region representing deep forest ecology.
All reported collections occurred in October and November 2012.
At each site, mosquitoes were collected using three different methods: human-landing capture,
indoor manual aspirator or pyrethroid spray catch,
and larval capture - where the first and second instar larvae were raised to adult in a field insectary
under standard insectary conditions prior to DNA isolation from the adults,
and the third and fourth instar larvae were preserved directly for DNA isolation,
without rearing in the insectary.</p>
<p>The two distinct methods of larval collection were used to
control for possible genetic bias inherent in lab rearing of captured larvae.
Across sites, all types of larval sites were sampled,
including both temporary and permanent sites.
Human-landing captures were performed both inside dwellings
and outside (&gt;10 m from dwelling) at night between 18:00 and 06:30.
The indoor aspirator or spray catches were done in the morning between 06:00 and 12:00.
Adult specimens or third and fourth instar larvae were preserved immediately in 80% ethanol until later DNA extraction.
First and second instar larvae were raised to adults in nearby field insectaries and upon emergence were preserved in 80% ethanol.
DNA was extracted from mosquitoes using DNAzol by the provided protocol (Invitrogen, CA, USA).</p>
<h4 id="related-publications-6">Related publications</h4>
<p>Coulibaly, Boubacar, et al. “Malaria Vector Populations across Ecological Zones in Guinea Conakry and Mali, West Africa.” Malaria Journal, vol. 15, no. 1, 8 Apr. 2016, 10.1186/s12936-016-1242-5. <span class="citation" data-cites="122DDCVbs">[<a href="#ref-122DDCVbs" role="doc-biblioref">15</a>]</span></p>
<h4 id="contributors-13">Contributors</h4>
<ul>
<li>Boubacar Coulibaly
<ul>
<li>Malaria Research and Training Centre (MRTC), Faculty of Medicine and Dentistry, University of Mali, BP: E 423 Bamako, Mali</li>
</ul></li>
<li>Kenneth D. Vernick (kenneth.vernick@pasteur.fr)
<ul>
<li>Unit for Genetics and Genomics of Insect Vectors, Institut Pasteur, 75015, Paris, France.</li>
</ul></li>
<li>Michelle M. Riehle
<ul>
<li>Department of Microbiology and Immunology, Medical College of Wisconsin, Milwaukee, WI 53226, USA.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="mali-2">Mali (2)</h3>
<h4 id="sample-sets-14">Sample sets</h4>
<p><code>AG1000G-ML-A</code></p>
<h4 id="study-information-14">Study information</h4>
<p>Collections were made in four villages in the Koulikoro region;
Tieneguebougou <strong><a href="https://www.openstreetmap.org/?mlat=12.810&amp;mlon=-8.080&amp;zoom=12&amp;layers=M">(12.810,-8.080)</a></strong> approximately 20 km north of Bamako,
and Kababougou <strong><a href="https://www.openstreetmap.org/?mlat=12.890&amp;mlon=-8.150&amp;zoom=12&amp;layers=M">(12.890, -8.150)</a></strong>,
Ouassorola <strong><a href="https://www.openstreetmap.org/?mlat=12.900&amp;mlon=-8.160&amp;zoom=12&amp;layers=M">(12.900, -8.160)</a></strong>,
Sogolombougou <strong><a href="https://www.openstreetmap.org/?mlat=12.880&amp;mlon=-8.140&amp;zoom=12&amp;layers=M">(12.880, -8.140)</a></strong>, approximately 30 km north of Bamako.</p>
<p>The collections were made in August 2014 by human landing catch and pyrethrum spray catch.
Both <em>An. gambiae</em> and <em>An. coluzzii</em> <span class="citation" data-cites="1G04AzsFo">[<a href="#ref-1G04AzsFo" role="doc-biblioref">8</a>]</span> were collected.
Specimens were stored in 80% ethanol.</p>
<h4 id="external-resources-1">External resources</h4>
<ul>
<li><strong><a href="%5Bhttps://targetmalaria.org/">https://targetmalaria.org/</a></strong></li>
</ul>
<h4 id="contributors-14">Contributors</h4>
<ul>
<li>Mamadou B. Coulibaly
<ul>
<li>Malaria Research and Training Centre (MRTC), Faculty of Medicine and Dentistry, University of Mali, BP: E 423 Bamako, Mali</li>
</ul></li>
<li>Amadou Guindo
<ul>
<li>Malaria Research and Training Centre (MRTC), Faculty of Medicine and Dentistry, University of Mali, BP: E 423 Bamako, Mali</li>
</ul></li>
<li>Sidy Doumbia
<ul>
<li>Malaria Research and Training Centre (MRTC), Faculty of Medicine and Dentistry, University of Mali, BP: E 423 Bamako, Mali</li>
</ul></li>
<li>Samantha O’Loughlin (s.oloughlin@imperial.ac.uk)
<ul>
<li>Department of Life Sciences, Imperial College, Silwood Park, Ascot, Berkshire SL5 7PY, UK.</li>
</ul></li>
<li>Austin Burt
<ul>
<li>Department of Life Sciences, Imperial College, Silwood Park, Ascot, Berkshire SL5 7PY, UK.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="mali-3">Mali (3)</h3>
<h4 id="sample-sets-15">Sample sets</h4>
<p><code>AG1000G-ML-B</code></p>
<h4 id="study-information-15">Study information</h4>
<p>Collections of indoor resting adults were made by spray catch from seven villages in the southern part of Mali in August-September 2004:
Banambani <strong><a href="">(12.800, -8.050)</a></strong>,
Bancoumana <strong><a href="">(12.200,-8.200)</a></strong>,
Douna <strong><a href="">(13.210, -5.900)</a></strong>,
Fanzana <strong><a href="">(13.200, -6.130)</a></strong>,
Kela <strong><a href="">(11.880, -8.450)</a></strong>,
Moribobougou <strong><a href="">(12.690, -7.870)</a></strong> and
N’Gabakoro <strong><a href="">(12.680, -7.840)</a></strong>.
Specimens were sorted morphologically to <em>An. gambiae s.l.</em>.
Ovaries of half-gravid females were dissected and placed in numbered individual micro-tubes containing
modified Carnoy’s solution (1:3 glacial acetic acid: 100% ethanol).
Carcasses were placed in correspondingly numbered micro-tubes over desiccant.
Genomic DNA was isolated from individual mosquitoes using one of the following:
DNeasy Extraction Kit (Qiagen, Valencia, CA), Puregene kit (Gentra Systems, Inc., Minneapolis, MN),
DNAzol kit (Molecular Research Center, Inc., Cincinnati, OH.) or Easy-DNA kit (Invitrogen, Carlsbad, CA).</p>
<p><em>An. gambiae s.s.</em> and its molecular forms were identified using one of two rDNA-based PCR/RFLP assays <span class="citation" data-cites="1G04AzsFo">[<a href="#ref-1G04AzsFo" role="doc-biblioref">8</a>]</span>.
Ovaries from specimens of the desired species were subject to polytene chromosome analysis.</p>
<h4 id="related-publications-7">Related publications</h4>
<ul>
<li>Coulibaly, Mamadou B., et al. “PCR-Based Karyotyping of Anopheles Gambiae Inversion 2Rj Identifies the BAMAKO Chromosomal Form.” Malaria Journal, vol. 6, 1 Oct. 2007, p. 133, pubmed.ncbi.nlm.nih.gov/17908310/, 10.1186/1475-2875-6-133. <span class="citation" data-cites="vmUWM43r">[<a href="#ref-vmUWM43r" role="doc-biblioref">31</a>]</span></li>
</ul>
<h4 id="contributors-15">Contributors</h4>
<ul>
<li>Mamadou B. Coulibaly
<ul>
<li>Malaria Research and Training Centre (MRTC), Faculty of Medicine and Dentistry, University of Mali, BP: E 423 Bamako, Mali</li>
</ul></li>
<li>Moussa Keita
<ul>
<li>Malaria Research and Training Centre (MRTC), Faculty of Medicine and Dentistry, University of Mali, BP: E 423 Bamako, Mali</li>
</ul></li>
<li>Cheick Traore
<ul>
<li>Malaria Research and Training Centre (MRTC), Faculty of Medicine and Dentistry, University of Mali, BP: E 423 Bamako, Mali</li>
</ul></li>
<li>Nora J. Besansky (nbesansk@nd.edu)
<ul>
<li>Eck Institute for Global Health &amp; Department of Biological Sciences, University of Notre Dame, IN 46556, USA.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="kenya">Kenya</h3>
<h4 id="sample-sets-16">Sample sets</h4>
<p><code>AG1000G-KE</code></p>
<h4 id="study-information-16">Study information</h4>
<p>Kenyan specimens were obtained from villages located in Kilifi County near the Kenyan coast between 2000 and 2014.
All <em>Anopheles</em> mosquito sampling was conducted indoors using CDC light traps which were hung at 6pm and collected
at 6am the following morning during the rainy season in September.
Specimens were stored in 80% ethanol.
All specimens contributed to the project were identified as <em>An. gambiae</em>
using the species complex diagnostic assay of <span class="citation" data-cites="14yZJJBPD">[<a href="#ref-14yZJJBPD" role="doc-biblioref">20</a>]</span>.
<em>An. gambiae</em>, <em>An. funestus</em>, <em>An. arabiensis</em>, <em>An. merus</em> were present at sampling locations.
Sporozoite rates for the area during previous studies were 1.47% <span class="citation" data-cites="1C7qEzk6S">[<a href="#ref-1C7qEzk6S" role="doc-biblioref">32</a>]</span>.</p>
<h4 id="related-publications-8">Related publications</h4>
<ul>
<li>Midega, Janet T., et al. “Wind Direction and Proximity to Larval Sites Determines Malaria Risk in Kilifi District in Kenya.” Nature Communications, vol. 3, no. 1, Jan. 2012, 10.1038/ncomms1672. <span class="citation" data-cites="1C7qEzk6S">[<a href="#ref-1C7qEzk6S" role="doc-biblioref">32</a>]</span></li>
</ul>
<h4 id="contributors-16">Contributors</h4>
<ul>
<li>Janet Midega (tjmidega@yahoo.com)
<ul>
<li>KEMRI-Wellcome Trust Research Programme, 80108 Kilifi, Kenya.</li>
</ul></li>
<li>Joseph Mwangangi
<ul>
<li>KEMRI-Wellcome Trust Research Programme, 80108 Kilifi, Kenya.</li>
</ul></li>
<li>Charles Mbogo
<ul>
<li>KEMRI-Wellcome Trust Research Programme, 80108 Kilifi, Kenya.</li>
</ul></li>
<li>Philip Bejon
<ul>
<li>KEMRI-Wellcome Trust Research Programme, 80108 Kilifi, Kenya.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="malawi">Malawi</h3>
<h4 id="sample-sets-17">Sample sets</h4>
<p><code>AG1000G-MW</code></p>
<h4 id="study-information-17">Study information</h4>
<p>Specimens were obtained from villages within the catchment of the Majete Malaria Project,
Chikhwawa District, Malawi <strong><a href="https://www.openstreetmap.org/?mlat=-15.933&amp;mlon=34.755&amp;zoom=12&amp;layers=M">(-15.933, 34.755)</a></strong>
<span class="citation" data-cites="xsFRxYp6">[<a href="#ref-xsFRxYp6" role="doc-biblioref">33</a>]</span>.
Mosquitoes were collected indoors and outdoors by Suna light trap from April through August 2015.
Chickhwawa District is an area with perennial and intense malaria transmission <span class="citation" data-cites="vckO98IK">[<a href="#ref-vckO98IK" role="doc-biblioref">34</a>]</span>
All specimens were <em>An. arabiensis</em> <span class="citation" data-cites="1G04AzsFo">[<a href="#ref-1G04AzsFo" role="doc-biblioref">8</a>]</span>.
Specimens were stored over silica and DNA was extracted using the Qiagen plate protocol.</p>
<h4 id="related-publications-9">Related publications</h4>
<ul>
<li>McCann, Robert S., et al. “Assessment of the Effect of Larval Source Management and House Improvement on Malaria Transmission When Added to Standard Malaria Control Strategies in Southern Malawi: Study Protocol for a Cluster-Randomised Controlled Trial.” BMC Infectious Diseases, vol. 17, no. 1, 22 Sept. 2017, 10.1186/s12879-017-2749-2. <span class="citation" data-cites="xsFRxYp6">[<a href="#ref-xsFRxYp6" role="doc-biblioref">33</a>]</span></li>
</ul>
<h4 id="external-resources-2">External resources</h4>
<ul>
<li><a href="https://mesamalaria.org/mesa-track/majete-integrated-malaria-control-project-mmp" class="uri">https://mesamalaria.org/mesa-track/majete-integrated-malaria-control-project-mmp</a></li>
</ul>
<h4 id="contributors-17">Contributors</h4>
<ul>
<li>Robert McCann
<ul>
<li>College of Medicine, University of Malawi, Blantyre, Malawi.</li>
<li>Laboratory of Entomology, Wageningen University and Research, Wageningen, The Netherlands.</li>
<li>Center for Vaccine Development and Global Health, University of Maryland School of Medicine, Baltimore, USA.</li>
</ul></li>
<li>Themba Mzilahowa
<ul>
<li>Department of Vector Biology, Liverpool School of Tropical Medicine, Liverpool, United Kingdom.</li>
<li>Malawi-Liverpool Wellcome Trust Clinical Research Programme, Blantyre, Malawi.</li>
<li>Malaria Alert Centre (MAC), P/ Bag 360, Chichiri, Blantyre 3, Malawi.</li>
</ul></li>
<li>Alinune Kabaghe
<ul>
<li>College of Medicine, University of Malawi, Blantyre, Malawi.</li>
</ul></li>
<li>Willem Takken
<ul>
<li>Laboratory of Entomology, Wageningen University and Research, Wageningen, The Netherlands.</li>
</ul></li>
<li>Martin J. Donnelly (Martin.Donnelly@lstmed.ac.uk)
<ul>
<li>Department of Vector Biology, Liverpool School of Tropical Medicine, Liverpool, United Kingdom.</li>
<li>Parasites and Microbes Programme, Wellcome Sanger Institute, Hinxton, Cambridge CB10 1SA, UK.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="mayotte-island">Mayotte Island</h3>
<h4 id="sample-sets-18">Sample sets</h4>
<p><code>AG1000G-FR</code></p>
<h4 id="study-information-18">Study information</h4>
<p>Collections were taken from multiple sites on the island of Mayotte.
Samples were collected as larvae during March-April 2011 in temporary pools by dipping.
Sites included Mtsanga Charifou <strong><a href="https://www.openstreetmap.org/?mlat=-12.991&amp;mlon=45.156&amp;zoom=12&amp;layers=M">(-12.991, 45.156)</a></strong>
and Combani <strong><a href="https://www.openstreetmap.org/?mlat=-12.779&amp;mlon=45.143&amp;zoom=12&amp;layers=M">(-12.779, 45.143)</a></strong>.
Larvae were stored in 80% ethanol prior to DNA extraction.
All specimens contributed were <em>An. gambiae s.s.</em> <span class="citation" data-cites="mTkFlBOi">[<a href="#ref-mTkFlBOi" role="doc-biblioref">13</a>]</span> with the standard 2L<sup>a+</sup>/2L<sup>a+</sup> or inverted 2L<sup>a</sup>/2L<sup>a</sup>
karyotype as determined by the molecular PCR diagnostics <span class="citation" data-cites="WKPmPPDE">[<a href="#ref-WKPmPPDE" role="doc-biblioref">21</a>]</span>.
Samples were identified as males or females by the sequencing read coverage of the X chromosome
using <code>LookSeq</code> <span class="citation" data-cites="19Z5CFAYk">[<a href="#ref-19Z5CFAYk" role="doc-biblioref">35</a>]</span>.</p>
<h4 id="contributors-18">Contributors</h4>
<ul>
<li>Gilbert Le Go
<ul>
<li>UMR MIVEGEC, Univ. Montpellier, CNRS, IRD, Montpellier, France.</li>
</ul></li>
<li>Vincent Robert (gilbert.lego@ird.fr)
<ul>
<li>UMR MIVEGEC, Univ. Montpellier, CNRS, IRD, Montpellier, France.</li>
</ul></li>
<li>Maryam Kamali
<ul>
<li>Department of Medical Entomology and Parasitology, Faculty of Medical Sciences, Tarbiat Modares University, Tehran, Iran.</li>
<li>Department of Entomology, Virginia Tech, Blacksburg, Virginia 24061, USA.</li>
</ul></li>
<li>Igor Sharakhov (igor@vt.edu)
<ul>
<li>Department of Entomology, Virginia Tech, Blacksburg, Virginia 24061, USA.</li>
<li>Department of Genetics and Cell Biology, Tomsk State University, Tomsk 634050, Russia.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="mozambique">Mozambique</h3>
<h4 id="sample-sets-19">Sample sets</h4>
<p><code>AG1000G-MZ</code></p>
<h4 id="study-information-19">Study information</h4>
<p>Mosquito samples were collected in Furvela <strong><a href="https://www.openstreetmap.org/?mlat=-23.716&amp;mlon=35.299&amp;zoom=12&amp;layers=M">(-23.716, 35.299)</a></strong>,
Mozambique, by CDC light traps between December 2003 and April 2004.
Specimens were stored on silica gel and DNA was extracted according to <span class="citation" data-cites="pSmuJkYQ">[<a href="#ref-pSmuJkYQ" role="doc-biblioref">36</a>]</span>.
Contributed specimens consisted of individuals identified according to <span class="citation" data-cites="1G04AzsFo">[<a href="#ref-1G04AzsFo" role="doc-biblioref">8</a>]</span>.
Furvela is a rural village located in Inhambane Province, where malaria is transmitted mainly by <em>An. gambiae</em> and <em>An. funestus</em> <span class="citation" data-cites="f8uRMyY6">[<a href="#ref-f8uRMyY6" role="doc-biblioref">37</a>]</span>.
<em>An. arabiensis</em> and <em>An. merus</em> are also found at low frequency.
Sporozoite rates around 4% have been reported in <em>An. gambiae</em> from Furvela <span class="citation" data-cites="f8uRMyY6">[<a href="#ref-f8uRMyY6" role="doc-biblioref">37</a>]</span>.</p>
<h4 id="contributors-19">Contributors</h4>
<ul>
<li>João Pinto (jpinto@ihmt.unl.pt)
<ul>
<li>Global Health and Tropical Medicine, GHTM, Instituto de Higiene e Medicina Tropical, IHMT, Universidade Nova de Lisboa, UNL, Rua da Junqueira 100, 1349-008 Lisbon, Portugal.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="tanzania">Tanzania</h3>
<h4 id="sample-sets-20">Sample sets</h4>
<p><code>AG1000G-TZ</code></p>
<h4 id="study-information-20">Study information</h4>
<p>Tanzanian samples were collected from four distinct locations.
Moshi samples came from lower Mabogini <strong><a href="https://www.openstreetmap.org/?mlat=-3.400&amp;mlon=37.350&amp;zoom=12&amp;layers=M">(-3.400, 37.350)</a></strong>,
rice fields near lower Moshi on the southern slope of Mount Kilimanjaro,
a region shown to have increasing resistance to pyrethroids <span class="citation" data-cites="FVN9nrFP">[<a href="#ref-FVN9nrFP" role="doc-biblioref">38</a>]</span>.
Mosquitoes were collected as larvae, during the rice growing season in August-September 2012,
raised to adults and females bioassayed in WHO tubes for one hour with 0.05% lambda cyhalothrin <span class="citation" data-cites="BIuyPabi">[<a href="#ref-BIuyPabi" role="doc-biblioref">39</a>]</span>.
Alive and dead mosquitoes were preserved over silica.
In Tanzanian samples screened in Kabula et al. <span class="citation" data-cites="ca2mjT0Y">[<a href="#ref-ca2mjT0Y" role="doc-biblioref">40</a>]</span>.
Moshi was the most pyrethroid resistant population,
they were found to be completely DDT susceptible,
only in one out of 642 mosquitoes assayed by <span class="citation" data-cites="FVN9nrFP">[<a href="#ref-FVN9nrFP" role="doc-biblioref">38</a>]</span>.
was found to carry a <em>kdr</em> resistance mutation (Vgsc-995F).</p>
<p>Tarime collections took place in the village of Komaswa <strong><a href="https://www.openstreetmap.org/?mlat=-1.417&amp;mlon=34.183&amp;zoom=12&amp;layers=M">(-1.417, 34.183)</a></strong>
about 410 km north west of Moshi, during August 2012.
Mosquito larvae were collected, raised to adults and females bioassayed
with a range of insecticides in WHO tubes for one hour <span class="citation" data-cites="BIuyPabi">[<a href="#ref-BIuyPabi" role="doc-biblioref">39</a>]</span>.
finding almost complete multi-insecticide susceptibility:
permethrin (100% mortality), lambda cyhalothrin (97%), fenitrothion (100%),
DDT (100%) and bendiocarb (100%) <span class="citation" data-cites="1HPbPGiiF">[<a href="#ref-1HPbPGiiF" role="doc-biblioref">41</a>]</span>.</p>
<p>Muheza samples were collected from Zeneti village <strong><a href="https://www.openstreetmap.org/?mlat=-5.217&amp;mlon=38.650&amp;zoom=12&amp;layers=M">(-5.217, 38.650)</a></strong>,
northeast Tanzania.
Malaria is intense and perennial with transmission peaking after the rainy season in May and June <span class="citation" data-cites="ca2mjT0Y">[<a href="#ref-ca2mjT0Y" role="doc-biblioref">40</a>]</span>.
Mosquitoes were sampled between November 2012 and May 2013.
Indoor resting collections were used to obtain live females for deltamethrin susceptibility testing
and pyrethrum spray catches were used for mosquitoes that were collected for blood meal analysis.
Collections were conducted between 06:00 and 09:00 from randomly selected houses.
Live mosquitoes collected for susceptibility testing were provided with
10% glucose solution and transported to the field insectary.
Mosquitoes were sorted and morphologically identified to species,
carcasses were stored individually over desiccant for laboratory processing.</p>
<p>Muleba <strong><a href="https://www.openstreetmap.org/?mlat=1.750&amp;mlon=31.667&amp;zoom=12&amp;layers=M">(1.750, 31.667)</a></strong>,
the final collection region, is in the North-western part of Tanzania.
The district is known to be a malaria epidemic prone area with unstable transmission of varying seasonality.
The highest peak of malaria transmission is usually reached between May-July and November-January,
which results from proceeding rain seasons.
There have been malaria vector control efforts since 2007
when indoor residual spraying using lambda cyhalothrin was introduced.
Insecticide resistance in this district is coupled with high frequency of
<em>kdr</em> pyrethroid target site mutations in the <em>An. gambiae s.s.</em> population <span class="citation" data-cites="FVN9nrFP dB1ebxOj">[<a href="#ref-FVN9nrFP" role="doc-biblioref">38</a>,<a href="#ref-dB1ebxOj" role="doc-biblioref">42</a>]</span>.
Sampling was conducted over six months,
which include both dry and rainy season and covers 6 villages selected to represent all major ecological systems in the district.</p>
<h4 id="contributors-20">Contributors</h4>
<ul>
<li>Bilali Kabula (Bika72@gmail.com)
<ul>
<li>National Institute for Medical Research (NIMR), Dar es Salaam, Tanzania.</li>
</ul></li>
<li>Johnson Matowo
<ul>
<li>Kilimanjaro Christian Medical University College, Tumaini University, Makumira, Tanzania.</li>
</ul></li>
<li>David Weetman (david.weetman@lstmed.ac.uk)
<ul>
<li>Department of Vector Biology, Liverpool School of Tropical Medicine, Liverpool, United Kingdom.</li>
</ul></li>
<li>Martin J. Donnelly (Martin.Donnelly@lstmed.ac.uk)
<ul>
<li>Department of Vector Biology, Liverpool School of Tropical Medicine, Liverpool, United Kingdom.</li>
<li>Parasites and Microbes Programme, Wellcome Sanger Institute, Hinxton, Cambridge CB10 1SA, UK.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="the-gambia-1">The Gambia (1)</h3>
<h4 id="sample-sets-21">Sample sets</h4>
<p><code>AG1000G-GM-A</code></p>
<h4 id="study-information-21">Study information</h4>
<p>Indoor resting female mosquitoes were collected by pyrethrum spray catch from four hamlets around
Njabakunda <strong><a href="https://www.openstreetmap.org/?mlat=13.55&amp;mlon=-15.90&amp;zoom=12&amp;layers=M">(13.55, -15.90)</a></strong>, North Bank Region,
The Gambia between August and October 2011.
The four hamlets were Maria Samba Nyado, Sare Illo Buya, Kerr Birom Kardo, and Kerr Sama Kuma; all are within 1 km of each other.
This is an area of unusually high hybridization rates
between <em>An. gambiae s.s.</em> and <em>An. coluzzii</em> <span class="citation" data-cites="1EjgFGIf9 mJ3VTNed">[<a href="#ref-1EjgFGIf9" role="doc-biblioref">43</a>,<a href="#ref-mJ3VTNed" role="doc-biblioref">44</a>]</span>.
Njabakunda village is approximately 30km to the west of Farafenni town and 4km away from the Gambia River.
The vegetation is a mix of open savannah woodland and farmland.
With apparent high gene-flow in the region, it is problematic to assign species to these samples.</p>
<h4 id="related-publications-10">Related publications</h4>
<ul>
<li>Nwakanma, Davis C., et al. “Breakdown in the Process of Incipient Speciation in Anopheles Gambiae.” Genetics, vol. 193, no. 4, 1 Apr. 2013, pp. 1221–1231, www.ncbi.nlm.nih.gov/pmc/articles/PMC3606099/, 10.1534/genetics.112.148718. <span class="citation" data-cites="mJ3VTNed">[<a href="#ref-mJ3VTNed" role="doc-biblioref">44</a>]</span></li>
</ul>
<h4 id="contributors-21">Contributors</h4>
<ul>
<li>Davis C. Nwakanma
<ul>
<li>Medical Research Council Unit, The Gambia at the London School of Hygiene &amp; Tropical Medicine (MRCG at LSHTM), Banjul, The Gambia.</li>
</ul></li>
<li>Musa Jawara
<ul>
<li>Medical Research Council Unit, The Gambia at the London School of Hygiene &amp; Tropical Medicine (MRCG at LSHTM), Banjul, The Gambia.</li>
</ul></li>
<li>David Conway
<ul>
<li>London School of Hygiene and Tropical Medicine, Keppel Street, Bloomsbury, London WC1E 7HT, UK.</li>
</ul></li>
<li>Martin J. Donnelly (Martin.Donnelly@lstmed.ac.uk)
<ul>
<li>Department of Vector Biology, Liverpool School of Tropical Medicine, Liverpool, United Kingdom.</li>
<li>Parasites and Microbes Programme, Wellcome Sanger Institute, Hinxton, Cambridge CB10 1SA, UK.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="the-gambia-2">The Gambia (2)</h3>
<h4 id="sample-sets-22">Sample sets</h4>
<p><code>AG1000G-GM-B</code></p>
<h4 id="study-information-22">Study information</h4>
<p>Specimens were collected along the Gambia River from the western coastal region of The Gambia <span class="citation" data-cites="1EjgFGIf9">[<a href="#ref-1EjgFGIf9" role="doc-biblioref">43</a>]</span>, in August 2006.
<em>An. gambiae</em> and <em>An. coluzzi</em> specimens were identified to species following the PCR-RFLP protocol <span class="citation" data-cites="1G04AzsFo">[<a href="#ref-1G04AzsFo" role="doc-biblioref">8</a>]</span> using DNA extracted from the mosquito leg.
Only <em>An. coluzzii</em> specimens were collected from villages of
Tankular <strong><a href="">(13.417, -16.033)</a></strong>
and Kalataba <strong><a href="">(13.550, -15.617)</a></strong>.
<em>An. gambiae</em> and <em>An. coluzzii</em> specimens were found in sympatry and collected from villages of
Yallal Tankonjala <strong><a href="">(13.550, -15.700)</a></strong>,
Sare Samba Sowe <strong><a href="">(13.583, -15.900)</a></strong> and
Hamdalai <strong><a href="">(13.567, -16.0167)</a></strong>.
PCR-RFLP protocol also revealed the presence of mosquitoes with hybrid <em>An. gambiae/coluzzii</em> genotype in Yallal Tankonjala and Sare Samba Sowe.
Collections of indoor daytime-resting half gravid mosquitoes were carried out mainly in human dwellings and, in few cases, in animal shelters.
Collections were carried out by pyrethroid and/or paper-cup mouth aspirators from 12 AM to sunset, and kept in vials with desiccant.
Ovaries were dissected, maintained into Carnoy fixative (three parts pure ethanol:one part glacial acetic acid)
and stored at -20<sup>o</sup>C before polytene chromosome preparations <span class="citation" data-cites="1EjgFGIf9">[<a href="#ref-1EjgFGIf9" role="doc-biblioref">43</a>]</span>.
Chromosome scoring was carried out under a phase‐contrast optical microscope.
Paracentric inversion karyotypes were scored according to the nomenclature and conventions
of <span class="citation" data-cites="eTFJNJEG Z6MC0qXg">[<a href="#ref-eTFJNJEG" role="doc-biblioref">45</a>,<a href="#ref-Z6MC0qXg" role="doc-biblioref">46</a>]</span>.</p>
<h4 id="related-publications-11">Related publications</h4>
<ul>
<li>Caputo, Beniamino, et al. “Anopheles Gambiae Complex along the Gambia River, with Particular Reference to the Molecular Forms of An. Gambiae S.s.” Malaria Journal, vol. 7, no. 1, 2008, p. 182, 10.1186/1475-2875-7-182. <span class="citation" data-cites="1EjgFGIf9">[<a href="#ref-1EjgFGIf9" role="doc-biblioref">43</a>]</span></li>
</ul>
<h4 id="contributors-22">Contributors</h4>
<ul>
<li>Beniamino Caputo (beniamino.caputo@uniroma1.it)
<ul>
<li>Istituto Pasteur Italia-Fondazione Cenci Bolognetti, Dipartimento di Sanita Pubblica e Malattie Infettive, Università di Roma SAPIENZA, Rome, Italy.</li>
</ul></li>
<li>Musa Jawara
<ul>
<li>Medical Research Council Unit, The Gambia at the London School of Hygiene &amp; Tropical Medicine (MRCG at LSHTM), Banjul, The Gambia.</li>
</ul></li>
<li>Davis C. Nwakanma
<ul>
<li>Medical Research Council Unit, The Gambia at the London School of Hygiene &amp; Tropical Medicine (MRCG at LSHTM), Banjul, The Gambia.</li>
</ul></li>
<li>David Conway
<ul>
<li>London School of Hygiene and Tropical Medicine, Keppel Street, Bloomsbury, London WC1E 7HT, UK.</li>
</ul></li>
<li>Alessandra della Torre (alessandra.dellatorre@uniroma1.it)
<ul>
<li>Istituto Pasteur Italia Fondazione Cenci Bolognetti, Dipartimento di Sanita Pubblica e Malattie Infettive, Università di Roma SAPIENZA, Rome, Italy.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="the-gambia-3">The Gambia (3)</h3>
<h4 id="sample-sets-23">Sample sets</h4>
<p><code>AG1000G-GM-C</code></p>
<h4 id="study-information-23">Study information</h4>
<p>Adult mosquitoes were collected at Wali Kunda in the rural,
central river region of The Gambia <strong><a href="https://www.openstreetmap.org/?mlat=13.567&amp;mlon=-14.917&amp;zoom=12&amp;layers=M">(13.567, -14.917)</a></strong>.
The area is 180 km from the sea, on the south bank of the River Gambia,
in flat Sudan savannah with a small fishing village (and a research field station)
as well as rice fields and swamplands.
The dominant <em>Anopheles</em> vector species in this region is <em>An. coluzzii</em> <span class="citation" data-cites="Px2G7gld">[<a href="#ref-Px2G7gld" role="doc-biblioref">47</a>]</span>.
Mosquitoes were captured using human landing collections both inside
and outside huts for 19 days in October and November 2012.
Mosquitoes were stored in RNAlater or dried over silica gel and stored at -20<sup>o</sup>C.</p>
<h4 id="contributors-23">Contributors</h4>
<ul>
<li>Danica Fabrigar
<ul>
<li>Department of Zoology, University of Oxford, Oxford OX1 3SZ, UK.</li>
</ul></li>
<li>Musa Jawara
<ul>
<li>Medical Research Council Unit, The Gambia at the London School of Hygiene &amp; Tropical Medicine (MRCG at LSHTM), Banjul, The Gambia.</li>
</ul></li>
<li>Steve Lindsay
<ul>
<li>Department of Biosciences, Durham University, Durham, UK.</li>
</ul></li>
<li>Umberto D’Alessandro
<ul>
<li>Medical Research Council Unit, The Gambia at the London School of Hygiene &amp; Tropical Medicine (MRCG at LSHTM), Banjul, The Gambia.</li>
</ul></li>
<li>H. Charles J. Godfray (charles.godfray@oxfordmartin.ox.ac.uk)
<ul>
<li>Department of Zoology, University of Oxford, Oxford OX1 3SZ, UK.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="uganda">Uganda</h3>
<h4 id="sample-sets-24">Sample sets</h4>
<p><code>AG1000G-UG</code></p>
<h4 id="study-information-24">Study information</h4>
<p>Specimens were obtained from two locations in Uganda: Nagongera,
30km to the North of Lake Victoria near the border with Kenya,
and Kihihi, in the very South-West of the country.
In Nagongera, Tororo District <strong><a href="https://www.openstreetmap.org/?mlat=0.77&amp;mlon=34.026&amp;zoom=12&amp;layers=M">(0.77, 34.026)</a></strong>,
mosquitoes were collected by CDC light trap, resting and window trap collections, during October 2012.
This is an area of intense perennial malaria transmission <span class="citation" data-cites="m3y4gwzb">[<a href="#ref-m3y4gwzb" role="doc-biblioref">48</a>]</span>.
Additional details of vector population bionomics may be found in <span class="citation" data-cites="7D6mNysA 19o13dRo9 16MD5kWB3 QHmnxjQ3">[<a href="#ref-7D6mNysA" role="doc-biblioref">49</a>,<a href="#ref-19o13dRo9" role="doc-biblioref">50</a>,<a href="#ref-16MD5kWB3" role="doc-biblioref">51</a>,<a href="#ref-QHmnxjQ3" role="doc-biblioref">52</a>]</span>.
Specimens were stored in 80% ethanol and DNA was extracted using the Qiagen plate protocol.
In Kihihi subcounty, Kanungu District <strong><a href="https://www.openstreetmap.org/?mlat=-0.751&amp;mlon=29.701&amp;zoom=12&amp;layers=M">(-0.751, 29.701)</a></strong>,
resting mosquitoes were collected during October and November 2012.
Kihihi is located in an upland area with seasonal malaria transmission <span class="citation" data-cites="m3y4gwzb">[<a href="#ref-m3y4gwzb" role="doc-biblioref">48</a>]</span>.
Specimens were stored in 80% ethanol and DNA was extracted using the Qiagen plate protocol.
All specimens from both collections were <em>An. gambiae</em> <span class="citation" data-cites="1G04AzsFo">[<a href="#ref-1G04AzsFo" role="doc-biblioref">8</a>]</span>.</p>
<h4 id="related-publications-12">Related publications</h4>
<ul>
<li>Kilama, Maxwell, et al. “Estimating the Annual Entomological Inoculation Rate for Plasmodium Falciparum Transmitted by Anopheles Gambiae S.l. Using Three Sampling Methods in Three Sites in Uganda.” Malaria Journal, vol. 13, no. 1, 21 Mar. 2014, 10.1186/1475-2875-13-111. <span class="citation" data-cites="m3y4gwzb">[<a href="#ref-m3y4gwzb" role="doc-biblioref">48</a>]</span></li>
</ul>
<h4 id="contributors-24">Contributors</h4>
<ul>
<li>Henry D. Mawejje
<ul>
<li>Infectious Diseases Research Collaboration, 2C Nakasero Hill Road, PO Box 7475, Kampala, Uganda.</li>
</ul></li>
<li>Martin J. Donnelly (Martin.Donnelly@lstmed.ac.uk)
<ul>
<li>Department of Vector Biology, Liverpool School of Tropical Medicine, Liverpool, United Kingdom.</li>
<li>Parasites and Microbes Programme, Wellcome Sanger Institute, Hinxton, Cambridge CB10 1SA, UK.</li>
</ul></li>
<li>David Weetman (david.weetman@lstmed.ac.uk)
<ul>
<li>Department of Vector Biology, Liverpool School of Tropical Medicine, Liverpool, United Kingdom.</li>
</ul></li>
<li>Craig S. Wilding (C.S.Wilding@ljmu.ac.uk)
<ul>
<li>School of Biological and Environmental Sciences, Liverpool John Moores University, Liverpool L3 3AF, UK.</li>
</ul></li>
</ul>
<h3 class="page_break_before" id="colony-crosses">Colony crosses</h3>
<h4 id="sample-sets-25">Sample sets</h4>
<p><code>AG1000G-X</code></p>
<h4 id="study-information-25">Study information</h4>
<p>15 crosses were contributed to Ag1000G phase 3,
11 of which were previously released in Ag1000G phase 2.
Crosses were generated using parents from eight different colonies:
G3 (MRA-112); Kisumu (MRA-762); Pimperena (canonical representative of <em>An. gambiae</em> species; MRA-861);
Ghana (recent colony of <em>An. coluzzii</em> from Okyereko, southern Ghana <span class="citation" data-cites="z9HqxYF3">[<a href="#ref-z9HqxYF3" role="doc-biblioref">53</a>]</span>.
Mali-NIH (canonical representative of <em>An. coluzzii</em> species; Niono, MRA-860);
(P)Akron (Benin, MRA-913); Nagongera (Tororo, Uganda); and Tiassalé (southern Côte d’Ivoire <span class="citation" data-cites="z9HqxYF3">[<a href="#ref-z9HqxYF3" role="doc-biblioref">53</a>]</span>.
The cross family labels, <em>e.g.</em> <code>29-2</code>, are identifiers
used for each of the crosses within the contributor project and have no special meaning.</p>
<p><em>Anopheles gambiae</em> is a swarm-mater and crosses were therefore undertaken in mixed groups
involving 4-10 females from a single colony with 1-4 males from each from a different colony
in plastic cups covered with netting with 10% sugar water provided ad libitum.
Females were fed on human blood. After 3 days, males were removed and individually preserved in 95% ethanol.
Gravid or half-gravid females were then removed and placed in 1.5ml Eppendorf tubes.
Females that did not appear gravid were given a second blood meal before placing in Eppendorf tubes for egg laying.
Following egg deposition, females were removed and stored in tubes containing ethanol for subsequent DNA extraction.
Eggs were floated in clear plastic trays (15x10x5cm)
and following hatching, larvae were raised on finely-ground fish food (Tetramin).
Trays were checked daily and pupae were placed individually into small, labelled centrifuge tubes.
Offspring were removed on eclosion and stored in individual tubes containing ethanol.
DNA was extracted from parents and offspring using the Qiagen DNeasy kit.</p>
<p>A preliminary assessment of the father of each cross was obtained by genotyping
seven microsatellite loci in the mother, potential fathers and five or six offspring.
Where possible, the colony of origin of each father was established using individual
clustering of the mothers and fathers in BAPS version 5.2,
with cluster identity mapped to colony of origin via the mothers
(for which the colony was known) <span class="citation" data-cites="Rho6nYsT">[<a href="#ref-Rho6nYsT" role="doc-biblioref">4</a>]</span>.</p>
<p>The four crosses that are novel to phase 3: B5, K2, K4 and K6,
required further analysis to ascertain the true father of each cross,
given mother and offspring.
For each cross for which the father was in doubt, the list of potential parental pairs was determined.
For each of these pairs Mendelian error was computed for every
sample of the progeny and the median value (among samples) was plotted.
In these four crosses (B5, K2, K4 and K6) one pair yielded median Mendelian errors
significantly lower for every autosome than all other pairs,
identifying the parsimonious parents.
Two of the novel crosses, K4 and K6, were found to be fathered by the same male, <code>AC0398</code>.</p>
<h4 id="contributors-25">Contributors</h4>
<ul>
<li>David Weetman (david.weetman@lstmed.ac.uk)
<ul>
<li>Department of Vector Biology, Liverpool School of Tropical Medicine, Liverpool, United Kingdom.</li>
</ul></li>
<li>Craig S. Wilding (C.S.Wilding@ljmu.ac.uk)
<ul>
<li>School of Biological and Environmental Sciences, Liverpool John Moores University, Liverpool L3 3AF, UK.</li>
</ul></li>
</ul>
<h2 id="whole-genome-sequencing">Whole Genome Sequencing</h2>
<p>All library preparation and sequencing was performed at the Wellcome Sanger Institute.</p>
<p>Paired-end multiplex libraries were prepared using the manufacturer’s protocol, with the exception that genomic DNA was fragmented using Covaris Adaptive Focused Acoustics rather than nebulization.</p>
<p>Multiplexes comprised 12 tagged individual mosquitoes and three lanes of sequencing were generated for each multiplex to even out variations in yield between sequencing runs.</p>
<p>Cluster generation and sequencing were undertaken according to the manufacturer’s protocol for paired-end sequence reads with insert size in the range 100-200 bp.</p>
<p>4,693 individual mosquitoes were sequenced in total, of which 3,130 were sequenced using the Illumina HiSeq 2000 platform and 1,563 were sequenced using the Illumina HiSeq X platform.</p>
<p>All individuals were sequenced to a target coverage of 30X.</p>
<p>HiSeq 2000 sequencing runs generated 100 bp paired-end reads, while HiSeq X sequencing runs generated 150 bp paired-end reads.</p>
<h2 id="alignment-and-snp-calling">Alignment and SNP calling</h2>
<p>Reads were aligned to the AgamP4 reference genome using <code>bwa</code> version 0.7.15.</p>
<p>Indel realignment was performed using GATK version 3.7-0 <code>RealignerTargetCreator</code> and <code>IndelRealigner</code>.</p>
<p>Single nucleotide polymorphisms were called using GATK version 3.7-0 <code>UnifiedGenotyper</code>.</p>
<p>Genotypes were called for each sample independently, in genotyping mode, given all possible alleles at all genomic sites where the reference base was not <code>N</code>.</p>
<p>Coverage was capped at 250X by random down-sampling.</p>
<p>Complete specifications of the <strong><a href="https://github.com/malariagen/pipelines/blob/v0.0.4/docs/specs/short-read-alignment-vector.md">alignment</a></strong> and <strong><a href="https://github.com/malariagen/pipelines/blob/v0.0.4/docs/specs/snp-genotyping-vector.md">genotyping</a></strong> pipelines are available from the <strong><a href="https://github.com/malariagen/pipelines">malariagen/pipelines</a></strong> GitHub repository.</p>
<p>Open source WDL implementations of the <strong><a href="https://github.com/malariagen/pipelines/tree/v0.0.4/pipelines/short-read-alignment-vector">alignment</a></strong> and <strong><a href="https://github.com/malariagen/pipelines/tree/v0.0.4/pipelines/SNP-genotyping-vector">genotyping</a></strong> pipelines are also available from GitHub.</p>
<p>Following successful completion of these pipelines, samples entered the sample quality control (QC) process.</p>
<h2 id="sample-qc-1">Sample QC</h2>
<p>The following subsections describe analyses performed to identify and exclude samples from the final dataset.</p>
<h3 id="coverage-1">Coverage</h3>
<p>For each sample, depth of coverage was computed at all genome positions.</p>
<p>Samples were excluded if median coverage across all chromosomes was less than 10X, or if less than 50% of the reference genome was covered by at least 1X.</p>
<h3 id="cross-contamination">Cross-contamination</h3>
<p>To identify samples affected by cross-contamination, we implemented the model for detecting contamination in NGS alignments described in <span class="citation" data-cites="cQaeXqlZ"><a href="#ref-cQaeXqlZ" role="doc-biblioref">54</a></span>.</p>
<p>Briefly, the method estimates the likelihood of the observed alternate and reference allele counts under different contamination fractions, given approximate population allele frequencies.</p>
<p>Population allele frequencies were estimated from the Ag1000G phase 2 data release <span class="citation" data-cites="Rho6nYsT"><a href="#ref-Rho6nYsT" role="doc-biblioref">4</a></span>.</p>
<p>The model computes a maximum likelihood value for a parameter α representing percentage contamination.</p>
<p>Samples were excluded if α was 4.5% or greater.</p>
<h3 id="technical-replicates">Technical replicates</h3>
<p>A number of samples were sequenced more than once within this project phase (technical replicates).</p>
<p>To create a final dataset without any replicates suitable for population genetic analysis, we performed an analysis to confirm all technical replicates, and to choose the sample within each replicate with the best sequencing data.</p>
<p>We computed pairwise genetic distance between all sample pairs within a submission set.</p>
<p>The distance metric used was city block distance between genotype allele counts, to allow for handling of multiallelic SNPs.</p>
<p>So, <em>e.g.</em>, distance between genotypes of <code>0/1</code> and <code>0/1</code> is 0, distance between <code>0/0</code> and <code>0/1</code> is 2, distance between <code>0/1</code> and <code>1/2</code> is 2, distance between <code>0/0</code> and <code>1/1</code> is 4, etc.</p>
<p>For each pair of samples, distance was averaged over all sites where both samples had a non-missing genotype call.</p>
<p>Computations were initially carried out on a down-sampled set of 10 x 100,000 contiguous genomic sites, to be computationally feasible.</p>
<p>Where a pair of samples fell beneath a conservative threshold of 0.012, the genetic distance was then recomputed across all genomic sites (<em>i.e.</em>, without down-sampling).</p>
<p>For each pair of samples that were expected to be technical replicates according to metadata records, we excluded both members of the pair if genetic distance was above 0.006.</p>
<p>Where an expected replicate pair had genetic distance below 0.006, we retained only one sample in the pair.</p>
<p>We also identified and excluded both samples in any pair where genetic distance was below 0.006, where samples were not expected to be replicates.</p>
<h3 id="population-outliers">Population outliers</h3>
<p>We used principal component analysis (PCA) to identify and exclude individual samples that were population outliers.</p>
<p>SNPs were down-sampled to use 100,000 segregating non-singleton sites from chromosomes 3R and 3L, to avoid regions complicated by known introgression loci or paracentric inversions.</p>
<p>PCA was computed using <code>scikit-allel</code> version 1.2.0.</p>
<p>We iteratively identified and excluded any individual samples that were outliers along a single principal component.</p>
<p>We then identified and excluded any individual samples or small sample groups that clustered together with other samples in a way that was not plausible given metadata regarding their collection location.</p>
<h3 id="colony-crosses-1">Colony crosses</h3>
<p>Samples in the <code>AG1000G-X</code> sample set were parents and progeny from colony crosses and were subject to a slightly different QC process.</p>
<p>For each cross, we performed an analysis of Mendelian inheritance and consistency to confirm the true parents and the validity of the cross.</p>
<p>Not all crosses were able to be successfully resolved, and samples that were not in a resolved cross were excluded.</p>
<p>From the samples originally submitted in the <code>AG1000G-X</code> sample set, 297 samples from 15 crosses were retained for release.</p>
<p>We did not include the colony crosses in the population outlier analysis due to their relatedness.</p>
<h3 id="sex-calling">Sex calling</h3>
<p>We called the sex of all samples based on the modal coverage ratio between the X chromosome and the autosomal chromosome arm 3R.</p>
<p>The sample was classed as male where the coverage ratio was between 0.4-0.6, and female between 0.8-1.2.</p>
<p>Where the ratio was outside these limits, the sample was excluded.</p>
<p>One of the sample sets from The Gambia, <code>AG1000G-GM-B</code>, included whole-genome amplified (WGA) samples which displayed some skew in their coverage ratios, which meant that sex could not be called via the same process.</p>
<p>These samples received a sex call where possible, but no samples were excluded based on uncertain sex call.</p>
<h2 id="species-assignment">Species assignment</h2>
<p>We assigned a species to each individual that passed sample QC using their genomic data, via two independent methods: ancestry-informative markers (AIMs) and principal components analysis (PCA).</p>
<h3 id="species-calling-via-ancestry-informative-markers">Species calling via ancestry-informative markers</h3>
<p>To derive AIMs between <em>A. arabiensis</em> and <em>A. gambiae</em>, we used publicly available data from the <em>Anopheles</em> 16 genomes project (<span class="citation" data-cites="9XFa7ZJB"><a href="#ref-9XFa7ZJB" role="doc-biblioref">1</a></span>).</p>
<p>Whole genome SNP calls for 12 <em>A. arabiensis</em> and 38 <em>A. gambiae</em> individuals were used.</p>
<p>Alleles were mapped onto the same alternate allele space, and allele frequencies were computed for both species.</p>
<p>Sites that were multiallelic in either group were excluded, as well as sites where any genotypes were missing.</p>
<p>565,329 SNPs were identified as potentially informative, where no shared alleles were present between groups.</p>
<p>These were spread throughout the genome, but were concentrated on the X chromosome (63.2%), particularly around the Xag inversion.</p>
<p>We randomly down-sampled these SNPs to a set of 50,000 AIMs, then computed the fraction of alleles at these SNPs that were arabiensis-like for each individual in the Ag1000G phase 3 cohort.</p>
<p>Given the relatively small number of <em>A. arabiensis</em> samples in the 16 genomes project, it was clear that a significant proportion of putative AIMs were not likely to be truly informative across the broader sampling in Ag1000G.</p>
<p>Individuals in Ag1000G were classed as <em>A. arabiensis</em> where a fraction &gt;0.8 of alleles were arabiensis-like.</p>
<p>To resolve the non-<em>A. arabiensis</em> individuals into <em>A. gambiae</em> and <em>A. coluzzii</em>, we applied the AIMs previously used in <span class="citation" data-cites="Rho6nYsT"><a href="#ref-Rho6nYsT" role="doc-biblioref">4</a></span>.</p>
<p>For each individual, we computed the fraction of coluzzii-like alleles at these AIMs.</p>
<p>Individuals were called as <em>A. gambiae</em> where this fraction was &lt;0.12 and <em>A. coluzzii</em> where this fraction was &gt;0.9, with individuals in between classed as intermediate.</p>
<h3 id="species-calling-via-principal-components-analysis">Species calling via principal components analysis</h3>
<p>To provide a complementary view of species assignments, we also used the results of the principal components analysis of Chromosome 3 computed during the outlier analysis described above.</p>
<p>Based on a comparison with the AIM species calls, it was apparent that the first two principal components could be used to assign species.</p>
<p>Individuals where PC1 &gt; 150 were called as <em>A. arabiensis</em>.</p>
<p>Individuals where PC1 &lt; 0 and PC2 &gt; -7 were called as <em>A. gambiae</em>.</p>
<p>Individuals where PC1 &lt; 0 and PC2 &lt; -24 were called as <em>A. coluzzii</em>.</p>
<p>All other individuals were called as intermediate.</p>
<p>The results of the PCA and AIM species calls were highly concordant in most sample sets, except for the Far West (Guinea-Bissau, The Gambiae) and Far East (Kenya, Tanzania).</p>
<p>Further investigation is required to resolve the species status of these individuals.</p>
<h2 id="site-filtering">Site filtering</h2>
<p>We developed filters that identify genomic sites where SNP calling and genotyping is likely to be less reliable in one or more mosquito species.</p>
<p>To guide the design and calibration of the site filters, we made use of the 15 colony crosses included in this release.</p>
<p>Each cross comprises two parents and up to 20 progeny, allowing identification of sites where genotypes in one or more progeny are not consistent with Mendelian inheritance (Mendelian errors).</p>
<p>A small number of Mendelian errors may be due to <em>de novo</em> mutation, but the vast majority of Mendelian errors are likely to be due to errors in sequencing, alignment or SNP calling.</p>
<p>The general approach we took was to use Mendelian consistency to identify sets of positive and negative training sites, then used these to train a machine learning model that classified all genomic sites as either PASS or FAIL.</p>
<h3 id="site-filters-for-use-with-a.-gambiae-andor-a.-coluzzii">Site filters for use with <em>A. gambiae</em> and/or <em>A. coluzzii</em></h3>
<p>All the 15 crosses involved <em>A. gambiae</em> and/or <em>A. coluzzii</em> parents, while none of the crosses involved <em>A. arabiensis</em>.</p>
<p>We therefore used the crosses to first develop site filters suitable for use with <em>A. gambiae</em> and/or <em>A. coluzzii</em>.</p>
<p>Hereafter we refer to these filters as the <code>gamb\_colu</code> site filters.</p>
<p>Five of the 15 crosses were held out for validation, so performance could be evaluated objectively.</p>
<p>Sites were assigned to the positive training set where all genotypes across all 10 training crosses were called, and no Mendelian errors were observed.</p>
<p>Sites were assigned to negative training set where one or more Mendelian errors were observed in any cross.</p>
<p>All other sites were not considered eligible for inclusion in model training.</p>
<p>A balanced training set was then generated containing 100,000 autosomal sites from each of the positive and negative training sets.</p>
<p>The inputs to the machine learning model were a set of per-site summary statistics computed from the sequence read alignments and SNP genotypes across all wild-caught <em>A. gambiae</em> and <em>A. coluzzii</em> individuals.</p>
<p>These input summary statistics are described further in the appendix.</p>
<p>Male individuals were excluded from the summary statistic calculations, so that the model could also be applied without modification to the X chromosome.</p>
<p>We used these summary statistics, together with the positive and negative training sites, to train a decision tree model.</p>
<p>We initially trained a set of trees with different hyperparameter values, exploring the depth of trees, and the number of samples allowed at a terminal node.</p>
<p>Each of these trees was evaluated on an unbalanced set of sites randomly sampled from the whole genome (2% of all sites, without replacement).</p>
<p>Leaves of these trees contained different proportions of positive and negative training sites, and by increasing the cutoff for these proportions required to label a leaf as PASS, we were able to compute the area under the receiver operating curve (AUROC) for each set of hyperparameter values.</p>
<p>The best performing hyperparameter set based on AUROC was selected as the final model, and the leaf classification cutoff used was optimised based on the Youden statistic.</p>
<p>The resulting model was a decision tree of depth 8, where leaves were assigned to PASS where &gt; 0.533 of training data in that leaf were positive training sites.</p>
<p>All sites in the genome were then assigned to PASS or FAIL via this model.</p>
<p>The 5 remaining cross pedigrees were used to perform a final evaluation of the approach.</p>
<p>For each of these crosses, we computed the Mendelian error rate (fraction of variants with one or more Mendelian errors among progeny) before and after applying the site filters, to provide five independent evaluation results.</p>
<p>We also evaluated performance on the X chromosome using heterozygote calls in males as indicator of error rates.</p>
<p>The fraction of variants with a heterozygous genotype call in or more males was computed before and after applying site filters.</p>
<p>Male error rates were estimated from genotype calls with a minimum Genotype Quality (GQ) value of 30.</p>
<h3 id="site-filters-for-use-with-a.-arabiensis">Site filters for use with <em>A. arabiensis</em></h3>
<p>To generate site filters for use with <em>A. arabiensis</em>, we recomputed site summary statistics using only wild-caught <em>A. arabiensis</em> individuals, then applied the decision tree model described above.</p>
<p>These filters, which we refer to as the <code>arab</code> site filters, are appropriate when working with <em>A. arabiensis</em> samples only.</p>
<h3 id="site-filters-for-joint-analyses-of-all-three-species">Site filters for joint analyses of all three species</h3>
<p>We created site filters suitable for joint analysis of individuals from all three species by taking the intersection of the <code>gamb\_colu</code> and the <code>arab</code> site filters.</p>
<p>We refer to these filters as the <code>gamb\_colu\_arab</code> site filters.</p>
<h2 id="acknowledgments">Acknowledgments</h2>
<p>We would like to thank the staff of the Wellcome Sanger Institute Sample Logistics, Sequencing and Informatics facilities for their contributions to the production of this data release.</p>
<p>We would like to thank the members of the Data Engineering team of the Broad Institute of Harvard and MIT for their work on open source implementations of the alignment and SNP calling pipelines used in Ag1000G phase 3.</p>
<h2 id="further-information">Further information</h2>
<p>For further information about the Ag1000G project, please visit <strong><a href="https://www.malariagen.net/ag1000g">https://www.malariagen.net/ag1000g</a></strong>.</p>
<p>For further information about the Ag1000G phase 3 SNP data release, please visit <strong><a href="www.malariagen.net/data/ag1000g-phase3-snp">www.malariagen.net/data/ag1000g-phase3-snp</a></strong>.</p>
<p>If you have any questions regarding the data release, please start a new discussion at <strong><a href="https://github.com/malariagen/vector-public-data/discussions">https://github.com/malariagen/vector-public-data/discussions</a></strong>.</p>
<h2 class="page_break_before" id="references">References</h2>
<!-- Explicitly insert bibliography here -->
<div id="refs" class="references hanging-indent" role="doc-bibliography">
<div id="ref-9XFa7ZJB">
<p>1. <strong>Highly evolvable malaria vectors: The genomes of 16 <em>Anopheles</em> mosquitoes</strong> <br />
Daniel E. Neafsey, Robert M. Waterhouse, Mohammad R. Abai, Sergey S. Aganezov, Max A. Alekseyev, James E. Allen, James Amon, Bruno Arcà, Peter Arensburger, Gleb Artemov, … Nora J. Besansky<br />
<em>Science</em> (2015-01-02) <a href="https://doi.org/gdkzt5">https://doi.org/gdkzt5</a> <br />
DOI: <a href="https://doi.org/10.1126/science.1258522">10.1126/science.1258522</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25554792">25554792</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4380271">PMC4380271</a></p>
</div>
<div id="ref-2xdynpZK">
<p>2. <strong>SNP genotyping defines complex gene-flow boundaries among African malaria vector mosquitoes.</strong> <br />
DE Neafsey, MKN Lawniczak, DJ Park, SN Redmond, MB Coulibaly, SF Traoré, N Sagnon, C Costantini, C Johnson, RC Wiegand, … MAT Muskavitch<br />
<em>Science (New York, N.Y.)</em> (2010-10-22) <a href="https://www.ncbi.nlm.nih.gov/pubmed/20966254">https://www.ncbi.nlm.nih.gov/pubmed/20966254</a> <br />
DOI: <a href="https://doi.org/10.1126/science.1193036">10.1126/science.1193036</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/20966254">20966254</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4811326">PMC4811326</a></p>
</div>
<div id="ref-nvrjDUOw">
<p>3. <strong>Genetic diversity of the African malaria vector Anopheles gambiae</strong> <br />
The Anopheles gambiae 1000 Genomes Consortium<br />
<em>Nature</em> (2017-11-29) <a href="https://doi.org/gcmd34">https://doi.org/gcmd34</a> <br />
DOI: <a href="https://doi.org/10.1038/nature24995">10.1038/nature24995</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/29186111">29186111</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6026373">PMC6026373</a></p>
</div>
<div id="ref-Rho6nYsT">
<p>4. <strong>Genome variation and population structure among 1142 mosquitoes of the African malaria vector species <em>Anopheles gambiae</em> and <em>Anopheles coluzzii</em></strong> <br />
The Anopheles gambiae 1000 Genomes Consortium<br />
<em>Genome Research</em> (2020-10) <a href="https://doi.org/ghvn76">https://doi.org/ghvn76</a> <br />
DOI: <a href="https://doi.org/10.1101/gr.262790.120">10.1101/gr.262790.120</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/32989001">32989001</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7605271">PMC7605271</a></p>
</div>
<div id="ref-zjUaokUj">
<p>5. <strong>Adaptive introgression between Anopheles sibling species eliminates a major genomic island but not reproductive isolation.</strong> <br />
Chris S Clarkson, David Weetman, John Essandoh, Alexander E Yawson, Gareth Maslen, Magnus Manske, Stuart G Field, Mark Webster, Tiago Antão, Bronwyn MacInnis, … Martin J Donnelly<br />
<em>Nature communications</em> (2014-06-25) <a href="https://www.ncbi.nlm.nih.gov/pubmed/24963649">https://www.ncbi.nlm.nih.gov/pubmed/24963649</a> <br />
DOI: <a href="https://doi.org/10.1038/ncomms5248">10.1038/ncomms5248</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24963649">24963649</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4086683">PMC4086683</a></p>
</div>
<div id="ref-PFIjKsQ5">
<p>6. <strong>Spatiotemporal dynamics of gene flow and hybrid fitness between the M and S forms of the malaria mosquito, Anopheles gambiae</strong> <br />
Y. Lee, C. D. Marsden, L. C. Norris, T. C. Collier, B. J. Main, A. Fofana, A. J. Cornel, G. C. Lanzaro<br />
<em>Proceedings of the National Academy of Sciences</em> (2013-11-18) <a href="https://doi.org/f5h939">https://doi.org/f5h939</a> <br />
DOI: <a href="https://doi.org/10.1073/pnas.1316851110">10.1073/pnas.1316851110</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24248386">24248386</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3856788">PMC3856788</a></p>
</div>
<div id="ref-R1m6hsR0">
<p>7. <strong>Anopheles coluzzii and Anopheles amharicus, new members of the Anopheles gambiae complex.</strong> <br />
Maureen Coetzee, Richard H Hunt, Richard Wilkerson, Alessandra Della Torre, Mamadou B Coulibaly, Nora J Besansky<br />
<em>Zootaxa</em> (2013) <a href="https://www.ncbi.nlm.nih.gov/pubmed/26131476">https://www.ncbi.nlm.nih.gov/pubmed/26131476</a> <br />
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26131476">26131476</a></p>
</div>
<div id="ref-1G04AzsFo">
<p>8. <strong>Simultaneous identification of species and molecular forms of the <em>Anopheles gambiae</em> complex by PCR-RFLP</strong> <br />
C. Fanello, F. Santolamazza, A. Della Torre<br />
<em>Medical and Veterinary Entomology</em> (2002-12) <a href="https://doi.org/ds5pmz">https://doi.org/ds5pmz</a> <br />
DOI: <a href="https://doi.org/10.1046/j.1365-2915.2002.00393.x">10.1046/j.1365-2915.2002.00393.x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/12510902">12510902</a></p>
</div>
<div id="ref-BAhAVfFc">
<p>9.<strong>:(unav)</strong> <br />
Nelson Cuamba, Kwang Choi, Harold Townson<br />
<em>Malaria Journal</em> (2006) <a href="https://doi.org/cd3n27">https://doi.org/cd3n27</a> <br />
DOI: <a href="https://doi.org/10.1186/1475-2875-5-2">10.1186/1475-2875-5-2</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/16420701">16420701</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1363361">PMC1363361</a></p>
</div>
<div id="ref-2Iw2A7Bd">
<p>10. <strong>Distribution and Chromosomal Characterization of the Anopheles gambiae Complex in Angola</strong> <br />
Pedro J. Cani, Maria Calzetta, Maria Angela Di Deco, Federica Santolamazza, Alessandra della Torre, Gian Carlo Carrara, Vincenzo Petrarca, Filomeno Fortes<br />
<em>The American Journal of Tropical Medicine and Hygiene</em> (2008-01-01) <a href="https://doi.org/ghv47t">https://doi.org/ghv47t</a> <br />
DOI: <a href="https://doi.org/10.4269/ajtmh.2008.78.169">10.4269/ajtmh.2008.78.169</a></p>
</div>
<div id="ref-DnnjILYG">
<p>11. <strong>Population structure in the malaria vector, Anopheles arabiensis Patton, in East Africa</strong> <br />
MJ Donnelly, N Cuamba, JD Charlwood, FH Collins, H Townson<br />
<em>Heredity</em> (1999-10-01) <a href="https://doi.org/bg4xmm">https://doi.org/bg4xmm</a> <br />
DOI: <a href="https://doi.org/10.1038/sj.hdy.6885930">10.1038/sj.hdy.6885930</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/10583542">10583542</a></p>
</div>
<div id="ref-QhOf6jCa">
<p>12. <strong>IMP PCR primers detect single nucleotide polymorphisms for Anopheles gambiae species identification, Mopti and Savanna rDNA types, and resistance to dieldrin in Anopheles arabiensis.</strong> <br />
Elien E Wilkins, Paul I Howell, Mark Q Benedict<br />
<em>Malaria journal</em> (2006-12-19) <a href="https://www.ncbi.nlm.nih.gov/pubmed/17177993">https://www.ncbi.nlm.nih.gov/pubmed/17177993</a> <br />
DOI: <a href="https://doi.org/10.1186/1475-2875-5-125">10.1186/1475-2875-5-125</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/17177993">17177993</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1769388">PMC1769388</a></p>
</div>
<div id="ref-mTkFlBOi">
<p>13. <strong>Short report: A new polymerase chain reaction-restriction fragment length polymorphism method to identify Anopheles arabiensis from An. gambiae and its two molecular forms from degraded DNA templates or museum samples.</strong> <br />
Federica Santolamazza, Alessandra Della Torre, Adalgisa Caccone<br />
<em>The American journal of tropical medicine and hygiene</em> (2004-06) <a href="https://www.ncbi.nlm.nih.gov/pubmed/15210999">https://www.ncbi.nlm.nih.gov/pubmed/15210999</a> <br />
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/15210999">15210999</a></p>
</div>
<div id="ref-1rFnXZkd">
<p>14. <strong>Breakpoint structure of the Anopheles gambiae 2Rb chromosomal inversion</strong> <br />
Neil F Lobo, Djibril M Sangaré, Allison A Regier, Kyanne R Reidenbach, David A Bretz, Maria V Sharakhova, Scott J Emrich, Sekou F Traore, Carlo Costantini, Nora J Besansky, Frank H Collins<br />
<em>Malaria Journal</em> (2010-10-25) <a href="https://doi.org/c8td4x">https://doi.org/c8td4x</a> <br />
DOI: <a href="https://doi.org/10.1186/1475-2875-9-293">10.1186/1475-2875-9-293</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/20974007">20974007</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2988034">PMC2988034</a></p>
</div>
<div id="ref-122DDCVbs">
<p>15. <strong>Malaria vector populations across ecological zones in Guinea Conakry and Mali, West Africa</strong> <br />
Boubacar Coulibaly, Raymond Kone, Mamadou S. Barry, Becky Emerson, Mamadou B. Coulibaly, Oumou Niare, Abdoul H. Beavogui, Sekou F. Traore, Kenneth D. Vernick, Michelle M. Riehle<br />
<em>Malaria Journal</em> (2016-04-08) <a href="https://doi.org/f8gzkb">https://doi.org/f8gzkb</a> <br />
DOI: <a href="https://doi.org/10.1186/s12936-016-1242-5">10.1186/s12936-016-1242-5</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/27059057">27059057</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4826509">PMC4826509</a></p>
</div>
<div id="ref-daG5Pj6c">
<p>16. <strong>Ecological niche partitioning between Anopheles gambiae molecular forms in Cameroon: the ecological side of speciation</strong> <br />
Frédéric Simard, Diego Ayala, Guy Kamdem, Marco Pombi, Joachim Etouna, Kenji Ose, Jean-Marie Fotsing, Didier Fontenille, Nora J Besansky, Carlo Costantini<br />
<em>BMC Ecology</em> (2009) <a href="https://doi.org/bd8bz5">https://doi.org/bd8bz5</a> <br />
DOI: <a href="https://doi.org/10.1186/1472-6785-9-17">10.1186/1472-6785-9-17</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19460146">19460146</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2698860">PMC2698860</a></p>
</div>
<div id="ref-14XMzb1n7">
<p>17. <strong>The Anophelinae of Africa south of the Sahara. Suppl: Afrotropical region</strong> <br />
M. T. Gillies, Botha de Meillon<br />
<em>Publications of the South African Institute for Medical Research</em> (1987) <br />
ISBN: <a href="https://worldcat.org/isbn/9780620103213">9780620103213</a></p>
</div>
<div id="ref-Zqohe5sj">
<p>18. <strong>The anophelinae of Africa south of the Sahara (Ethiopian zoogeographical region)</strong> <br />
M. T. Gillies, Botha de Meillon</p>
</div>
<div id="ref-kGCikEHW">
<p>19. <strong>Pollutants and Insecticides Drive Local Adaptation in African Malaria Mosquitoes.</strong> <br />
Colince Kamdem, Caroline Fouet, Stephanie Gamez, Bradley J White<br />
<em>Molecular biology and evolution</em> (2017-05-01) <a href="https://www.ncbi.nlm.nih.gov/pubmed/28204524">https://www.ncbi.nlm.nih.gov/pubmed/28204524</a> <br />
DOI: <a href="https://doi.org/10.1093/molbev/msx087">10.1093/molbev/msx087</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28204524">28204524</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5400387">PMC5400387</a></p>
</div>
<div id="ref-14yZJJBPD">
<p>20. <strong>Identification of Single Specimens of the Anopheles Gambiae Complex by the Polymerase Chain Reaction</strong> <br />
Julie A. Scott, William G. Brogdon, Frank H. Collins<br />
<em>The American Journal of Tropical Medicine and Hygiene</em> (1993-10-01) <a href="https://doi.org/ghcqgk">https://doi.org/ghcqgk</a> <br />
DOI: <a href="https://doi.org/10.4269/ajtmh.1993.49.520">10.4269/ajtmh.1993.49.520</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/8214283">8214283</a></p>
</div>
<div id="ref-WKPmPPDE">
<p>21. <strong>Molecular karyotyping of the 2La inversion in Anopheles gambiae.</strong> <br />
Bradley J White, Federica Santolamazza, Luna Kamau, Marco Pombi, Olga Grushko, Karine Mouline, Cecile Brengues, Wamdaogo Guelbeogo, Mamadou Coulibaly, Jonathan K Kayondo, … Nora J Besansky<br />
<em>The American journal of tropical medicine and hygiene</em> (2007-02) <a href="https://www.ncbi.nlm.nih.gov/pubmed/17297045">https://www.ncbi.nlm.nih.gov/pubmed/17297045</a> <br />
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/17297045">17297045</a></p>
</div>
<div id="ref-EfCseGxd">
<p>22. <strong>Malaria vector control by indoor residual insecticide spraying on the tropical island of Bioko, Equatorial Guinea</strong> <br />
Brian L Sharp, Frances C Ridl, Dayanandan Govender, Jaime Kuklinski, Immo Kleinschmidt<br />
<em>Malaria Journal</em> (2007) <a href="https://doi.org/czzjf5">https://doi.org/czzjf5</a> <br />
DOI: <a href="https://doi.org/10.1186/1475-2875-6-52">10.1186/1475-2875-6-52</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/17474975">17474975</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1868751">PMC1868751</a></p>
</div>
<div id="ref-DaNgNJ6t">
<p>23. <strong>Light traps fail to estimate reliable malaria mosquito biting rates on Bioko Island, Equatorial Guinea</strong> <br />
Hans J Overgaard, Solve Sæbø, Michael R Reddy, Vamsi P Reddy, Simon Abaga, Abrahan Matias, Michel A Slotman<br />
<em>Malaria Journal</em> (2012-02-24) <a href="https://doi.org/ghwxxm">https://doi.org/ghwxxm</a> <br />
DOI: <a href="https://doi.org/10.1186/1475-2875-11-56">10.1186/1475-2875-11-56</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22364588">22364588</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3384454">PMC3384454</a></p>
</div>
<div id="ref-meNim6ix">
<p>24. <strong>Resistance to pirimiphos-methyl in West African Anopheles is spreading via duplication and introgression of the Ace1 locus</strong> <br />
Xavier Grau-Bové, Eric Lucas, Dimitra Pipini, Emily Rippon, Arjèn E. van ‘t Hof, Edi Constant, Samuel Dadzie, Alexander Egyir-Yawson, John Essandoh, Joseph Chabi, … The Anopheles gambiae 1000 Genomes Consortium<br />
<em>PLOS Genetics</em> (2021-01-21) <a href="https://doi.org/ghwzd9">https://doi.org/ghwzd9</a> <br />
DOI: <a href="https://doi.org/10.1371/journal.pgen.1009253">10.1371/journal.pgen.1009253</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/33476334">33476334</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7853456">PMC7853456</a></p>
</div>
<div id="ref-szUswio1">
<p>25. <strong>Acetylcholinesterase (Ace-1) target site mutation 119S is strongly diagnostic of carbamate and organophosphate resistance in Anopheles gambiae s.s. and Anopheles coluzzii across southern Ghana</strong> <br />
John Essandoh, Alexander E Yawson, David Weetman<br />
<em>Malaria Journal</em> (2013) <a href="https://doi.org/gbfcqt">https://doi.org/gbfcqt</a> <br />
DOI: <a href="https://doi.org/10.1186/1475-2875-12-404">10.1186/1475-2875-12-404</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24206629">24206629</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3842805">PMC3842805</a></p>
</div>
<div id="ref-TPRu2NBx">
<p>26. <strong>Co-occurrence of East and West African kdr mutations suggests high levels of resistance to pyrethroid insecticides in Anopheles gambiae from Libreville, Gabon</strong> <br />
J. Pinto, A. Lynd, N. Elissa, M. J. Donnelly, C. Costa, G. Gentile, A. Caccone, V. E. DO Rosario<br />
<em>Medical and Veterinary Entomology</em> (2006-03) <a href="https://doi.org/brg9qk">https://doi.org/brg9qk</a> <br />
DOI: <a href="https://doi.org/10.1111/j.1365-2915.2006.00611.x">10.1111/j.1365-2915.2006.00611.x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/16608487">16608487</a></p>
</div>
<div id="ref-s65aawfm">
<p>27. <strong>Malaria transmission in Libreville: results of a one year survey</strong> <br />
Jean-Romain Mourou, Thierry Coffinet, Fanny Jarjaval, Christelle Cotteaux, Eve Pradines, Lydie Godefroy, Maryvonne Kombila, Frédéric Pagès<br />
<em>Malaria Journal</em> (2012) <a href="https://doi.org/gdkzp5">https://doi.org/gdkzp5</a> <br />
DOI: <a href="https://doi.org/10.1186/1475-2875-11-40">10.1186/1475-2875-11-40</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22321336">22321336</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3310827">PMC3310827</a></p>
</div>
<div id="ref-8S1Lluej">
<p>28. <strong>Population structure in the malaria vector, Anopheles arabiensis patton, in East Africa.</strong> <br />
MJ Donnelly, N Cuamba, JD Charlwood, FH Collins, H Townson<br />
<em>Heredity</em> (1999-10) <a href="https://www.ncbi.nlm.nih.gov/pubmed/10583542">https://www.ncbi.nlm.nih.gov/pubmed/10583542</a> <br />
DOI: <a href="https://doi.org/10.1038/sj.hdy.6885930">10.1038/sj.hdy.6885930</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/10583542">10583542</a></p>
</div>
<div id="ref-YX2M3jrq">
<p>29. <strong>Massive introgression drives species radiation at the range limit of Anopheles gambiae</strong> <br />
José L. Vicente, Christopher S. Clarkson, Beniamino Caputo, Bruno Gomes, Marco Pombi, Carla A. Sousa, Tiago Antao, João Dinis, Giordano Bottà, Emiliano Mancini, … João Pinto<br />
<em>Scientific Reports</em> (2017-04-18) <a href="https://doi.org/f93m36">https://doi.org/f93m36</a> <br />
DOI: <a href="https://doi.org/10.1038/srep46451">10.1038/srep46451</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28417969">28417969</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5394460">PMC5394460</a></p>
</div>
<div id="ref-16as3jwaa">
<p>30. <strong>TRANSMISSION OF MIXED PLASMODIUM SPECIES AND PLASMODIUM FALCIPARUM GENOTYPES</strong> <br />
VIRGÍLIO E. DO ROSÁRIO, KATINKA PÅLSSON, THOMAS G. T. JAENSON, GEORGES SNOUNOU, JOÃO PINTO, ANA PAULA AREZ<br />
<em>The American Journal of Tropical Medicine and Hygiene</em> (2003-02-01) <a href="https://doi.org/ghcqgm">https://doi.org/ghcqgm</a> <br />
DOI: <a href="https://doi.org/10.4269/ajtmh.2003.68.2.0680161">10.4269/ajtmh.2003.68.2.0680161</a></p>
</div>
<div id="ref-vmUWM43r">
<p>31. <strong>PCR-based karyotyping of Anopheles gambiae inversion 2Rj identifies the BAMAKO chromosomal form.</strong> <br />
Mamadou B Coulibaly, Marco Pombi, Beniamino Caputo, Davis Nwakanma, Musa Jawara, Lassana Konate, Ibrahima Dia, Abdrahamane Fofana, Marcia Kern, Frédéric Simard, … Nora J Besansky<br />
<em>Malaria journal</em> (2007-10-01) <a href="https://www.ncbi.nlm.nih.gov/pubmed/17908310">https://www.ncbi.nlm.nih.gov/pubmed/17908310</a> <br />
DOI: <a href="https://doi.org/10.1186/1475-2875-6-133">10.1186/1475-2875-6-133</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/17908310">17908310</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2134931">PMC2134931</a></p>
</div>
<div id="ref-1C7qEzk6S">
<p>32. <strong>Wind direction and proximity to larval sites determines malaria risk in Kilifi District in Kenya</strong> <br />
Janet T. Midega, Dave L. Smith, Ally Olotu, Joseph M. Mwangangi, Joseph G. Nzovu, Juliana Wambua, George Nyangweso, Charles M. Mbogo, George K. Christophides, Kevin Marsh, Philip Bejon<br />
<em>Nature Communications</em> (2012-02-14) <a href="https://doi.org/ghcqgd">https://doi.org/ghcqgd</a> <br />
DOI: <a href="https://doi.org/10.1038/ncomms1672">10.1038/ncomms1672</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22334077">22334077</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3292715">PMC3292715</a></p>
</div>
<div id="ref-xsFRxYp6">
<p>33. <strong>Assessment of the effect of larval source management and house improvement on malaria transmission when added to standard malaria control strategies in southern Malawi: study protocol for a cluster-randomised controlled trial</strong> <br />
Robert S. McCann, Henk van den Berg, Peter J. Diggle, Michèle van Vugt, Dianne J. Terlouw, Kamija S. Phiri, Aurelio Di Pasquale, Nicolas Maire, Steven Gowelo, Monicah M. Mburu, … Willem Takken<br />
<em>BMC Infectious Diseases</em> (2017-09-22) <a href="https://doi.org/ggr5g7">https://doi.org/ggr5g7</a> <br />
DOI: <a href="https://doi.org/10.1186/s12879-017-2749-2">10.1186/s12879-017-2749-2</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/28938876">28938876</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5610449">PMC5610449</a></p>
</div>
<div id="ref-vckO98IK">
<p>34. <strong>Mapping Malaria Transmission Intensity in Malawi, 2000–2010</strong> <br />
Adam Bennett, Lawrence Kazembe, Don P. Mathanga, Damaris Kinyoki, Doreen Ali, Robert W. Snow, Abdisalan M. Noor<br />
<em>The American Journal of Tropical Medicine and Hygiene</em> (2013-11-06) <a href="https://doi.org/f5h9zz">https://doi.org/f5h9zz</a> <br />
DOI: <a href="https://doi.org/10.4269/ajtmh.13-0028">10.4269/ajtmh.13-0028</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24062477">24062477</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3820324">PMC3820324</a></p>
</div>
<div id="ref-19Z5CFAYk">
<p>35. <strong>LookSeq: A browser-based viewer for deep sequencing data</strong> <br />
H. M. Manske, D. P. Kwiatkowski<br />
<em>Genome Research</em> (2009-08-13) <a href="https://doi.org/b5hmbh">https://doi.org/b5hmbh</a> <br />
DOI: <a href="https://doi.org/10.1101/gr.093443.109">10.1101/gr.093443.109</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19679872">19679872</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2775587">PMC2775587</a></p>
</div>
<div id="ref-pSmuJkYQ">
<p>36. <strong>A Ribosomal RNA Gene Probe Differentiates Member Species of the Anopheles gambiae Complex</strong> <br />
Frank H. Collins, Nora J. Besansky, M. Alina Mendez, Melissa O. Rasmussen, Victoria Finnerty, Philip C. Mehaffey<br />
<em>The American Journal of Tropical Medicine and Hygiene</em> (1987-07-01) <a href="https://doi.org/ghcqgj">https://doi.org/ghcqgj</a> <br />
DOI: <a href="https://doi.org/10.4269/ajtmh.1987.37.37">10.4269/ajtmh.1987.37.37</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/2886070">2886070</a></p>
</div>
<div id="ref-f8uRMyY6">
<p>37. <strong>Analysis of the sporozoite ELISA for estimating infection rates in Mozambican anophelines</strong> <br />
J. D. CHARLWOOD, E. V. E. TOMÁS, N. CUAMBA, J. PINTO<br />
<em>Medical and Veterinary Entomology</em> (2015-03) <a href="https://doi.org/f62ckd">https://doi.org/f62ckd</a> <br />
DOI: <a href="https://doi.org/10.1111/mve.12084">10.1111/mve.12084</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25088021">25088021</a></p>
</div>
<div id="ref-FVN9nrFP">
<p>38. <strong>Genetic basis of pyrethroid resistance in a population of Anopheles arabiensis, the primary malaria vector in Lower Moshi, north-eastern Tanzania</strong> <br />
Johnson Matowo, Christopher M Jones, Bilali Kabula, Hilary Ranson, Keith Steen, Franklin Mosha, Mark Rowland, David Weetman<br />
<em>Parasites &amp; Vectors</em> (2014) <a href="https://doi.org/ghcqgg">https://doi.org/ghcqgg</a> <br />
DOI: <a href="https://doi.org/10.1186/1756-3305-7-274">10.1186/1756-3305-7-274</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24946780">24946780</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4082164">PMC4082164</a></p>
</div>
<div id="ref-BIuyPabi">
<p>39. <strong>WHO | Test procedures for insecticide resistance monitoring in malaria vector mosquitoes (Second edition)</strong> <br />
WHO<br />
<a href="http://www.who.int/malaria/publications/atoz/9789241511575/en/">http://www.who.int/malaria/publications/atoz/9789241511575/en/</a></p>
</div>
<div id="ref-ca2mjT0Y">
<p>40. <strong>Susceptibility status of malaria vectors to insecticides commonly used for malaria control in Tanzania</strong> <br />
Bilali Kabula, Patrick Tungu, Johnson Matowo, Jovin Kitau, Clement Mweya, Basiliana Emidi, Denis Masue, Calvin Sindato, Robert Malima, Jubilate Minja, … William Kisinza<br />
<em>Tropical Medicine &amp; International Health</em> (2012-06) <a href="https://doi.org/f3zsbt">https://doi.org/f3zsbt</a> <br />
DOI: <a href="https://doi.org/10.1111/j.1365-3156.2012.02986.x">10.1111/j.1365-3156.2012.02986.x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/22519840">22519840</a></p>
</div>
<div id="ref-1HPbPGiiF">
<p>41. <strong>Dynamics and monitoring of insecticide resistance in malaria vectors across mainland Tanzania from 1997 to 2017: a systematic review</strong> <br />
Deokary Joseph Matiya, Anitha B. Philbert, Winifrida Kidima, Johnson J. Matowo<br />
<em>Malaria Journal</em> (2019-03-26) <a href="https://doi.org/ghxc2s">https://doi.org/ghxc2s</a> <br />
DOI: <a href="https://doi.org/10.1186/s12936-019-2738-6">10.1186/s12936-019-2738-6</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/30914051">30914051</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6434877">PMC6434877</a></p>
</div>
<div id="ref-dB1ebxOj">
<p>42. <strong>High level of resistance in the mosquito Anopheles gambiae to pyrethroid insecticides and reduced susceptibility to bendiocarb in north-western Tanzania</strong> <br />
Natacha Protopopoff, Johnson Matowo, Robert Malima, Reginald Kavishe, Robert Kaaya, Alexandra Wright, Philippa A West, Immo Kleinschmidt, William Kisinza, Franklin W Mosha, Mark Rowland<br />
<em>Malaria Journal</em> (2013-05-02) <a href="https://doi.org/ghcqgf">https://doi.org/ghcqgf</a> <br />
DOI: <a href="https://doi.org/10.1186/1475-2875-12-149">10.1186/1475-2875-12-149</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23638757">23638757</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3655935">PMC3655935</a></p>
</div>
<div id="ref-1EjgFGIf9">
<p>43. <strong>Anopheles gambiae complex along The Gambia river, with particular reference to the molecular forms of An. gambiae s.s</strong> <br />
Beniamino Caputo, Davis Nwakanma, Musa Jawara, Majidah Adiamoh, Ibrahima Dia, Lassana Konate, Vincenzo Petrarca, David J Conway, Alessandra della Torre<br />
<em>Malaria Journal</em> (2008) <a href="https://doi.org/b97c6c">https://doi.org/b97c6c</a> <br />
DOI: <a href="https://doi.org/10.1186/1475-2875-7-182">10.1186/1475-2875-7-182</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/18803885">18803885</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2569043">PMC2569043</a></p>
</div>
<div id="ref-mJ3VTNed">
<p>44. <strong>Breakdown in the process of incipient speciation in Anopheles gambiae</strong> <br />
Davis C Nwakanma, Daniel E Neafsey, Musa Jawara, Majidah Adiamoh, Emily Lund, Amabelia Rodrigues, Kovana M Loua, Lassana Konate, Ngayo Sy, Ibrahima Dia, … David J Conway<br />
<em>Genetics</em> (2013-04) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3606099/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3606099/</a> <br />
DOI: <a href="https://doi.org/10.1534/genetics.112.148718">10.1534/genetics.112.148718</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23335339">23335339</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3606099">PMC3606099</a></p>
</div>
<div id="ref-eTFJNJEG">
<p>45. <strong>Chromosomal differentiation and adaptation to human environments in the Anopheles gambiae complex.</strong> <br />
M Coluzzi, A Sabatini, V Petrarca, MA Di Deco<br />
<em>Transactions of the Royal Society of Tropical Medicine and Hygiene</em> (1979) <a href="https://www.ncbi.nlm.nih.gov/pubmed/394408">https://www.ncbi.nlm.nih.gov/pubmed/394408</a> <br />
DOI: <a href="https://doi.org/10.1016/0035-9203(79)90036-1">10.1016/0035-9203(79)90036-1</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/394408">394408</a></p>
</div>
<div id="ref-Z6MC0qXg">
<p>46. <strong>The distribution and inversion polymorphism of chromosomally recognized taxa of the Anopheles gambiae complex in Mali, West Africa.</strong> <br />
YT Touré, V Petrarca, SF Traoré, A Coulibaly, HM Maiga, O Sankaré, M Sow, MA Di Deco, M Coluzzi<br />
<em>Parassitologia</em> (1998-12) <a href="https://www.ncbi.nlm.nih.gov/pubmed/10645562">https://www.ncbi.nlm.nih.gov/pubmed/10645562</a> <br />
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/10645562">10645562</a></p>
</div>
<div id="ref-Px2G7gld">
<p>47. <strong>Does insecticide resistance contribute to heterogeneities in malaria transmission in The Gambia?</strong> <br />
Kevin Ochieng’ Opondo, David Weetman, Musa Jawara, Mathurin Diatta, Amfaal Fofana, Florence Crombe, Julia Mwesigwa, Umberto D’Alessandro, Martin James Donnelly<br />
<em>Malaria Journal</em> (2016-03-15) <a href="https://doi.org/ghcqgh">https://doi.org/ghcqgh</a> <br />
DOI: <a href="https://doi.org/10.1186/s12936-016-1203-z">10.1186/s12936-016-1203-z</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/26980461">26980461</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4793517">PMC4793517</a></p>
</div>
<div id="ref-m3y4gwzb">
<p>48. <strong>Estimating the annual entomological inoculation rate for Plasmodium falciparum transmitted by Anopheles gambiae s.l. using three sampling methods in three sites in Uganda</strong> <br />
Maxwell Kilama, David L Smith, Robert Hutchinson, Ruth Kigozi, Adoke Yeka, Geoff Lavoy, Moses R Kamya, Sarah G Staedke, Martin J Donnelly, Chris Drakeley, … Steve W Lindsay<br />
<em>Malaria Journal</em> (2014-03-21) <a href="https://doi.org/gdkzsz">https://doi.org/gdkzsz</a> <br />
DOI: <a href="https://doi.org/10.1186/1475-2875-13-111">10.1186/1475-2875-13-111</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24656206">24656206</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4001112">PMC4001112</a></p>
</div>
<div id="ref-7D6mNysA">
<p>49. <strong>Variation in malaria transmission intensity in seven sites throughout Uganda.</strong> <br />
Paul Edward Okello, Wim Van Bortel, Anatol Maranda Byaruhanga, Anne Correwyn, Patricia Roelants, Ambrose Talisuna, Umberto D’Alessandro, Marc Coosemans<br />
<em>The American journal of tropical medicine and hygiene</em> (2006-08) <a href="https://www.ncbi.nlm.nih.gov/pubmed/16896122">https://www.ncbi.nlm.nih.gov/pubmed/16896122</a> <br />
PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/16896122">16896122</a></p>
</div>
<div id="ref-19o13dRo9">
<p>50. <strong>Insecticide resistance and its association with target-site mutations in natural populations of Anopheles gambiae from eastern Uganda.</strong> <br />
Urvashi Ramphul, Thomas Boase, Chris Bass, Loyce M Okedi, Martin J Donnelly, Pie Müller<br />
<em>Transactions of the Royal Society of Tropical Medicine and Hygiene</em> (2009-03-19) <a href="https://www.ncbi.nlm.nih.gov/pubmed/19303125">https://www.ncbi.nlm.nih.gov/pubmed/19303125</a> <br />
DOI: <a href="https://doi.org/10.1016/j.trstmh.2009.02.014">10.1016/j.trstmh.2009.02.014</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/19303125">19303125</a></p>
</div>
<div id="ref-16MD5kWB3">
<p>51. <strong>Insecticide resistance monitoring of field-collected Anopheles gambiae s.l. populations from Jinja, eastern Uganda, identifies high levels of pyrethroid resistance</strong> <br />
HD Mawejje, CS Wilding, EJ Rippon, A Hughes, D Weetman, MJ Donnelly<br />
<em>Medical and veterinary entomology</em> (2013-09) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3543752/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3543752/</a> <br />
DOI: <a href="https://doi.org/10.1111/j.1365-2915.2012.01055.x">10.1111/j.1365-2915.2012.01055.x</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23046446">23046446</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3543752">PMC3543752</a></p>
</div>
<div id="ref-QHmnxjQ3">
<p>52. <strong>Contemporary gene flow between wild An. gambiae s.s. and An. arabiensis</strong> <br />
David Weetman, Keith Steen, Emily J Rippon, Henry D Mawejje, Martin J Donnelly, Craig S Wilding<br />
<em>Parasites &amp; vectors</em> (2014-07-24) <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4124135/">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4124135/</a> <br />
DOI: <a href="https://doi.org/10.1186/1756-3305-7-345">10.1186/1756-3305-7-345</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/25060488">25060488</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4124135">PMC4124135</a></p>
</div>
<div id="ref-z9HqxYF3">
<p>53. <strong>CYP6 P450 Enzymes and ACE-1 Duplication Produce Extreme and Multiple Insecticide Resistance in the Malaria Mosquito Anopheles gambiae</strong> <br />
Constant V. Edi, Luc Djogbénou, Adam M. Jenkins, Kimberly Regna, Marc A. T. Muskavitch, Rodolphe Poupardin, Christopher M. Jones, John Essandoh, Guillaume K. Kétoh, Mark J. I. Paine, … David Weetman<br />
<em>PLoS Genetics</em> (2014-03-20) <a href="https://doi.org/f56k77">https://doi.org/f56k77</a> <br />
DOI: <a href="https://doi.org/10.1371/journal.pgen.1004236">10.1371/journal.pgen.1004236</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/24651294">24651294</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3961184">PMC3961184</a></p>
</div>
<div id="ref-cQaeXqlZ">
<p>54. <strong>Detecting and Estimating Contamination of Human DNA Samples in Sequencing and Array-Based Genotype Data</strong> <br />
Goo Jun, Matthew Flickinger, Kurt N. Hetrick, Jane M. Romm, Kimberly F. Doheny, Gonçalo R. Abecasis, Michael Boehnke, Hyun Min Kang<br />
<em>The American Journal of Human Genetics</em> (2012-11) <a href="https://doi.org/f4cf5g">https://doi.org/f4cf5g</a> <br />
DOI: <a href="https://doi.org/10.1016/j.ajhg.2012.09.004">10.1016/j.ajhg.2012.09.004</a> · PMID: <a href="https://www.ncbi.nlm.nih.gov/pubmed/23103226">23103226</a> · PMCID: <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3487130">PMC3487130</a></p>
</div>
</div>
<!-- default theme -->

<style>
    /* import google fonts */
    @import url("https://fonts.googleapis.com/css?family=Open+Sans:400,600,700");
    @import url("https://fonts.googleapis.com/css?family=Source+Code+Pro");

    /* -------------------------------------------------- */
    /* global */
    /* -------------------------------------------------- */

    /* all elements */
    * {
        /* force sans-serif font unless specified otherwise */
        font-family: "Open Sans", "Helvetica", sans-serif;

        /* prevent text inflation on some mobile browsers */
        -webkit-text-size-adjust: none !important;
        -moz-text-size-adjust: none !important;
        -o-text-size-adjust: none !important;
        text-size-adjust: none !important;
    }

    @media only screen {
        /* "page" element */
        body {
            position: relative;
            box-sizing: border-box;
            font-size: 12pt;
            line-height: 1.5;
            max-width: 8.5in;
            margin: 20px auto;
            padding: 40px;
            border-radius: 5px;
            border: solid 1px #bdbdbd;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            background: #ffffff;
        }
    }

    /* when on screen < 8.5in wide */
    @media only screen and (max-width: 8.5in) {
        /* "page" element */
        body {
            padding: 20px;
            margin: 0;
            border-radius: 0;
            border: none;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05) inset;
            background: none;
        }
    }

    /* -------------------------------------------------- */
    /* headings */
    /* -------------------------------------------------- */

    /* all headings */
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
        margin: 20px 0;
        padding: 0;
        font-weight: bold;
    }

    /* biggest heading */
    h1 {
        margin: 40px 0;
        text-align: center;
    }

    /* second biggest heading */
    h2 {
        margin-top: 30px;
        padding-bottom: 5px;
        border-bottom: solid 1px #bdbdbd;
    }

    /* heading font sizes */
    h1 {
        font-size: 2em;
    }
    h2 {
        font-size: 1.5em;
    }
    h3{
        font-size: 1.35em;
    }
    h4 {
        font-size: 1.25em;
    }
    h5 {
        font-size: 1.15em;
    }
    h6 {
        font-size: 1em;
    }

    /* -------------------------------------------------- */
    /* manuscript header */
    /* -------------------------------------------------- */

    /* manuscript title */
    header > h1 {
        margin: 0;
    }

    /* manuscript title caption text (ie "automatically generated on") */
    header + p {
        text-align: center;
        margin-top: 10px;
    }

    /* -------------------------------------------------- */
    /* text elements */
    /* -------------------------------------------------- */

    /* links */
    a {
        color: #2196f3;
        overflow-wrap: break-word;
    }

    /* normal links (not empty, not button link, not syntax highlighting link) */
    a:not(:empty):not(.button):not(.sourceLine) {
        padding-left: 1px;
        padding-right: 1px;
    }

    /* superscripts and subscripts */
    sub,
    sup {
        /* prevent from affecting line height */
        line-height: 0;
    }

    /* unordered and ordered lists*/
    ul,
    ol {
        padding-left: 20px;
    }

    /* class for styling text semibold */
    .semibold {
        font-weight: 600;
    }

    /* class for styling elements horizontally left aligned */
    .left {
        display: block;
        text-align: left;
        margin-left: auto;
        margin-right: 0;
        justify-content: left;
    }

    /* class for styling elements horizontally centered */
    .center {
        display: block;
        text-align: center;
        margin-left: auto;
        margin-right: auto;
        justify-content: center;
    }

    /* class for styling elements horizontally right aligned */
    .right {
        display: block;
        text-align: right;
        margin-left: 0;
        margin-right: auto;
        justify-content: right;
    }

    /* -------------------------------------------------- */
    /* section elements */
    /* -------------------------------------------------- */

    /* horizontal divider line */
    hr {
        border: none;
        height: 1px;
        background: #bdbdbd;
    }

    /* paragraphs, horizontal dividers, figures, tables, code */
    p,
    hr,
    figure,
    table,
    pre {
        /* treat all as "paragraphs", with consistent vertical margins */
        margin-top: 20px;
        margin-bottom: 20px;
    }

    /* -------------------------------------------------- */
    /* figures */
    /* -------------------------------------------------- */

    /* figure */
    figure {
        max-width: 100%;
        margin-left: auto;
        margin-right: auto;
    }

    /* figure caption */
    figcaption {
        padding: 0;
        padding-top: 10px;
    }

    /* figure image element */
    figure > img,
    figure > svg {
        max-width: 100%;
        display: block;
        margin-left: auto;
        margin-right: auto;
    }

    /* figure auto-number */
    img + figcaption > span:first-of-type,
    svg + figcaption > span:first-of-type {
        font-weight: bold;
        margin-right: 5px;
    }

    /* -------------------------------------------------- */
    /* tables */
    /* -------------------------------------------------- */

    /* table */
    table {
        border-collapse: collapse;
        border-spacing: 0;
        width: 100%;
        margin-left: auto;
        margin-right: auto;
    }

    /* table cells */
    th,
    td {
        border: solid 1px #bdbdbd;
        padding: 10px;
        /* squash table if too wide for page by forcing line breaks */
        overflow-wrap: break-word;
        word-break: break-word;
    }

    /* header row and even rows */
    th,
    tr:nth-child(2n) {
        background-color: #fafafa;
    }

    /* odd rows */
    tr:nth-child(2n + 1) {
        background-color: #ffffff;
    }

    /* table caption */
    caption {
        text-align: left;
        padding: 0;
        padding-bottom: 10px;
    }

    /* table auto-number */
    table > caption > span:first-of-type,
    div.table_wrapper > table > caption > span:first-of-type {
        font-weight: bold;
        margin-right: 5px;
    }

    /* -------------------------------------------------- */
    /* code */
    /* -------------------------------------------------- */

    /* multi-line code block */
    pre {
        padding: 10px;
        background-color: #eeeeee;
        color: #000000;
        border-radius: 5px;
        break-inside: avoid;
        text-align: left;
    }

    /* inline code, ie code within normal text */
    :not(pre) > code {
        padding: 0 4px;
        background-color: #eeeeee;
        color: #000000;
        border-radius: 5px;
    }

    /* code text */
    /* apply all children, to reach syntax highlighting sub-elements */
    code,
    code * {
        /* force monospace font */
        font-family: "Source Code Pro", "Courier New", monospace;
    }

    /* -------------------------------------------------- */
    /* quotes */
    /* -------------------------------------------------- */

    /* quoted text */
    blockquote {
        margin: 0;
        padding: 0;
        border-left: 4px solid #bdbdbd;
        padding-left: 16px;
        break-inside: avoid;
    }

    /* -------------------------------------------------- */
    /* banners */
    /* -------------------------------------------------- */

    /* info banners */
    .banner {
        box-sizing: border-box;
        display: block;
        position: relative;
        width: 100%;
        margin-top: 20px;
        margin-bottom: 20px;
        padding: 20px;
        text-align: center;
    }

    /* paragraph in banner */
    .banner > p {
        margin: 0;
    }

    /* -------------------------------------------------- */
    /* highlight colors */
    /* -------------------------------------------------- */

    .white {
        background: #ffffff;
    }
    .lightgrey {
        background: #eeeeee;
    }
    .grey {
        background: #757575;
    }
    .darkgrey {
        background: #424242;
    }
    .black {
        background: #000000;
    }
    .lightred {
        background: #ffcdd2;
    }
    .lightyellow {
        background: #ffecb3;
    }
    .lightgreen {
        background: #dcedc8;
    }
    .lightblue {
        background: #e3f2fd;
    }
    .lightpurple {
        background: #f3e5f5;
    }
    .red {
        background: #f44336;
    }
    .orange {
        background: #ff9800;
    }
    .yellow {
        background: #ffeb3b;
    }
    .green {
        background: #4caf50;
    }
    .blue {
        background: #2196f3;
    }
    .purple {
        background: #9c27b0;
    }
    .white,
    .lightgrey,
    .lightred,
    .lightyellow,
    .lightgreen,
    .lightblue,
    .lightpurple,
    .orange,
    .yellow,
    .white a,
    .lightgrey a,
    .lightred a,
    .lightyellow a,
    .lightgreen a,
    .lightblue a,
    .lightpurple a,
    .orange a,
    .yellow a {
        color: #000000;
    }
    .grey,
    .darkgrey,
    .black,
    .red,
    .green,
    .blue,
    .purple,
    .grey a,
    .darkgrey a,
    .black a,
    .red a,
    .green a,
    .blue a,
    .purple a {
        color: #ffffff;
    }

    /* -------------------------------------------------- */
    /* buttons */
    /* -------------------------------------------------- */

    /* class for styling links like buttons */
    .button {
        display: inline-flex;
        justify-content: center;
        align-items: center;
        margin: 5px;
        padding: 10px 20px;
        font-size: 0.75em;
        font-weight: 600;
        text-transform: uppercase;
        text-decoration: none;
        letter-spacing: 1px;
        background: none;
        color: #2196f3;
        border: solid 1px #bdbdbd;
        border-radius: 5px;
    }

    /* buttons when hovered */
    .button:hover:not([disabled]),
    .icon_button:hover:not([disabled]) {
        cursor: pointer;
        background: #f5f5f5;
    }

    /* buttons when disabled */
    .button[disabled],
    .icon_button[disabled] {
        opacity: 0.35;
        pointer-events: none;
    }

    /* class for styling buttons containg only single icon */
    .icon_button {
        display: inline-flex;
        justify-content: center;
        align-items: center;
        text-decoration: none;
        margin: 0;
        padding: 0;
        background: none;
        border-radius: 5px;
        border: none;
        width: 20px;
        height: 20px;
        min-width: 20px;
        min-height: 20px;
    }

    /* icon button inner svg image */
    .icon_button > svg {
        height: 16px;
    }

    /* -------------------------------------------------- */
    /* icons */
    /* -------------------------------------------------- */

    /* class for styling icons inline with text */
    .inline_icon {
        height: 1em;
        position: relative;
        top: 0.125em;
    }

    /* -------------------------------------------------- */
    /* print control */
    /* -------------------------------------------------- */

    @media print {
        @page {
            /* suggested printing margin */
            margin: 0.5in;
        }

        /* document and "page" elements */
        html, body {
            margin: 0;
            padding: 0;
            width: 100%;
            height: 100%;
        }

        /* "page" element */
        body {
            font-size: 11pt !important;
            line-height: 1.35;
        }

        /* all headings */
        h1,
        h2,
        h3,
        h4,
        h5,
        h6 {
            margin: 15px 0;
        }

        /* figures and tables */
        figure, table {
            font-size: 0.85em;
        }

        /* table cells */
        th,
        td {
            padding: 5px;
        }

        /* shrink font awesome icons */
        i.fas,
        i.fab,
        i.far,
        i.fal {
            transform: scale(0.85);
        }

        /* decrease banner margins */
        .banner {
            margin-top: 15px;
            margin-bottom: 15px;
            padding: 15px;
        }

        /* class for centering an element vertically on its own page */
        .page_center {
            margin: auto;
            width: 100%;
            height: 100%;
            display: flex;
            align-items: center;
            vertical-align: middle;
            break-before: page;
            break-after: page;
        }

        /* always insert a page break before the element */
        .page_break_before {
            break-before: page;
        }

        /* always insert a page break after the element */
        .page_break_after {
            break-after: page;
        }

        /* avoid page break before the element */
        .page_break_before_avoid {
            break-before: avoid;
        }

        /* avoid page break after the element */
        .page_break_after_avoid {
            break-after: avoid;
        }

        /* avoid page break inside the element */
        .page_break_inside_avoid {
            break-inside: avoid;
        }
    }

    /* -------------------------------------------------- */
    /* override pandoc css quirks */
    /* -------------------------------------------------- */

    .sourceCode {
        /* prevent unsightly overflow in wide code blocks */
        overflow: auto !important;
    }

    div.sourceCode {
        /* prevent background fill on top-most code block  container */
        background: none !important;
    }

    .sourceCode * {
        /* force consistent line spacing */
        line-height: 1.5 !important;
    }

    div.sourceCode {
        /* style code block margins same as <pre> element */
        margin-top: 20px;
        margin-bottom: 20px;
    }

    /* -------------------------------------------------- */
    /* tablenos */
    /* -------------------------------------------------- */

    /* tablenos wrapper */
    .tablenos {
        /* show scrollbar on tables if necessary to prevent overflow */
        width: 100%;
        margin: 20px 0;
    }

    .tablenos > table {
        /* move margins from table to table_wrapper to allow margin collapsing */
        margin: 0;
    }

    @media only screen {
        /* tablenos wrapper */
        .tablenos {
            /* show scrollbar on tables if necessary to prevent overflow */
            overflow-x: auto !important;
        }

        .tablenos th,
        .tablenos td {
            overflow-wrap: unset !important;
            word-break: unset !important;
        }

        /* table in wrapper */
        .tablenos table,
        .tablenos table * {
            /* don't break table words */
            overflow-wrap: normal !important;
        }
    }

    /* -------------------------------------------------- */
    /* mathjax */
    /* -------------------------------------------------- */

    /* mathjax containers */
    .math.display > span:not(.MathJax_Preview) {
        /* turn inline element (no dimensions) into block (allows fixed width and thus scrolling) */
        display: flex !important;
        overflow-x: auto !important;
        overflow-y: hidden !important;
        justify-content: center;
        align-items: center;
        margin: 0 !important;
    }

    /* right click menu */
    .MathJax_Menu {
        border-radius: 5px !important;
        border: solid 1px #bdbdbd !important;
        box-shadow: none !important;
    }

    /* equation auto-number */
    span[id^="eq:"] > span.math.display + span {
        font-weight: 600;
    }

    /* equation */
    span[id^="eq:"] > span.math.display > span {
        /* nudge to make room for equation auto-number and anchor */
        margin-right: 60px !important;
    }

    /* -------------------------------------------------- */
    /* anchors plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* anchor button */
        .anchor {
            opacity: 0;
            margin-left: 5px;
        }

        /* anchor buttons within <h2>'s */
        h2 .anchor {
            margin-left: 10px;
        }

        /* anchor buttons when hovered/focused and anything containing an anchor button when hovered */
        *:hover > .anchor,
        .anchor:hover,
        .anchor:focus {
            opacity: 1;
        }

        /* anchor button when hovered */
        .anchor:hover {
            cursor: pointer;
        }
    }

    /* always show anchor button on devices with no mouse/hover ability */
    @media (hover: none) {
        .anchor {
            opacity: 1;
        }
    }

    /* always hide anchor button on print */
    @media only print {
        .anchor {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* accordion plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* accordion arrow button */
        .accordion_arrow {
            margin-right: 10px;
        }

        /* arrow icon when <h2> data-collapsed attribute true */
        h2[data-collapsed="true"] > .accordion_arrow > svg {
            transform: rotate(-90deg);
        }

        /* all elements (except <h2>'s) when data-collapsed attribute true */
        *:not(h2)[data-collapsed="true"] {
            display: none;
        }

        /* accordion arrow button when hovered and <h2>'s when hovered */
        .accordion_arrow:hover,
        h2[data-collapsed="true"]:hover,
        h2[data-collapsed="false"]:hover {
            cursor: pointer;
        }
    }

    /* always hide accordion arrow button on print */
    @media only print {
        .accordion_arrow {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* tooltips plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* tooltip container */
        #tooltip {
            position: absolute;
            width: 50%;
            min-width: 240px;
            max-width: 75%;
            z-index: 1;
        }

        /* tooltip content */
        #tooltip_content {
            margin-bottom: 5px;
            padding: 20px;
            border-radius: 5px;
            border: solid 1px #bdbdbd;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            background: #ffffff;
            overflow-wrap: break-word;
        }

        /* tooltip copy of paragraphs and figures */
        #tooltip_content > p,
        #tooltip_content > figure {
            margin: 0;
            max-height: 320px;
            overflow-y: auto;
        }

        /* tooltip copy of <img> */
        #tooltip_content > figure > img,
        #tooltip_content > figure > svg {
            max-height: 260px;
        }

        /* navigation bar */
        #tooltip_nav_bar {
            margin-top: 10px;
            text-align: center;
        }

        /* navigation bar previous/next buton */
        #tooltip_nav_bar > .icon_button {
            position: relative;
            top: 3px;
        }

        /* navigation bar previous button */
        #tooltip_nav_bar > .icon_button:first-of-type {
            margin-right: 5px;
        }

        /* navigation bar next button */
        #tooltip_nav_bar > .icon_button:last-of-type {
            margin-left: 5px;
        }
    }

    /* always hide tooltip on print */
    @media only print {
        #tooltip {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* jump to first plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* jump button */
        .jump_arrow {
            position: relative;
            top: 0.125em;
            margin-right: 5px;
        }
    }

    /* always hide jump button on print */
    @media only print {
        .jump_arrow {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* link highlight plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* anything with data-highlighted attribute true */
        [data-highlighted="true"] {
            background: #ffeb3b;
        }

        /* anything with data-selected attribute true */
        [data-selected="true"] {
            background: #ff8a65 !important;
        }

        /* animation definition for glow */
        @keyframes highlight_glow {
            0% {
                background: none;
            }
            10% {
                background: #bbdefb;
            }
            100% {
                background: none;
            }
        }

        /* anything with data-glow attribute true */
        [data-glow="true"] {
            animation: highlight_glow 2s;
        }
    }

    /* -------------------------------------------------- */
    /* table of contents plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* toc panel */
        #toc_panel {
            box-sizing: border-box;
            position: fixed;
            top: 0;
            left: 0;
            background: #ffffff;
            box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
            z-index: 2;
        }

        /* toc panel when closed */
        #toc_panel[data-open="false"] {
            min-width: 60px;
            width: 60px;
            height: 60px;
            border-right: solid 1px #bdbdbd;
            border-bottom: solid 1px #bdbdbd;
        }

        /* toc panel when open */
        #toc_panel[data-open="true"] {
            min-width: 260px;
            max-width: 480px;
            /* keep panel edge consistent distance away from "page" edge */
            width: calc(((100vw - 8.5in) / 2) - 30px - 40px);
            bottom: 0;
            border-right: solid 1px #bdbdbd;
        }

        /* toc panel header */
        #toc_header {
            box-sizing: border-box;
            display: flex;
            flex-direction: row;
            align-items: center;
            height: 60px;
            margin: 0;
            padding: 20px;
        }

        /* toc panel header when hovered */
        #toc_header:hover {
            cursor: pointer;
        }

        /* toc panel header when panel open */
        #toc_panel[data-open="true"] > #toc_header {
            border-bottom: solid 1px #bdbdbd;
        }

        /* toc open/close header button */
        #toc_button {
            margin-right: 20px;
        }

        /* hide toc list and header text when closed */
        #toc_panel[data-open="false"] > #toc_header > *:not(#toc_button),
        #toc_panel[data-open="false"] > #toc_list {
            display: none;
        }

        /* toc list of entries */
        #toc_list {
            box-sizing: border-box;
            width: 100%;
            padding: 20px;
            position: absolute;
            top: calc(60px + 1px);
            bottom: 0;
            overflow: auto;
        }

        /* toc entry, link to section in document */
        .toc_link {
            display: block;
            padding: 5px;
            position: relative;
            font-weight: 600;
            text-decoration: none;
        }

        /* toc entry when hovered or when "viewed" */
        .toc_link:hover,
        .toc_link[data-viewing="true"] {
            background: #f5f5f5;
        }

        /* toc entry, level 1 indentation */
        .toc_link[data-level="1"] {
            margin-left: 0;
        }

        /* toc entry, level 2 indentation */
        .toc_link[data-level="2"] {
            margin-left: 20px;
        }

        /* toc entry, level 3 indentation */
        .toc_link[data-level="3"] {
            margin-left: 40px;
        }

        /* toc entry, level 4 indentation */
        .toc_link[data-level="4"] {
            margin-left: 60px;
        }

        /* toc entry bullets */
        #toc_panel[data-bullets="true"] .toc_link[data-level]:before {
            position: absolute;
            left: -15px;
            top: -1px;
            font-size: 1.5em;
        }

        /* toc entry, level 2 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="2"]:before {
            content: "\2022";
        }

        /* toc entry, level 3 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="3"]:before {
            content: "\25AB";
        }

        /* toc entry, level 4 bullet */
        #toc_panel[data-bullets="true"] .toc_link[data-level="4"]:before {
            content: "-";
        }
    }

    /* when on screen < 8.5in wide */
    @media only screen and (max-width: 8.5in) {
        /* push <body> ("page") element down to make room for toc icon */
        .toc_body_nudge {
            padding-top: 60px;
        }

        /* toc icon when panel closed and not hovered */
        #toc_panel[data-open="false"]:not(:hover) {
            background: rgba(255, 255, 255, 0.75);
        }
    }

    /* always hide toc panel on print */
    @media only print {
        #toc_panel {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* lightbox plugin */
    /* -------------------------------------------------- */

    @media only screen {
        /* regular <img> in document when hovered */
        img.lightbox_document_img:hover {
            cursor: pointer;
        }

        .body_no_scroll {
            overflow: hidden !important;
        }

        /* screen overlay */
        #lightbox_overlay {
            display: flex;
            flex-direction: column;
            position: fixed;
            left: 0;
            top: 0;
            right: 0;
            bottom: 0;
            background: rgba(0, 0, 0, 0.75);
            z-index: 3;
        }

        /* middle area containing lightbox image */
        #lightbox_image_container {
            flex-grow: 1;
            display: flex;
            justify-content: center;
            align-items: center;
            overflow: hidden;
            position: relative;
            padding: 20px;
        }

        /* bottom area containing caption */
        #lightbox_bottom_container {
            display: flex;
            justify-content: center;
            align-items: center;
            height: 100px;
            min-height: 100px;
            max-height: 100px;
            background: rgba(0, 0, 0, 0.5);
        }

        /* image number info text box */
        #lightbox_number_info {
            position: absolute;
            color: #ffffff;
            font-weight: 600;
            left: 2px;
            top: 0;
            z-index: 4;
        }

        /* zoom info text box */
        #lightbox_zoom_info {
            position: absolute;
            color: #ffffff;
            font-weight: 600;
            right: 2px;
            top: 0;
            z-index: 4;
        }

        /* copy of image caption */
        #lightbox_caption {
            box-sizing: border-box;
            display: inline-block;
            width: 100%;
            max-height: 100%;
            padding: 10px 0;
            text-align: center;
            overflow-y: auto;
            color: #ffffff;
        }

        /* navigation previous/next button */
        .lightbox_button {
            width: 100px;
            height: 100%;
            min-width: 100px;
            min-height: 100%;
            color: #ffffff;
        }

        /* navigation previous/next button when hovered */
        .lightbox_button:hover {
            background: none !important;
        }

        /* navigation button icon */
        .lightbox_button > svg {
            height: 25px;
        }

        /* figure auto-number */
        #lightbox_caption > span:first-of-type {
            font-weight: bold;
            margin-right: 5px;
        }

        /* lightbox image when hovered */
        #lightbox_img:hover {
            cursor: grab;
        }

        /* lightbox image when grabbed */
        #lightbox_img:active {
            cursor: grabbing;
        }
    }

    /* when on screen < 480px wide */
    @media only screen and (max-width: 480px) {
        /* make navigation buttons skinnier on small screens to make more room for caption text */
        .lightbox_button {
            width: 50px;
            min-width: 50px;
        }
    }

    /* always hide lightbox on print */
    @media only print {
        #lightbox_overlay {
            display: none;
        }
    }

    /* -------------------------------------------------- */
    /* hypothesis (annotations) plugin */
    /* -------------------------------------------------- */

    /* hypothesis activation button */
    #hypothesis_button {
        box-sizing: border-box;
        position: fixed;
        top: 0;
        right: 0;
        width: 60px;
        height: 60px;
        background: #ffffff;
        border-radius: 0;
        border-left: solid 1px #bdbdbd;
        border-bottom: solid 1px #bdbdbd;
        box-shadow: 0 0 20px rgba(0, 0, 0, 0.05);
        z-index: 2;
    }

    /* hypothesis button svg */
    #hypothesis_button > svg {
        position: relative;
        top: -4px;
    }

    /* hypothesis annotation count */
    #hypothesis_count {
        position: absolute;
        left: 0;
        right: 0;
        bottom: 5px;
    }

    /* side panel */
    .annotator-frame {
        width: 280px !important;
    }

    /* match highlight color to rest of theme */
    .annotator-highlights-always-on .annotator-hl {
        background-color: #ffeb3b !important;
    }

    /* match focused color to rest of theme */
    .annotator-hl.annotator-hl-focused {
        background-color: #ff8a65 !important;
    }

    /* match bucket bar color to rest of theme */
    .annotator-bucket-bar {
        background: #f5f5f5 !important;
    }

    /* always hide button, toolbar, and tooltip on print */
    @media only print {
        #hypothesis_button {
            display: none;
        }

        .annotator-frame {
            display: none !important;
        }

        hypothesis-adder {
            display: none !important;
        }
    }
</style>
<!-- anchors plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin adds an anchor next to each of a certain type
        // of element that provides a human-readable url to that specific
        // item/position in the document (eg "manuscript.html#abstract"). It
        // also makes it such that scrolling out of view of a target removes
        // its identifier from the url.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'anchors';

        // default plugin options
        const options = {
            // which types of elements to add anchors next to, in
            // "document.querySelector" format
            typesQuery: 'h1, h2, h3, [id^="fig:"], [id^="tbl:"], [id^="eq:"]',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // add anchor to each element of specified types
            const elements = document.querySelectorAll(options.typesQuery);
            for (const element of elements)
                addAnchor(element);

            // attach scroll listener to window
            window.addEventListener('scroll', onScroll);
        }

        // when window is scrolled
        function onScroll() {
            // if url has hash and user has scrolled out of view of hash
            // target, remove hash from url
            const tolerance = 100;
            const target = getHashTarget();
            if (target) {
                if (
                    target.getBoundingClientRect().top >
                        window.innerHeight + tolerance ||
                    target.getBoundingClientRect().bottom < 0 - tolerance
                )
                    history.pushState(null, null, ' ');
            }
        }

        // add anchor to element
        function addAnchor(element) {
            let addTo; // element to add anchor button to

            // if figure or table, modify withId and addTo to get expected
            // elements
            if (element.id.indexOf('fig:') === 0) {
                addTo = element.querySelector('figcaption');
            } else if (element.id.indexOf('tbl:') === 0) {
                addTo = element.querySelector('caption');
            } else if (element.id.indexOf('eq:') === 0) {
                addTo = element.querySelector('.eqnos-number');
            }

            addTo = addTo || element;
            const id = element.id || null;

            // do not add anchor if element doesn't have assigned id.
            // id is generated by pandoc and is assumed to be unique and
            // human-readable
            if (!id)
                return;

            // create anchor button
            const anchor = document.createElement('a');
            anchor.innerHTML = document.querySelector('.icon_link').innerHTML;
            anchor.title = 'Link to this part of the document';
            anchor.classList.add('icon_button', 'anchor');
            anchor.dataset.ignore = 'true';
            anchor.href = '#' + id;
            addTo.appendChild(anchor);
        }

        // get element that is target of link or url hash
        function getHashTarget() {
            const hash = window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            // if figure or table, modify target to get expected element
            if (id.indexOf('fig:') === 0)
                target = target.querySelector('figure');
            if (id.indexOf('tbl:') === 0)
                target = target.querySelector('table');

            return target;
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- link icon -->

<template class="icon_link">
    <!-- modified from: https://fontawesome.com/icons/link -->
    <svg width="16" height="16" viewBox="0 0 512 512">
        <path
            fill="currentColor"
            d="M326.612 185.391c59.747 59.809 58.927 155.698.36 214.59-.11.12-.24.25-.36.37l-67.2 67.2c-59.27 59.27-155.699 59.262-214.96 0-59.27-59.26-59.27-155.7 0-214.96l37.106-37.106c9.84-9.84 26.786-3.3 27.294 10.606.648 17.722 3.826 35.527 9.69 52.721 1.986 5.822.567 12.262-3.783 16.612l-13.087 13.087c-28.026 28.026-28.905 73.66-1.155 101.96 28.024 28.579 74.086 28.749 102.325.51l67.2-67.19c28.191-28.191 28.073-73.757 0-101.83-3.701-3.694-7.429-6.564-10.341-8.569a16.037 16.037 0 0 1-6.947-12.606c-.396-10.567 3.348-21.456 11.698-29.806l21.054-21.055c5.521-5.521 14.182-6.199 20.584-1.731a152.482 152.482 0 0 1 20.522 17.197zM467.547 44.449c-59.261-59.262-155.69-59.27-214.96 0l-67.2 67.2c-.12.12-.25.25-.36.37-58.566 58.892-59.387 154.781.36 214.59a152.454 152.454 0 0 0 20.521 17.196c6.402 4.468 15.064 3.789 20.584-1.731l21.054-21.055c8.35-8.35 12.094-19.239 11.698-29.806a16.037 16.037 0 0 0-6.947-12.606c-2.912-2.005-6.64-4.875-10.341-8.569-28.073-28.073-28.191-73.639 0-101.83l67.2-67.19c28.239-28.239 74.3-28.069 102.325.51 27.75 28.3 26.872 73.934-1.155 101.96l-13.087 13.087c-4.35 4.35-5.769 10.79-3.783 16.612 5.864 17.194 9.042 34.999 9.69 52.721.509 13.906 17.454 20.446 27.294 10.606l37.106-37.106c59.271-59.259 59.271-155.699.001-214.959z"
        ></path>
    </svg>
</template>
<!-- accordion plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin allows sections of content under <h2> headings
        // to be collapsible.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'accordion';

        // default plugin options
        const options = {
            // whether to always start expanded ('false'), always start
            // collapsed ('true'), or start collapsed when screen small ('auto')
            startCollapsed: 'auto',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // run through each <h2> heading
            const headings = document.querySelectorAll('h2');
            for (const heading of headings) {
                addArrow(heading);

                // start expanded/collapsed based on option
                if (
                    options.startCollapsed === 'true' ||
                    (options.startCollapsed === 'auto' && isSmallScreen())
                )
                    collapseHeading(heading);
                else
                    expandHeading(heading);
            }

            // attach hash change listener to window
            window.addEventListener('hashchange', onHashChange);
        }

        // when hash (eg manuscript.html#introduction) changes
        function onHashChange() {
            const target = getHashTarget();
            if (target)
                goToElement(target);
        }

        // add arrow to heading
        function addArrow(heading) {
            // add arrow button
            const arrow = document.createElement('button');
            arrow.innerHTML = document.querySelector(
                '.icon_angle_down'
            ).innerHTML;
            arrow.classList.add('icon_button', 'accordion_arrow');
            heading.insertBefore(arrow, heading.firstChild);

            // attach click listener to heading and button
            heading.addEventListener('click', onHeadingClick);
            arrow.addEventListener('click', onArrowClick);
        }

        // determine if on mobile-like device with small screen
        function isSmallScreen() {
            return Math.min(window.innerWidth, window.innerHeight) < 480;
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // get element that is target of hash
        function getHashTarget(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            // if figure or table, modify target to get expected element
            if (id.indexOf('fig:') === 0)
                target = target.querySelector('figure');
            if (id.indexOf('tbl:') === 0)
                target = target.querySelector('table');

            return target;
        }

        // when <h2> heading is clicked
        function onHeadingClick(event) {
            // only collapse if <h2> itself is target of click (eg, user did
            // not click on anchor within <h2>)
            if (event.target === this)
                toggleCollapse(this);
        }

        // when arrow button is clicked
        function onArrowClick() {
            toggleCollapse(this.parentNode);
        }

        // collapse section if expanded, expand if collapsed
        function toggleCollapse(heading) {
            if (heading.dataset.collapsed === 'false')
                collapseHeading(heading);
            else
                expandHeading(heading);
        }

        // elements to exclude from collapse, such as table of contents panel,
        // hypothesis panel, etc
        const exclude = '#toc_panel, div.annotator-frame, #lightbox_overlay';

        // collapse section
        function collapseHeading(heading) {
            heading.setAttribute('data-collapsed', 'true');
            const children = getChildren(heading);
            for (const child of children)
                child.setAttribute('data-collapsed', 'true');
        }

        // expand section
        function expandHeading(heading) {
            heading.setAttribute('data-collapsed', 'false');
            const children = getChildren(heading);
            for (const child of children)
                child.setAttribute('data-collapsed', 'false');
        }

        // get list of elements between this <h2> and next <h2> or <h1>
        // ("children" of the <h2> section)
        function getChildren(heading) {
            return nextUntil(heading, 'h2, h1', exclude);
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get list of elements after a start element up to element matching
        // query
        function nextUntil(element, query, exclude) {
            const elements = [];
            while (element = element.nextElementSibling, element) {
                if (element.matches(query))
                    break;
                if (!element.matches(exclude))
                    elements.push(element);
            }
            return elements;
        }

        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- angle down icon -->

<template class="icon_angle_down">
    <!-- modified from: https://fontawesome.com/icons/angle-down -->
    <svg width="16" height="16" viewBox="0 0 448 512">
        <path
            fill="currentColor"
            d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
        ></path>
    </svg>
</template>
<!-- tooltips plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when the user hovers or
        // focuses a link to a citation or figure, a tooltip appears with a
        // preview of the reference content, along with arrows to navigate
        // between instances of the same reference in the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'tooltips';

        // default plugin options
        const options = {
            // whether user must click off to close tooltip instead of just
            // un-hovering
            clickClose: 'false',
            // delay (in ms) between opening and closing tooltip
            delay: '100',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            const links = getLinks();
            for (const link of links) {
                // attach hover and focus listeners to link
                link.addEventListener('mouseover', onLinkHover);
                link.addEventListener('mouseleave', onLinkUnhover);
                link.addEventListener('focus', onLinkFocus);
                link.addEventListener('touchend', onLinkTouch);
            }

            // attach mouse, key, and resize listeners to window
            window.addEventListener('mousedown', onClick);
            window.addEventListener('touchstart', onClick);
            window.addEventListener('keyup', onKeyUp);
            window.addEventListener('resize', onResize);
        }

        // when link is hovered
        function onLinkHover() {
            // function to open tooltip
            const delayOpenTooltip = function() {
                openTooltip(this);
            }.bind(this);

            // run open function after delay
            this.openTooltipTimer = window.setTimeout(
                delayOpenTooltip,
                options.delay
            );
        }

        // when mouse leaves link
        function onLinkUnhover() {
            // cancel opening tooltip
            window.clearTimeout(this.openTooltipTimer);

            // don't close on unhover if option specifies
            if (options.clickClose === 'true')
                return;

            // function to close tooltip
            const delayCloseTooltip = function() {
                // if tooltip open and if mouse isn't over tooltip, close
                const tooltip = document.getElementById('tooltip');
                if (tooltip && !tooltip.matches(':hover'))
                    closeTooltip();
            };

            // run close function after delay
            this.closeTooltipTimer = window.setTimeout(
                delayCloseTooltip,
                options.delay
            );
        }

        // when link is focused (tabbed to)
        function onLinkFocus(event) {
            openTooltip(this);
        }

        // when link is touched on touch screen
        function onLinkTouch(event) {
            // attempt to force hover state on first tap always, and trigger
            // regular link click (and navigation) on second tap
            if (event.target === document.activeElement)
                event.target.click();
            else {
                document.activeElement.blur();
                event.target.focus();
            }
            if (event.cancelable)
                event.preventDefault();
            event.stopPropagation();
            return false;
        }

        // when mouse is clicked anywhere in window
        function onClick(event) {
            closeTooltip();
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            switch (event.key) {
                // trigger click of prev button
                case 'ArrowLeft':
                    const prevButton = document.getElementById(
                        'tooltip_prev_button'
                    );
                    if (prevButton)
                        prevButton.click();
                    break;
                // trigger click of next button
                case 'ArrowRight':
                    const nextButton = document.getElementById(
                        'tooltip_next_button'
                    );
                    if (nextButton)
                        nextButton.click();
                    break;
                // close on esc
                case 'Escape':
                    closeTooltip();
                    break;
            }
        }

        // when window is resized or zoomed
        function onResize() {
            closeTooltip();
        }

        // get all links of types we wish to handle
        function getLinks() {
            const queries = [];
            // exclude buttons, anchor links, toc links, etc
            const exclude =
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            queries.push('a[href^="#ref-"]' + exclude); // citation links
            queries.push('a[href^="#fig:"]' + exclude); // figure links
            const query = queries.join(', ');
            return document.querySelectorAll(query);
        }

        // get links with same target, get index of link in set, get total
        // same links
        function getSameLinks(link) {
            const sameLinks = [];
            const links = getLinks();
            for (const otherLink of links) {
                if (
                    otherLink.getAttribute('href') === link.getAttribute('href')
                )
                    sameLinks.push(otherLink);
            }

            return {
                elements: sameLinks,
                index: sameLinks.indexOf(link),
                total: sameLinks.length
            };
        }

        // open tooltip
        function openTooltip(link) {
            // delete tooltip if it exists, start fresh
            closeTooltip();

            // make tooltip element
            const tooltip = makeTooltip(link);

            // if source couldn't be found and tooltip not made, exit
            if (!tooltip)
                return;

            // make navbar elements
            const navBar = makeNavBar(link);
            if (navBar)
                tooltip.firstElementChild.appendChild(navBar);

            // attach tooltip to page
            document.body.appendChild(tooltip);

            // position tooltip
            const position = function() {
                positionTooltip(link);
            };
            position();

            // if tooltip contains images, position again after they've loaded
            const imgs = tooltip.querySelectorAll('img');
            for (const img of imgs)
                img.addEventListener('load', position);
        }

        // close (delete) tooltip
        function closeTooltip() {
            const tooltip = document.getElementById('tooltip');
            if (tooltip)
                tooltip.remove();
        }

        // make tooltip
        function makeTooltip(link) {
            // get target element that link points to
            const source = getSource(link);

            // if source can't be found, exit
            if (!source)
                return;

            // create new tooltip
            const tooltip = document.createElement('div');
            tooltip.id = 'tooltip';
            const tooltipContent = document.createElement('div');
            tooltipContent.id = 'tooltip_content';
            tooltip.appendChild(tooltipContent);

            // make copy of source node and put in tooltip
            const sourceCopy = makeCopy(source);
            tooltipContent.appendChild(sourceCopy);

            // attach mouse event listeners
            tooltip.addEventListener('click', onTooltipClick);
            tooltip.addEventListener('mousedown', onTooltipClick);
            tooltip.addEventListener('touchstart', onTooltipClick);
            tooltip.addEventListener('mouseleave', onTooltipUnhover);

            // (for interaction with lightbox plugin)
            // transfer click on tooltip copied img to original img
            const sourceImg = source.querySelector('img');
            const sourceCopyImg = sourceCopy.querySelector('img');
            if (sourceImg && sourceCopyImg) {
                const clickImg = function() {
                    sourceImg.click();
                    closeTooltip();
                };
                sourceCopyImg.addEventListener('click', clickImg);
            }

            return tooltip;
        }

        // make carbon copy of html dom element
        function makeCopy(source) {
            const sourceCopy = source.cloneNode(true);

            // delete elements marked with ignore (eg anchor and jump buttons)
            const deleteFromCopy = sourceCopy.querySelectorAll(
                '[data-ignore="true"]'
            );
            for (const element of deleteFromCopy)
                element.remove();

            // delete certain element attributes
            const attributes = [
                'id',
                'data-collapsed',
                'data-selected',
                'data-highlighted',
                'data-glow'
            ];
            for (const attribute of attributes) {
                sourceCopy.removeAttribute(attribute);
                const elements = sourceCopy.querySelectorAll(
                    '[' + attribute + ']'
                );
                for (const element of elements)
                    element.removeAttribute(attribute);
            }

            return sourceCopy;
        }

        // when tooltip is clicked
        function onTooltipClick(event) {
            // when user clicks on tooltip, stop click from transferring
            // outside of tooltip (eg, click off to close tooltip, or eg click
            // off to unhighlight same refs)
            event.stopPropagation();
        }

        // when tooltip is unhovered
        function onTooltipUnhover(event) {
            if (options.clickClose === 'true')
                return;

            // make sure new mouse/touch/focus no longer over tooltip or any
            // element within it
            const tooltip = document.getElementById('tooltip');
            if (!tooltip)
                return;
            if (this.contains(event.relatedTarget))
                return;

            closeTooltip();
        }

        // make nav bar to go betwen prev/next instances of same reference
        function makeNavBar(link) {
            // find other links to the same source
            const sameLinks = getSameLinks(link);

            // don't show nav bar when singular reference
            if (sameLinks.total <= 1)
                return;

            // find prev/next links with same target
            const prevLink = getPrevLink(link, sameLinks);
            const nextLink = getNextLink(link, sameLinks);

            // create nav bar
            const navBar = document.createElement('div');
            navBar.id = 'tooltip_nav_bar';
            const text = sameLinks.index + 1 + ' of ' + sameLinks.total;

            // create nav bar prev/next buttons
            const prevButton = document.createElement('button');
            const nextButton = document.createElement('button');
            prevButton.id = 'tooltip_prev_button';
            nextButton.id = 'tooltip_next_button';
            prevButton.title =
                'Jump to the previous occurence of this item in the document [←]';
            nextButton.title =
                'Jump to the next occurence of this item in the document [→]';
            prevButton.classList.add('icon_button');
            nextButton.classList.add('icon_button');
            prevButton.innerHTML = document.querySelector(
                '.icon_caret_left'
            ).innerHTML;
            nextButton.innerHTML = document.querySelector(
                '.icon_caret_right'
            ).innerHTML;
            navBar.appendChild(prevButton);
            navBar.appendChild(document.createTextNode(text));
            navBar.appendChild(nextButton);

            // attach click listeners to buttons
            prevButton.addEventListener('click', function() {
                onPrevNextClick(link, prevLink);
            });
            nextButton.addEventListener('click', function() {
                onPrevNextClick(link, nextLink);
            });

            return navBar;
        }

        // get previous link with same target
        function getPrevLink(link, sameLinks) {
            if (!sameLinks)
                sameLinks = getSameLinks(link);
            // wrap index to other side if < 1
            let index;
            if (sameLinks.index - 1 >= 0)
                index = sameLinks.index - 1;
            else
                index = sameLinks.total - 1;
            return sameLinks.elements[index];
        }

        // get next link with same target
        function getNextLink(link, sameLinks) {
            if (!sameLinks)
                sameLinks = getSameLinks(link);
            // wrap index to other side if > total
            let index;
            if (sameLinks.index + 1 <= sameLinks.total - 1)
                index = sameLinks.index + 1;
            else
                index = 0;
            return sameLinks.elements[index];
        }

        // get element that is target of link or url hash
        function getSource(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            // if ref or figure, modify target to get expected element
            if (id.indexOf('ref-') === 0)
                target = target.querySelector('p');
            else if (id.indexOf('fig:') === 0)
                target = target.querySelector('figure');

            return target;
        }

        // when prev/next arrow button is clicked
        function onPrevNextClick(link, prevNextLink) {
            if (link && prevNextLink)
                goToElement(prevNextLink, window.innerHeight * 0.5);
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // determine position to place tooltip based on link position in
        // viewport and tooltip size
        function positionTooltip(link, left, top) {
            const tooltipElement = document.getElementById('tooltip');
            if (!tooltipElement)
                return;

            // get convenient vars for position/dimensions of
            // link/tooltip/page/view
            link = getRectInPage(link);
            const tooltip = getRectInPage(tooltipElement);
            const view = getRectInPage();

            // horizontal positioning
            if (left)
                // use explicit value
                left = left;
            else if (link.left + tooltip.width < view.right)
                // fit tooltip to right of link
                left = link.left;
            else if (link.right - tooltip.width > view.left)
                // fit tooltip to left of link
                left = link.right - tooltip.width;
            // center tooltip in view
            else
                left = (view.right - view.left) / 2 - tooltip.width / 2;

            // vertical positioning
            if (top)
                // use explicit value
                top = top;
            else if (link.top - tooltip.height > view.top)
                // fit tooltip above link
                top = link.top - tooltip.height;
            else if (link.bottom + tooltip.height < view.bottom)
                // fit tooltip below link
                top = link.bottom;
            else {
                // center tooltip in view
                top = view.top + view.height / 2 - tooltip.height / 2;
                // nudge off of link to left/right if possible
                if (link.right + tooltip.width < view.right)
                    left = link.right;
                else if (link.left - tooltip.width > view.left)
                    left = link.left - tooltip.width;
            }

            tooltipElement.style.left = left + 'px';
            tooltipElement.style.top = top + 'px';
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get position of element relative to page
        function getRectInPage(element) {
            const rect = getRectInView(element);
            const body = getRectInView(document.body);

            const newRect = {};
            newRect.left = rect.left - body.left;
            newRect.top = rect.top - body.top;
            newRect.right = rect.right - body.left;
            newRect.bottom = rect.bottom - body.top;
            newRect.width = rect.width;
            newRect.height = rect.height;

            return newRect;
        }

        // (for interaction with accordion plugin)
        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // (for interaction with accordion plugin)
        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // (for interaction with accordion plugin)
        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
    <!-- modified from: https://fontawesome.com/icons/caret-left -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
        ></path>
    </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
    <!-- modified from: https://fontawesome.com/icons/caret-right -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        ></path>
    </svg>
</template>
<!-- jump to first plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin adds a button next to each reference entry,
        // figure, and table that jumps the page to the first occurrence of a
        // link to that item in the manuscript.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'jumpToFirst';

        // default plugin options
        const options = {
            // whether to add buttons next to reference entries
            references: 'true',
            // whether to add buttons next to figures
            figures: 'true',
            // whether to add buttons next to tables
            tables: 'true',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            if (options.references !== 'false')
                makeReferenceButtons();
            if (options.figures !== 'false')
                makeFigureButtons();
            if (options.tables !== 'false')
                makeTableButtons();
        }

        // when jump button clicked
        function onButtonClick() {
            const first = getFirstOccurrence(this.dataset.id);
            if (!first)
                return;

            // update url hash so navigating "back" in history will return
            // user to jump button
            window.location.hash = this.dataset.id;
            // scroll to link
            window.setTimeout(function() {
                goToElement(first, window.innerHeight * 0.5);
            }, 0);
        }

        // get first occurence of link to item in document
        function getFirstOccurrence(id) {
            let query = 'a';
            query += '[href="#' + id + '"]';
            // exclude buttons, anchor links, toc links, etc
            query +=
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            return document.querySelector(query);
        }

        // add button next to each reference entry
        function makeReferenceButtons() {
            const references = document.querySelectorAll('div[id^="ref-"]');
            for (const reference of references) {
                // get reference id and element to add button to
                const id = reference.id;
                const container = reference.firstElementChild;
                const first = getFirstOccurrence(id);

                // if can't find link to reference, ignore
                if (!first)
                    continue;

                // make jump button
                let button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this reference in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.innerHTML = button.outerHTML + container.innerHTML;
                button = container.firstElementChild;
                button.addEventListener('click', onButtonClick);
            }
        }

        // add button next to each figure
        function makeFigureButtons() {
            const figures = document.querySelectorAll('[id^="fig:"]');
            for (const figure of figures) {
                // get figure id and element to add button to
                const id = figure.id;
                const container = figure.querySelector('figcaption') || figure;
                const first = getFirstOccurrence(id);

                // if can't find link to figure, ignore
                if (!first)
                    continue;

                // make jump button
                const button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this figure in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.insertBefore(button, container.firstElementChild);
                button.addEventListener('click', onButtonClick);
            }
        }

        // add button next to each figure
        function makeTableButtons() {
            const tables = document.querySelectorAll('[id^="tbl:"]');
            for (const table of tables) {
                // get ref id and element to add button to
                const id = table.id;
                const container = table.querySelector('caption') || table;
                const first = getFirstOccurrence(id);

                // if can't find link to table, ignore
                if (!first)
                    continue;

                // make jump button
                const button = document.createElement('button');
                button.classList.add('icon_button', 'jump_arrow');
                button.title =
                    'Jump to the first occurence of this table in the document';
                button.innerHTML = document.querySelector(
                    '.icon_angle_double_up'
                ).innerHTML;
                button.dataset.id = id;
                button.dataset.ignore = 'true';
                container.insertBefore(button, container.firstElementChild);
                button.addEventListener('click', onButtonClick);
            }
        }

        // scroll to and focus element
        function goToElement(element, offset) {
            // expand accordion section if collapsed
            expandElement(element);
            const y =
                getRectInView(element).top -
                getRectInView(document.documentElement).top -
                (offset || 0);
            // trigger any function listening for "onscroll" event
            window.dispatchEvent(new Event('scroll'));
            window.scrollTo(0, y);
            document.activeElement.blur();
            element.focus();
        }

        // get position/dimensions of element or viewport
        function getRectInView(element) {
            let rect = {};
            rect.left = 0;
            rect.top = 0;
            rect.right = document.documentElement.clientWidth;
            rect.bottom = document.documentElement.clientHeight;
            let style = {};

            if (element instanceof HTMLElement) {
                rect = element.getBoundingClientRect();
                style = window.getComputedStyle(element);
            }

            const margin = {};
            margin.left = parseFloat(style.marginLeftWidth) || 0;
            margin.top = parseFloat(style.marginTopWidth) || 0;
            margin.right = parseFloat(style.marginRightWidth) || 0;
            margin.bottom = parseFloat(style.marginBottomWidth) || 0;

            const border = {};
            border.left = parseFloat(style.borderLeftWidth) || 0;
            border.top = parseFloat(style.borderTopWidth) || 0;
            border.right = parseFloat(style.borderRightWidth) || 0;
            border.bottom = parseFloat(style.borderBottomWidth) || 0;

            const newRect = {};
            newRect.left = rect.left + margin.left + border.left;
            newRect.top = rect.top + margin.top + border.top;
            newRect.right = rect.right + margin.right + border.right;
            newRect.bottom = rect.bottom + margin.bottom + border.bottom;
            newRect.width = newRect.right - newRect.left;
            newRect.height = newRect.bottom - newRect.top;

            return newRect;
        }

        // get closest element before specified element that matches query
        function firstBefore(element, query) {
            while (
                element &&
                element !== document.body &&
                !element.matches(query)
            )
                element = element.previousElementSibling || element.parentNode;

            return element;
        }

        // check if element is part of collapsed heading
        function isCollapsed(element) {
            while (element && element !== document.body) {
                if (element.dataset.collapsed === 'true')
                    return true;
                element = element.parentNode;
            }
            return false;
        }

        // (for interaction with accordion plugin)
        // expand heading containing element if necesary
        function expandElement(element) {
            if (isCollapsed(element)) {
                const heading = firstBefore(element, 'h2');
                if (heading)
                    heading.click();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- angle double up icon -->

<template class="icon_angle_double_up">
    <!-- modified from: https://fontawesome.com/icons/angle-double-up -->
    <svg width="16" height="16" viewBox="0 0 320 512">
        <path
            fill="currentColor"
            d="M177 255.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 351.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 425.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1zm-34-192L7 199.7c-9.4 9.4-9.4 24.6 0 33.9l22.6 22.6c9.4 9.4 24.6 9.4 33.9 0l96.4-96.4 96.4 96.4c9.4 9.4 24.6 9.4 33.9 0l22.6-22.6c9.4-9.4 9.4-24.6 0-33.9l-136-136c-9.2-9.4-24.4-9.4-33.8 0z"
        ></path>
    </svg>
</template>
<!-- link highlight plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when a user hovers or
        // focuses a link, other links that have the same target will be
        // highlighted. It also makes it such that when clicking a link, the
        // target of the link (eg reference, figure, table) is briefly
        // highlighted.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'linkHighlight';

        // default plugin options
        const options = {
            // whether to also highlight links that go to external urls
            externalLinks: 'false',
            // whether user must click off to unhighlight instead of just
            // un-hovering
            clickUnhighlight: 'false',
            // whether to also highlight links that are unique
            highlightUnique: 'true',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            const links = getLinks();
            for (const link of links) {
                // attach mouse and focus listeners to link
                link.addEventListener('mouseenter', onLinkFocus);
                link.addEventListener('focus', onLinkFocus);
                link.addEventListener('mouseleave', onLinkUnhover);
            }

            // attach click and hash change listeners to window
            window.addEventListener('click', onClick);
            window.addEventListener('touchstart', onClick);
            window.addEventListener('hashchange', onHashChange);

            // run hash change on window load in case user has navigated
            // directly to hash
            onHashChange();
        }

        // when link is focused (tabbed to) or hovered
        function onLinkFocus() {
            highlight(this);
        }

        // when link is unhovered
        function onLinkUnhover() {
            if (options.clickUnhighlight !== 'true')
                unhighlightAll();
        }

        // when the mouse is clicked anywhere in window
        function onClick(event) {
            unhighlightAll();
        }

        // when hash (eg manuscript.html#introduction) changes
        function onHashChange() {
            const target = getHashTarget();
            if (target)
                glowElement(target);
        }

        // get element that is target of link or url hash
        function getHashTarget(link) {
            const hash = link ? link.hash : window.location.hash;
            const id = hash.slice(1);
            let target = document.querySelector('[id="' + id + '"]');
            if (!target)
                return;

            return target;
        }

        // start glow sequence on an element
        function glowElement(element) {
            const startGlow = function() {
                onGlowEnd();
                element.dataset.glow = 'true';
                element.addEventListener('animationend', onGlowEnd);
            };
            const onGlowEnd = function() {
                element.removeAttribute('data-glow');
                element.removeEventListener('animationend', onGlowEnd);
            };
            startGlow();
        }

        // highlight link and all others with same target
        function highlight(link) {
            // force unhighlight all to start fresh
            unhighlightAll();

            // get links with same target
            if (!link)
                return;
            const sameLinks = getSameLinks(link);

            // if link unique and option is off, exit and don't highlight
            if (sameLinks.length <= 1 && options.highlightUnique !== 'true')
                return;

            // highlight all same links, and "select" (special highlight) this
            // one
            for (const sameLink of sameLinks) {
                if (sameLink === link)
                    sameLink.setAttribute('data-selected', 'true');
                else
                    sameLink.setAttribute('data-highlighted', 'true');
            }
        }

        // unhighlight all links
        function unhighlightAll() {
            const links = getLinks();
            for (const link of links) {
                link.setAttribute('data-selected', 'false');
                link.setAttribute('data-highlighted', 'false');
            }
        }

        // get links with same target
        function getSameLinks(link) {
            const results = [];
            const links = getLinks();
            for (const otherLink of links) {
                if (
                    otherLink.getAttribute('href') === link.getAttribute('href')
                )
                    results.push(otherLink);
            }
            return results;
        }

        // get all links of types we wish to handle
        function getLinks() {
            let query = 'a';
            if (options.externalLinks !== 'true')
                query += '[href^="#"]';
            // exclude buttons, anchor links, toc links, etc
            query +=
                ':not(.button):not(.icon_button):not(.anchor):not(.toc_link)';
            return document.querySelectorAll(query);
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>
<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin provides a "table of contents" (toc) panel on
        // the side of the document that allows the user to conveniently
        // navigate between sections of the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'tableOfContents';

        // default plugin options
        const options = {
            // which types of elements to add links for, in
            // "document.querySelector" format
            typesQuery: 'h1, h2, h3',
            // whether toc starts open. use 'true' or 'false', or 'auto' to
            // use 'true' behavior when screen wide enough and 'false' when not
            startOpen: 'false',
            // whether toc closes when clicking on toc link. use 'true' or
            // 'false', or 'auto' to use 'false' behavior when screen wide
            // enough and 'true' when not
            clickClose: 'auto',
            // if list item is more than this many characters, text will be
            // truncated
            charLimit: '50',
            // whether or not to show bullets next to each toc item
            bullets: 'false',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // make toc panel and populate with entries (links to document
            // sections)
            const panel = makePanel();
            if (!panel)
                return;
            makeEntries(panel);
            // attach panel to document after making entries, so 'toc' heading
            // in panel isn't included in toc
            document.body.insertBefore(panel, document.body.firstChild);

            // initial panel state
            if (
                options.startOpen === 'true' ||
                (options.startOpen === 'auto' && !isSmallScreen())
            )
                openPanel();
            else
                closePanel();

            // attach click, scroll, and hash change listeners to window
            window.addEventListener('click', onClick);
            window.addEventListener('scroll', onScroll);
            window.addEventListener('hashchange', onScroll);
            window.addEventListener('keyup', onKeyUp);
            onScroll();

            // add class to push document body down out of way of toc button
            document.body.classList.add('toc_body_nudge');
        }

        // determine if screen wide enough to fit toc panel
        function isSmallScreen() {
            // in default theme:
            // 816px = 8.5in = width of "page" (<body>) element
            // 260px = min width of toc panel (*2 for both sides of <body>)
            return window.innerWidth < 816 + 260 * 2;
        }

        // when mouse is clicked anywhere in window
        function onClick() {
            if (isSmallScreen())
                closePanel();
        }

        // when window is scrolled or hash changed
        function onScroll() {
            highlightViewed();
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            // close on esc
            if (event.key === 'Escape')
                closePanel();
        }

        // find entry of currently viewed document section in toc and highlight
        function highlightViewed() {
            const firstId = getFirstInView(options.typesQuery);

            // get toc entries (links), unhighlight all, then highlight viewed
            const list = document.getElementById('toc_list');
            if (!firstId || !list)
                return;
            const links = list.querySelectorAll('a');
            for (const link of links)
                link.dataset.viewing = 'false';
            const link = list.querySelector('a[href="#' + firstId + '"]');
            if (!link)
                return;
            link.dataset.viewing = 'true';
        }

        // get first or previous toc listed element in top half of view
        function getFirstInView(query) {
            // get all elements matching query and with id
            const elements = document.querySelectorAll(query);
            const elementsWithIds = [];
            for (const element of elements) {
                if (element.id)
                    elementsWithIds.push(element);
            }


            // get first or previous element in top half of view
            for (let i = 0; i < elementsWithIds.length; i++) {
                const element = elementsWithIds[i];
                const prevElement = elementsWithIds[Math.max(0, i - 1)];
                if (element.getBoundingClientRect().top >= 0) {
                    if (
                        element.getBoundingClientRect().top <
                        window.innerHeight / 2
                    )
                        return element.id;
                    else
                        return prevElement.id;
                }
            }
        }

        // make panel
        function makePanel() {
            // create panel
            const panel = document.createElement('div');
            panel.id = 'toc_panel';
            if (options.bullets === 'true')
                panel.dataset.bullets = 'true';

            // create header
            const header = document.createElement('div');
            header.id = 'toc_header';

            // create toc button
            const button = document.createElement('button');
            button.id = 'toc_button';
            button.innerHTML = document.querySelector('.icon_th_list').innerHTML;
            button.title = 'Table of Contents';
            button.classList.add('icon_button');

            // create header text
            const text = document.createElement('h4');
            text.innerHTML = 'Table of Contents';

            // create container for toc list
            const list = document.createElement('div');
            list.id = 'toc_list';

            // attach click listeners
            panel.addEventListener('click', onPanelClick);
            header.addEventListener('click', onHeaderClick);
            button.addEventListener('click', onButtonClick);

            // attach elements
            header.appendChild(button);
            header.appendChild(text);
            panel.appendChild(header);
            panel.appendChild(list);

            return panel;
        }

        // create toc entries (links) to each element of the specified types
        function makeEntries(panel) {
            const elements = document.querySelectorAll(options.typesQuery);
            for (const element of elements) {
                // do not add link if element doesn't have assigned id
                if (!element.id)
                    continue;

                // create link/list item
                const link = document.createElement('a');
                link.classList.add('toc_link');
                switch (element.tagName.toLowerCase()) {
                    case 'h1':
                        link.dataset.level = '1';
                        break;
                    case 'h2':
                        link.dataset.level = '2';
                        break;
                    case 'h3':
                        link.dataset.level = '3';
                        break;
                    case 'h4':
                        link.dataset.level = '4';
                        break;
                }
                link.title = element.innerText;
                let text = element.innerText;
                if (text.length > options.charLimit)
                    text = text.slice(0, options.charLimit) + '...';
                link.innerHTML = text;
                link.href = '#' + element.id;
                link.addEventListener('click', onLinkClick);

                // attach link
                panel.querySelector('#toc_list').appendChild(link);
            }
        }

        // when panel is clicked
        function onPanelClick(event) {
            // stop click from propagating to window/document and closing panel
            event.stopPropagation();
        }

        // when header itself is clicked
        function onHeaderClick(event) {
            togglePanel();
        }

        // when button is clicked
        function onButtonClick(event) {
            togglePanel();
            // stop header underneath button from also being clicked
            event.stopPropagation();
        }

        // when link is clicked
        function onLinkClick(event) {
            if (
                options.clickClose === 'true' ||
                (options.clickClose === 'auto' && isSmallScreen())
            )
                closePanel();
            else
                openPanel();
        }

        // open panel if closed, close if opened
        function togglePanel() {
            const panel = document.getElementById('toc_panel');
            if (!panel)
                return;

            if (panel.dataset.open === 'true')
                closePanel();
            else
                openPanel();
        }

        // open panel
        function openPanel() {
            const panel = document.getElementById('toc_panel');
            if (panel)
                panel.dataset.open = 'true';
        }

        // close panel
        function closePanel() {
            const panel = document.getElementById('toc_panel');
            if (panel)
                panel.dataset.open = 'false';
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- th list icon -->

<template class="icon_th_list">
    <!-- modified from: https://fontawesome.com/icons/th-list -->
    <svg width="16" height="16" viewBox="0 0 512 512" tabindex="-1">
        <path
            fill="currentColor"
            d="M96 96c0 26.51-21.49 48-48 48S0 122.51 0 96s21.49-48 48-48 48 21.49 48 48zM48 208c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm0 160c-26.51 0-48 21.49-48 48s21.49 48 48 48 48-21.49 48-48-21.49-48-48-48zm96-236h352c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h352c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H144c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z"
            tabindex="-1"
        ></path>
    </svg>
</template>
<!-- lightbox plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin makes it such that when a user clicks on an
        // image, the image fills the screen and the user can pan/drag/zoom
        // the image and navigate between other images in the document.

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'lightbox';

        // default plugin options
        const options = {
            // list of possible zoom/scale factors
            zoomSteps:
                '0.1, 0.25, 0.333333, 0.5, 0.666666, 0.75, 1,' +
                '1.25, 1.5, 1.75, 2, 2.5, 3, 3.5, 4, 5, 6, 7, 8',
            // whether to fit image to view ('fit'), display at 100% and shrink
            // if necessary ('shrink'), or always display at 100% ('100')
            defaultZoom: 'fit',
            // whether to zoom in/out toward center of view ('true') or mouse
            // ('false')
            centerZoom: 'false',
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // run through each <img> element
            const imgs = document.querySelectorAll('figure > img');
            let count = 1;
            for (const img of imgs) {
                img.classList.add('lightbox_document_img');
                img.dataset.number = count;
                img.dataset.total = imgs.length;
                img.addEventListener('click', openLightbox);
                count++;
            }

            // attach mouse and key listeners to window
            window.addEventListener('mousemove', onWindowMouseMove);
            window.addEventListener('keyup', onKeyUp);
        }

        // when mouse is moved anywhere in window
        function onWindowMouseMove(event) {
            window.mouseX = event.clientX;
            window.mouseY = event.clientY;
        }

        // when key pressed
        function onKeyUp(event) {
            if (!event || !event.key)
                return;

            switch (event.key) {
                // trigger click of prev button
                case 'ArrowLeft':
                    const prevButton = document.getElementById(
                        'lightbox_prev_button'
                    );
                    if (prevButton)
                        prevButton.click();
                    break;
                // trigger click of next button
                case 'ArrowRight':
                    const nextButton = document.getElementById(
                        'lightbox_next_button'
                    );
                    if (nextButton)
                        nextButton.click();
                    break;
                // close on esc
                case 'Escape':
                    closeLightbox();
                    break;
            }
        }

        // open lightbox
        function openLightbox() {
            const lightbox = makeLightbox(this);
            if (!lightbox)
                return;

            blurBody(lightbox);
            document.body.appendChild(lightbox);
        }

        // make lightbox
        function makeLightbox(img) {
            // delete lightbox if it exists, start fresh
            closeLightbox();

            // create screen overlay containing lightbox
            const overlay = document.createElement('div');
            overlay.id = 'lightbox_overlay';

            // create image info boxes
            const numberInfo = document.createElement('div');
            const zoomInfo = document.createElement('div');
            numberInfo.id = 'lightbox_number_info';
            zoomInfo.id = 'lightbox_zoom_info';

            // create container for image
            const imageContainer = document.createElement('div');
            imageContainer.id = 'lightbox_image_container';
            const lightboxImg = makeLightboxImg(
                img,
                imageContainer,
                numberInfo,
                zoomInfo
            );
            imageContainer.appendChild(lightboxImg);

            // create bottom container for caption and navigation buttons
            const bottomContainer = document.createElement('div');
            bottomContainer.id = 'lightbox_bottom_container';
            const caption = makeCaption(img);
            const prevButton = makePrevButton(img);
            const nextButton = makeNextButton(img);
            bottomContainer.appendChild(prevButton);
            bottomContainer.appendChild(caption);
            bottomContainer.appendChild(nextButton);

            // attach top middle and bottom to overlay
            overlay.appendChild(numberInfo);
            overlay.appendChild(zoomInfo);
            overlay.appendChild(imageContainer);
            overlay.appendChild(bottomContainer);

            return overlay;
        }

        // make <img> object that is intuitively draggable and zoomable
        function makeLightboxImg(
            sourceImg,
            container,
            numberInfoBox,
            zoomInfoBox
        ) {
            // create copy of source <img>
            const img = sourceImg.cloneNode(true);
            img.classList.remove('lightbox_document_img');
            img.removeAttribute('id');
            img.removeAttribute('width');
            img.removeAttribute('height');
            img.style.position = 'unset';
            img.style.margin = '0';
            img.style.padding = '0';
            img.style.width = '';
            img.style.height = '';
            img.style.minWidth = '';
            img.style.minHeight = '';
            img.style.maxWidth = '';
            img.style.maxHeight = '';
            img.id = 'lightbox_img';

            // build sorted list of unique zoomSteps, always including a 100%
            let zoomSteps = [];
            const optionsZooms = options.zoomSteps.split(/[^0-9.]/);
            for (const optionZoom of optionsZooms) {
                const newZoom = parseFloat(optionZoom);
                if (newZoom && !zoomSteps.includes(newZoom))
                    zoomSteps.push(newZoom);
            }
            if (!zoomSteps.includes(1))
                zoomSteps.push(1);
            zoomSteps = zoomSteps.sort(function sortNumber(a, b) {
                return a - b;
            });

            // <img> object property variables
            let zoom = 1;
            let translateX = 0;
            let translateY = 0;
            let clickMouseX = undefined;
            let clickMouseY = undefined;
            let clickTranslateX = undefined;
            let clickTranslateY = undefined;

            updateNumberInfo();

            // update image numbers displayed in info box
            function updateNumberInfo() {
                numberInfoBox.innerHTML =
                    sourceImg.dataset.number + ' of ' + sourceImg.dataset.total;
            }

            // update zoom displayed in info box
            function updateZoomInfo() {
                let zoomInfo = zoom * 100;
                if (!Number.isInteger(zoomInfo))
                    zoomInfo = zoomInfo.toFixed(2);
                zoomInfoBox.innerHTML = zoomInfo + '%';
            }

            // move to closest zoom step above current zoom
            const zoomIn = function() {
                for (const zoomStep of zoomSteps) {
                    if (zoomStep > zoom) {
                        zoom = zoomStep;
                        break;
                    }
                }
                updateTransform();
            };

            // move to closest zoom step above current zoom
            const zoomOut = function() {
                zoomSteps.reverse();
                for (const zoomStep of zoomSteps) {
                    if (zoomStep < zoom) {
                        zoom = zoomStep;
                        break;
                    }
                }
                zoomSteps.reverse();

                updateTransform();
            };

            // update display of <img> based on scale/translate properties
            const updateTransform = function() {
                // set transform
                img.style.transform =
                    'translate(' +
                    (translateX || 0) +
                    'px,' +
                    (translateY || 0) +
                    'px) scale(' +
                    (zoom || 1) +
                    ')';

                // get new width/height after scale
                const rect = img.getBoundingClientRect();
                // limit translate
                translateX = Math.max(translateX, -rect.width / 2);
                translateX = Math.min(translateX, rect.width / 2);
                translateY = Math.max(translateY, -rect.height / 2);
                translateY = Math.min(translateY, rect.height / 2);

                // set transform
                img.style.transform =
                    'translate(' +
                    (translateX || 0) +
                    'px,' +
                    (translateY || 0) +
                    'px) scale(' +
                    (zoom || 1) +
                    ')';

                updateZoomInfo();
            };

            // fit <img> to container
            const fit = function() {
                // no x/y offset, 100% zoom by default
                translateX = 0;
                translateY = 0;
                zoom = 1;

                // widths of <img> and container
                const imgWidth = img.naturalWidth;
                const imgHeight = img.naturalHeight;
                const containerWidth = parseFloat(
                    window.getComputedStyle(container).width
                );
                const containerHeight = parseFloat(
                    window.getComputedStyle(container).height
                );

                // how much zooming is needed to fit <img> to container
                const xRatio = imgWidth / containerWidth;
                const yRatio = imgHeight / containerHeight;
                const maxRatio = Math.max(xRatio, yRatio);
                const newZoom = 1 / maxRatio;

                // fit <img> to container according to option
                if (options.defaultZoom === 'shrink') {
                    if (maxRatio > 1)
                        zoom = newZoom;
                } else if (options.defaultZoom === 'fit')
                    zoom = newZoom;

                updateTransform();
            };

            // when mouse wheel is rolled anywhere in container
            const onContainerWheel = function(event) {
                if (!event)
                    return;

                // let ctrl + mouse wheel to zoom behave as normal
                if (event.ctrlKey)
                    return;

                // prevent normal scroll behavior
                event.preventDefault();
                event.stopPropagation();

                // point around which to scale img
                const viewRect = container.getBoundingClientRect();
                const viewX = (viewRect.left + viewRect.right) / 2;
                const viewY = (viewRect.top + viewRect.bottom) / 2;
                const originX = options.centerZoom === 'true' ? viewX : mouseX;
                const originY = options.centerZoom === 'true' ? viewY : mouseY;

                // get point on image under origin
                const oldRect = img.getBoundingClientRect();
                const oldPercentX = (originX - oldRect.left) / oldRect.width;
                const oldPercentY = (originY - oldRect.top) / oldRect.height;

                // increment/decrement zoom
                if (event.deltaY < 0)
                    zoomIn();
                if (event.deltaY > 0)
                    zoomOut();

                // get offset between previous image point and origin
                const newRect = img.getBoundingClientRect();
                const offsetX =
                    originX - (newRect.left + newRect.width * oldPercentX);
                const offsetY =
                    originY - (newRect.top + newRect.height * oldPercentY);

                // translate image to keep image point under origin
                translateX += offsetX;
                translateY += offsetY;

                // perform translate
                updateTransform();
            };

            // when container is clicked
            function onContainerClick(event) {
                // if container itself is target of click, and not other
                // element above it
                if (event.target === this)
                    closeLightbox();
            }

            // when mouse button is pressed on image
            const onImageMouseDown = function(event) {
                // store original mouse position relative to image
                clickMouseX = window.mouseX;
                clickMouseY = window.mouseY;
                clickTranslateX = translateX;
                clickTranslateY = translateY;
                event.stopPropagation();
                event.preventDefault();
            };

            // when mouse button is released anywhere in window
            const onWindowMouseUp = function(event) {
                // reset original mouse position
                clickMouseX = undefined;
                clickMouseY = undefined;
                clickTranslateX = undefined;
                clickTranslateY = undefined;

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('mouseup', onWindowMouseUp);
            };

            // when mouse is moved anywhere in window
            const onWindowMouseMove = function(event) {
                if (
                    clickMouseX === undefined ||
                    clickMouseY === undefined ||
                    clickTranslateX === undefined ||
                    clickTranslateY === undefined
                )
                    return;

                // offset image based on original and current mouse position
                translateX = clickTranslateX + window.mouseX - clickMouseX;
                translateY = clickTranslateY + window.mouseY - clickMouseY;
                updateTransform();
                event.preventDefault();

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('mousemove', onWindowMouseMove);
            };

            // when window is resized
            const onWindowResize = function(event) {
                fit();

                // remove global listener if lightbox removed from document
                if (!document.body.contains(container))
                    window.removeEventListener('resize', onWindowResize);
            };

            // attach the necessary event listeners
            img.addEventListener('dblclick', fit);
            img.addEventListener('mousedown', onImageMouseDown);
            container.addEventListener('wheel', onContainerWheel);
            container.addEventListener('mousedown', onContainerClick);
            container.addEventListener('touchstart', onContainerClick);
            window.addEventListener('mouseup', onWindowMouseUp);
            window.addEventListener('mousemove', onWindowMouseMove);
            window.addEventListener('resize', onWindowResize);

            // run fit() after lightbox atttached to document and <img> Loaded
            // so needed container and img dimensions available
            img.addEventListener('load', fit);

            return img;
        }

        // make caption
        function makeCaption(img) {
            const caption = document.createElement('div');
            caption.id = 'lightbox_caption';
            const captionSource = img.nextElementSibling;
            if (captionSource.tagName.toLowerCase() === 'figcaption') {
                const captionCopy = makeCopy(captionSource);
                caption.innerHTML = captionCopy.innerHTML;
            }

            caption.addEventListener('touchstart', function(event) {
                event.stopPropagation();
            });

            return caption;
        }

        // make carbon copy of html dom element
        function makeCopy(source) {
            const sourceCopy = source.cloneNode(true);

            // delete elements marked with ignore (eg anchor and jump buttons)
            const deleteFromCopy = sourceCopy.querySelectorAll(
                '[data-ignore="true"]'
            );
            for (const element of deleteFromCopy)
                element.remove();

            // delete certain element attributes
            const attributes = [
                'id',
                'data-collapsed',
                'data-selected',
                'data-highlighted',
                'data-glow'
            ];
            for (const attribute of attributes) {
                sourceCopy.removeAttribute(attribute);
                const elements = sourceCopy.querySelectorAll(
                    '[' + attribute + ']'
                );
                for (const element of elements)
                    element.removeAttribute(attribute);
            }

            return sourceCopy;
        }

        // make button to jump to previous image in document
        function makePrevButton(img) {
            const prevButton = document.createElement('button');
            prevButton.id = 'lightbox_prev_button';
            prevButton.title = 'Jump to the previous image in the document [←]';
            prevButton.classList.add('icon_button', 'lightbox_button');
            prevButton.innerHTML = document.querySelector(
                '.icon_caret_left'
            ).innerHTML;

            // attach click listeners to button
            prevButton.addEventListener('click', function() {
                getPrevImg(img).click();
            });

            return prevButton;
        }

        // make button to jump to next image in document
        function makeNextButton(img) {
            const nextButton = document.createElement('button');
            nextButton.id = 'lightbox_next_button';
            nextButton.title = 'Jump to the next image in the document [→]';
            nextButton.classList.add('icon_button', 'lightbox_button');
            nextButton.innerHTML = document.querySelector(
                '.icon_caret_right'
            ).innerHTML;

            // attach click listeners to button
            nextButton.addEventListener('click', function() {
                getNextImg(img).click();
            });

            return nextButton;
        }

        // get previous image in document
        function getPrevImg(img) {
            const imgs = document.querySelectorAll('.lightbox_document_img');

            // find index of provided img
            let index;
            for (index = 0; index < imgs.length; index++) {
                if (imgs[index] === img)
                    break;
            }


            // wrap index to other side if < 1
            if (index - 1 >= 0)
                index--;
            else
                index = imgs.length - 1;
            return imgs[index];
        }

        // get next image in document
        function getNextImg(img) {
            const imgs = document.querySelectorAll('.lightbox_document_img');

            // find index of provided img
            let index;
            for (index = 0; index < imgs.length; index++) {
                if (imgs[index] === img)
                    break;
            }


            // wrap index to other side if > total
            if (index + 1 <= imgs.length - 1)
                index++;
            else
                index = 0;
            return imgs[index];
        }

        // close lightbox
        function closeLightbox() {
            focusBody();

            const lightbox = document.getElementById('lightbox_overlay');
            if (lightbox)
                lightbox.remove();
        }

        // make all elements behind lightbox non-focusable
        function blurBody(overlay) {
            const all = document.querySelectorAll('*');
            for (const element of all)
                element.tabIndex = -1;
            document.body.classList.add('body_no_scroll');
        }

        // make all elements focusable again
        function focusBody() {
            const all = document.querySelectorAll('*');
            for (const element of all)
                element.removeAttribute('tabIndex');
            document.body.classList.remove('body_no_scroll');
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>

<!-- caret left icon -->

<template class="icon_caret_left">
    <!-- modified from: https://fontawesome.com/icons/caret-left -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M192 127.338v257.324c0 17.818-21.543 26.741-34.142 14.142L29.196 270.142c-7.81-7.81-7.81-20.474 0-28.284l128.662-128.662c12.599-12.6 34.142-3.676 34.142 14.142z"
        ></path>
    </svg>
</template>

<!-- caret right icon -->

<template class="icon_caret_right">
    <!-- modified from: https://fontawesome.com/icons/caret-right -->
    <svg width="16" height="16" viewBox="0 0 192 512">
        <path
            fill="currentColor"
            d="M0 384.662V127.338c0-17.818 21.543-26.741 34.142-14.142l128.662 128.662c7.81 7.81 7.81 20.474 0 28.284L34.142 398.804C21.543 411.404 0 402.48 0 384.662z"
        ></path>
    </svg>
</template>
<!-- attributes plugin -->

<script>
    (function() {
        // /////////////////////////
        // DESCRIPTION
        // /////////////////////////

        // This Manubot plugin allows arbitrary HTML attributes to be attached
        // to (almost) any element. Place an HTML comment inside or next to the
        // desired element in the format <!-- $attribute="value" -->

        // /////////////////////////
        // OPTIONS
        // /////////////////////////

        // plugin name prefix for url parameters
        const pluginName = 'attributes';

        // default plugin options
        const options = {
            // whether plugin is on or not
            enabled: 'true'
        };

        // change options above, or override with url parameter, eg:
        // 'manuscript.html?pluginName-enabled=false'

        // /////////////////////////
        // SCRIPT
        // /////////////////////////

        // start script
        function start() {
            // get list of comments in document
            const comments = findComments();

            for(const comment of comments)
                if (comment.parentElement)
                    addAttributes(
                        comment.parentElement,
                        comment.nodeValue.trim()
                    );
        }

        // add html attributes to specified element based on string of 
        // html attributes and values
        function addAttributes(element, text) {
            // regex's for finding attribute/value pairs in the format of
            // attribute="value" or attribute='value
            const regex2 = /\$([a-zA-Z\-]+)?=\"(.+?)\"/;
            const regex1 = /\$([a-zA-Z\-]+)?=\'(.+?)\'/;

            // loop through attribute/value pairs
            let match;
            while(match = text.match(regex2) || text.match(regex1)) {
                // get attribute and value from regex capture groups
                let attribute = match[1];
                let value = match[2];

                // remove from string
                text = text.substring(match.index + match[0].length);

                if (!attribute || !value)
                    break;

                // set attribute of parent element
                try {
                    element.setAttribute(attribute, value);
                } catch(error) {
                    console.log(error);
                }

                // special case for colspan
                if (attribute === 'colspan')
                    removeTableCells(element, value);
            }
        }

        // get list of comment elements in document
        function findComments() {
            const comments = [];

            // iterate over comment nodes in document
            function acceptNode(node) {
                return NodeFilter.FILTER_ACCEPT;
            }
            const iterator = document.createNodeIterator(
                document.body,
                NodeFilter.SHOW_COMMENT,
                acceptNode
            );
            let node;
            while(node = iterator.nextNode())
                comments.push(node);

            return comments;
        }

        // remove certain number of cells after specified cell
        function removeTableCells(cell, number) {
            number = parseInt(number);
            if (!number)
                return;

            // remove elements
            for(; number > 1; number--) {
                if (cell.nextElementSibling)
                    cell.nextElementSibling.remove();
            }
        }

        // load options from url parameters
        function loadOptions() {
            const url = window.location.search;
            const params = new URLSearchParams(url);
            for (const optionName of Object.keys(options)) {
                const paramName = pluginName + '-' + optionName;
                const param = params.get(paramName);
                if (param !== '' && param !== null)
                    options[optionName] = param;
            }
        }
        loadOptions();

        // start script when document is finished loading
        if (options.enabled === 'true')
            window.addEventListener('load', start);
    })();
</script>
<!-- mathjax plugin configuration -->

<script type="text/x-mathjax-config">
    MathJax.Hub.Config({
        "CommonHTML": { linebreaks: { automatic: true } },
        "HTML-CSS": { linebreaks: { automatic: true } },
        "SVG": { linebreaks: { automatic: true } },
        "fast-preview": { disabled: true }
  });
</script>

<!-- mathjax plugin -->

<script
    src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-MML-AM_CHTML"
    integrity="sha512-0doc9hKxR3PYwso42RD1p5ySZpzzuDiOwMrdCEh2WdJZCjcmFKc/wEnL+z8fBQrnHoiNWbo+3fiGkOYXBdQp4A=="
    crossorigin="anonymous"
>
    // /////////////////////////
    // DESCRIPTION
    // /////////////////////////

    // This third-party plugin 'MathJax' allows the proper rendering of
    // math/equations written in LaTeX.

    // https://www.mathjax.org/
</script>
<!-- annotations plugin -->

<script>
    // /////////////////////////
    // DESCRIPTION
    // /////////////////////////

    // This third-party plugin 'Hypothesis' allows public annotation of the
    // manuscript.

    // https://web.hypothes.is/

    // plugin configuration
    window.hypothesisConfig = function() {
        return {
            branding: {
                accentColor: '#2196f3',
                appBackgroundColor: '#f8f8f8',
                ctaBackgroundColor: '#f8f8f8',
                ctaTextColor: '#000000',
                selectionFontFamily: 'Open Sans, Helvetica, sans serif',
                annotationFontFamily: 'Open Sans, Helvetica, sans serif'
            }
        };
    };

    // hypothesis client script
    const embed = 'https://hypothes.is/embed.js';
    // hypothesis annotation count query url
    const query = 'https://api.hypothes.is/api/search?limit=0&url='

    
    // start script
    function start() {
        const button = makeButton();
        document.body.insertBefore(button, document.body.firstChild);
        insertCount(button);
    }

    // make button
    function makeButton() {
        // create button
        const button = document.createElement('button');
        button.id = 'hypothesis_button';
        button.innerHTML = document.querySelector('.icon_hypothesis').innerHTML;
        button.title = 'Hypothesis annotations';
        button.classList.add('icon_button');

        function onClick(event) {
            onButtonClick(event, button);
        }

        // attach click listeners
        button.addEventListener('click', onClick);

        return button;
    }

    // insert annotations count
    async function insertCount(button) {
        // get annotation count from Hypothesis based on url
        let count = '-';
        try {
            const canonical = document.querySelector('link[rel="canonical"]');
            const location = window.location;
            const url = encodeURIComponent((canonical || location).href);
            const response = await fetch(query + url);
            const json = await response.json();
            count = json.total || '-';
        } catch(error) {
            console.log(error);
        }
        
        // put count into button
        const counter = document.createElement('span');
        counter.id = 'hypothesis_count';
        counter.innerHTML = count;
        button.title = 'View ' + count + ' Hypothesis annotations';
        button.append(counter);
    }

    // when button is clicked
    function onButtonClick(event, button) {
        const script = document.createElement('script');
        script.src = embed;
        document.body.append(script);
        button.remove();
    }

    window.addEventListener('load', start);
</script>

<!-- hypothesis icon -->

<template class="icon_hypothesis">
    <!-- modified from: https://simpleicons.org/icons/hypothesis.svg / https://git.io/Jf1VB -->
    <svg width="16" height="16" viewBox="0 0 24 24" tabindex="-1">
        <path
            fill="currentColor"
            d="M3.43 0C2.5 0 1.72 .768 1.72 1.72V18.86C1.72 19.8 2.5 20.57 3.43 20.57H9.38L12 24L14.62 20.57H20.57C21.5 20.57 22.29 19.8 22.29 18.86V1.72C22.29 .77 21.5 0 20.57 0H3.43M5.14 3.43H7.72V9.43S8.58 7.72 10.28 7.72C12 7.72 13.74 8.57 13.74 11.24V17.14H11.16V12C11.16 10.61 10.28 10.07 9.43 10.29C8.57 10.5 7.72 11.41 7.72 13.29V17.14H5.14V3.43M18 13.72C18.95 13.72 19.72 14.5 19.72 15.42A1.71 1.71 0 0 1 18 17.13A1.71 1.71 0 0 1 16.29 15.42C16.29 14.5 17.05 13.71 18 13.71Z"
            tabindex="-1"
        ></path>
    </svg>
</template>
<!-- analytics plugin -->

<!-- copy and paste code from Google Analytics or similar service here -->
</body>
</html>
